{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pickle\n",
    "import pandas as pd\n",
    "import spacy\n",
    "import matplotlib.pyplot as plt\n",
    "import numpy as np\n",
    "import seaborn as sns\n",
    "import scipy\n",
    "from spacy.vectors import Vectors\n",
    "from spacy.strings import StringStore\n",
    "from scipy.cluster.hierarchy import dendrogram, linkage\n",
    "from wordcloud import WordCloud\n",
    "from nltk.corpus import stopwords\n",
    "\n",
    "from six import iteritems\n",
    "from gensim import corpora\n",
    "from gensim.models.doc2vec import Doc2Vec, TaggedDocument\n",
    "from nltk.tokenize import word_tokenize\n",
    "from nltk.corpus import stopwords\n",
    "\n",
    "from retrieve_data import get_descriptive_fields\n",
    "\n",
    "%matplotlib inline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/ryohayama/Python/b13logy/nlp'"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "os.getcwd()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Retrive with sql query\n",
    "df = get_descriptive_fields(num_rows=6000, db_file='../database_files/b13logy.ini')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>dataset_id</th>\n",
       "      <th>sample_protocol</th>\n",
       "      <th>data_protocol</th>\n",
       "      <th>description</th>\n",
       "      <th>instruments</th>\n",
       "      <th>exp_types</th>\n",
       "      <th>quant_methods</th>\n",
       "      <th>labhead_fullname</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PXD000001</td>\n",
       "      <td>Not available</td>\n",
       "      <td>Two extra files have been added post-publicati...</td>\n",
       "      <td>Expected reporter ion ratios: Erwinia peptides...</td>\n",
       "      <td>LTQ Orbitrap Velos, instrument model</td>\n",
       "      <td>Bottom-up proteomics</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PXD000002</td>\n",
       "      <td>Not available</td>\n",
       "      <td>Not available</td>\n",
       "      <td>Human saliva samples from a 36 year old male w...</td>\n",
       "      <td>instrument model, LTQ Velos</td>\n",
       "      <td>Bottom-up proteomics</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  dataset_id sample_protocol  \\\n",
       "0  PXD000001   Not available   \n",
       "1  PXD000002   Not available   \n",
       "\n",
       "                                       data_protocol  \\\n",
       "0  Two extra files have been added post-publicati...   \n",
       "1                                      Not available   \n",
       "\n",
       "                                         description  \\\n",
       "0  Expected reporter ion ratios: Erwinia peptides...   \n",
       "1  Human saliva samples from a 36 year old male w...   \n",
       "\n",
       "                            instruments             exp_types quant_methods  \\\n",
       "0  LTQ Orbitrap Velos, instrument model  Bottom-up proteomics          None   \n",
       "1           instrument model, LTQ Velos  Bottom-up proteomics          None   \n",
       "\n",
       "  labhead_fullname  \n",
       "0             None  \n",
       "1             None  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "5452"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get descriptive text\n",
    "sample_proto = df['sample_protocol'].dropna()\n",
    "sample_proto = sample_proto[sample_proto != 'Not available']\n",
    "\n",
    "data_proto = df['data_protocol'].dropna()\n",
    "data_proto = data_proto[data_proto != 'Not available']\n",
    "\n",
    "descrip = df['description'].dropna()\n",
    "\n",
    "corpus = list(sample_proto) + list(data_proto) + list(descrip)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['The crude membranes from 5 P56-P70 Glun1TAP/TAP mouse forebrains were re-suspended in 12.5\\u2009ml buffer H and extracted with 12.5\\u2009ml 2% deoxycholate, 100\\u2009mM NaCl, 50\\u2009mM Tris.Cl pH8 for 1\\u2009h at 6\\u2009°C. Total extract was centrifuged at 120,000g. for 40\\u2009min at 8\\u2009°C. Conditions for immuno-capture, wash and peptide-antigen exchange elution were screened using a high-throughput purification robot (MAGic sample processor, Invitrogen). For 25\\u2009ml Glun1TAP/TAP extract supernatant, 80\\u2009μg mouse Flag antibody was coupled to 30\\u2009mg (500\\u2009μl) protein G magnetic beads (Invitrogen). Receptor was captured from extract supernatant for 2\\u2009h. The beads were washed three times with 5\\u2009ml wash buffer (0.37% w/v sodium deoxycholate, 0.05\\u2009mg.ml−1 lipids (1:1:3 POPC:POPE:POG), 150\\u2009mM NaCl, 50\\u2009mM Tris.Cl pH8). Flag captured complexes were eluted with 600\\u2009μl wash buffer supplemented with 0.2\\u2009mg.ml−1 Flag peptide for 2\\u2009h at 6\\u2009°C. Eluate was buffer exchanged to remove the Flag elution peptide and concentrated with a 100-kDa MWCO centrifugal filter (Amicon) to 35\\u2009μl 0.1–0.25\\u2009mg.ml−1 native NMDAR. BNP was run according to Schägger. Excised BNP gel slices containing the purified proteins were prepared for mass spectrometric analysis by manual in situ enzymatic digestion. Briefly, the excised protein gel pieces were placed in a well of a 96-well microtitre plate and de-stained with 50% v/v acetonitrile and 50\\u2009mM ammonium bicarbonate, reduced with 10\\u2009mM DTT and alkylated with 55\\u2009mM iodoacetamide. After alkylation, proteins were digested with 6\\u2009ng\\u2009μl−1 Trypsin (Promega, UK) overnight at 37\\u2009°C. The resulting peptides were extracted in 2% v/v formic acid, 2% v/v acetonitrile. The digest was analysed by nano-scale capillary LC-MS/MS using an Ultimate U3000 HPLC (ThermoScientific Dionex, San Jose, USA) to deliver a flow of ~300\\u2009nl\\u2009min−1. A C18 Acclaim PepMap100 5\\u2009μm, 100\\u2009μm × 20\\u2009mm nanoViper (ThermoScientific Dionex, San Jose, USA), trapped the peptides before separation on a C18 Acclaim PepMap100 3\\u2009μm, 75\\u2009μm × 150\\u2009mm nanoViper (ThermoScientific Dionex, San Jose, USA). Peptides were eluted with a gradient of acetonitrile. The analytical column outlet was directly interfaced via a modified nano-flow electrospray ionization source, with a hybrid dual pressure linear ion trap mass spectrometer (Orbitrap Velos, ThermoScientific, San Jose, USA).',\n",
       " 'Breast cancer tissue lysates, reduction, alkylation by methyl-methanethiosulfonate, trypsin digestion, iTRAQ 8-plex labeling, HILIC fractionation, LC-MS/MS on Orbitrap Velos.  Transcriptomics analysis, immunohistochemistry. Detailed description provided in the manuscript.',\n",
       " 'In-gel digests were performed as described in standard protocols. Briefly, following the SDS-PAGE and washing of the excised gel slices proteins were reduced by adding 10 mM DTT (Sigma Aldrich) prior to alkylation with 55 mM iodoacetamide (Sigma Aldrich). After washing and shrinking of the gel pieces with 100% acetonitrile trypsin (Sequencing Grade Modified, Promega) was added and proteins were digested overnight in 40 mM ammoniumbicarbonate at 37°C. For protein identification samples were directly used for nano-ESI-LC-MS/MS. Each sample was first desalted on-line by a C18 microcolumn (300 µm i.d. x 5 mm, packed with C18 PepMap™, 5 µm, 100 Å by LC Packings). Peptides were then separated on a C18 reversed phase column via a linear acetonitrile gradient (UltiMate 3000 system (Dionex) and column (75 µm i.d. x 15 cm, packed with C18 PepMap™, 3 µm, 100 Å by LC Packings) before MS and MS/MS spectra were recorded on an Oribitrap mass spectrometer (Thermo Electron). ',\n",
       " 'Conditioned media preparation and HPLC-isobaric tags for relative and absolute quantitation mass-spectrometry (iTRAQ MS). Twenty million cells were transfected with either control or Gαs siRNA, then were seeded in 100-mm cell culture plates in DMEM + 10% FBS. Two days later, cells were washed twice with pre-warmed PBS. Conditioned media was harvested another 24 h later, and centrifuged at 20,000 g for 10 min at 4°C to remove cell debris. To enrich for secreted proteins in the conditioned media, conditioned media samples were centrifuged using Amicon centrifugal filters with a 3kDa cutoff (Millipore, Billerica, MA). A total of 8 concentrated conditioned media samples were independently prepared: 4 samples from control siRNA-treated cells, and 4 samples from Gαs siRNA-treated cells. Samples were stored at –70°C until they were analyzed by iTRAQ labeling and LC-MS/MS. iTRAQ labeling – Total protein concentration of each sample was measured using the Bradford assay. A quantity of 100 µg was taken from each sample, denatured in 0.1% SDS, reduced in 50 mM Tris(2-carboxyethyl)phosphine (TCEP), alkylated in 200 mM methyl methanethiosulfonate (MMTS), and digested overnight with trypsin. Each sample’s peptides were labeled with different iTRAQ reagents (AB Sciex, Framingham, MA) following the manufacturer’s protocol, combined, dried down to a 50 µL volume, acidified with 1% trifluoroacetic acid (TFA), and desalted with a 100 mg-C18 SepPak cartridge (Waters, Milford, MA). Peptide fractionation – Combined iTRAQ labeled peptides were first separated into 12 fractions using a 3100 OFFGEL Fractionator (Agilent, Santa Clara, CA) with a 12-well setup and a 12-cm pH 3–10 immobilized pH-gradient (IPG) strip, according to manufacturer’s recommendations. Peptides were collected after the focusing voltage reached a maximum plateau (> 50 kVh). The recovered fractions were lyophilized. LC-MALDI-TOF/TOF analysis – All solvents used in this section were aqueous unless specified and obtained from ThermoFisher Scientific (San Jose, CA). Dried samples were reconstituted in 30 µL of solvent A (2% v/v acetonitrile [ACN], 0.1% v/v TFA), and 10 µL of the resulting solution was injected into an Ultimate™ HPLC system with UV detection (LC Packings, Sunnyvale, CA). Solvent B contained 98% v/v ACN, 0.1% v/v TFA. A C18 trap (5.0 mm x 300 µm ID, LC Packings) and a 15 cm x 100 µm ID column packed in-house with Magic C18 5 µm beads (Michrom Bioresources, Auburn, CA) were used for chromatography. Solvent flow rate was set to 0.700 µL/min. After injection, each sample was loaded and washed with solvent A for 5 min at a flow rate of 0.030 mL/min, followed by a 10% to 50% gradient over 55 min, then ramped to 90% by 65 min. Each sample run was maintained in 90% solvent B until 75 min, returned to 10% at 75.1 min, followed by a 10 min equilibration in 10% solvent B. The HPLC system is connected to a modified Probot™ automated MALDI spotter (LC Packings) to spot eluates on an LC MALDI plate with on-line mixing of α-Cyano-4-hydroxycinnamic acid (CHCA) matrix (7.5 mg/mL in 70% ACN). Considering the dead time for the HPLC and spotting system, spot collection time was started 10 min from injection time with a spotting time of 0.33 min per spot. A total of 176 spots were collected. The sample plate with the resulting spots was loaded onto a 5800 MALDI time-of-flight/time-of-flight (TOF/TOF) mass spectrometer (AB Sciex, Framingham, MA) for MS and MS/MS analysis. Batch-mode mass calibration with plate alignment were done before data acquisition, thus a separate plate calibration file was generated for each individual plate containing sample spots. Full scan MS spectra were obtained with a total of 1000 laser shots per spectrum. From each spot, the top 20 most intense peaks, weakest precursor first, that were above a signal-to-noise threshold of 10, were selected for successful MS/MS analysis. Each MS/MS spectrum was an average of 1600 laser shots. MS/MS spectra were acquired in the presence of collision gas. The sequence of the VGF-NERP1 peptide was confirmed using the LTQ-Orbitrap mass spectrometer (ThermoFisher Scientific, San Jose, CA).',\n",
       " 'Proteins were eluted from washed beads by adding 2-fold-concentrated SDS-gel sample loading buffer and heating to 80°C for 10 min. The supernatant was loaded onto a Novex gel (10% Bis-Tris SDS gel; Life Technologies/Invitrogen, Carlsbad, CA) and run about 1/3 of the length of the lane. The lanes were cut out and stained with InstantBlue (Expedeon Ltd., Harston, United Kingdom) in separate trays until bands were just visible. The area covering the bands was cut into 5 to 6 slices, which were washed, reduced, and alkylated and treated with trypsin according to standard procedures. Peptides were extracted with 5% formic acid–50% acetonitrile, dried down, and redissolved in 0.1% trifluoroacetic acid (TFA). For liquid chromatography-tandem mass spectrometry (LC-MS/MS) analysis, a sample aliquot was applied via a nanoAcquity (Waters, Manchester, United Kingdom) ultraperformance liquid chromatography (UPLC) system running at a flow rate of 250 nl min−1 to an LTQ-Orbitrap mass spectrometer (Thermo Fisher, Waltham, MA). Peptides were trapped using a precolumn (Symmetry C18, 5 μm, 180 μm by 20 mm; Waters), which was then switched in-line to an analytical column (BEH C18, 1.7 μm, 75 μm by 250 mm; Waters) for separation. Peptides were eluted with a gradient of 3 to 40% acetonitrile in water–0.1% formic acid at a rate of 0.67% min−1. The column was connected to a 10-μm SilicaTip nanospray emitter (New Objective, Woburn, MA, USA) attached to a nanospray interface (Proxeon, Odense, Denmark) for infusion into the mass spectrometer. The mass spectrometer was operated in positive ion mode at a capillary temperature of 200°C. The source voltage and focusing voltages were tuned for the transmission of Met-Arg-Phe-Ala (MRFA) peptide (m/z 524) (Sigma-Aldrich, St. Louis, MO). Data-dependent analysis was carried out in orbitrap-ion trap parallel mode using collision-induced fragmentation (CID) on the 6 most abundant ions in each cycle. The orbitrap was run with a resolution of 30,000 over the MS range from m/z 350 to m/z 1,800 and an MS target of 106 and 1-s maximum scan time. Collision energy was 35, and an isolation width of 2 was used. Only monoisotopic 2+ and 3+ charged precursors were selected for MS2 fragmentation stage. The MS2 was triggered by a minimal signal of 1,000 with an automatic gain control target of 3 × 104 ions and 150-ms scan time using the chromatography function for peak apex detection. Dynamic exclusion was set to 1 count and 60-s exclusion with an exclusion mass window of ±20 ppm. MS scans were saved in profile mode, while MS/MS scans were saved in centroid mode.']"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "corpus[:5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "nlp = spacy.load('en_core_web_lg')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "def lemmatize_text(text):\n",
    "    doc = nlp(text)\n",
    "    lemm_text = [\n",
    "            token.lemma_.lower() for token in doc \n",
    "            if token.text != '\\u2009'\n",
    "            if not token.is_stop\n",
    "            if not token.is_punct\n",
    "            if not token.is_bracket\n",
    "            if not token.is_quote\n",
    "            if token.pos_ != 'PRON'\n",
    "            if token.tag_ != 'BES'\n",
    "            if token.tag_ != 'IN'\n",
    "            if token.tag_ !='HVS'\n",
    "            if token.tag_ !='PDT'\n",
    "            if token.tag_ !='TO'\n",
    "            if token.tag_ !='UH'\n",
    "            \n",
    "    ]\n",
    "    return lemm_text\n",
    "\n",
    "def lemmatize_corpus(corpus):\n",
    "    lemm_corpus = []\n",
    "    for text in corpus:\n",
    "        lemm_text = lemmatize_text(text)\n",
    "        lemm_corpus.append(lemm_text)\n",
    "    return lemm_corpus"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 49min 58s, sys: 47min 56s, total: 1h 37min 54s\n",
      "Wall time: 12min 32s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "lemm_corpus = lemmatize_corpus(corpus)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['The crude membranes from 5 P56-P70 Glun1TAP/TAP mouse forebrains were re-suspended in 12.5\\u2009ml buffer H and extracted with 12.5\\u2009ml 2% deoxycholate, 100\\u2009mM NaCl, 50\\u2009mM Tris.Cl pH8 for 1\\u2009h at 6\\u2009°C. Total extract was centrifuged at 120,000g. for 40\\u2009min at 8\\u2009°C. Conditions for immuno-capture, wash and peptide-antigen exchange elution were screened using a high-throughput purification robot (MAGic sample processor, Invitrogen). For 25\\u2009ml Glun1TAP/TAP extract supernatant, 80\\u2009μg mouse Flag antibody was coupled to 30\\u2009mg (500\\u2009μl) protein G magnetic beads (Invitrogen). Receptor was captured from extract supernatant for 2\\u2009h. The beads were washed three times with 5\\u2009ml wash buffer (0.37% w/v sodium deoxycholate, 0.05\\u2009mg.ml−1 lipids (1:1:3 POPC:POPE:POG), 150\\u2009mM NaCl, 50\\u2009mM Tris.Cl pH8). Flag captured complexes were eluted with 600\\u2009μl wash buffer supplemented with 0.2\\u2009mg.ml−1 Flag peptide for 2\\u2009h at 6\\u2009°C. Eluate was buffer exchanged to remove the Flag elution peptide and concentrated with a 100-kDa MWCO centrifugal filter (Amicon) to 35\\u2009μl 0.1–0.25\\u2009mg.ml−1 native NMDAR. BNP was run according to Schägger. Excised BNP gel slices containing the purified proteins were prepared for mass spectrometric analysis by manual in situ enzymatic digestion. Briefly, the excised protein gel pieces were placed in a well of a 96-well microtitre plate and de-stained with 50% v/v acetonitrile and 50\\u2009mM ammonium bicarbonate, reduced with 10\\u2009mM DTT and alkylated with 55\\u2009mM iodoacetamide. After alkylation, proteins were digested with 6\\u2009ng\\u2009μl−1 Trypsin (Promega, UK) overnight at 37\\u2009°C. The resulting peptides were extracted in 2% v/v formic acid, 2% v/v acetonitrile. The digest was analysed by nano-scale capillary LC-MS/MS using an Ultimate U3000 HPLC (ThermoScientific Dionex, San Jose, USA) to deliver a flow of ~300\\u2009nl\\u2009min−1. A C18 Acclaim PepMap100 5\\u2009μm, 100\\u2009μm × 20\\u2009mm nanoViper (ThermoScientific Dionex, San Jose, USA), trapped the peptides before separation on a C18 Acclaim PepMap100 3\\u2009μm, 75\\u2009μm × 150\\u2009mm nanoViper (ThermoScientific Dionex, San Jose, USA). Peptides were eluted with a gradient of acetonitrile. The analytical column outlet was directly interfaced via a modified nano-flow electrospray ionization source, with a hybrid dual pressure linear ion trap mass spectrometer (Orbitrap Velos, ThermoScientific, San Jose, USA).',\n",
       " 'Breast cancer tissue lysates, reduction, alkylation by methyl-methanethiosulfonate, trypsin digestion, iTRAQ 8-plex labeling, HILIC fractionation, LC-MS/MS on Orbitrap Velos.  Transcriptomics analysis, immunohistochemistry. Detailed description provided in the manuscript.']"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "corpus[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[['the',\n",
       "  'crude',\n",
       "  'membrane',\n",
       "  '5',\n",
       "  'p56-p70',\n",
       "  'glun1tap',\n",
       "  'tap',\n",
       "  'mouse',\n",
       "  'forebrain',\n",
       "  'be',\n",
       "  're',\n",
       "  'suspend',\n",
       "  '12.5',\n",
       "  'ml',\n",
       "  'buffer',\n",
       "  'h',\n",
       "  'and',\n",
       "  'extract',\n",
       "  '12.5',\n",
       "  'ml',\n",
       "  '2',\n",
       "  'deoxycholate',\n",
       "  '100',\n",
       "  'mm',\n",
       "  'nacl',\n",
       "  '50',\n",
       "  'mm',\n",
       "  'tris',\n",
       "  'cl',\n",
       "  'ph8',\n",
       "  '1',\n",
       "  'h',\n",
       "  '6',\n",
       "  '°',\n",
       "  'c',\n",
       "  'total',\n",
       "  'extract',\n",
       "  'be',\n",
       "  'centrifuge',\n",
       "  '120,000',\n",
       "  'g',\n",
       "  '40',\n",
       "  'min',\n",
       "  '8',\n",
       "  '°',\n",
       "  'c',\n",
       "  'condition',\n",
       "  'immuno',\n",
       "  'capture',\n",
       "  'wash',\n",
       "  'and',\n",
       "  'peptide',\n",
       "  'antigen',\n",
       "  'exchange',\n",
       "  'elution',\n",
       "  'be',\n",
       "  'screen',\n",
       "  'use',\n",
       "  'a',\n",
       "  'high',\n",
       "  'throughput',\n",
       "  'purification',\n",
       "  'robot',\n",
       "  'magic',\n",
       "  'sample',\n",
       "  'processor',\n",
       "  'invitrogen',\n",
       "  '25',\n",
       "  'glun1tap',\n",
       "  'tap',\n",
       "  'extract',\n",
       "  'supernatant',\n",
       "  '80',\n",
       "  'μg',\n",
       "  'mouse',\n",
       "  'flag',\n",
       "  'antibody',\n",
       "  'be',\n",
       "  'couple',\n",
       "  '30',\n",
       "  'mg',\n",
       "  '500',\n",
       "  'protein',\n",
       "  'g',\n",
       "  'magnetic',\n",
       "  'bead',\n",
       "  'invitrogen',\n",
       "  'receptor',\n",
       "  'be',\n",
       "  'capture',\n",
       "  'extract',\n",
       "  'supernatant',\n",
       "  '2',\n",
       "  'h.',\n",
       "  'the',\n",
       "  'bead',\n",
       "  'be',\n",
       "  'wash',\n",
       "  'three',\n",
       "  'time',\n",
       "  '5',\n",
       "  'ml',\n",
       "  'wash',\n",
       "  'buffer',\n",
       "  '0.37',\n",
       "  'v',\n",
       "  'sodium',\n",
       "  'deoxycholate',\n",
       "  '0.05',\n",
       "  'mg.ml−1',\n",
       "  'lipid',\n",
       "  '1:1:3',\n",
       "  'popc',\n",
       "  'pope',\n",
       "  'pog',\n",
       "  '150',\n",
       "  'mm',\n",
       "  'nacl',\n",
       "  '50',\n",
       "  'mm',\n",
       "  'tris',\n",
       "  'cl',\n",
       "  'ph8',\n",
       "  'flag',\n",
       "  'capture',\n",
       "  'complex',\n",
       "  'be',\n",
       "  'elute',\n",
       "  '600',\n",
       "  'wash',\n",
       "  'buffer',\n",
       "  'supplement',\n",
       "  '0.2',\n",
       "  'mg.ml−1',\n",
       "  'flag',\n",
       "  'peptide',\n",
       "  '2',\n",
       "  'h',\n",
       "  '6',\n",
       "  '°',\n",
       "  'c',\n",
       "  'eluate',\n",
       "  'be',\n",
       "  'buffer',\n",
       "  'exchange',\n",
       "  'remove',\n",
       "  'the',\n",
       "  'flag',\n",
       "  'elution',\n",
       "  'peptide',\n",
       "  'and',\n",
       "  'concentrate',\n",
       "  'a',\n",
       "  '100-kda',\n",
       "  'mwco',\n",
       "  'centrifugal',\n",
       "  'filter',\n",
       "  'amicon',\n",
       "  '35',\n",
       "  '0.1–0.25',\n",
       "  'mg.ml−1',\n",
       "  'native',\n",
       "  'nmdar',\n",
       "  'bnp',\n",
       "  'be',\n",
       "  'run',\n",
       "  'accord',\n",
       "  'schägger',\n",
       "  'excise',\n",
       "  'bnp',\n",
       "  'gel',\n",
       "  'slice',\n",
       "  'contain',\n",
       "  'the',\n",
       "  'purify',\n",
       "  'protein',\n",
       "  'be',\n",
       "  'prepare',\n",
       "  'mass',\n",
       "  'spectrometric',\n",
       "  'analysis',\n",
       "  'manual',\n",
       "  'situ',\n",
       "  'enzymatic',\n",
       "  'digestion',\n",
       "  'briefly',\n",
       "  'the',\n",
       "  'excise',\n",
       "  'protein',\n",
       "  'gel',\n",
       "  'piece',\n",
       "  'be',\n",
       "  'place',\n",
       "  'a',\n",
       "  'well',\n",
       "  'a',\n",
       "  '96-well',\n",
       "  'microtitre',\n",
       "  'plate',\n",
       "  'and',\n",
       "  'de',\n",
       "  'stain',\n",
       "  '50',\n",
       "  'v',\n",
       "  'v',\n",
       "  'acetonitrile',\n",
       "  'and',\n",
       "  '50',\n",
       "  'mm',\n",
       "  'ammonium',\n",
       "  'bicarbonate',\n",
       "  'reduce',\n",
       "  '10',\n",
       "  'mm',\n",
       "  'dtt',\n",
       "  'and',\n",
       "  'alkylat',\n",
       "  '55',\n",
       "  'mm',\n",
       "  'iodoacetamide',\n",
       "  'alkylation',\n",
       "  'protein',\n",
       "  'be',\n",
       "  'digest',\n",
       "  '6',\n",
       "  'ng',\n",
       "  'μl−1',\n",
       "  'trypsin',\n",
       "  'promega',\n",
       "  'uk',\n",
       "  'overnight',\n",
       "  '37',\n",
       "  '°',\n",
       "  'c',\n",
       "  'the',\n",
       "  'result',\n",
       "  'peptide',\n",
       "  'be',\n",
       "  'extract',\n",
       "  '2',\n",
       "  'v',\n",
       "  'v',\n",
       "  'formic',\n",
       "  'acid',\n",
       "  '2',\n",
       "  'v',\n",
       "  'v',\n",
       "  'acetonitrile',\n",
       "  'the',\n",
       "  'digest',\n",
       "  'be',\n",
       "  'analyse',\n",
       "  'nano',\n",
       "  'scale',\n",
       "  'capillary',\n",
       "  'lc',\n",
       "  'ms',\n",
       "  'ms',\n",
       "  'use',\n",
       "  'an',\n",
       "  'ultimate',\n",
       "  'u3000',\n",
       "  'hplc',\n",
       "  'thermoscientific',\n",
       "  'dionex',\n",
       "  'san',\n",
       "  'jose',\n",
       "  'usa',\n",
       "  'deliver',\n",
       "  'a',\n",
       "  'flow',\n",
       "  '~300',\n",
       "  'nl',\n",
       "  'min−1',\n",
       "  'a',\n",
       "  'c18',\n",
       "  'acclaim',\n",
       "  'pepmap100',\n",
       "  '5',\n",
       "  'μm',\n",
       "  '100',\n",
       "  'μm',\n",
       "  '×',\n",
       "  '20',\n",
       "  'mm',\n",
       "  'nanoviper',\n",
       "  'thermoscientific',\n",
       "  'dionex',\n",
       "  'san',\n",
       "  'jose',\n",
       "  'usa',\n",
       "  'trap',\n",
       "  'the',\n",
       "  'peptide',\n",
       "  'separation',\n",
       "  'a',\n",
       "  'c18',\n",
       "  'acclaim',\n",
       "  'pepmap100',\n",
       "  '3',\n",
       "  'μm',\n",
       "  '75',\n",
       "  'μm',\n",
       "  '×',\n",
       "  '150',\n",
       "  'mm',\n",
       "  'nanoviper',\n",
       "  'thermoscientific',\n",
       "  'dionex',\n",
       "  'san',\n",
       "  'jose',\n",
       "  'usa',\n",
       "  'peptide',\n",
       "  'be',\n",
       "  'elute',\n",
       "  'a',\n",
       "  'gradient',\n",
       "  'acetonitrile',\n",
       "  'the',\n",
       "  'analytical',\n",
       "  'column',\n",
       "  'outlet',\n",
       "  'be',\n",
       "  'directly',\n",
       "  'interfac',\n",
       "  'a',\n",
       "  'modify',\n",
       "  'nano',\n",
       "  'flow',\n",
       "  'electrospray',\n",
       "  'ionization',\n",
       "  'source',\n",
       "  'a',\n",
       "  'hybrid',\n",
       "  'dual',\n",
       "  'pressure',\n",
       "  'linear',\n",
       "  'ion',\n",
       "  'trap',\n",
       "  'mass',\n",
       "  'spectrometer',\n",
       "  'orbitrap',\n",
       "  'velos',\n",
       "  'thermoscientific',\n",
       "  'san',\n",
       "  'jose',\n",
       "  'usa'],\n",
       " ['breast',\n",
       "  'cancer',\n",
       "  'tissue',\n",
       "  'lysate',\n",
       "  'reduction',\n",
       "  'alkylation',\n",
       "  'methyl',\n",
       "  'methanethiosulfonate',\n",
       "  'trypsin',\n",
       "  'digestion',\n",
       "  'itraq',\n",
       "  '8-plex',\n",
       "  'labeling',\n",
       "  'hilic',\n",
       "  'fractionation',\n",
       "  'lc',\n",
       "  'ms',\n",
       "  'ms',\n",
       "  'orbitrap',\n",
       "  'velos',\n",
       "  ' ',\n",
       "  'transcriptomics',\n",
       "  'analysis',\n",
       "  'immunohistochemistry',\n",
       "  'detailed',\n",
       "  'description',\n",
       "  'provide',\n",
       "  'the',\n",
       "  'manuscript']]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "lemm_corpus[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "outfile1 = open('nlp09_data/nlp09_original_corpus.pickle', 'wb')\n",
    "pickle.dump(corpus, outfile1)\n",
    "outfile1.close()\n",
    "\n",
    "outfile2 = open('nlp09_data/nlp09_lemmatized_corpus.pickle', 'wb')\n",
    "pickle.dump(lemm_corpus, outfile2)\n",
    "outfile2.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create dictionary for the lemmatized_corpus\n",
    "dictionary = corpora.Dictionary(lemm_corpus)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "stop_words = stopwords.words('english')\n",
    "once_ids = [tokenid for tokenid, docfreq in iteritems(dictionary.dfs) if docfreq == 1]\n",
    "stop_ids = [dictionary.token2id[stopword] for stopword in stop_words\n",
    "                   if stopword in dictionary.token2id]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "dictionary.filter_tokens(stop_ids + once_ids)\n",
    "dictionary.compactify()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "dictionary.save('serialized_data/nlp09_lemmatized_corpus.dict')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create BoW\n",
    "bow_corpus = [dictionary.doc2bow(text) for text in lemm_corpus]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "corpora.MmCorpus.serialize('serialized_data/nlp09_bow_corpus.mm', bow_corpus)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dictionary(27077 unique tokens: ['0.05', '0.2', '1', '10', '100']...)\n"
     ]
    }
   ],
   "source": [
    "print(dictionary)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
