{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Use the retrieving function that queries descriptive texts from the PX database (on postgres) as well as their categorical attributes. Then just take the descriptive text part, combine them, split them into sentences and store in a list. Then save the list as a pickle object."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "import spacy\n",
    "import matplotlib.pyplot as plt\n",
    "import numpy as np\n",
    "import seaborn as sns\n",
    "import scipy\n",
    "from spacy.vectors import Vectors\n",
    "from spacy.strings import StringStore\n",
    "from scipy.cluster.hierarchy import dendrogram, linkage\n",
    "from wordcloud import WordCloud\n",
    "from nltk.corpus import stopwords\n",
    "\n",
    "from gensim.models.doc2vec import Doc2Vec, TaggedDocument\n",
    "from nltk.tokenize import word_tokenize\n",
    "\n",
    "from retrieve_data import get_descriptive_fields\n",
    "\n",
    "%matplotlib inline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/ryohayama/Python/b13logy/nlp'"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "os.getcwd()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Retrive with sql query\n",
    "df = get_descriptive_fields(num_rows=6000, db_file='../database_files/b13logy.ini')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>dataset_id</th>\n",
       "      <th>sample_protocol</th>\n",
       "      <th>data_protocol</th>\n",
       "      <th>description</th>\n",
       "      <th>instruments</th>\n",
       "      <th>exp_types</th>\n",
       "      <th>quant_methods</th>\n",
       "      <th>labhead_fullname</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PXD000001</td>\n",
       "      <td>Not available</td>\n",
       "      <td>Two extra files have been added post-publicati...</td>\n",
       "      <td>Expected reporter ion ratios: Erwinia peptides...</td>\n",
       "      <td>LTQ Orbitrap Velos, instrument model</td>\n",
       "      <td>Bottom-up proteomics</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PXD000002</td>\n",
       "      <td>Not available</td>\n",
       "      <td>Not available</td>\n",
       "      <td>Human saliva samples from a 36 year old male w...</td>\n",
       "      <td>instrument model, LTQ Velos</td>\n",
       "      <td>Bottom-up proteomics</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PXD000003</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PXD000004</td>\n",
       "      <td>Not available</td>\n",
       "      <td>Not available</td>\n",
       "      <td>Prefrontal cortex tissue slices from a cogniti...</td>\n",
       "      <td>instrument model, Q Exactive</td>\n",
       "      <td>Bottom-up proteomics</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PXD000005</td>\n",
       "      <td>Not available</td>\n",
       "      <td>Not available</td>\n",
       "      <td>Proteomics on HCT116 cells. 3 samples wild-typ...</td>\n",
       "      <td>LTQ Orbitrap Velos</td>\n",
       "      <td>Bottom-up proteomics</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>PXD000006</td>\n",
       "      <td>Not available</td>\n",
       "      <td>Not available</td>\n",
       "      <td>15-deoxy-delta12,14-prostaglandin J2 (15d-PGJ2...</td>\n",
       "      <td>LTQ Orbitrap, instrument model</td>\n",
       "      <td>Bottom-up proteomics</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>PXD000008</td>\n",
       "      <td>Not available</td>\n",
       "      <td>Not available</td>\n",
       "      <td>2.3  Cell wall and secretome preparation.  C. ...</td>\n",
       "      <td>instrument model, Q-Tof ultima</td>\n",
       "      <td>Bottom-up proteomics</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>PXD000009</td>\n",
       "      <td>Not available</td>\n",
       "      <td>Not available</td>\n",
       "      <td>Top down identification of proteins detected b...</td>\n",
       "      <td>LTQ Orbitrap</td>\n",
       "      <td>Bottom-up proteomics</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>PXD000011</td>\n",
       "      <td>The crude membranes from 5 P56-P70 Glun1TAP/TA...</td>\n",
       "      <td>Data-dependent analysis was carried out, using...</td>\n",
       "      <td>TAP-GluN1 (840 kDa and 1.5 MDa), PSD95-TAP (1....</td>\n",
       "      <td>LTQ Orbitrap, instrument model</td>\n",
       "      <td>Bottom-up proteomics</td>\n",
       "      <td>None</td>\n",
       "      <td>Seth Grant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>PXD000012</td>\n",
       "      <td>Not available</td>\n",
       "      <td>Not available</td>\n",
       "      <td>Enlarged white adipose tissue (WAT) is a featu...</td>\n",
       "      <td>instrument model</td>\n",
       "      <td>Bottom-up proteomics</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>PXD000013</td>\n",
       "      <td>Not available</td>\n",
       "      <td>Not available</td>\n",
       "      <td>Study on the MS1 interference problem for repo...</td>\n",
       "      <td>LTQ Orbitrap, instrument model</td>\n",
       "      <td>Bottom-up proteomics</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>PXD000014</td>\n",
       "      <td>Not available</td>\n",
       "      <td>Not available</td>\n",
       "      <td>Experiment was performed on 293 cells, details...</td>\n",
       "      <td>LTQ Orbitrap</td>\n",
       "      <td>Bottom-up proteomics</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>PXD000015</td>\n",
       "      <td>Not available</td>\n",
       "      <td>Not available</td>\n",
       "      <td>Physiological changes in trunk wood of Vitis v...</td>\n",
       "      <td>instrument model, Bruker Daltonics amaZon series</td>\n",
       "      <td>Bottom-up proteomics</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>PXD000016</td>\n",
       "      <td>Not available</td>\n",
       "      <td>Not available</td>\n",
       "      <td>Rhodococcus jostii RHA1 is a catabolically ver...</td>\n",
       "      <td>LTQ Orbitrap, instrument model</td>\n",
       "      <td>Bottom-up proteomics</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>PXD000017</td>\n",
       "      <td>Not available</td>\n",
       "      <td>Not available</td>\n",
       "      <td>Studies of neuronal, endocrine and metabolic d...</td>\n",
       "      <td>Q-Tof Premier, microchannel plate detector, el...</td>\n",
       "      <td>Bottom-up proteomics</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>PXD000018</td>\n",
       "      <td>Not available</td>\n",
       "      <td>Not available</td>\n",
       "      <td>Integrin adhesion complexes were isolated from...</td>\n",
       "      <td>instrument model, LTQ Velos</td>\n",
       "      <td>Bottom-up proteomics</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>PXD000021</td>\n",
       "      <td>Not available</td>\n",
       "      <td>Not available</td>\n",
       "      <td>Processing of the dataset of synthetic phospho...</td>\n",
       "      <td>LTQ Orbitrap, instrument model</td>\n",
       "      <td>Bottom-up proteomics</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>PXD000022</td>\n",
       "      <td>Not available</td>\n",
       "      <td>Not available</td>\n",
       "      <td>During the last decade, skeletal muscle-secret...</td>\n",
       "      <td>LTQ Orbitrap</td>\n",
       "      <td>Bottom-up proteomics</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>PXD000023</td>\n",
       "      <td>Not available</td>\n",
       "      <td>Not available</td>\n",
       "      <td>Comparative analysis of EHEC O157:H7 Sakai vs....</td>\n",
       "      <td>LTQ Orbitrap, instrument model</td>\n",
       "      <td>Bottom-up proteomics</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>PXD000025</td>\n",
       "      <td>Not available</td>\n",
       "      <td>Not available</td>\n",
       "      <td>This study is to compare the phosphoproteome o...</td>\n",
       "      <td>LTQ Orbitrap, instrument model, TripleTOF 5600</td>\n",
       "      <td>Bottom-up proteomics</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>PXD000026</td>\n",
       "      <td>Not available</td>\n",
       "      <td>Not available</td>\n",
       "      <td>In an attempt to identify proteins interacting...</td>\n",
       "      <td>instrument model, Q-Tof micro</td>\n",
       "      <td>Bottom-up proteomics</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>PXD000027</td>\n",
       "      <td>Not available</td>\n",
       "      <td>Not available</td>\n",
       "      <td>Comparative analysis of salt-responsive phosph...</td>\n",
       "      <td>instrument model</td>\n",
       "      <td>Bottom-up proteomics</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>PXD000028</td>\n",
       "      <td>Not available</td>\n",
       "      <td>Not available</td>\n",
       "      <td>Rats were fed either a control diet or high fa...</td>\n",
       "      <td>LTQ</td>\n",
       "      <td>Bottom-up proteomics</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>PXD000029</td>\n",
       "      <td>Breast cancer tissue lysates, reduction, alkyl...</td>\n",
       "      <td>Proteomics data analysis: Proteome Discoverer ...</td>\n",
       "      <td>Current prognostic factors are insufficient fo...</td>\n",
       "      <td>LTQ Orbitrap Velos</td>\n",
       "      <td>Shotgun proteomics</td>\n",
       "      <td>iTRAQ</td>\n",
       "      <td>Pavel Bouchal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>PXD000030</td>\n",
       "      <td>Not available</td>\n",
       "      <td>Not available</td>\n",
       "      <td>In the present study, we used a functional pro...</td>\n",
       "      <td>instrument model</td>\n",
       "      <td>Bottom-up proteomics</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>PXD000031</td>\n",
       "      <td>Not available</td>\n",
       "      <td>Not available</td>\n",
       "      <td>Proteomics on affinity purified and mock purif...</td>\n",
       "      <td>LTQ Orbitrap Velos</td>\n",
       "      <td>Bottom-up proteomics</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>PXD000033</td>\n",
       "      <td>Not available</td>\n",
       "      <td>Not available</td>\n",
       "      <td>Pilotexp_IMAC_TiO2_eluate_ATM_ATR_dep_phos_15m...</td>\n",
       "      <td>LTQ Orbitrap Velos</td>\n",
       "      <td>Bottom-up proteomics</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>PXD000035</td>\n",
       "      <td>Not available</td>\n",
       "      <td>Not available</td>\n",
       "      <td>Listeria monocytogenes cells (strain LI0521) w...</td>\n",
       "      <td>MALDI Synapt HDMS, instrument model</td>\n",
       "      <td>Bottom-up proteomics</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>PXD000036</td>\n",
       "      <td>Not available</td>\n",
       "      <td>Not available</td>\n",
       "      <td>Albuminome from human serum was isolated using...</td>\n",
       "      <td>LTQ Orbitrap Elite</td>\n",
       "      <td>Bottom-up proteomics</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>PXD000038</td>\n",
       "      <td>Not available</td>\n",
       "      <td>Not available</td>\n",
       "      <td>Two female pigs weighing 40-55 kg were used. E...</td>\n",
       "      <td>LCQ Deca XP Plus</td>\n",
       "      <td>Bottom-up proteomics</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5415</th>\n",
       "      <td>PXD009730</td>\n",
       "      <td>OMVs, whole cell lysates and whole culture lys...</td>\n",
       "      <td>Raw MS files were analysed by MaxQuant (Ver 1....</td>\n",
       "      <td>Porphyromonas gingivalis is an anaerobic, Gram...</td>\n",
       "      <td>Q Exactive</td>\n",
       "      <td>Shotgun proteomics</td>\n",
       "      <td>None</td>\n",
       "      <td>Eric Charles Reynolds</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5416</th>\n",
       "      <td>PXD009742</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5417</th>\n",
       "      <td>PXD009747</td>\n",
       "      <td>HeLa cells were grown in DMEM (Gibco) suppleme...</td>\n",
       "      <td>The resulting MS raw files were submitted for ...</td>\n",
       "      <td>The orientation of the mitotic spindle (MS) is...</td>\n",
       "      <td>Q Exactive</td>\n",
       "      <td>Gel-based experiment</td>\n",
       "      <td>None</td>\n",
       "      <td>Tuula Nyman</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5418</th>\n",
       "      <td>PXD009778</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5419</th>\n",
       "      <td>PXD009801</td>\n",
       "      <td>Dynamic SILAC Labeling was done with three iso...</td>\n",
       "      <td>Data Analysis. Peptides were identified, and t...</td>\n",
       "      <td>MHC class I peptides are products of endogenou...</td>\n",
       "      <td>LTQ Orbitrap, Q Exactive</td>\n",
       "      <td>Affinity purification coupled with mass spectr...</td>\n",
       "      <td>SILAC</td>\n",
       "      <td>Arie Admon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5420</th>\n",
       "      <td>PXD009803</td>\n",
       "      <td>HEK293 cells were lysed in protein lysis buffe...</td>\n",
       "      <td>All mass spectrometric raw data were processed...</td>\n",
       "      <td>Pluripotent stem cells undergo unlimited self-...</td>\n",
       "      <td>LTQ Orbitrap</td>\n",
       "      <td>Shotgun proteomics</td>\n",
       "      <td>None</td>\n",
       "      <td>David Vilchez</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5421</th>\n",
       "      <td>PXD009816</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5422</th>\n",
       "      <td>PXD009830</td>\n",
       "      <td>HEK293T cells were cultured in SILAC media con...</td>\n",
       "      <td>Peptide fractions were analyzed on a quadrupol...</td>\n",
       "      <td>BRCA1 deficiencies cause breast, ovarian and o...</td>\n",
       "      <td>Q Exactive</td>\n",
       "      <td>Affinity purification coupled with mass spectr...</td>\n",
       "      <td>SILAC</td>\n",
       "      <td>Petra Beli</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5423</th>\n",
       "      <td>PXD009841</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5424</th>\n",
       "      <td>PXD009842</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5425</th>\n",
       "      <td>PXD009863</td>\n",
       "      <td>Cells were processed as described in Serandour...</td>\n",
       "      <td>Raw MS/MS data in Thermo Fisher .RAW format an...</td>\n",
       "      <td>Identification of TRPS1 interactors in MCF7, T...</td>\n",
       "      <td>LTQ Orbitrap</td>\n",
       "      <td>Affinity purification coupled with mass spectr...</td>\n",
       "      <td>Peptide counting</td>\n",
       "      <td>Jason Carroll</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5426</th>\n",
       "      <td>PXD009872</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5427</th>\n",
       "      <td>PXD009889</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5428</th>\n",
       "      <td>PXD009947</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5429</th>\n",
       "      <td>PXD009972</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5430</th>\n",
       "      <td>PXD009985</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5431</th>\n",
       "      <td>PXD009987</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5432</th>\n",
       "      <td>PXD010000</td>\n",
       "      <td>Samples were digested with trypsin then analyz...</td>\n",
       "      <td>Data was searched with MSGF+ using PNNL's DMS ...</td>\n",
       "      <td>A benchmark set of bottom-up proteomics data f...</td>\n",
       "      <td>Q Exactive</td>\n",
       "      <td>Shotgun proteomics</td>\n",
       "      <td>None</td>\n",
       "      <td>Samuel Payne</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5433</th>\n",
       "      <td>PXD010011</td>\n",
       "      <td>Worms that lack the natural smo-1 gene (encodi...</td>\n",
       "      <td>The tryptic peptides were analyzed by LC-MS/MS...</td>\n",
       "      <td>In this project we asked how reducing the acti...</td>\n",
       "      <td>LTQ Orbitrap XL</td>\n",
       "      <td>Gel-based experiment</td>\n",
       "      <td>Label free</td>\n",
       "      <td>Ehud Cohen</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5434</th>\n",
       "      <td>PXD010043</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5435</th>\n",
       "      <td>PXD010049</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5436</th>\n",
       "      <td>PXD010050</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5437</th>\n",
       "      <td>PXD010051</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5438</th>\n",
       "      <td>PXD010079</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5439</th>\n",
       "      <td>RPXD000650</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5440</th>\n",
       "      <td>RPXD000665</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5441</th>\n",
       "      <td>RPXD000667</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5442</th>\n",
       "      <td>RPXD000668</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5443</th>\n",
       "      <td>RPXD000669</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5444</th>\n",
       "      <td>RPXD006668</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5445 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      dataset_id                                    sample_protocol  \\\n",
       "0      PXD000001                                      Not available   \n",
       "1      PXD000002                                      Not available   \n",
       "2      PXD000003                                               None   \n",
       "3      PXD000004                                      Not available   \n",
       "4      PXD000005                                      Not available   \n",
       "5      PXD000006                                      Not available   \n",
       "6      PXD000008                                      Not available   \n",
       "7      PXD000009                                      Not available   \n",
       "8      PXD000011  The crude membranes from 5 P56-P70 Glun1TAP/TA...   \n",
       "9      PXD000012                                      Not available   \n",
       "10     PXD000013                                      Not available   \n",
       "11     PXD000014                                      Not available   \n",
       "12     PXD000015                                      Not available   \n",
       "13     PXD000016                                      Not available   \n",
       "14     PXD000017                                      Not available   \n",
       "15     PXD000018                                      Not available   \n",
       "16     PXD000021                                      Not available   \n",
       "17     PXD000022                                      Not available   \n",
       "18     PXD000023                                      Not available   \n",
       "19     PXD000025                                      Not available   \n",
       "20     PXD000026                                      Not available   \n",
       "21     PXD000027                                      Not available   \n",
       "22     PXD000028                                      Not available   \n",
       "23     PXD000029  Breast cancer tissue lysates, reduction, alkyl...   \n",
       "24     PXD000030                                      Not available   \n",
       "25     PXD000031                                      Not available   \n",
       "26     PXD000033                                      Not available   \n",
       "27     PXD000035                                      Not available   \n",
       "28     PXD000036                                      Not available   \n",
       "29     PXD000038                                      Not available   \n",
       "...          ...                                                ...   \n",
       "5415   PXD009730  OMVs, whole cell lysates and whole culture lys...   \n",
       "5416   PXD009742                                               None   \n",
       "5417   PXD009747  HeLa cells were grown in DMEM (Gibco) suppleme...   \n",
       "5418   PXD009778                                               None   \n",
       "5419   PXD009801  Dynamic SILAC Labeling was done with three iso...   \n",
       "5420   PXD009803  HEK293 cells were lysed in protein lysis buffe...   \n",
       "5421   PXD009816                                               None   \n",
       "5422   PXD009830  HEK293T cells were cultured in SILAC media con...   \n",
       "5423   PXD009841                                               None   \n",
       "5424   PXD009842                                               None   \n",
       "5425   PXD009863  Cells were processed as described in Serandour...   \n",
       "5426   PXD009872                                               None   \n",
       "5427   PXD009889                                               None   \n",
       "5428   PXD009947                                               None   \n",
       "5429   PXD009972                                               None   \n",
       "5430   PXD009985                                               None   \n",
       "5431   PXD009987                                               None   \n",
       "5432   PXD010000  Samples were digested with trypsin then analyz...   \n",
       "5433   PXD010011  Worms that lack the natural smo-1 gene (encodi...   \n",
       "5434   PXD010043                                               None   \n",
       "5435   PXD010049                                               None   \n",
       "5436   PXD010050                                               None   \n",
       "5437   PXD010051                                               None   \n",
       "5438   PXD010079                                               None   \n",
       "5439  RPXD000650                                               None   \n",
       "5440  RPXD000665                                               None   \n",
       "5441  RPXD000667                                               None   \n",
       "5442  RPXD000668                                               None   \n",
       "5443  RPXD000669                                               None   \n",
       "5444  RPXD006668                                               None   \n",
       "\n",
       "                                          data_protocol  \\\n",
       "0     Two extra files have been added post-publicati...   \n",
       "1                                         Not available   \n",
       "2                                                  None   \n",
       "3                                         Not available   \n",
       "4                                         Not available   \n",
       "5                                         Not available   \n",
       "6                                         Not available   \n",
       "7                                         Not available   \n",
       "8     Data-dependent analysis was carried out, using...   \n",
       "9                                         Not available   \n",
       "10                                        Not available   \n",
       "11                                        Not available   \n",
       "12                                        Not available   \n",
       "13                                        Not available   \n",
       "14                                        Not available   \n",
       "15                                        Not available   \n",
       "16                                        Not available   \n",
       "17                                        Not available   \n",
       "18                                        Not available   \n",
       "19                                        Not available   \n",
       "20                                        Not available   \n",
       "21                                        Not available   \n",
       "22                                        Not available   \n",
       "23    Proteomics data analysis: Proteome Discoverer ...   \n",
       "24                                        Not available   \n",
       "25                                        Not available   \n",
       "26                                        Not available   \n",
       "27                                        Not available   \n",
       "28                                        Not available   \n",
       "29                                        Not available   \n",
       "...                                                 ...   \n",
       "5415  Raw MS files were analysed by MaxQuant (Ver 1....   \n",
       "5416                                               None   \n",
       "5417  The resulting MS raw files were submitted for ...   \n",
       "5418                                               None   \n",
       "5419  Data Analysis. Peptides were identified, and t...   \n",
       "5420  All mass spectrometric raw data were processed...   \n",
       "5421                                               None   \n",
       "5422  Peptide fractions were analyzed on a quadrupol...   \n",
       "5423                                               None   \n",
       "5424                                               None   \n",
       "5425  Raw MS/MS data in Thermo Fisher .RAW format an...   \n",
       "5426                                               None   \n",
       "5427                                               None   \n",
       "5428                                               None   \n",
       "5429                                               None   \n",
       "5430                                               None   \n",
       "5431                                               None   \n",
       "5432  Data was searched with MSGF+ using PNNL's DMS ...   \n",
       "5433  The tryptic peptides were analyzed by LC-MS/MS...   \n",
       "5434                                               None   \n",
       "5435                                               None   \n",
       "5436                                               None   \n",
       "5437                                               None   \n",
       "5438                                               None   \n",
       "5439                                               None   \n",
       "5440                                               None   \n",
       "5441                                               None   \n",
       "5442                                               None   \n",
       "5443                                               None   \n",
       "5444                                               None   \n",
       "\n",
       "                                            description  \\\n",
       "0     Expected reporter ion ratios: Erwinia peptides...   \n",
       "1     Human saliva samples from a 36 year old male w...   \n",
       "2                                                  None   \n",
       "3     Prefrontal cortex tissue slices from a cogniti...   \n",
       "4     Proteomics on HCT116 cells. 3 samples wild-typ...   \n",
       "5     15-deoxy-delta12,14-prostaglandin J2 (15d-PGJ2...   \n",
       "6     2.3  Cell wall and secretome preparation.  C. ...   \n",
       "7     Top down identification of proteins detected b...   \n",
       "8     TAP-GluN1 (840 kDa and 1.5 MDa), PSD95-TAP (1....   \n",
       "9     Enlarged white adipose tissue (WAT) is a featu...   \n",
       "10    Study on the MS1 interference problem for repo...   \n",
       "11    Experiment was performed on 293 cells, details...   \n",
       "12    Physiological changes in trunk wood of Vitis v...   \n",
       "13    Rhodococcus jostii RHA1 is a catabolically ver...   \n",
       "14    Studies of neuronal, endocrine and metabolic d...   \n",
       "15    Integrin adhesion complexes were isolated from...   \n",
       "16    Processing of the dataset of synthetic phospho...   \n",
       "17    During the last decade, skeletal muscle-secret...   \n",
       "18    Comparative analysis of EHEC O157:H7 Sakai vs....   \n",
       "19    This study is to compare the phosphoproteome o...   \n",
       "20    In an attempt to identify proteins interacting...   \n",
       "21    Comparative analysis of salt-responsive phosph...   \n",
       "22    Rats were fed either a control diet or high fa...   \n",
       "23    Current prognostic factors are insufficient fo...   \n",
       "24    In the present study, we used a functional pro...   \n",
       "25    Proteomics on affinity purified and mock purif...   \n",
       "26    Pilotexp_IMAC_TiO2_eluate_ATM_ATR_dep_phos_15m...   \n",
       "27    Listeria monocytogenes cells (strain LI0521) w...   \n",
       "28    Albuminome from human serum was isolated using...   \n",
       "29    Two female pigs weighing 40-55 kg were used. E...   \n",
       "...                                                 ...   \n",
       "5415  Porphyromonas gingivalis is an anaerobic, Gram...   \n",
       "5416                                               None   \n",
       "5417  The orientation of the mitotic spindle (MS) is...   \n",
       "5418                                               None   \n",
       "5419  MHC class I peptides are products of endogenou...   \n",
       "5420  Pluripotent stem cells undergo unlimited self-...   \n",
       "5421                                               None   \n",
       "5422  BRCA1 deficiencies cause breast, ovarian and o...   \n",
       "5423                                               None   \n",
       "5424                                               None   \n",
       "5425  Identification of TRPS1 interactors in MCF7, T...   \n",
       "5426                                               None   \n",
       "5427                                               None   \n",
       "5428                                               None   \n",
       "5429                                               None   \n",
       "5430                                               None   \n",
       "5431                                               None   \n",
       "5432  A benchmark set of bottom-up proteomics data f...   \n",
       "5433  In this project we asked how reducing the acti...   \n",
       "5434                                               None   \n",
       "5435                                               None   \n",
       "5436                                               None   \n",
       "5437                                               None   \n",
       "5438                                               None   \n",
       "5439                                               None   \n",
       "5440                                               None   \n",
       "5441                                               None   \n",
       "5442                                               None   \n",
       "5443                                               None   \n",
       "5444                                               None   \n",
       "\n",
       "                                            instruments  \\\n",
       "0                  LTQ Orbitrap Velos, instrument model   \n",
       "1                           instrument model, LTQ Velos   \n",
       "2                                                  None   \n",
       "3                          instrument model, Q Exactive   \n",
       "4                                    LTQ Orbitrap Velos   \n",
       "5                        LTQ Orbitrap, instrument model   \n",
       "6                        instrument model, Q-Tof ultima   \n",
       "7                                          LTQ Orbitrap   \n",
       "8                        LTQ Orbitrap, instrument model   \n",
       "9                                      instrument model   \n",
       "10                       LTQ Orbitrap, instrument model   \n",
       "11                                         LTQ Orbitrap   \n",
       "12     instrument model, Bruker Daltonics amaZon series   \n",
       "13                       LTQ Orbitrap, instrument model   \n",
       "14    Q-Tof Premier, microchannel plate detector, el...   \n",
       "15                          instrument model, LTQ Velos   \n",
       "16                       LTQ Orbitrap, instrument model   \n",
       "17                                         LTQ Orbitrap   \n",
       "18                       LTQ Orbitrap, instrument model   \n",
       "19       LTQ Orbitrap, instrument model, TripleTOF 5600   \n",
       "20                        instrument model, Q-Tof micro   \n",
       "21                                     instrument model   \n",
       "22                                                  LTQ   \n",
       "23                                   LTQ Orbitrap Velos   \n",
       "24                                     instrument model   \n",
       "25                                   LTQ Orbitrap Velos   \n",
       "26                                   LTQ Orbitrap Velos   \n",
       "27                  MALDI Synapt HDMS, instrument model   \n",
       "28                                   LTQ Orbitrap Elite   \n",
       "29                                     LCQ Deca XP Plus   \n",
       "...                                                 ...   \n",
       "5415                                         Q Exactive   \n",
       "5416                                               None   \n",
       "5417                                         Q Exactive   \n",
       "5418                                               None   \n",
       "5419                           LTQ Orbitrap, Q Exactive   \n",
       "5420                                       LTQ Orbitrap   \n",
       "5421                                               None   \n",
       "5422                                         Q Exactive   \n",
       "5423                                               None   \n",
       "5424                                               None   \n",
       "5425                                       LTQ Orbitrap   \n",
       "5426                                               None   \n",
       "5427                                               None   \n",
       "5428                                               None   \n",
       "5429                                               None   \n",
       "5430                                               None   \n",
       "5431                                               None   \n",
       "5432                                         Q Exactive   \n",
       "5433                                    LTQ Orbitrap XL   \n",
       "5434                                               None   \n",
       "5435                                               None   \n",
       "5436                                               None   \n",
       "5437                                               None   \n",
       "5438                                               None   \n",
       "5439                                               None   \n",
       "5440                                               None   \n",
       "5441                                               None   \n",
       "5442                                               None   \n",
       "5443                                               None   \n",
       "5444                                               None   \n",
       "\n",
       "                                              exp_types     quant_methods  \\\n",
       "0                                  Bottom-up proteomics              None   \n",
       "1                                  Bottom-up proteomics              None   \n",
       "2                                                  None              None   \n",
       "3                                  Bottom-up proteomics              None   \n",
       "4                                  Bottom-up proteomics              None   \n",
       "5                                  Bottom-up proteomics              None   \n",
       "6                                  Bottom-up proteomics              None   \n",
       "7                                  Bottom-up proteomics              None   \n",
       "8                                  Bottom-up proteomics              None   \n",
       "9                                  Bottom-up proteomics              None   \n",
       "10                                 Bottom-up proteomics              None   \n",
       "11                                 Bottom-up proteomics              None   \n",
       "12                                 Bottom-up proteomics              None   \n",
       "13                                 Bottom-up proteomics              None   \n",
       "14                                 Bottom-up proteomics              None   \n",
       "15                                 Bottom-up proteomics              None   \n",
       "16                                 Bottom-up proteomics              None   \n",
       "17                                 Bottom-up proteomics              None   \n",
       "18                                 Bottom-up proteomics              None   \n",
       "19                                 Bottom-up proteomics              None   \n",
       "20                                 Bottom-up proteomics              None   \n",
       "21                                 Bottom-up proteomics              None   \n",
       "22                                 Bottom-up proteomics              None   \n",
       "23                                   Shotgun proteomics             iTRAQ   \n",
       "24                                 Bottom-up proteomics              None   \n",
       "25                                 Bottom-up proteomics              None   \n",
       "26                                 Bottom-up proteomics              None   \n",
       "27                                 Bottom-up proteomics              None   \n",
       "28                                 Bottom-up proteomics              None   \n",
       "29                                 Bottom-up proteomics              None   \n",
       "...                                                 ...               ...   \n",
       "5415                                 Shotgun proteomics              None   \n",
       "5416                                               None              None   \n",
       "5417                               Gel-based experiment              None   \n",
       "5418                                               None              None   \n",
       "5419  Affinity purification coupled with mass spectr...             SILAC   \n",
       "5420                                 Shotgun proteomics              None   \n",
       "5421                                               None              None   \n",
       "5422  Affinity purification coupled with mass spectr...             SILAC   \n",
       "5423                                               None              None   \n",
       "5424                                               None              None   \n",
       "5425  Affinity purification coupled with mass spectr...  Peptide counting   \n",
       "5426                                               None              None   \n",
       "5427                                               None              None   \n",
       "5428                                               None              None   \n",
       "5429                                               None              None   \n",
       "5430                                               None              None   \n",
       "5431                                               None              None   \n",
       "5432                                 Shotgun proteomics              None   \n",
       "5433                               Gel-based experiment        Label free   \n",
       "5434                                               None              None   \n",
       "5435                                               None              None   \n",
       "5436                                               None              None   \n",
       "5437                                               None              None   \n",
       "5438                                               None              None   \n",
       "5439                                               None              None   \n",
       "5440                                               None              None   \n",
       "5441                                               None              None   \n",
       "5442                                               None              None   \n",
       "5443                                               None              None   \n",
       "5444                                               None              None   \n",
       "\n",
       "           labhead_fullname  \n",
       "0                      None  \n",
       "1                      None  \n",
       "2                      None  \n",
       "3                      None  \n",
       "4                      None  \n",
       "5                      None  \n",
       "6                      None  \n",
       "7                      None  \n",
       "8                Seth Grant  \n",
       "9                      None  \n",
       "10                     None  \n",
       "11                     None  \n",
       "12                     None  \n",
       "13                     None  \n",
       "14                     None  \n",
       "15                     None  \n",
       "16                     None  \n",
       "17                     None  \n",
       "18                     None  \n",
       "19                     None  \n",
       "20                     None  \n",
       "21                     None  \n",
       "22                     None  \n",
       "23            Pavel Bouchal  \n",
       "24                     None  \n",
       "25                     None  \n",
       "26                     None  \n",
       "27                     None  \n",
       "28                     None  \n",
       "29                     None  \n",
       "...                     ...  \n",
       "5415  Eric Charles Reynolds  \n",
       "5416                   None  \n",
       "5417            Tuula Nyman  \n",
       "5418                   None  \n",
       "5419             Arie Admon  \n",
       "5420          David Vilchez  \n",
       "5421                   None  \n",
       "5422             Petra Beli  \n",
       "5423                   None  \n",
       "5424                   None  \n",
       "5425          Jason Carroll  \n",
       "5426                   None  \n",
       "5427                   None  \n",
       "5428                   None  \n",
       "5429                   None  \n",
       "5430                   None  \n",
       "5431                   None  \n",
       "5432           Samuel Payne  \n",
       "5433             Ehud Cohen  \n",
       "5434                   None  \n",
       "5435                   None  \n",
       "5436                   None  \n",
       "5437                   None  \n",
       "5438                   None  \n",
       "5439                   None  \n",
       "5440                   None  \n",
       "5441                   None  \n",
       "5442                   None  \n",
       "5443                   None  \n",
       "5444                   None  \n",
       "\n",
       "[5445 rows x 8 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get descriptive text\n",
    "sample_proto = df['sample_protocol'].dropna()\n",
    "sample_proto = sample_proto[sample_proto != 'Not available']\n",
    "\n",
    "data_proto = df['data_protocol'].dropna()\n",
    "data_proto = data_proto[data_proto != 'Not available']\n",
    "\n",
    "descrip = df['description'].dropna()\n",
    "\n",
    "descriptive_text = list(sample_proto) + list(data_proto) + list(descrip)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['The crude membranes from 5 P56-P70 Glun1TAP/TAP mouse forebrains were re-suspended in 12.5\\u2009ml buffer H and extracted with 12.5\\u2009ml 2% deoxycholate, 100\\u2009mM NaCl, 50\\u2009mM Tris.Cl pH8 for 1\\u2009h at 6\\u2009°C. Total extract was centrifuged at 120,000g. for 40\\u2009min at 8\\u2009°C. Conditions for immuno-capture, wash and peptide-antigen exchange elution were screened using a high-throughput purification robot (MAGic sample processor, Invitrogen). For 25\\u2009ml Glun1TAP/TAP extract supernatant, 80\\u2009μg mouse Flag antibody was coupled to 30\\u2009mg (500\\u2009μl) protein G magnetic beads (Invitrogen). Receptor was captured from extract supernatant for 2\\u2009h. The beads were washed three times with 5\\u2009ml wash buffer (0.37% w/v sodium deoxycholate, 0.05\\u2009mg.ml−1 lipids (1:1:3 POPC:POPE:POG), 150\\u2009mM NaCl, 50\\u2009mM Tris.Cl pH8). Flag captured complexes were eluted with 600\\u2009μl wash buffer supplemented with 0.2\\u2009mg.ml−1 Flag peptide for 2\\u2009h at 6\\u2009°C. Eluate was buffer exchanged to remove the Flag elution peptide and concentrated with a 100-kDa MWCO centrifugal filter (Amicon) to 35\\u2009μl 0.1–0.25\\u2009mg.ml−1 native NMDAR. BNP was run according to Schägger. Excised BNP gel slices containing the purified proteins were prepared for mass spectrometric analysis by manual in situ enzymatic digestion. Briefly, the excised protein gel pieces were placed in a well of a 96-well microtitre plate and de-stained with 50% v/v acetonitrile and 50\\u2009mM ammonium bicarbonate, reduced with 10\\u2009mM DTT and alkylated with 55\\u2009mM iodoacetamide. After alkylation, proteins were digested with 6\\u2009ng\\u2009μl−1 Trypsin (Promega, UK) overnight at 37\\u2009°C. The resulting peptides were extracted in 2% v/v formic acid, 2% v/v acetonitrile. The digest was analysed by nano-scale capillary LC-MS/MS using an Ultimate U3000 HPLC (ThermoScientific Dionex, San Jose, USA) to deliver a flow of ~300\\u2009nl\\u2009min−1. A C18 Acclaim PepMap100 5\\u2009μm, 100\\u2009μm × 20\\u2009mm nanoViper (ThermoScientific Dionex, San Jose, USA), trapped the peptides before separation on a C18 Acclaim PepMap100 3\\u2009μm, 75\\u2009μm × 150\\u2009mm nanoViper (ThermoScientific Dionex, San Jose, USA). Peptides were eluted with a gradient of acetonitrile. The analytical column outlet was directly interfaced via a modified nano-flow electrospray ionization source, with a hybrid dual pressure linear ion trap mass spectrometer (Orbitrap Velos, ThermoScientific, San Jose, USA).',\n",
       " 'Breast cancer tissue lysates, reduction, alkylation by methyl-methanethiosulfonate, trypsin digestion, iTRAQ 8-plex labeling, HILIC fractionation, LC-MS/MS on Orbitrap Velos.  Transcriptomics analysis, immunohistochemistry. Detailed description provided in the manuscript.',\n",
       " 'In-gel digests were performed as described in standard protocols. Briefly, following the SDS-PAGE and washing of the excised gel slices proteins were reduced by adding 10 mM DTT (Sigma Aldrich) prior to alkylation with 55 mM iodoacetamide (Sigma Aldrich). After washing and shrinking of the gel pieces with 100% acetonitrile trypsin (Sequencing Grade Modified, Promega) was added and proteins were digested overnight in 40 mM ammoniumbicarbonate at 37°C. For protein identification samples were directly used for nano-ESI-LC-MS/MS. Each sample was first desalted on-line by a C18 microcolumn (300 µm i.d. x 5 mm, packed with C18 PepMap™, 5 µm, 100 Å by LC Packings). Peptides were then separated on a C18 reversed phase column via a linear acetonitrile gradient (UltiMate 3000 system (Dionex) and column (75 µm i.d. x 15 cm, packed with C18 PepMap™, 3 µm, 100 Å by LC Packings) before MS and MS/MS spectra were recorded on an Oribitrap mass spectrometer (Thermo Electron). ',\n",
       " 'Conditioned media preparation and HPLC-isobaric tags for relative and absolute quantitation mass-spectrometry (iTRAQ MS). Twenty million cells were transfected with either control or Gαs siRNA, then were seeded in 100-mm cell culture plates in DMEM + 10% FBS. Two days later, cells were washed twice with pre-warmed PBS. Conditioned media was harvested another 24 h later, and centrifuged at 20,000 g for 10 min at 4°C to remove cell debris. To enrich for secreted proteins in the conditioned media, conditioned media samples were centrifuged using Amicon centrifugal filters with a 3kDa cutoff (Millipore, Billerica, MA). A total of 8 concentrated conditioned media samples were independently prepared: 4 samples from control siRNA-treated cells, and 4 samples from Gαs siRNA-treated cells. Samples were stored at –70°C until they were analyzed by iTRAQ labeling and LC-MS/MS. iTRAQ labeling – Total protein concentration of each sample was measured using the Bradford assay. A quantity of 100 µg was taken from each sample, denatured in 0.1% SDS, reduced in 50 mM Tris(2-carboxyethyl)phosphine (TCEP), alkylated in 200 mM methyl methanethiosulfonate (MMTS), and digested overnight with trypsin. Each sample’s peptides were labeled with different iTRAQ reagents (AB Sciex, Framingham, MA) following the manufacturer’s protocol, combined, dried down to a 50 µL volume, acidified with 1% trifluoroacetic acid (TFA), and desalted with a 100 mg-C18 SepPak cartridge (Waters, Milford, MA). Peptide fractionation – Combined iTRAQ labeled peptides were first separated into 12 fractions using a 3100 OFFGEL Fractionator (Agilent, Santa Clara, CA) with a 12-well setup and a 12-cm pH 3–10 immobilized pH-gradient (IPG) strip, according to manufacturer’s recommendations. Peptides were collected after the focusing voltage reached a maximum plateau (> 50 kVh). The recovered fractions were lyophilized. LC-MALDI-TOF/TOF analysis – All solvents used in this section were aqueous unless specified and obtained from ThermoFisher Scientific (San Jose, CA). Dried samples were reconstituted in 30 µL of solvent A (2% v/v acetonitrile [ACN], 0.1% v/v TFA), and 10 µL of the resulting solution was injected into an Ultimate™ HPLC system with UV detection (LC Packings, Sunnyvale, CA). Solvent B contained 98% v/v ACN, 0.1% v/v TFA. A C18 trap (5.0 mm x 300 µm ID, LC Packings) and a 15 cm x 100 µm ID column packed in-house with Magic C18 5 µm beads (Michrom Bioresources, Auburn, CA) were used for chromatography. Solvent flow rate was set to 0.700 µL/min. After injection, each sample was loaded and washed with solvent A for 5 min at a flow rate of 0.030 mL/min, followed by a 10% to 50% gradient over 55 min, then ramped to 90% by 65 min. Each sample run was maintained in 90% solvent B until 75 min, returned to 10% at 75.1 min, followed by a 10 min equilibration in 10% solvent B. The HPLC system is connected to a modified Probot™ automated MALDI spotter (LC Packings) to spot eluates on an LC MALDI plate with on-line mixing of α-Cyano-4-hydroxycinnamic acid (CHCA) matrix (7.5 mg/mL in 70% ACN). Considering the dead time for the HPLC and spotting system, spot collection time was started 10 min from injection time with a spotting time of 0.33 min per spot. A total of 176 spots were collected. The sample plate with the resulting spots was loaded onto a 5800 MALDI time-of-flight/time-of-flight (TOF/TOF) mass spectrometer (AB Sciex, Framingham, MA) for MS and MS/MS analysis. Batch-mode mass calibration with plate alignment were done before data acquisition, thus a separate plate calibration file was generated for each individual plate containing sample spots. Full scan MS spectra were obtained with a total of 1000 laser shots per spectrum. From each spot, the top 20 most intense peaks, weakest precursor first, that were above a signal-to-noise threshold of 10, were selected for successful MS/MS analysis. Each MS/MS spectrum was an average of 1600 laser shots. MS/MS spectra were acquired in the presence of collision gas. The sequence of the VGF-NERP1 peptide was confirmed using the LTQ-Orbitrap mass spectrometer (ThermoFisher Scientific, San Jose, CA).',\n",
       " 'Proteins were eluted from washed beads by adding 2-fold-concentrated SDS-gel sample loading buffer and heating to 80°C for 10 min. The supernatant was loaded onto a Novex gel (10% Bis-Tris SDS gel; Life Technologies/Invitrogen, Carlsbad, CA) and run about 1/3 of the length of the lane. The lanes were cut out and stained with InstantBlue (Expedeon Ltd., Harston, United Kingdom) in separate trays until bands were just visible. The area covering the bands was cut into 5 to 6 slices, which were washed, reduced, and alkylated and treated with trypsin according to standard procedures. Peptides were extracted with 5% formic acid–50% acetonitrile, dried down, and redissolved in 0.1% trifluoroacetic acid (TFA). For liquid chromatography-tandem mass spectrometry (LC-MS/MS) analysis, a sample aliquot was applied via a nanoAcquity (Waters, Manchester, United Kingdom) ultraperformance liquid chromatography (UPLC) system running at a flow rate of 250 nl min−1 to an LTQ-Orbitrap mass spectrometer (Thermo Fisher, Waltham, MA). Peptides were trapped using a precolumn (Symmetry C18, 5 μm, 180 μm by 20 mm; Waters), which was then switched in-line to an analytical column (BEH C18, 1.7 μm, 75 μm by 250 mm; Waters) for separation. Peptides were eluted with a gradient of 3 to 40% acetonitrile in water–0.1% formic acid at a rate of 0.67% min−1. The column was connected to a 10-μm SilicaTip nanospray emitter (New Objective, Woburn, MA, USA) attached to a nanospray interface (Proxeon, Odense, Denmark) for infusion into the mass spectrometer. The mass spectrometer was operated in positive ion mode at a capillary temperature of 200°C. The source voltage and focusing voltages were tuned for the transmission of Met-Arg-Phe-Ala (MRFA) peptide (m/z 524) (Sigma-Aldrich, St. Louis, MO). Data-dependent analysis was carried out in orbitrap-ion trap parallel mode using collision-induced fragmentation (CID) on the 6 most abundant ions in each cycle. The orbitrap was run with a resolution of 30,000 over the MS range from m/z 350 to m/z 1,800 and an MS target of 106 and 1-s maximum scan time. Collision energy was 35, and an isolation width of 2 was used. Only monoisotopic 2+ and 3+ charged precursors were selected for MS2 fragmentation stage. The MS2 was triggered by a minimal signal of 1,000 with an automatic gain control target of 3 × 104 ions and 150-ms scan time using the chromatography function for peak apex detection. Dynamic exclusion was set to 1 count and 60-s exclusion with an exclusion mass window of ±20 ppm. MS scans were saved in profile mode, while MS/MS scans were saved in centroid mode.']"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "descriptive_text[:5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Use spacy and split into sentences"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "nlp = spacy.load('en_core_web_lg')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "def process_descriptive_text(corpus):\n",
    "    sentences = []\n",
    "    for text in corpus:\n",
    "        doc = nlp(text)\n",
    "        s = [sent.text for sent in doc.sents]\n",
    "        sentences.extend(s)\n",
    "    return sentences"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "sentences = process_descriptive_text(descriptive_text)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['The crude membranes from 5 P56-P70 Glun1TAP/TAP mouse forebrains were re-suspended in 12.5\\u2009ml buffer H and extracted with 12.5\\u2009ml 2% deoxycholate, 100\\u2009mM NaCl, 50\\u2009mM Tris.',\n",
       " 'Cl pH8 for 1\\u2009h at 6\\u2009°C.',\n",
       " 'Total extract was centrifuged at 120,000g.',\n",
       " 'for 40\\u2009min at 8\\u2009°C.',\n",
       " 'Conditions for immuno-capture, wash and peptide-antigen exchange elution were screened using a high-throughput purification robot (MAGic sample processor, Invitrogen).',\n",
       " 'For 25\\u2009ml Glun1TAP/TAP extract supernatant, 80\\u2009μg mouse Flag antibody was coupled to 30\\u2009mg (500\\u2009μl)',\n",
       " 'protein G magnetic beads (Invitrogen).',\n",
       " 'Receptor was captured from extract supernatant for 2\\u2009h.',\n",
       " 'The beads were washed three times with 5\\u2009ml wash buffer (0.37% w/v sodium deoxycholate, 0.05\\u2009mg.ml−1 lipids (1:1:3',\n",
       " 'POPC:POPE:POG), 150\\u2009mM NaCl, 50\\u2009mM Tris.']"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sentences[:10]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Save as pickle object"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pickle"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "outfile = open('descriptive_text.pickle', 'wb')\n",
    "pickle.dump(sentences, outfile)\n",
    "outfile.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
