{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<a id='sec0'></a>\n",
    "# Predictive annotation of quantification methods with Noun Chunks 3\n",
    "It appears there are some weird noun chunks and here I'm re-acessing the vocabulary, so that improved text-processing scheme can be established.<br>\n",
    "- <a href='#sec1'><b></b></a>\n",
    "- <a href='#sec2'><b></b></a>\n",
    "- <a href='#sec3'><b></b></a>\n",
    "- <a href='#sec4'><b></b></a>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import pickle\n",
    "import spacy\n",
    "import re\n",
    "\n",
    "from nltk.corpus import stopwords\n",
    "from gensim import corpora, models\n",
    "from gensim import matutils\n",
    "import matplotlib.pyplot as plt\n",
    "%matplotlib inline"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<a id='sec1-1'></a>\n",
    "#### 1. Read in data and process df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dictionary(125116 unique tokens: [') luminal a tumors', '1 tumors', '24 lymph node', '24 lymph node negative grade 1 luminal a primary breast tumors', '4405 proteins']...)\n"
     ]
    }
   ],
   "source": [
    "with open('nlp15_data/dfs/all_fields_nchunks_df_quant_dummies.pickle', 'rb') as infile_df:\n",
    "    df = pickle.load(infile_df)\n",
    "    \n",
    "my_dictionary = corpora.Dictionary.load('nlp15_data/whole_dictionary_nchunks.dict')\n",
    "print(my_dictionary)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>dataset_id</th>\n",
       "      <th>sample_protocol</th>\n",
       "      <th>data_protocol</th>\n",
       "      <th>description</th>\n",
       "      <th>instruments</th>\n",
       "      <th>exp_types</th>\n",
       "      <th>quant_methods</th>\n",
       "      <th>labhead_fullname</th>\n",
       "      <th>silac</th>\n",
       "      <th>ms1_label_free</th>\n",
       "      <th>spectrum_counting</th>\n",
       "      <th>tmt</th>\n",
       "      <th>itraq</th>\n",
       "      <th>label_free</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PXD000029</td>\n",
       "      <td>[breast cancer tissue lysates, reduction, alky...</td>\n",
       "      <td>[proteomics data, proteome discoverer, fdr&lt;0.0...</td>\n",
       "      <td>[current prognostic factors, precise risk-disc...</td>\n",
       "      <td>LTQ Orbitrap Velos</td>\n",
       "      <td>Shotgun proteomics</td>\n",
       "      <td>itraq</td>\n",
       "      <td>Pavel Bouchal</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PXD000164</td>\n",
       "      <td>[protein extraction, catheter biofilm small pi...</td>\n",
       "      <td>[tryptic digest, reversed phase, rp) chromatog...</td>\n",
       "      <td>[term-catheterization, catheter-associated bac...</td>\n",
       "      <td>LTQ Orbitrap Velos</td>\n",
       "      <td>Shotgun proteomics</td>\n",
       "      <td>label free</td>\n",
       "      <td>Katharina Riedel</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  dataset_id                                    sample_protocol  \\\n",
       "0  PXD000029  [breast cancer tissue lysates, reduction, alky...   \n",
       "1  PXD000164  [protein extraction, catheter biofilm small pi...   \n",
       "\n",
       "                                       data_protocol  \\\n",
       "0  [proteomics data, proteome discoverer, fdr<0.0...   \n",
       "1  [tryptic digest, reversed phase, rp) chromatog...   \n",
       "\n",
       "                                         description         instruments  \\\n",
       "0  [current prognostic factors, precise risk-disc...  LTQ Orbitrap Velos   \n",
       "1  [term-catheterization, catheter-associated bac...  LTQ Orbitrap Velos   \n",
       "\n",
       "            exp_types quant_methods  labhead_fullname  silac  ms1_label_free  \\\n",
       "0  Shotgun proteomics         itraq     Pavel Bouchal      0               0   \n",
       "1  Shotgun proteomics    label free  Katharina Riedel      0               0   \n",
       "\n",
       "   spectrum_counting  tmt  itraq  label_free  \n",
       "0                  0    0      1           0  \n",
       "1                  0    0      0           1  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "125116"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tokens = sorted(list(my_dictionary.token2id.keys()))\n",
    "len(tokens)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u00020.1\n",
      "\u00021.9\n",
      "\u00022.2\n",
      "\u00023.5\n",
      "\u0006cn\n",
      "\" molecular & cellular proteomics\n",
      "\"\"cessation\n",
      "\"\"heavy\"\" media\n",
      "\"\"light\n",
      "\"\"lock mass\"\" function\n",
      "\"\"mouse complete proteome\n",
      "\"\"switch\n",
      "\"071812_glit3t\n",
      "\"ab initio\n",
      "\"annotation matrix\n",
      "\"b\n",
      "\"background\n",
      "\"bait\n",
      "\"complete\" t. cruzi proteome\n",
      "\"database\n",
      "\"deep\" proteome\n",
      "\"experimental design proteomexchange.txt\n",
      "\"experiments\n",
      "\"filter assisted sample preparation\n",
      "\"gene name & annotations\" section\n",
      "\"global normalization\n",
      "\"good outcome\n",
      "\"heavy\" amino-acids\n",
      "\"heavy\" capture assay samples\n",
      "\"heavy\" deuterated versions\n",
      "\"heavy\" isotope-labeled amino acids\n",
      "\"heavy\" media\n",
      "\"input\" material\n",
      "\"large-scale targeted proteomics\n",
      "\"light\n",
      "\"light\" l-arginine and l-lysine\n",
      "\"light\" media\n",
      "\"light\" silac media\n",
      "\"lock mass\" function\n",
      "\"match\n",
      "\"medium-heavy\" and \"heavy\" cells\n",
      "\"mixo\n",
      "\"mouse complete proteome\n",
      "\"multicomp\" package\n",
      "\"neg\n",
      "\"negative\" weight acetylation\n",
      "\"poor outcome\n",
      "\"precursor ion quantifier\" node\n",
      "\"protein identification and network analysis\" methods section\n",
      "\"proteinpilot mgf\" parameters\n",
      "\"proteomic ruler\" method\n",
      "\"pseudomature\n",
      "\"ql060112_c1b\n",
      "\"ql060112_c2b\n",
      "\"ql060112_g1\n",
      "\"ql060112_g1b\" \"ql060112_g2\n",
      "\"quantile normalization\" procedure\n",
      "\"r\n",
      "\"rank\n",
      "\"rapid\" settings\n",
      "\"re-quantify\" \"filter labeled amino acids\n",
      "\"reporter ions quantifier\" node\n",
      "\"requantify\n",
      "\"sartogosm\n",
      "\"significance threshold\n",
      "\"sijimi\" longan\n",
      "\"sixia\n",
      "\"spectrum exporter\" nodes\n",
      "\"spectrum selector\n",
      "\"switch\n",
      "\"system suitability (compref) study\n",
      "\"thorough\" search effort\n",
      "\"three to five\" vectors\n",
      "\"toxodb-26_ncaninumliv_annotatedproteins\n",
      "\"uniprot-reviewed musmusculus\n",
      "\"unique\" quantitative results\n",
      "\"use\n",
      "\"variable modification\n",
      "#\n",
      "# i1149\n",
      "# ki-3003\n",
      "# lc6025\n",
      "# mrcprt010 ultracel\n",
      "# v5111\n",
      "# vn01h23eto\n",
      "# wat094225\n",
      "## may\n",
      "#12686}. peptide preparation\n",
      "#53}. proteins\n",
      "#c10416\n",
      "#es803\n",
      "#f4799\n",
      "#psms\n",
      "#reverse peptide\n",
      "#reverse proteins\n",
      "%\n",
      "% acetonitrile\n",
      "% acn\n",
      "% b\n",
      "% confluence\n",
      "% confluency\n",
      "% deuteration\n",
      "% formic acid\n",
      "% globin\n",
      "% incorporation\n",
      "% isopropanol\n",
      "% nh3\n",
      "% tfa\n",
      "%a/%b\n",
      "%b\n",
      "%vol\n",
      "'bimodalitytest' package\n",
      "'contaminants\n",
      "'control' ratios\n",
      "'cor\n",
      "'core' human pr\n",
      "'earliest' fraction\n",
      "'early risers\n",
      "'gel-top' approach\n",
      "'hartigan’s dip test\n",
      "'heavy' cells\n",
      "'heavy' isotope-labelled lysates\n",
      "'heavy' lysine\n",
      "'heavy' silac\n",
      "'ibaq\n",
      "'integrin family cell surface interactions\n",
      "'last-resort' antibiotic\n",
      "'lfq\n",
      "'light\n",
      "'ls\n",
      "'match\n",
      "'missing proteins\n",
      "'multiple hit' model\n",
      "'par-1-mediated thrombin signal events\n",
      "'platelet degranulation\n",
      "'proportion-of-total' value\n",
      "'proteomic ruler' plugin\n",
      "'proteomic ruler' strategy\n",
      "'requantify\n",
      "'second peptide search' option\n",
      "'short rna\n",
      "'short rna' improved rna stability\n",
      "'significance\n",
      "'software lock mass’ option\n",
      "'stats' package\n",
      "'tat\n",
      "'tmt n-term\n",
      "'treated' silac ratios\n",
      "'unique' quantification results\n",
      "'virtual western blots\n",
      "(+) and (-) manassantin a samples\n",
      "(+) manassantin a sample\n",
      "(+) manassantin a samplewas\n",
      "(-) control\n",
      "(-) control sample\n",
      "(0.1% formic acid) b (80% acetonitrile\n",
      "(10 mm ammonium formate\n",
      "(10 mm ammonium hydroxide\n",
      "(10mm teab ph8\n",
      "(16o vial and 18o vial\n",
      "(1:2 peptide\n",
      "(2% acetonitrile\n",
      "(2% acn\n",
      "(20 mm ammonium formate\n",
      "(3k molecular weight cut-off\n",
      "(48 nce\n",
      "(5 mm k2hpo4\n",
      "(5% acetonitrile\n",
      "(5% dmso\n",
      "(7 mm kh2po4 ph 2.65 30% acn\n",
      "(8 fractions\n",
      "(8 m urea\n",
      "(80% acetonitrile\n",
      "(90% acetonitrile\n",
      "(95% water\n",
      "(98% acetonitrile\n",
      "(aqueous 2% acetonitrile\n",
      "(bpa) exposure\n",
      "(c2h6sio)6\n",
      "(di)methylated hla class\n",
      "(dmso\n",
      "(forward and reverse) experiment\n",
      "(h)-silac-labeled wild type yeast cells\n",
      "(l)-silac-labeled wild type yeast cells\n",
      "(long-term) former smoker group\n",
      "(mainly ms-based) proteomic data\n",
      "(modified) proteome\n",
      "(moi\n",
      "(n=21\n",
      "(non-resolved) protein band\n",
      "(phospho) proteome\n",
      "(phospho)proteome and 105 min\n",
      "(pngase f-catalyzed) conversion\n",
      "(previously sequenced) prophage\n",
      "(raw) intensities\n",
      "(resolution\n",
      "(saline\n",
      "(sample s1\n",
      "(sarco)endoplasmic reticulum\n",
      "(sham) exposure\n"
     ]
    }
   ],
   "source": [
    "for token in tokens[:200]:\n",
    "    print(token)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " 'a/%b',\n",
       " 'ab initio',\n",
       " 'acetonitrile',\n",
       " 'acetonitrile',\n",
       " 'acetonitrile',\n",
       " 'acetonitrile',\n",
       " 'acetonitrile',\n",
       " 'acetonitrile',\n",
       " 'acn',\n",
       " 'acn',\n",
       " 'and (-) manassantin a samples',\n",
       " 'annotation matrix',\n",
       " 'aqueous 2% acetonitrile',\n",
       " 'b',\n",
       " 'b',\n",
       " 'b',\n",
       " 'background',\n",
       " 'bait',\n",
       " \"bimodalitytest' package\",\n",
       " 'bpa) exposure',\n",
       " 'c10416',\n",
       " 'c2h6sio)6',\n",
       " 'cessation',\n",
       " 'cn',\n",
       " 'complete\" t. cruzi proteome',\n",
       " 'confluence',\n",
       " 'confluency',\n",
       " 'contaminants',\n",
       " 'control',\n",
       " 'control sample',\n",
       " \"control' ratios\",\n",
       " 'cor',\n",
       " \"core' human pr\",\n",
       " 'database',\n",
       " 'deep\" proteome',\n",
       " 'deuteration',\n",
       " 'di)methylated hla class',\n",
       " 'dmso',\n",
       " 'dmso',\n",
       " \"earliest' fraction\",\n",
       " 'early risers',\n",
       " 'es803',\n",
       " 'experimental design proteomexchange.txt',\n",
       " 'experiments',\n",
       " 'f4799',\n",
       " 'filter assisted sample preparation',\n",
       " 'formic acid',\n",
       " 'formic acid) b (80% acetonitrile',\n",
       " 'forward and reverse) experiment',\n",
       " 'fractions',\n",
       " \"gel-top' approach\",\n",
       " 'gene name & annotations\" section',\n",
       " 'glit3t',\n",
       " 'global normalization',\n",
       " 'globin',\n",
       " 'good outcome',\n",
       " 'h)-silac-labeled wild type yeast cells',\n",
       " 'hartigan’s dip test',\n",
       " 'heavy\" amino-acids',\n",
       " 'heavy\" capture assay samples',\n",
       " 'heavy\" deuterated versions',\n",
       " 'heavy\" isotope-labeled amino acids',\n",
       " 'heavy\" media',\n",
       " 'heavy\"\" media',\n",
       " \"heavy' cells\",\n",
       " \"heavy' isotope-labelled lysates\",\n",
       " \"heavy' lysine\",\n",
       " \"heavy' silac\",\n",
       " 'i1149',\n",
       " 'ibaq',\n",
       " 'incorporation',\n",
       " 'input\" material',\n",
       " 'integrin family cell surface interactions',\n",
       " 'isopropanol',\n",
       " 'k molecular weight cut-off',\n",
       " 'ki-3003',\n",
       " 'l)-silac-labeled wild type yeast cells',\n",
       " 'large-scale targeted proteomics',\n",
       " \"last-resort' antibiotic\",\n",
       " 'lc6025',\n",
       " 'lfq',\n",
       " 'light',\n",
       " 'light',\n",
       " 'light',\n",
       " 'light\" l-arginine and l-lysine',\n",
       " 'light\" media',\n",
       " 'light\" silac media',\n",
       " 'lock mass\" function',\n",
       " 'lock mass\"\" function',\n",
       " 'long-term) former smoker group',\n",
       " 'ls',\n",
       " 'm urea',\n",
       " 'mainly ms-based) proteomic data',\n",
       " 'manassantin a sample',\n",
       " 'manassantin a samplewas',\n",
       " 'match',\n",
       " 'match',\n",
       " 'may',\n",
       " 'medium-heavy\" and \"heavy\" cells',\n",
       " 'missing proteins',\n",
       " 'mixo',\n",
       " 'mm ammonium formate',\n",
       " 'mm ammonium formate',\n",
       " 'mm ammonium hydroxide',\n",
       " 'mm k2hpo4',\n",
       " 'mm kh2po4 ph 2.65 30% acn',\n",
       " 'mm teab ph8',\n",
       " 'modified) proteome',\n",
       " 'moi',\n",
       " 'molecular & cellular proteomics',\n",
       " 'mouse complete proteome',\n",
       " 'mouse complete proteome',\n",
       " 'mrcprt010 ultracel',\n",
       " 'multicomp\" package',\n",
       " \"multiple hit' model\",\n",
       " 'n=21',\n",
       " 'nce',\n",
       " 'neg',\n",
       " 'negative\" weight acetylation',\n",
       " 'nh3',\n",
       " 'non-resolved) protein band',\n",
       " 'o vial and 18o vial',\n",
       " 'par-1-mediated thrombin signal events',\n",
       " 'peptide',\n",
       " 'peptide preparation',\n",
       " 'phospho) proteome',\n",
       " 'phospho)proteome and 105 min',\n",
       " 'platelet degranulation',\n",
       " 'pngase f-catalyzed) conversion',\n",
       " 'poor outcome',\n",
       " 'precursor ion quantifier\" node',\n",
       " 'previously sequenced) prophage',\n",
       " \"proportion-of-total' value\",\n",
       " 'protein identification and network analysis\" methods section',\n",
       " 'proteinpilot mgf\" parameters',\n",
       " 'proteins',\n",
       " 'proteomic ruler\" method',\n",
       " \"proteomic ruler' plugin\",\n",
       " \"proteomic ruler' strategy\",\n",
       " 'pseudomature',\n",
       " 'psms',\n",
       " 'ql060112_c1b',\n",
       " 'ql060112_c2b',\n",
       " 'ql060112_g1',\n",
       " 'ql060112_g1b\" \"ql060112_g2',\n",
       " 'quantile normalization\" procedure',\n",
       " 'r',\n",
       " 'rank',\n",
       " 'rapid\" settings',\n",
       " 'raw) intensities',\n",
       " 're-quantify\" \"filter labeled amino acids',\n",
       " 'reporter ions quantifier\" node',\n",
       " 'requantify',\n",
       " 'requantify',\n",
       " 'resolution',\n",
       " 'reverse peptide',\n",
       " 'reverse proteins',\n",
       " 'saline',\n",
       " 'sample s1',\n",
       " 'sarco)endoplasmic reticulum',\n",
       " 'sartogosm',\n",
       " \"second peptide search' option\",\n",
       " 'sham) exposure',\n",
       " 'short rna',\n",
       " \"short rna' improved rna stability\",\n",
       " 'significance',\n",
       " 'significance threshold',\n",
       " 'sijimi\" longan',\n",
       " 'sixia',\n",
       " 'software lock mass’ option',\n",
       " 'spectrum exporter\" nodes',\n",
       " 'spectrum selector',\n",
       " \"stats' package\",\n",
       " 'switch',\n",
       " 'switch',\n",
       " 'system suitability (compref) study',\n",
       " 'tat',\n",
       " 'tfa',\n",
       " 'thorough\" search effort',\n",
       " 'three to five\" vectors',\n",
       " 'tmt n-term',\n",
       " 'toxodb-26_ncaninumliv_annotatedproteins',\n",
       " \"treated' silac ratios\",\n",
       " 'uniprot-reviewed musmusculus',\n",
       " 'unique\" quantitative results',\n",
       " \"unique' quantification results\",\n",
       " 'use',\n",
       " 'v5111',\n",
       " 'variable modification',\n",
       " 'virtual western blots',\n",
       " 'vn01h23eto',\n",
       " 'vol',\n",
       " 'wat094225',\n",
       " 'water']"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "regex1 = r'^[^A-Za-z]+'\n",
    "sorted([re.sub(regex1, '', s).strip() for s in tokens[:200]])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " 'a/%b',\n",
       " 'ab initio',\n",
       " 'acetonitrile',\n",
       " 'acetonitrile',\n",
       " 'acetonitrile',\n",
       " 'acetonitrile',\n",
       " 'acetonitrile',\n",
       " 'acetonitrile',\n",
       " 'acn',\n",
       " 'acn',\n",
       " 'and  manassantin a samples',\n",
       " 'annotation matrix',\n",
       " 'aqueous 2% acetonitrile',\n",
       " 'b',\n",
       " 'b',\n",
       " 'b',\n",
       " 'background',\n",
       " 'bait',\n",
       " 'bimodalitytest package',\n",
       " 'bpa) exposure',\n",
       " 'c10416',\n",
       " 'c2h6sio)6',\n",
       " 'cessation',\n",
       " 'cn',\n",
       " 'complete t. cruzi proteome',\n",
       " 'confluence',\n",
       " 'confluency',\n",
       " 'contaminants',\n",
       " 'control',\n",
       " 'control sample',\n",
       " 'control ratios',\n",
       " 'cor',\n",
       " 'core human pr',\n",
       " 'database',\n",
       " 'deep proteome',\n",
       " 'deuteration',\n",
       " 'di)methylated hla class',\n",
       " 'dmso',\n",
       " 'dmso',\n",
       " 'earliest fraction',\n",
       " 'early risers',\n",
       " 'es803',\n",
       " 'experimental design proteomexchange.txt',\n",
       " 'experiments',\n",
       " 'f4799',\n",
       " 'filter assisted sample preparation',\n",
       " 'formic acid',\n",
       " 'formic acid) b (80% acetonitrile',\n",
       " 'forward and reverse) experiment',\n",
       " 'fractions',\n",
       " 'gel-top approach',\n",
       " 'gene name & annotations section',\n",
       " 'glit3t',\n",
       " 'global normalization',\n",
       " 'globin',\n",
       " 'good outcome',\n",
       " 'h)-silac-labeled wild type yeast cells',\n",
       " 'hartigan’s dip test',\n",
       " 'heavy amino-acids',\n",
       " 'heavy capture assay samples',\n",
       " 'heavy deuterated versions',\n",
       " 'heavy isotope-labeled amino acids',\n",
       " 'heavy media',\n",
       " 'heavy media',\n",
       " 'heavy cells',\n",
       " 'heavy isotope-labelled lysates',\n",
       " 'heavy lysine',\n",
       " 'heavy silac',\n",
       " 'i1149',\n",
       " 'ibaq',\n",
       " 'incorporation',\n",
       " 'input material',\n",
       " 'integrin family cell surface interactions',\n",
       " 'isopropanol',\n",
       " 'k molecular weight cut-off',\n",
       " 'ki-3003',\n",
       " 'l)-silac-labeled wild type yeast cells',\n",
       " 'large-scale targeted proteomics',\n",
       " 'last-resort antibiotic',\n",
       " 'lc6025',\n",
       " 'lfq',\n",
       " 'light',\n",
       " 'light',\n",
       " 'light',\n",
       " 'light l-arginine and l-lysine',\n",
       " 'light media',\n",
       " 'light silac media',\n",
       " 'lock mass function',\n",
       " 'lock mass function',\n",
       " 'long-term) former smoker group',\n",
       " 'ls',\n",
       " 'm urea',\n",
       " 'mainly ms-based) proteomic data',\n",
       " 'manassantin a sample',\n",
       " 'manassantin a samplewas',\n",
       " 'match',\n",
       " 'match',\n",
       " 'may',\n",
       " 'medium-heavy and heavy cells',\n",
       " 'missing proteins',\n",
       " 'mixo',\n",
       " 'mm ammonium formate',\n",
       " 'mm ammonium formate',\n",
       " 'mm ammonium hydroxide',\n",
       " 'mm k2hpo4',\n",
       " 'mm kh2po4 ph 2.65 30% acn',\n",
       " 'mm teab ph8',\n",
       " 'modified) proteome',\n",
       " 'moi',\n",
       " 'molecular & cellular proteomics',\n",
       " 'mouse complete proteome',\n",
       " 'mouse complete proteome',\n",
       " 'mrcprt010 ultracel',\n",
       " 'multicomp package',\n",
       " 'multiple hit model',\n",
       " 'n=21',\n",
       " 'nce',\n",
       " 'neg',\n",
       " 'negative weight acetylation',\n",
       " 'nh3',\n",
       " 'non-resolved) protein band',\n",
       " 'o vial and 18o vial',\n",
       " 'par-1-mediated thrombin signal events',\n",
       " 'peptide',\n",
       " 'peptide preparation',\n",
       " 'phospho) proteome',\n",
       " 'phospho)proteome and 105 min',\n",
       " 'platelet degranulation',\n",
       " 'pngase f-catalyzed) conversion',\n",
       " 'poor outcome',\n",
       " 'precursor ion quantifier node',\n",
       " 'previously sequenced) prophage',\n",
       " 'proportion-of-total value',\n",
       " 'protein identification and network analysis methods section',\n",
       " 'proteinpilot mgf parameters',\n",
       " 'proteins',\n",
       " 'proteomic ruler method',\n",
       " 'proteomic ruler plugin',\n",
       " 'proteomic ruler strategy',\n",
       " 'pseudomature',\n",
       " 'psms',\n",
       " 'ql060112_c1b',\n",
       " 'ql060112_c2b',\n",
       " 'ql060112_g1',\n",
       " 'ql060112_g1b ql060112_g2',\n",
       " 'quantile normalization procedure',\n",
       " 'r',\n",
       " 'rank',\n",
       " 'rapid settings',\n",
       " 'raw) intensities',\n",
       " 're-quantify filter labeled amino acids',\n",
       " 'reporter ions quantifier node',\n",
       " 'requantify',\n",
       " 'requantify',\n",
       " 'resolution',\n",
       " 'reverse peptide',\n",
       " 'reverse proteins',\n",
       " 'saline',\n",
       " 'sample s1',\n",
       " 'sarco)endoplasmic reticulum',\n",
       " 'sartogosm',\n",
       " 'second peptide search option',\n",
       " 'sham) exposure',\n",
       " 'short rna',\n",
       " 'short rna improved rna stability',\n",
       " 'significance',\n",
       " 'significance threshold',\n",
       " 'sijimi longan',\n",
       " 'sixia',\n",
       " 'software lock mass’ option',\n",
       " 'spectrum exporter nodes',\n",
       " 'spectrum selector',\n",
       " 'stats package',\n",
       " 'switch',\n",
       " 'switch',\n",
       " 'system suitability  study',\n",
       " 'tat',\n",
       " 'tfa',\n",
       " 'thorough search effort',\n",
       " 'three to five vectors',\n",
       " 'tmt n-term',\n",
       " 'toxodb-26_ncaninumliv_annotatedproteins',\n",
       " 'treated silac ratios',\n",
       " 'uniprot-reviewed musmusculus',\n",
       " 'unique quantitative results',\n",
       " 'unique quantification results',\n",
       " 'use',\n",
       " 'v5111',\n",
       " 'variable modification',\n",
       " 'virtual western blots',\n",
       " 'vn01h23eto',\n",
       " 'vol',\n",
       " 'wat094225',\n",
       " 'water']"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "regex2 = r'\\'|\\\"|\\(.+\\)'\n",
    "tokens2 = sorted([re.sub(regex1, '', s).strip() for s in tokens[:200]])\n",
    "[re.sub(regex2, '', s).strip() for s in tokens2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "regex3 = r'^[^A-Za-z]+|\\'|\\\"|\\(.+\\)'\n",
    "tokens3 = sorted([re.sub(regex3, '', s).strip() for s in tokens[:200]])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " 'a/%b',\n",
       " 'ab initio',\n",
       " 'acetonitrile',\n",
       " 'acetonitrile',\n",
       " 'acetonitrile',\n",
       " 'acetonitrile',\n",
       " 'acetonitrile',\n",
       " 'acetonitrile',\n",
       " 'acn',\n",
       " 'acn',\n",
       " 'and  manassantin a samples',\n",
       " 'annotation matrix',\n",
       " 'aqueous 2% acetonitrile',\n",
       " 'b',\n",
       " 'b',\n",
       " 'b',\n",
       " 'background',\n",
       " 'bait',\n",
       " 'bimodalitytest package',\n",
       " 'bpa) exposure',\n",
       " 'c10416',\n",
       " 'c2h6sio)6',\n",
       " 'cessation',\n",
       " 'cn',\n",
       " 'complete t. cruzi proteome',\n",
       " 'confluence',\n",
       " 'confluency',\n",
       " 'contaminants',\n",
       " 'control',\n",
       " 'control ratios',\n",
       " 'control sample',\n",
       " 'cor',\n",
       " 'core human pr',\n",
       " 'database',\n",
       " 'deep proteome',\n",
       " 'deuteration',\n",
       " 'di)methylated hla class',\n",
       " 'dmso',\n",
       " 'dmso',\n",
       " 'earliest fraction',\n",
       " 'early risers',\n",
       " 'es803',\n",
       " 'experimental design proteomexchange.txt',\n",
       " 'experiments',\n",
       " 'f4799',\n",
       " 'filter assisted sample preparation',\n",
       " 'formic acid',\n",
       " 'formic acid) b (80% acetonitrile',\n",
       " 'forward and reverse) experiment',\n",
       " 'fractions',\n",
       " 'gel-top approach',\n",
       " 'gene name & annotations section',\n",
       " 'glit3t',\n",
       " 'global normalization',\n",
       " 'globin',\n",
       " 'good outcome',\n",
       " 'h)-silac-labeled wild type yeast cells',\n",
       " 'hartigan’s dip test',\n",
       " 'heavy amino-acids',\n",
       " 'heavy capture assay samples',\n",
       " 'heavy cells',\n",
       " 'heavy deuterated versions',\n",
       " 'heavy isotope-labeled amino acids',\n",
       " 'heavy isotope-labelled lysates',\n",
       " 'heavy lysine',\n",
       " 'heavy media',\n",
       " 'heavy media',\n",
       " 'heavy silac',\n",
       " 'i1149',\n",
       " 'ibaq',\n",
       " 'incorporation',\n",
       " 'input material',\n",
       " 'integrin family cell surface interactions',\n",
       " 'isopropanol',\n",
       " 'k molecular weight cut-off',\n",
       " 'ki-3003',\n",
       " 'l)-silac-labeled wild type yeast cells',\n",
       " 'large-scale targeted proteomics',\n",
       " 'last-resort antibiotic',\n",
       " 'lc6025',\n",
       " 'lfq',\n",
       " 'light',\n",
       " 'light',\n",
       " 'light',\n",
       " 'light l-arginine and l-lysine',\n",
       " 'light media',\n",
       " 'light silac media',\n",
       " 'lock mass function',\n",
       " 'lock mass function',\n",
       " 'long-term) former smoker group',\n",
       " 'ls',\n",
       " 'm urea',\n",
       " 'mainly ms-based) proteomic data',\n",
       " 'manassantin a sample',\n",
       " 'manassantin a samplewas',\n",
       " 'match',\n",
       " 'match',\n",
       " 'may',\n",
       " 'medium-heavy and heavy cells',\n",
       " 'missing proteins',\n",
       " 'mixo',\n",
       " 'mm ammonium formate',\n",
       " 'mm ammonium formate',\n",
       " 'mm ammonium hydroxide',\n",
       " 'mm k2hpo4',\n",
       " 'mm kh2po4 ph 2.65 30% acn',\n",
       " 'mm teab ph8',\n",
       " 'modified) proteome',\n",
       " 'moi',\n",
       " 'molecular & cellular proteomics',\n",
       " 'mouse complete proteome',\n",
       " 'mouse complete proteome',\n",
       " 'mrcprt010 ultracel',\n",
       " 'multicomp package',\n",
       " 'multiple hit model',\n",
       " 'n=21',\n",
       " 'nce',\n",
       " 'neg',\n",
       " 'negative weight acetylation',\n",
       " 'nh3',\n",
       " 'non-resolved) protein band',\n",
       " 'o vial and 18o vial',\n",
       " 'par-1-mediated thrombin signal events',\n",
       " 'peptide',\n",
       " 'peptide preparation',\n",
       " 'phospho) proteome',\n",
       " 'phospho)proteome and 105 min',\n",
       " 'platelet degranulation',\n",
       " 'pngase f-catalyzed) conversion',\n",
       " 'poor outcome',\n",
       " 'precursor ion quantifier node',\n",
       " 'previously sequenced) prophage',\n",
       " 'proportion-of-total value',\n",
       " 'protein identification and network analysis methods section',\n",
       " 'proteinpilot mgf parameters',\n",
       " 'proteins',\n",
       " 'proteomic ruler method',\n",
       " 'proteomic ruler plugin',\n",
       " 'proteomic ruler strategy',\n",
       " 'pseudomature',\n",
       " 'psms',\n",
       " 'ql060112_c1b',\n",
       " 'ql060112_c2b',\n",
       " 'ql060112_g1',\n",
       " 'ql060112_g1b ql060112_g2',\n",
       " 'quantile normalization procedure',\n",
       " 'r',\n",
       " 'rank',\n",
       " 'rapid settings',\n",
       " 'raw) intensities',\n",
       " 're-quantify filter labeled amino acids',\n",
       " 'reporter ions quantifier node',\n",
       " 'requantify',\n",
       " 'requantify',\n",
       " 'resolution',\n",
       " 'reverse peptide',\n",
       " 'reverse proteins',\n",
       " 'saline',\n",
       " 'sample s1',\n",
       " 'sarco)endoplasmic reticulum',\n",
       " 'sartogosm',\n",
       " 'second peptide search option',\n",
       " 'sham) exposure',\n",
       " 'short rna',\n",
       " 'short rna improved rna stability',\n",
       " 'significance',\n",
       " 'significance threshold',\n",
       " 'sijimi longan',\n",
       " 'sixia',\n",
       " 'software lock mass’ option',\n",
       " 'spectrum exporter nodes',\n",
       " 'spectrum selector',\n",
       " 'stats package',\n",
       " 'switch',\n",
       " 'switch',\n",
       " 'system suitability  study',\n",
       " 'tat',\n",
       " 'tfa',\n",
       " 'thorough search effort',\n",
       " 'three to five vectors',\n",
       " 'tmt n-term',\n",
       " 'toxodb-26_ncaninumliv_annotatedproteins',\n",
       " 'treated silac ratios',\n",
       " 'uniprot-reviewed musmusculus',\n",
       " 'unique quantification results',\n",
       " 'unique quantitative results',\n",
       " 'use',\n",
       " 'v5111',\n",
       " 'variable modification',\n",
       " 'virtual western blots',\n",
       " 'vn01h23eto',\n",
       " 'vol',\n",
       " 'wat094225',\n",
       " 'water']"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tokens3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "regex4 = r'^.*\\)\\W*|\\(.*$'\n",
    "tokens4 = sorted([re.sub(regex4, '', s).strip() for s in tokens3])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '',\n",
       " '6',\n",
       " 'a/%b',\n",
       " 'ab initio',\n",
       " 'acetonitrile',\n",
       " 'acetonitrile',\n",
       " 'acetonitrile',\n",
       " 'acetonitrile',\n",
       " 'acetonitrile',\n",
       " 'acetonitrile',\n",
       " 'acn',\n",
       " 'acn',\n",
       " 'and  manassantin a samples',\n",
       " 'annotation matrix',\n",
       " 'aqueous 2% acetonitrile',\n",
       " 'b',\n",
       " 'b',\n",
       " 'b',\n",
       " 'b',\n",
       " 'background',\n",
       " 'bait',\n",
       " 'bimodalitytest package',\n",
       " 'c10416',\n",
       " 'cessation',\n",
       " 'cn',\n",
       " 'complete t. cruzi proteome',\n",
       " 'confluence',\n",
       " 'confluency',\n",
       " 'contaminants',\n",
       " 'control',\n",
       " 'control ratios',\n",
       " 'control sample',\n",
       " 'conversion',\n",
       " 'cor',\n",
       " 'core human pr',\n",
       " 'database',\n",
       " 'deep proteome',\n",
       " 'deuteration',\n",
       " 'dmso',\n",
       " 'dmso',\n",
       " 'earliest fraction',\n",
       " 'early risers',\n",
       " 'endoplasmic reticulum',\n",
       " 'es803',\n",
       " 'experiment',\n",
       " 'experimental design proteomexchange.txt',\n",
       " 'experiments',\n",
       " 'exposure',\n",
       " 'exposure',\n",
       " 'f4799',\n",
       " 'filter assisted sample preparation',\n",
       " 'former smoker group',\n",
       " 'formic acid',\n",
       " 'fractions',\n",
       " 'gel-top approach',\n",
       " 'gene name & annotations section',\n",
       " 'glit3t',\n",
       " 'global normalization',\n",
       " 'globin',\n",
       " 'good outcome',\n",
       " 'hartigan’s dip test',\n",
       " 'heavy amino-acids',\n",
       " 'heavy capture assay samples',\n",
       " 'heavy cells',\n",
       " 'heavy deuterated versions',\n",
       " 'heavy isotope-labeled amino acids',\n",
       " 'heavy isotope-labelled lysates',\n",
       " 'heavy lysine',\n",
       " 'heavy media',\n",
       " 'heavy media',\n",
       " 'heavy silac',\n",
       " 'i1149',\n",
       " 'ibaq',\n",
       " 'incorporation',\n",
       " 'input material',\n",
       " 'integrin family cell surface interactions',\n",
       " 'intensities',\n",
       " 'isopropanol',\n",
       " 'k molecular weight cut-off',\n",
       " 'ki-3003',\n",
       " 'large-scale targeted proteomics',\n",
       " 'last-resort antibiotic',\n",
       " 'lc6025',\n",
       " 'lfq',\n",
       " 'light',\n",
       " 'light',\n",
       " 'light',\n",
       " 'light l-arginine and l-lysine',\n",
       " 'light media',\n",
       " 'light silac media',\n",
       " 'lock mass function',\n",
       " 'lock mass function',\n",
       " 'ls',\n",
       " 'm urea',\n",
       " 'manassantin a sample',\n",
       " 'manassantin a samplewas',\n",
       " 'match',\n",
       " 'match',\n",
       " 'may',\n",
       " 'medium-heavy and heavy cells',\n",
       " 'methylated hla class',\n",
       " 'missing proteins',\n",
       " 'mixo',\n",
       " 'mm ammonium formate',\n",
       " 'mm ammonium formate',\n",
       " 'mm ammonium hydroxide',\n",
       " 'mm k2hpo4',\n",
       " 'mm kh2po4 ph 2.65 30% acn',\n",
       " 'mm teab ph8',\n",
       " 'moi',\n",
       " 'molecular & cellular proteomics',\n",
       " 'mouse complete proteome',\n",
       " 'mouse complete proteome',\n",
       " 'mrcprt010 ultracel',\n",
       " 'multicomp package',\n",
       " 'multiple hit model',\n",
       " 'n=21',\n",
       " 'nce',\n",
       " 'neg',\n",
       " 'negative weight acetylation',\n",
       " 'nh3',\n",
       " 'o vial and 18o vial',\n",
       " 'par-1-mediated thrombin signal events',\n",
       " 'peptide',\n",
       " 'peptide preparation',\n",
       " 'platelet degranulation',\n",
       " 'poor outcome',\n",
       " 'precursor ion quantifier node',\n",
       " 'prophage',\n",
       " 'proportion-of-total value',\n",
       " 'protein band',\n",
       " 'protein identification and network analysis methods section',\n",
       " 'proteinpilot mgf parameters',\n",
       " 'proteins',\n",
       " 'proteome',\n",
       " 'proteome',\n",
       " 'proteome and 105 min',\n",
       " 'proteomic data',\n",
       " 'proteomic ruler method',\n",
       " 'proteomic ruler plugin',\n",
       " 'proteomic ruler strategy',\n",
       " 'pseudomature',\n",
       " 'psms',\n",
       " 'ql060112_c1b',\n",
       " 'ql060112_c2b',\n",
       " 'ql060112_g1',\n",
       " 'ql060112_g1b ql060112_g2',\n",
       " 'quantile normalization procedure',\n",
       " 'r',\n",
       " 'rank',\n",
       " 'rapid settings',\n",
       " 're-quantify filter labeled amino acids',\n",
       " 'reporter ions quantifier node',\n",
       " 'requantify',\n",
       " 'requantify',\n",
       " 'resolution',\n",
       " 'reverse peptide',\n",
       " 'reverse proteins',\n",
       " 'saline',\n",
       " 'sample s1',\n",
       " 'sartogosm',\n",
       " 'second peptide search option',\n",
       " 'short rna',\n",
       " 'short rna improved rna stability',\n",
       " 'significance',\n",
       " 'significance threshold',\n",
       " 'sijimi longan',\n",
       " 'silac-labeled wild type yeast cells',\n",
       " 'silac-labeled wild type yeast cells',\n",
       " 'sixia',\n",
       " 'software lock mass’ option',\n",
       " 'spectrum exporter nodes',\n",
       " 'spectrum selector',\n",
       " 'stats package',\n",
       " 'switch',\n",
       " 'switch',\n",
       " 'system suitability  study',\n",
       " 'tat',\n",
       " 'tfa',\n",
       " 'thorough search effort',\n",
       " 'three to five vectors',\n",
       " 'tmt n-term',\n",
       " 'toxodb-26_ncaninumliv_annotatedproteins',\n",
       " 'treated silac ratios',\n",
       " 'uniprot-reviewed musmusculus',\n",
       " 'unique quantification results',\n",
       " 'unique quantitative results',\n",
       " 'use',\n",
       " 'v5111',\n",
       " 'variable modification',\n",
       " 'virtual western blots',\n",
       " 'vn01h23eto',\n",
       " 'vol',\n",
       " 'wat094225',\n",
       " 'water']"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tokens4"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "regex5 = r'^.$|.*=.*|^\\w |^mm '\n",
    "tokens5 = sorted([re.sub(regex5, '', s) for s in tokens4])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['a/%b',\n",
       " 'ab initio',\n",
       " 'acetonitrile',\n",
       " 'acetonitrile',\n",
       " 'acetonitrile',\n",
       " 'acetonitrile',\n",
       " 'acetonitrile',\n",
       " 'acetonitrile',\n",
       " 'acn',\n",
       " 'acn',\n",
       " 'ammonium formate',\n",
       " 'ammonium formate',\n",
       " 'ammonium hydroxide',\n",
       " 'and  manassantin a samples',\n",
       " 'annotation matrix',\n",
       " 'aqueous 2% acetonitrile',\n",
       " 'background',\n",
       " 'bait',\n",
       " 'bimodalitytest package',\n",
       " 'c10416',\n",
       " 'cessation',\n",
       " 'cn',\n",
       " 'complete t. cruzi proteome',\n",
       " 'confluence',\n",
       " 'confluency',\n",
       " 'contaminants',\n",
       " 'control',\n",
       " 'control ratios',\n",
       " 'control sample',\n",
       " 'conversion',\n",
       " 'cor',\n",
       " 'core human pr',\n",
       " 'database',\n",
       " 'deep proteome',\n",
       " 'deuteration',\n",
       " 'dmso',\n",
       " 'dmso',\n",
       " 'earliest fraction',\n",
       " 'early risers',\n",
       " 'endoplasmic reticulum',\n",
       " 'es803',\n",
       " 'experiment',\n",
       " 'experimental design proteomexchange.txt',\n",
       " 'experiments',\n",
       " 'exposure',\n",
       " 'exposure',\n",
       " 'f4799',\n",
       " 'filter assisted sample preparation',\n",
       " 'former smoker group',\n",
       " 'formic acid',\n",
       " 'fractions',\n",
       " 'gel-top approach',\n",
       " 'gene name & annotations section',\n",
       " 'glit3t',\n",
       " 'global normalization',\n",
       " 'globin',\n",
       " 'good outcome',\n",
       " 'hartigan’s dip test',\n",
       " 'heavy amino-acids',\n",
       " 'heavy capture assay samples',\n",
       " 'heavy cells',\n",
       " 'heavy deuterated versions',\n",
       " 'heavy isotope-labeled amino acids',\n",
       " 'heavy isotope-labelled lysates',\n",
       " 'heavy lysine',\n",
       " 'heavy media',\n",
       " 'heavy media',\n",
       " 'heavy silac',\n",
       " 'i1149',\n",
       " 'ibaq',\n",
       " 'incorporation',\n",
       " 'input material',\n",
       " 'integrin family cell surface interactions',\n",
       " 'intensities',\n",
       " 'isopropanol',\n",
       " 'k2hpo4',\n",
       " 'kh2po4 ph 2.65 30% acn',\n",
       " 'ki-3003',\n",
       " 'large-scale targeted proteomics',\n",
       " 'last-resort antibiotic',\n",
       " 'lc6025',\n",
       " 'lfq',\n",
       " 'light',\n",
       " 'light',\n",
       " 'light',\n",
       " 'light l-arginine and l-lysine',\n",
       " 'light media',\n",
       " 'light silac media',\n",
       " 'lock mass function',\n",
       " 'lock mass function',\n",
       " 'ls',\n",
       " 'manassantin a sample',\n",
       " 'manassantin a samplewas',\n",
       " 'match',\n",
       " 'match',\n",
       " 'may',\n",
       " 'medium-heavy and heavy cells',\n",
       " 'methylated hla class',\n",
       " 'missing proteins',\n",
       " 'mixo',\n",
       " 'moi',\n",
       " 'molecular & cellular proteomics',\n",
       " 'molecular weight cut-off',\n",
       " 'mouse complete proteome',\n",
       " 'mouse complete proteome',\n",
       " 'mrcprt010 ultracel',\n",
       " 'multicomp package',\n",
       " 'multiple hit model',\n",
       " 'nce',\n",
       " 'neg',\n",
       " 'negative weight acetylation',\n",
       " 'nh3',\n",
       " 'par-1-mediated thrombin signal events',\n",
       " 'peptide',\n",
       " 'peptide preparation',\n",
       " 'platelet degranulation',\n",
       " 'poor outcome',\n",
       " 'precursor ion quantifier node',\n",
       " 'prophage',\n",
       " 'proportion-of-total value',\n",
       " 'protein band',\n",
       " 'protein identification and network analysis methods section',\n",
       " 'proteinpilot mgf parameters',\n",
       " 'proteins',\n",
       " 'proteome',\n",
       " 'proteome',\n",
       " 'proteome and 105 min',\n",
       " 'proteomic data',\n",
       " 'proteomic ruler method',\n",
       " 'proteomic ruler plugin',\n",
       " 'proteomic ruler strategy',\n",
       " 'pseudomature',\n",
       " 'psms',\n",
       " 'ql060112_c1b',\n",
       " 'ql060112_c2b',\n",
       " 'ql060112_g1',\n",
       " 'ql060112_g1b ql060112_g2',\n",
       " 'quantile normalization procedure',\n",
       " 'rank',\n",
       " 'rapid settings',\n",
       " 're-quantify filter labeled amino acids',\n",
       " 'reporter ions quantifier node',\n",
       " 'requantify',\n",
       " 'requantify',\n",
       " 'resolution',\n",
       " 'reverse peptide',\n",
       " 'reverse proteins',\n",
       " 'saline',\n",
       " 'sample s1',\n",
       " 'sartogosm',\n",
       " 'second peptide search option',\n",
       " 'short rna',\n",
       " 'short rna improved rna stability',\n",
       " 'significance',\n",
       " 'significance threshold',\n",
       " 'sijimi longan',\n",
       " 'silac-labeled wild type yeast cells',\n",
       " 'silac-labeled wild type yeast cells',\n",
       " 'sixia',\n",
       " 'software lock mass’ option',\n",
       " 'spectrum exporter nodes',\n",
       " 'spectrum selector',\n",
       " 'stats package',\n",
       " 'switch',\n",
       " 'switch',\n",
       " 'system suitability  study',\n",
       " 'tat',\n",
       " 'teab ph8',\n",
       " 'tfa',\n",
       " 'thorough search effort',\n",
       " 'three to five vectors',\n",
       " 'tmt n-term',\n",
       " 'toxodb-26_ncaninumliv_annotatedproteins',\n",
       " 'treated silac ratios',\n",
       " 'uniprot-reviewed musmusculus',\n",
       " 'unique quantification results',\n",
       " 'unique quantitative results',\n",
       " 'urea',\n",
       " 'use',\n",
       " 'v5111',\n",
       " 'variable modification',\n",
       " 'vial and 18o vial',\n",
       " 'virtual western blots',\n",
       " 'vn01h23eto',\n",
       " 'vol',\n",
       " 'wat094225',\n",
       " 'water']"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "[s for s in tokens5 if s != '']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "def extract_noun_chuncks2(tokens):\n",
    "    # Remove quote marks and closed brackets and clauses in half brackets\n",
    "    regex1 = r'^[^A-Za-z]+|\\'|\\\"|\\(.+\\)'\n",
    "    tokens = [re.sub(regex1, '', s).strip() for s in tokens]\n",
    "    \n",
    "    # Remove half open brackets\n",
    "    regex2 = r'^.*\\)\\W*|\\(.*$'\n",
    "    tokens = [re.sub(regex2, '', s).strip() for s in tokens]\n",
    "    \n",
    "    # Remove one letter words, ones that start with one latter and a space,\n",
    "    # and reminiscence of mM\n",
    "    regex3 = r'^.$|.*=.*|^\\w |^mm '\n",
    "    tokens = [re.sub(regex3, '', s).strip() for s in tokens]\n",
    "    \n",
    "    tokens = [s for s in tokens if s != '']\n",
    "    \n",
    "    return tokens"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['cn',\n",
       " 'molecular & cellular proteomics',\n",
       " 'cessation',\n",
       " 'heavy media',\n",
       " 'light',\n",
       " 'lock mass function',\n",
       " 'mouse complete proteome',\n",
       " 'switch',\n",
       " 'glit3t',\n",
       " 'ab initio',\n",
       " 'annotation matrix',\n",
       " 'background',\n",
       " 'bait',\n",
       " 'complete t. cruzi proteome',\n",
       " 'database',\n",
       " 'deep proteome',\n",
       " 'experimental design proteomexchange.txt',\n",
       " 'experiments',\n",
       " 'filter assisted sample preparation',\n",
       " 'gene name & annotations section',\n",
       " 'global normalization',\n",
       " 'good outcome',\n",
       " 'heavy amino-acids',\n",
       " 'heavy capture assay samples',\n",
       " 'heavy deuterated versions',\n",
       " 'heavy isotope-labeled amino acids',\n",
       " 'heavy media',\n",
       " 'input material',\n",
       " 'large-scale targeted proteomics',\n",
       " 'light',\n",
       " 'light l-arginine and l-lysine',\n",
       " 'light media',\n",
       " 'light silac media',\n",
       " 'lock mass function',\n",
       " 'match',\n",
       " 'medium-heavy and heavy cells',\n",
       " 'mixo',\n",
       " 'mouse complete proteome',\n",
       " 'multicomp package',\n",
       " 'neg',\n",
       " 'negative weight acetylation',\n",
       " 'poor outcome',\n",
       " 'precursor ion quantifier node',\n",
       " 'protein identification and network analysis methods section',\n",
       " 'proteinpilot mgf parameters',\n",
       " 'proteomic ruler method',\n",
       " 'pseudomature',\n",
       " 'ql060112_c1b',\n",
       " 'ql060112_c2b',\n",
       " 'ql060112_g1',\n",
       " 'ql060112_g1b ql060112_g2',\n",
       " 'quantile normalization procedure',\n",
       " 'rank',\n",
       " 'rapid settings',\n",
       " 're-quantify filter labeled amino acids',\n",
       " 'reporter ions quantifier node',\n",
       " 'requantify',\n",
       " 'sartogosm',\n",
       " 'significance threshold',\n",
       " 'sijimi longan',\n",
       " 'sixia',\n",
       " 'spectrum exporter nodes',\n",
       " 'spectrum selector',\n",
       " 'switch',\n",
       " 'system suitability  study',\n",
       " 'thorough search effort',\n",
       " 'three to five vectors',\n",
       " 'toxodb-26_ncaninumliv_annotatedproteins',\n",
       " 'uniprot-reviewed musmusculus',\n",
       " 'unique quantitative results',\n",
       " 'use',\n",
       " 'variable modification',\n",
       " 'i1149',\n",
       " 'ki-3003',\n",
       " 'lc6025',\n",
       " 'mrcprt010 ultracel',\n",
       " 'v5111',\n",
       " 'vn01h23eto',\n",
       " 'wat094225',\n",
       " 'may',\n",
       " 'peptide preparation',\n",
       " 'proteins',\n",
       " 'c10416',\n",
       " 'es803',\n",
       " 'f4799',\n",
       " 'psms',\n",
       " 'reverse peptide',\n",
       " 'reverse proteins',\n",
       " 'acetonitrile',\n",
       " 'acn',\n",
       " 'confluence',\n",
       " 'confluency',\n",
       " 'deuteration',\n",
       " 'formic acid',\n",
       " 'globin',\n",
       " 'incorporation',\n",
       " 'isopropanol',\n",
       " 'nh3',\n",
       " 'tfa',\n",
       " 'a/%b',\n",
       " 'vol',\n",
       " 'bimodalitytest package',\n",
       " 'contaminants',\n",
       " 'control ratios',\n",
       " 'cor',\n",
       " 'core human pr',\n",
       " 'earliest fraction',\n",
       " 'early risers',\n",
       " 'gel-top approach',\n",
       " 'hartigan’s dip test',\n",
       " 'heavy cells',\n",
       " 'heavy isotope-labelled lysates',\n",
       " 'heavy lysine',\n",
       " 'heavy silac',\n",
       " 'ibaq',\n",
       " 'integrin family cell surface interactions',\n",
       " 'last-resort antibiotic',\n",
       " 'lfq',\n",
       " 'light',\n",
       " 'ls',\n",
       " 'match',\n",
       " 'missing proteins',\n",
       " 'multiple hit model',\n",
       " 'par-1-mediated thrombin signal events',\n",
       " 'platelet degranulation',\n",
       " 'proportion-of-total value',\n",
       " 'proteomic ruler plugin',\n",
       " 'proteomic ruler strategy',\n",
       " 'requantify',\n",
       " 'second peptide search option',\n",
       " 'short rna',\n",
       " 'short rna improved rna stability',\n",
       " 'significance',\n",
       " 'software lock mass’ option',\n",
       " 'stats package',\n",
       " 'tat',\n",
       " 'tmt n-term',\n",
       " 'treated silac ratios',\n",
       " 'unique quantification results',\n",
       " 'virtual western blots',\n",
       " 'and  manassantin a samples',\n",
       " 'manassantin a sample',\n",
       " 'manassantin a samplewas',\n",
       " 'control',\n",
       " 'control sample',\n",
       " 'ammonium formate',\n",
       " 'ammonium hydroxide',\n",
       " 'teab ph8',\n",
       " 'vial and 18o vial',\n",
       " 'peptide',\n",
       " 'acetonitrile',\n",
       " 'acn',\n",
       " 'ammonium formate',\n",
       " 'molecular weight cut-off',\n",
       " 'nce',\n",
       " 'k2hpo4',\n",
       " 'acetonitrile',\n",
       " 'dmso',\n",
       " 'kh2po4 ph 2.65 30% acn',\n",
       " 'fractions',\n",
       " 'urea',\n",
       " 'acetonitrile',\n",
       " 'acetonitrile',\n",
       " 'water',\n",
       " 'acetonitrile',\n",
       " 'aqueous 2% acetonitrile',\n",
       " 'exposure',\n",
       " 'methylated hla class',\n",
       " 'dmso',\n",
       " 'experiment',\n",
       " 'silac-labeled wild type yeast cells',\n",
       " 'silac-labeled wild type yeast cells',\n",
       " 'former smoker group',\n",
       " 'proteomic data',\n",
       " 'proteome',\n",
       " 'moi',\n",
       " 'protein band',\n",
       " 'proteome',\n",
       " 'proteome and 105 min',\n",
       " 'conversion',\n",
       " 'prophage',\n",
       " 'intensities',\n",
       " 'resolution',\n",
       " 'saline',\n",
       " 'sample s1',\n",
       " 'endoplasmic reticulum',\n",
       " 'exposure']"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "extract_noun_chuncks2(tokens[:200])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing sample_protocol...\n",
      "Processing data_protocol...\n",
      "Processing description...\n",
      "CPU times: user 1.61 s, sys: 11.5 ms, total: 1.62 s\n",
      "Wall time: 1.62 s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "with open('nlp15_data/dfs/all_fields_nchunks_df_quant_dummies.pickle', 'rb') as infile_df:\n",
    "    df = pickle.load(infile_df)\n",
    "    \n",
    "# extrac noun_chunks from the text\n",
    "if True:\n",
    "    print('Processing sample_protocol...')\n",
    "    df.loc[:, 'sample_protocol'] = df.loc[:, 'sample_protocol'].apply(lambda x: extract_noun_chuncks2(x))\n",
    "    \n",
    "    print('Processing data_protocol...')\n",
    "    df.loc[:, 'data_protocol'] = df.loc[:, 'data_protocol'].apply(lambda x: extract_noun_chuncks2(x))\n",
    "    \n",
    "    print('Processing description...')\n",
    "    df.loc[:, 'description'] = df.loc[:, 'description'].apply(lambda x: extract_noun_chuncks2(x))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>dataset_id</th>\n",
       "      <th>sample_protocol</th>\n",
       "      <th>data_protocol</th>\n",
       "      <th>description</th>\n",
       "      <th>instruments</th>\n",
       "      <th>exp_types</th>\n",
       "      <th>quant_methods</th>\n",
       "      <th>labhead_fullname</th>\n",
       "      <th>silac</th>\n",
       "      <th>ms1_label_free</th>\n",
       "      <th>spectrum_counting</th>\n",
       "      <th>tmt</th>\n",
       "      <th>itraq</th>\n",
       "      <th>label_free</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PXD000029</td>\n",
       "      <td>[breast cancer tissue lysates, reduction, alky...</td>\n",
       "      <td>[proteomics data, proteome discoverer, fdr&lt;0.0...</td>\n",
       "      <td>[current prognostic factors, precise risk-disc...</td>\n",
       "      <td>LTQ Orbitrap Velos</td>\n",
       "      <td>Shotgun proteomics</td>\n",
       "      <td>itraq</td>\n",
       "      <td>Pavel Bouchal</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PXD000164</td>\n",
       "      <td>[protein extraction, catheter biofilm small pi...</td>\n",
       "      <td>[tryptic digest, reversed phase, chromatograph...</td>\n",
       "      <td>[term-catheterization, catheter-associated bac...</td>\n",
       "      <td>LTQ Orbitrap Velos</td>\n",
       "      <td>Shotgun proteomics</td>\n",
       "      <td>label free</td>\n",
       "      <td>Katharina Riedel</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PXD000175</td>\n",
       "      <td>[generation, mdia2-based immunocomplexes and m...</td>\n",
       "      <td>[ms data, peak lists, raw data files, proteome...</td>\n",
       "      <td>[mdia2, auto-inhibited formin, actin dynamics,...</td>\n",
       "      <td>LTQ Orbitrap</td>\n",
       "      <td>Affinity purification coupled with mass spectr...</td>\n",
       "      <td>spectrum counting</td>\n",
       "      <td>Maarten Altelaar</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PXD000376</td>\n",
       "      <td>[naïve cd4+ t cells, magnetic beads, manufactu...</td>\n",
       "      <td>[ms data, proteome discoverer, thermo fisher s...</td>\n",
       "      <td>[technological advances, genomics, epigenomics...</td>\n",
       "      <td>LTQ Orbitrap Velos, LTQ Orbitrap Elite</td>\n",
       "      <td>Shotgun proteomics</td>\n",
       "      <td>itraq</td>\n",
       "      <td>Akhilesh Pandey</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PXD000397</td>\n",
       "      <td>[cells, hours, serum, free dmem, phenol red, e...</td>\n",
       "      <td>[raw data, maxquant, results, scripts, house, ...</td>\n",
       "      <td>[project, expression patterns, hypoxia, secret...</td>\n",
       "      <td>Q Exactive</td>\n",
       "      <td>Shotgun proteomics</td>\n",
       "      <td>silac</td>\n",
       "      <td>Janine T Erler</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  dataset_id                                    sample_protocol  \\\n",
       "0  PXD000029  [breast cancer tissue lysates, reduction, alky...   \n",
       "1  PXD000164  [protein extraction, catheter biofilm small pi...   \n",
       "2  PXD000175  [generation, mdia2-based immunocomplexes and m...   \n",
       "3  PXD000376  [naïve cd4+ t cells, magnetic beads, manufactu...   \n",
       "4  PXD000397  [cells, hours, serum, free dmem, phenol red, e...   \n",
       "\n",
       "                                       data_protocol  \\\n",
       "0  [proteomics data, proteome discoverer, fdr<0.0...   \n",
       "1  [tryptic digest, reversed phase, chromatograph...   \n",
       "2  [ms data, peak lists, raw data files, proteome...   \n",
       "3  [ms data, proteome discoverer, thermo fisher s...   \n",
       "4  [raw data, maxquant, results, scripts, house, ...   \n",
       "\n",
       "                                         description  \\\n",
       "0  [current prognostic factors, precise risk-disc...   \n",
       "1  [term-catheterization, catheter-associated bac...   \n",
       "2  [mdia2, auto-inhibited formin, actin dynamics,...   \n",
       "3  [technological advances, genomics, epigenomics...   \n",
       "4  [project, expression patterns, hypoxia, secret...   \n",
       "\n",
       "                              instruments  \\\n",
       "0                      LTQ Orbitrap Velos   \n",
       "1                      LTQ Orbitrap Velos   \n",
       "2                            LTQ Orbitrap   \n",
       "3  LTQ Orbitrap Velos, LTQ Orbitrap Elite   \n",
       "4                              Q Exactive   \n",
       "\n",
       "                                           exp_types      quant_methods  \\\n",
       "0                                 Shotgun proteomics              itraq   \n",
       "1                                 Shotgun proteomics         label free   \n",
       "2  Affinity purification coupled with mass spectr...  spectrum counting   \n",
       "3                                 Shotgun proteomics              itraq   \n",
       "4                                 Shotgun proteomics              silac   \n",
       "\n",
       "   labhead_fullname  silac  ms1_label_free  spectrum_counting  tmt  itraq  \\\n",
       "0     Pavel Bouchal      0               0                  0    0      1   \n",
       "1  Katharina Riedel      0               0                  0    0      0   \n",
       "2  Maarten Altelaar      0               0                  1    0      0   \n",
       "3   Akhilesh Pandey      0               0                  0    0      1   \n",
       "4    Janine T Erler      1               0                  0    0      0   \n",
       "\n",
       "   label_free  \n",
       "0           0  \n",
       "1           1  \n",
       "2           0  \n",
       "3           0  \n",
       "4           0  "
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['breast cancer tissue lysates',\n",
       " 'reduction',\n",
       " 'alkylation',\n",
       " 'methyl-methanethiosulfonate trypsin digestion',\n",
       " 'plex labeling',\n",
       " 'hilic fractionation',\n",
       " 'lc-ms/ms',\n",
       " 'orbitrap velos',\n",
       " 'transcriptomics',\n",
       " 'immunohistochemistry',\n",
       " 'detailed description',\n",
       " 'manuscript']"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.sample_protocol.iloc[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Serialize and save the processed DF\n",
    "if True:\n",
    "    with open('nlp15_data/dfs/all_fields_nchunks_df_quant_dummies2.pickle', 'wb') as out_df:\n",
    "        pickle.dump(df, out_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save the corpora\n",
    "# This time I'll only use combined protocols and that combined with description field (i.e. two corpora) as in nlp13\n",
    "if True:\n",
    "    # Protocols combined\n",
    "    protocols_corpus = list(df.sample_protocol + df.data_protocol)\n",
    "    with open('nlp15_data/corpora/protocols_corpus_nchunks2.pickle', 'wb') as outfile:\n",
    "        pickle.dump(protocols_corpus, outfile)\n",
    "\n",
    "    # All combined\n",
    "    whole_corpus = list(df.sample_protocol + df.data_protocol + df.description)\n",
    "    with open('nlp15_data/corpora/whole_corpus_nchunks2.pickle', 'wb') as outfile:\n",
    "        pickle.dump(whole_corpus, outfile)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create and save dictionary using whole_corpus\n",
    "if True:\n",
    "    my_dictionary = corpora.Dictionary(whole_corpus)\n",
    "    my_dictionary.save('nlp15_data/whole_dictionary_nchunks2.dict')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save BoW and Tf-Idf\n",
    "if True:\n",
    "    # BoW transformations and save\n",
    "    protocols_bow = [my_dictionary.doc2bow(text) for text in protocols_corpus]\n",
    "    whole_bow = [my_dictionary.doc2bow(text) for text in whole_corpus]\n",
    "    corpora.MmCorpus.serialize('nlp15_data/bow_and_tfidf/protocols_bow_nchunks2.mm', protocols_bow)\n",
    "    corpora.MmCorpus.serialize('nlp15_data/bow_and_tfidf/whole_bow_nchunks2.mm', whole_bow)\n",
    "    \n",
    "    # Tf-Idf transformations  and save\n",
    "    protocols_tfidf_model = models.TfidfModel(protocols_bow)\n",
    "    protocols_tfidf = protocols_tfidf_model[protocols_bow]\n",
    "    corpora.MmCorpus.serialize('nlp15_data/bow_and_tfidf/protocols_tfidf_nchunks2.mm', protocols_tfidf)\n",
    "    \n",
    "    whole_tfidf_model = models.TfidfModel(whole_bow)\n",
    "    whole_tfidf = whole_tfidf_model[whole_bow]\n",
    "    corpora.MmCorpus.serialize('nlp15_data/bow_and_tfidf/whole_tfidf_nchunks2.mm', whole_tfidf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
