{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import re\n",
    "import numpy\n",
    "import pickle"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "df =  pd.read_csv('base_data/pride_table.csv').astype(str)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>dataset_id</th>\n",
       "      <th>sample_protocol</th>\n",
       "      <th>data_protocol</th>\n",
       "      <th>description</th>\n",
       "      <th>instruments</th>\n",
       "      <th>exp_types</th>\n",
       "      <th>quant_methods</th>\n",
       "      <th>labhead_fullname</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PXD000001</td>\n",
       "      <td>Not available</td>\n",
       "      <td>Two extra files have been added post-publicati...</td>\n",
       "      <td>Expected reporter ion ratios: Erwinia peptides...</td>\n",
       "      <td>LTQ Orbitrap Velos, instrument model</td>\n",
       "      <td>Bottom-up proteomics</td>\n",
       "      <td>nan</td>\n",
       "      <td>nan</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PXD000002</td>\n",
       "      <td>Not available</td>\n",
       "      <td>Not available</td>\n",
       "      <td>Human saliva samples from a 36 year old male w...</td>\n",
       "      <td>instrument model, LTQ Velos</td>\n",
       "      <td>Bottom-up proteomics</td>\n",
       "      <td>nan</td>\n",
       "      <td>nan</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  dataset_id sample_protocol  \\\n",
       "0  PXD000001   Not available   \n",
       "1  PXD000002   Not available   \n",
       "\n",
       "                                       data_protocol  \\\n",
       "0  Two extra files have been added post-publicati...   \n",
       "1                                      Not available   \n",
       "\n",
       "                                         description  \\\n",
       "0  Expected reporter ion ratios: Erwinia peptides...   \n",
       "1  Human saliva samples from a 36 year old male w...   \n",
       "\n",
       "                            instruments             exp_types quant_methods  \\\n",
       "0  LTQ Orbitrap Velos, instrument model  Bottom-up proteomics           nan   \n",
       "1           instrument model, LTQ Velos  Bottom-up proteomics           nan   \n",
       "\n",
       "  labhead_fullname  \n",
       "0              nan  \n",
       "1              nan  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "method = df['sample_protocol'][(df['sample_protocol'] != 'Not available') & (df['sample_protocol'] != 'nan')].reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Conditioned media preparation and HPLC-isobaric tags for relative and absolute quantitation mass-spectrometry (iTRAQ MS). Twenty million cells were transfected with either control or Gαs siRNA, then were seeded in 100-mm cell culture plates in DMEM + 10% FBS. Two days later, cells were washed twice with pre-warmed PBS. Conditioned media was harvested another 24 h later, and centrifuged at 20,000 g for 10 min at 4°C to remove cell debris. To enrich for secreted proteins in the conditioned media, conditioned media samples were centrifuged using Amicon centrifugal filters with a 3kDa cutoff (Millipore, Billerica, MA). A total of 8 concentrated conditioned media samples were independently prepared: 4 samples from control siRNA-treated cells, and 4 samples from Gαs siRNA-treated cells. Samples were stored at –70°C until they were analyzed by iTRAQ labeling and LC-MS/MS. iTRAQ labeling – Total protein concentration of each sample was measured using the Bradford assay. A quantity of 100 µg was taken from each sample, denatured in 0.1% SDS, reduced in 50 mM Tris(2-carboxyethyl)phosphine (TCEP), alkylated in 200 mM methyl methanethiosulfonate (MMTS), and digested overnight with trypsin. Each sample’s peptides were labeled with different iTRAQ reagents (AB Sciex, Framingham, MA) following the manufacturer’s protocol, combined, dried down to a 50 µL volume, acidified with 1% trifluoroacetic acid (TFA), and desalted with a 100 mg-C18 SepPak cartridge (Waters, Milford, MA). Peptide fractionation – Combined iTRAQ labeled peptides were first separated into 12 fractions using a 3100 OFFGEL Fractionator (Agilent, Santa Clara, CA) with a 12-well setup and a 12-cm pH 3–10 immobilized pH-gradient (IPG) strip, according to manufacturer’s recommendations. Peptides were collected after the focusing voltage reached a maximum plateau (> 50 kVh). The recovered fractions were lyophilized. LC-MALDI-TOF/TOF analysis – All solvents used in this section were aqueous unless specified and obtained from ThermoFisher Scientific (San Jose, CA). Dried samples were reconstituted in 30 µL of solvent A (2% v/v acetonitrile [ACN], 0.1% v/v TFA), and 10 µL of the resulting solution was injected into an Ultimate™ HPLC system with UV detection (LC Packings, Sunnyvale, CA). Solvent B contained 98% v/v ACN, 0.1% v/v TFA. A C18 trap (5.0 mm x 300 µm ID, LC Packings) and a 15 cm x 100 µm ID column packed in-house with Magic C18 5 µm beads (Michrom Bioresources, Auburn, CA) were used for chromatography. Solvent flow rate was set to 0.700 µL/min. After injection, each sample was loaded and washed with solvent A for 5 min at a flow rate of 0.030 mL/min, followed by a 10% to 50% gradient over 55 min, then ramped to 90% by 65 min. Each sample run was maintained in 90% solvent B until 75 min, returned to 10% at 75.1 min, followed by a 10 min equilibration in 10% solvent B. The HPLC system is connected to a modified Probot™ automated MALDI spotter (LC Packings) to spot eluates on an LC MALDI plate with on-line mixing of α-Cyano-4-hydroxycinnamic acid (CHCA) matrix (7.5 mg/mL in 70% ACN). Considering the dead time for the HPLC and spotting system, spot collection time was started 10 min from injection time with a spotting time of 0.33 min per spot. A total of 176 spots were collected. The sample plate with the resulting spots was loaded onto a 5800 MALDI time-of-flight/time-of-flight (TOF/TOF) mass spectrometer (AB Sciex, Framingham, MA) for MS and MS/MS analysis. Batch-mode mass calibration with plate alignment were done before data acquisition, thus a separate plate calibration file was generated for each individual plate containing sample spots. Full scan MS spectra were obtained with a total of 1000 laser shots per spectrum. From each spot, the top 20 most intense peaks, weakest precursor first, that were above a signal-to-noise threshold of 10, were selected for successful MS/MS analysis. Each MS/MS spectrum was an average of 1600 laser shots. MS/MS spectra were acquired in the presence of collision gas. The sequence of the VGF-NERP1 peptide was confirmed using the LTQ-Orbitrap mass spectrometer (ThermoFisher Scientific, San Jose, CA).'"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "method.iloc[3]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10%\n",
      "0.1%\n",
      "50 mM\n",
      "200 mM\n",
      "1%\n",
      "2%\n",
      "0.1%\n",
      "98%\n",
      "0.1%\n",
      "10%\n",
      "50%\n",
      "90%\n",
      "90%\n",
      "10%\n",
      "10%\n",
      "70%\n"
     ]
    }
   ],
   "source": [
    "# finding concentrations\n",
    "regex1 = r'((?:0\\.)?(?:\\d{1,4}) ?(?:mM|µM|nM|%))'\n",
    "text = method.iloc[3]\n",
    "res = [s for s in re.findall(regex1, text)]\n",
    "for conc in res:\n",
    "    print(conc)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10% FBS\n",
      "0.1% SDS\n",
      "50 mM Tris\n",
      "200 mM methyl\n",
      "1% trifluoroacetic\n",
      "2% v\n",
      "0.1% v\n",
      "98% v\n",
      "0.1% v\n",
      "10% to\n",
      "50% gradient\n",
      "90% by\n",
      "90% solvent\n",
      "10% at\n",
      "10% solvent\n",
      "70% ACN\n"
     ]
    }
   ],
   "source": [
    "# finding buffers1\n",
    "regex2 = r'((?:0\\.)?(?:\\d{1,4}) ?(?:mM|µM|nM|%) [A-Za-z]+)'\n",
    "text = method.iloc[3]\n",
    "res = [s for s in re.findall(regex2, text)]\n",
    "for buffer in res:\n",
    "    print(buffer)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10% FBS\n",
      "0.1% SDS\n",
      "50 mM Tris\n",
      "200 mM methyl\n",
      "1% trifluoroacetic\n",
      "2% v/v acetonitrile\n",
      "0.1% v/v TFA\n",
      "98% v/v ACN\n",
      "0.1% v/v TFA\n",
      "10% to\n",
      "50% gradient\n",
      "90% by\n",
      "90% solvent\n",
      "10% at\n",
      "10% solvent\n",
      "70% ACN\n"
     ]
    }
   ],
   "source": [
    "# finding buffers2: add w/v, v/v\n",
    "regex2 = r'((?:0\\.)?(?:\\d{1,4}) ?(?:mM|µM|nM|%) \\(?(?:w/v |v/v )?\\)?[A-Za-z]+)'\n",
    "text = method.iloc[3]\n",
    "res = [s for s in re.findall(regex2, text)]\n",
    "for buffer in res:\n",
    "    print(buffer)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Except '10% at' it seems to work!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10% Bis\n",
      "5% formic\n",
      "50% acetonitrile\n",
      "0.1% trifluoroacetic\n",
      "40% acetonitrile\n",
      "0.1% formic\n",
      "0.67% min\n"
     ]
    }
   ],
   "source": [
    "# Test on another text\n",
    "regex2 = r'((?:0\\.)?(?:\\d{1,4}) ?(?:mM|µM|nM|%) \\(?(?:w/v |v/v )?\\)?[A-Za-z]+)'\n",
    "text = method.iloc[4]\n",
    "res = [s for s in re.findall(regex2, text)]\n",
    "for buffer in res:\n",
    "    print(buffer)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "===== for text 0 =====\n",
      "2% deoxycholate\n",
      "0.37% w/v sodium\n",
      "50% v/v acetonitrile\n",
      "2% v/v formic\n",
      "2% v/v acetonitrile\n",
      "===== for text 1 =====\n",
      "===== for text 2 =====\n",
      "10 mM DTT\n",
      "55 mM iodoacetamide\n",
      "100% acetonitrile\n",
      "40 mM ammoniumbicarbonate\n",
      "===== for text 3 =====\n",
      "10% FBS\n",
      "0.1% SDS\n",
      "50 mM Tris\n",
      "200 mM methyl\n",
      "1% trifluoroacetic\n",
      "2% v/v acetonitrile\n",
      "0.1% v/v TFA\n",
      "98% v/v ACN\n",
      "0.1% v/v TFA\n",
      "10% to\n",
      "50% gradient\n",
      "90% by\n",
      "90% solvent\n",
      "10% at\n",
      "10% solvent\n",
      "70% ACN\n",
      "===== for text 4 =====\n",
      "10% Bis-Tris\n",
      "5% formic\n",
      "50% acetonitrile\n",
      "0.1% trifluoroacetic\n",
      "40% acetonitrile\n",
      "0.1% formic\n",
      "0.67% min\n",
      "===== for text 5 =====\n",
      "100 mM solution\n",
      "250 mM sucrose\n",
      "10 mM triethanolamine\n",
      "12% SDS\n",
      "0.1% formic\n",
      "0.1% formic\n",
      "50% B\n",
      "===== for text 6 =====\n",
      "2% acetonitrile\n",
      "0.1% formic\n",
      "2% to\n",
      "50% Acetonitrile\n",
      "0.1% formic\n",
      "20 mM KH2PO4\n",
      "20% ACN\n",
      "1000 mM KCl\n",
      "2% ACN\n",
      "0.1% FA\n",
      "2% to\n",
      "36% ACN\n",
      "0.1% FA\n",
      "===== for text 7 =====\n",
      "===== for text 8 =====\n",
      "20 mM Tris\n",
      "20 mM NaCl\n",
      "12% NuPAGE\n",
      "25 mM ammonium\n",
      "30% acetonitrile\n",
      "25 mM ABC\n",
      "80% acetonitrile\n",
      "25 mM ABC\n",
      "10 mM DTT\n",
      "25 mM ABC\n",
      "55 mM iodoacetamide\n",
      "25 mM ABC\n",
      "5% trifluoroacetic\n",
      "0.1% TFA\n",
      "===== for text 9 =====\n",
      "===== for text 10 =====\n",
      "70 mM dichlorodiphenyltrichloroethane\n",
      "1% SDS\n",
      "70% EtOH\n",
      "1% SDS\n",
      "12% SDS\n",
      "50% (v/v) methanol\n",
      "100 mM NH4HCO3\n",
      "50 mM NH4HCO3\n",
      "10 mM DTT\n",
      "50 mM NH4HCO3\n",
      "50 mM iodoacetamide\n",
      "100% ACN\n",
      "50 mM Tris\n",
      "1% (v/v) formic\n",
      "10% (v/v) formic\n",
      "5% (v/v) ACN\n",
      "0.1% (v/v) formic\n",
      "===== for text 11 =====\n",
      "50 mM Tris-HCl\n",
      "150 mM NaCl\n",
      "5 mM EDTA\n",
      "0.1% Triton\n",
      "5 mM NaF\n",
      "1 mM NaVO4\n",
      "12% Bis-Tris\n",
      "100% ACN\n",
      "80% acetonitrile\n",
      "28% solvent\n",
      "50% solvent\n",
      "100% solvent\n",
      "100% solvent\n",
      "35% for\n",
      "===== for text 12 =====\n",
      "10 mM dithiothreitol\n",
      "55 mM iodoacetamide\n",
      "25 mM ammonium\n",
      "===== for text 13 =====\n",
      "0.1% RapiGest\n",
      "5 mM TCEP\n",
      "10 mM iodoacetamide\n",
      "5 mM N-acetyl-cysteine\n",
      "100 mM ammoniumbicarbonate\n",
      "0.5% and\n",
      "===== for text 14 =====\n",
      "===== for text 15 =====\n",
      "===== for text 16 =====\n",
      "12% gradient\n",
      "50 mM ammonium\n",
      "50 mM ammonium\n",
      "50% acetonitrile\n",
      "1% formic\n",
      "5% acetonitrile\n",
      "0.1% formic\n",
      "5% acetonitrile\n",
      "0.1% formic\n",
      "===== for text 17 =====\n",
      "0.1% RapiGest\n",
      "===== for text 18 =====\n",
      "10% acetonitrile\n",
      "5% acetonitrile-0\n",
      "1% trifluoroacetic\n",
      "3% acetonitrile\n",
      "0.1% trifluoroacetic\n",
      "===== for text 19 =====\n",
      "4% CHAPS\n",
      "2% DTT\n",
      "0.5% Bio-Lyte\n",
      "0.5% CHAPS\n",
      "0.2% DTT\n",
      "0.2% Bio-Lyte\n",
      "12% gel\n",
      "0.1% TFA\n",
      "5% acetonitrile\n",
      "0.1% formic\n",
      "95% acetonitrile\n",
      "0.1% formic\n",
      "===== for text 20 =====\n",
      "===== for text 21 =====\n",
      "1% PSPEP\n",
      "===== for text 22 =====\n",
      "===== for text 23 =====\n",
      "0.5% SDS\n",
      "===== for text 24 =====\n",
      "===== for text 25 =====\n",
      "200mM of IAM\n",
      "100 mM of IAM\n",
      "20 mM of ascorbate\n",
      "1 mM of iodoTMT1\n",
      "5mM of TCEP\n",
      "1mM of iodoTMT2\n",
      "25mM of ammonium\n",
      "0.5% of RapiGest\n",
      "===== for text 26 =====\n",
      "5% charcoal-treated\n",
      "4% FA\n",
      "===== for text 27 =====\n",
      "5 mM glucose\n",
      "0.05% bovine\n",
      "10 % SDS-polyacrylamide\n",
      "35% and\n",
      "===== for text 28 =====\n",
      "===== for text 29 =====\n",
      "10% fetal\n",
      "10% fetal\n",
      "50 mM ammonium\n",
      "1 mM sodium\n",
      "4 mM at\n",
      "8 mM at\n",
      "50 mM ammonium\n",
      "5% formic\n",
      "80% acetonitrile\n",
      "6% trifluoroacetic\n",
      "80% ACN\n",
      "6% TFA\n",
      "50% ACN\n",
      "0.5% TFA\n",
      "200 mM NaCl\n",
      "50% ACN\n",
      "0.1% TFA\n",
      "10% formic\n",
      "10% ammonia\n",
      "80% ACN\n",
      "2% formic\n",
      "100% formic\n",
      "100% solvent\n",
      "0.1 % formic\n",
      "7% to\n",
      "30% solvent\n",
      "0.1% formic\n",
      "35% and\n"
     ]
    }
   ],
   "source": [
    "# some more improvements\n",
    "buffer_pattern = r'((?:0\\.)?(?:\\d{1,4}) ?(?:mM|µM|nM|%) (?:w/v\\)? |\\(?v/v\\)? )?(?:of )?[A-Za-z0-9-]+)'\n",
    "for i in range(30):\n",
    "    text = method.iloc[i]\n",
    "    res = [s for s in re.findall(buffer_pattern, text)]\n",
    "    print('===== for text %d =====' % i)\n",
    "    for b in res:\n",
    "        print(b)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### cases to remove\n",
    "'10% at'\n",
    "'0.67% min' (min-1)\n",
    "'90% by'\n",
    "'10% to'\n",
    "'50% B' (could be included)\n",
    "'35% for'\n",
    "'12% gel' (could be included in a different category, not as buffer)\n",
    "'12% gradient' (could be included in a different category, not as buffer)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "===== for text 0 =====\n",
      "2% deoxycholate, \n",
      "0.37% w/v sodium\n",
      "50% v/v acetonitrile\n",
      "2% v/v formic acid, 2% v/v acetonitrile\n",
      "===== for text 1 =====\n",
      "===== for text 2 =====\n",
      "10 mM DTT\n",
      "55 mM iodoacetamide\n",
      "100% acetonitrile\n",
      "40 mM ammoniumbicarbonate\n",
      "===== for text 3 =====\n",
      "10% FBS\n",
      "0.1% SDS, \n",
      "50 mM Tris\n",
      "200 mM methyl\n",
      "1% trifluoroacetic acid\n",
      "2% v/v acetonitrile\n",
      "0.1% v/v TFA\n",
      "98% v/v ACN, 0.1% v/v TFA\n",
      "10% to\n",
      "50% gradient\n",
      "90% by\n",
      "90% solvent\n",
      "10% at\n",
      "10% solvent\n",
      "70% ACN\n",
      "===== for text 4 =====\n",
      "10% Bis-Tris\n",
      "5% formic acid\n",
      "50% acetonitrile, \n",
      "0.1% trifluoroacetic acid\n",
      "40% acetonitrile\n",
      "0.1% formic acid\n",
      "0.67% min\n",
      "===== for text 5 =====\n",
      "100 mM solution\n",
      "250 mM sucrose\n",
      "10 mM triethanolamine\n",
      "12% SDS\n",
      "0.1% formic acid\n",
      "0.1% formic acid\n",
      "50% B\n",
      "===== for text 6 =====\n",
      "2% acetonitrile\n",
      "0.1% formic acid\n",
      "2% to\n",
      "50% Acetonitrile with 0.1% formic acid\n",
      "20 mM KH2PO4, 20% ACN, \n",
      "1000 mM KCl\n",
      "2% ACN, 0.1% FA\n",
      "2% to\n",
      "36% ACN, 0.1% FA\n",
      "===== for text 7 =====\n",
      "===== for text 8 =====\n",
      "20 mM Tris\n",
      "20 mM NaCl\n",
      "12% NuPAGE\n",
      "25 mM ammonium\n",
      "30% acetonitrile\n",
      "25 mM ABC\n",
      "80% acetonitrile\n",
      "25 mM ABC\n",
      "10 mM DTT\n",
      "25 mM ABC\n",
      "55 mM iodoacetamide\n",
      "25 mM ABC with \n",
      "5% trifluoroacetic acid\n",
      "0.1% TFA\n",
      "===== for text 9 =====\n",
      "===== for text 10 =====\n",
      "70 mM dichlorodiphenyltrichloroethane\n",
      "1% SDS\n",
      "70% EtOH\n",
      "1% SDS\n",
      "12% SDS\n",
      "50% (v/v) methanol\n",
      "100 mM NH4HCO3\n",
      "50 mM NH4HCO3\n",
      "10 mM DTT\n",
      "50 mM NH4HCO3\n",
      "50 mM iodoacetamide\n",
      "100% ACN\n",
      "50 mM Tris\n",
      "1% (v/v) formic acid\n",
      "10% (v/v) formic acid\n",
      "5% (v/v) ACN, 0.1% (v/v) formic acid\n",
      "===== for text 11 =====\n",
      "50 mM Tris-HCl\n",
      "150 mM NaCl, 5 mM EDTA\n",
      "0.1% Triton\n",
      "5 mM NaF\n",
      "1 mM NaVO4\n",
      "12% Bis-Tris\n",
      "100% ACN\n",
      "80% acetonitrile\n",
      "28% solvent\n",
      "50% solvent\n",
      "100% solvent\n",
      "100% solvent\n",
      "35% for\n",
      "===== for text 12 =====\n",
      "10 mM dithiothreitol\n",
      "55 mM iodoacetamide\n",
      "25 mM ammonium\n",
      "===== for text 13 =====\n",
      "0.1% RapiGest, \n",
      "5 mM TCEP\n",
      "10 mM iodoacetamide\n",
      "5 mM N-acetyl-cysteine\n",
      "100 mM ammoniumbicarbonate\n",
      "0.5% and HCl\n",
      "===== for text 14 =====\n",
      "===== for text 15 =====\n",
      "===== for text 16 =====\n",
      "12% gradient\n",
      "50 mM ammonium\n",
      "50 mM ammonium\n",
      "50% acetonitrile\n",
      "1% formic acid\n",
      "5% acetonitrile, 0.1% formic acid\n",
      "5% acetonitrile, 0.1% formic acid\n",
      "===== for text 17 =====\n",
      "0.1% RapiGest\n",
      "===== for text 18 =====\n",
      "10% acetonitrile\n",
      "5% acetonitrile-0\n",
      "1% trifluoroacetic acid\n",
      "3% acetonitrile\n",
      "0.1% trifluoroacetic acid\n",
      "===== for text 19 =====\n",
      "4% CHAPS, 2% DTT, 0.5% Bio-Lyte\n",
      "0.5% CHAPS, 0.2% DTT\n",
      "0.2% Bio-Lyte\n",
      "12% gel\n",
      "0.1% TFA\n",
      "5% acetonitrile\n",
      "0.1% formic acid\n",
      "95% acetonitrile\n",
      "0.1% formic acid\n",
      "===== for text 20 =====\n",
      "===== for text 21 =====\n",
      "1% PSPEP\n",
      "===== for text 22 =====\n",
      "===== for text 23 =====\n",
      "0.5% SDS\n",
      "===== for text 24 =====\n",
      "===== for text 25 =====\n",
      "200mM of IAM\n",
      "100 mM of IAM\n",
      "20 mM of ascorbate\n",
      "1 mM of iodoTMT1\n",
      "5mM of TCEP\n",
      "1mM of iodoTMT2\n",
      "25mM of ammonium\n",
      "0.5% of RapiGest\n",
      "===== for text 26 =====\n",
      "5% charcoal-treated\n",
      "4% FA\n",
      "===== for text 27 =====\n",
      "5 mM glucose\n",
      "0.05% bovine\n",
      "10 % SDS-polyacrylamide\n",
      "35% and wideband-activation\n",
      "===== for text 28 =====\n",
      "===== for text 29 =====\n",
      "10% fetal\n",
      "10% fetal\n",
      "50 mM ammonium\n",
      "1 mM sodium\n",
      "4 mM at\n",
      "8 mM at\n",
      "50 mM ammonium\n",
      "5% formic acid\n",
      "80% acetonitrile\n",
      "6% trifluoroacetic acid\n",
      "80% ACN\n",
      "6% TFA\n",
      "50% ACN, 0.5% TFA\n",
      "200 mM NaCl\n",
      "50% ACN\n",
      "0.1% TFA, \n",
      "10% formic acid\n",
      "10% ammonia\n",
      "80% ACN\n",
      "2% formic acid\n",
      "100% formic acid\n",
      "100% solvent\n",
      "0.1 % formic acid\n",
      "7% to\n",
      "30% solvent\n",
      "0.1% formic acid\n",
      "35% and supplemental\n",
      "===== for text 30 =====\n",
      "10 mM DTT\n",
      "55mM IA\n",
      "===== for text 31 =====\n",
      "12% (v/v) acrylamide\n",
      "12% (v/v) acrylamide\n",
      "===== for text 32 =====\n",
      "===== for text 33 =====\n",
      "20 mM ammonium\n",
      "2% w/v CHAPS\n",
      "65 mM DTT\n",
      "0.5% (v/v) IPG\n",
      "===== for text 34 =====\n",
      "50 mM DTT\n",
      "120 mM iodoacetamide\n",
      "12% Bis-Tris\n",
      "100 mM ammonium\n",
      "5% formic acid\n",
      "3% acetonitrile\n",
      "===== for text 35 =====\n",
      "===== for text 36 =====\n",
      "20% (v/v) H2O2\n",
      "50 mM Tris-HCl, \n",
      "10% glycerol, 1% Triton\n",
      "150 mM NaCl, 0.1% SDS, \n",
      "5 mM EDTA\n",
      "1 mM PMSF\n",
      "10 mM DTT\n",
      "50 mM iodoacetamide\n",
      "===== for text 37 =====\n",
      "===== for text 38 =====\n",
      "50 mM TEAB\n",
      "0.1% RapiGest\n",
      "500mM TCEP\n",
      "5% buffer\n",
      "40 % buffer\n",
      "===== for text 39 =====\n",
      "20% dialyzed\n",
      "1% of tryptic\n",
      "0.3% of identified\n",
      "===== for text 40 =====\n",
      "0.1% Rapigest\n",
      "50 mM ammonium\n",
      "5 mM dithiothreitol\n",
      "15 mM iodoacetamide\n",
      "1% trifluoroacetic acid\n",
      "1% acetonitrile, 0.1% formic acid\n",
      "40% acetonitrile, 0.1% formic acid\n",
      "70% of starting\n",
      "1% of retention\n",
      "40% acetonitrile\n",
      "0.1% formic acid\n",
      "===== for text 41 =====\n",
      "===== for text 42 =====\n",
      "95% as\n",
      "===== for text 43 =====\n",
      "0.25% sodium\n",
      "0.2 mM iodoacetamide, 1 mM EDTA, \n",
      "1 mM PMSF, 1% octyl-\n",
      "150 mM NaCl, 20 mM Tris\n",
      "400 mM NaCl, 20 mM Tris\n",
      "20 mM Tris\n",
      "12% SDS-PAGE\n",
      "80% acetonitrile\n",
      "0.1% trifluoroacetic acid\n",
      "0.1% TFA\n",
      "0.1% TFA\n",
      "30% ACN\n",
      "0.1% TFA\n",
      "30% ACN\n",
      "0.1% TFA\n",
      "0.5% acetic acid\n",
      "30% buffer\n",
      "80% ACN\n",
      "0.5% acetic acid\n",
      "===== for text 44 =====\n",
      "35% normalized\n",
      "===== for text 45 =====\n",
      "21% oxygen\n",
      "1% oxygen\n",
      "98% incorporation, \n",
      "10mM Hepes\n",
      "1mM DTT, \n",
      "5mM Chloroacetamide, \n",
      "60% of 80\n",
      "0.1% Formic acid, \n",
      "===== for text 46 =====\n",
      "===== for text 47 =====\n",
      "40% in\n",
      "0.1% formic acid\n",
      "===== for text 48 =====\n",
      "0.2% formic acid, 1% acetonitrile\n",
      "0.2% formic acid, 100% acetonitrile\n",
      "0.8% to\n",
      "8% of B\n",
      "35% B\n",
      "48% B\n",
      "97% at\n",
      "0% around\n",
      "40% and an\n",
      "5% to\n",
      "26% acetonitrile\n",
      "26% to\n",
      "48% acetonitrile\n",
      "0.5% acetic acid\n",
      "===== for text 49 =====\n"
     ]
    }
   ],
   "source": [
    "# evensome more improvements\n",
    "buffer_pattern = r'(?:(?:0\\.)?(?:\\d{1,4}) ?(?:mM|µM|nM|%) (?:w/v\\)? |\\(?v/v\\)? )?(?:and )?(?:of )?[A-Za-z0-9-]+(?: [Aa]cid)?(?: with | ?, )?)+'\n",
    "\n",
    "for i in range(50):\n",
    "    text = method.iloc[i]\n",
    "    res = [s for s in re.findall(buffer_pattern, text)]\n",
    "    print('===== for text %d =====' % i)\n",
    "    for b in res:\n",
    "        print(b)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "===== for text 0 =====\n",
      "2% v/v for\n",
      "===== for text 1 =====\n",
      "===== for text 2 =====\n",
      "===== for text 3 =====\n",
      "98% v/v A\n",
      "10% to\n",
      "50% gradient\n",
      "90% by\n",
      "90% solvent\n",
      "10% at\n",
      "10% solvent\n",
      "70% A\n",
      "===== for text 4 =====\n",
      "10% B\n",
      "5% for\n",
      "0.1% for\n",
      "===== for text 5 =====\n",
      "0.1% for\n",
      "0.1% for\n",
      "50% B\n",
      "===== for text 6 =====\n",
      "0.1% for\n",
      "2% to\n",
      "50% A\n",
      "0.1% for\n",
      "20% A\n",
      "2% A\n",
      "2% to\n",
      "36% A\n",
      "===== for text 7 =====\n",
      "===== for text 8 =====\n",
      "25 mM A\n",
      "25 mM A\n",
      "25 mM A\n",
      "25 mM A\n",
      "===== for text 9 =====\n",
      "===== for text 10 =====\n",
      "100% A\n",
      "1% (v/v) for\n",
      "10% (v/v) for\n",
      "5% (v/v) A\n",
      "0.1% (v/v) for\n",
      "===== for text 11 =====\n",
      "12% B\n",
      "100% A\n",
      "28% solvent\n",
      "50% solvent\n",
      "100% solvent\n",
      "100% solvent\n",
      "35% for\n",
      "===== for text 12 =====\n",
      "===== for text 13 =====\n",
      "===== for text 14 =====\n",
      "===== for text 15 =====\n",
      "===== for text 16 =====\n",
      "12% gradient\n",
      "1% for\n",
      "0.1% for\n",
      "0.1% for\n",
      "===== for text 17 =====\n",
      "===== for text 18 =====\n",
      "===== for text 19 =====\n",
      "0.5% B\n",
      "0.2% B\n",
      "12% gel\n",
      "0.1% for\n",
      "0.1% for\n",
      "===== for text 20 =====\n",
      "===== for text 21 =====\n",
      "===== for text 22 =====\n",
      "===== for text 23 =====\n",
      "===== for text 24 =====\n",
      "===== for text 25 =====\n",
      "===== for text 26 =====\n",
      "===== for text 27 =====\n",
      "===== for text 28 =====\n",
      "===== for text 29 =====\n",
      "4 mM at\n",
      "8 mM at\n",
      "5% for\n",
      "80% A\n",
      "50% A\n",
      "50% A\n",
      "10% for\n",
      "80% A\n",
      "2% for\n",
      "100% for\n",
      "100% solvent\n",
      "0.1 % for\n",
      "7% to\n",
      "30% solvent\n",
      "0.1% for\n",
      "===== for text 30 =====\n",
      "===== for text 31 =====\n",
      "===== for text 32 =====\n",
      "===== for text 33 =====\n",
      "===== for text 34 =====\n",
      "12% B\n",
      "5% for\n",
      "===== for text 35 =====\n",
      "===== for text 36 =====\n",
      "===== for text 37 =====\n",
      "===== for text 38 =====\n",
      "===== for text 39 =====\n",
      "===== for text 40 =====\n",
      "0.1% for\n",
      "0.1% for\n",
      "0.1% for\n",
      "===== for text 41 =====\n",
      "===== for text 42 =====\n",
      "===== for text 43 =====\n",
      "30% A\n",
      "30% A\n",
      "80% A\n",
      "===== for text 44 =====\n",
      "===== for text 45 =====\n",
      "80% A\n",
      "===== for text 46 =====\n",
      "===== for text 47 =====\n",
      "0.1% for\n",
      "===== for text 48 =====\n",
      "0.2% for\n",
      "0.2% for\n",
      "0.8% to\n",
      "35% B\n",
      "48% B\n",
      "97% at\n",
      "0% around\n",
      "5% to\n",
      "26% to\n",
      "===== for text 49 =====\n"
     ]
    }
   ],
   "source": [
    "# pattern to remove\n",
    "remove_pattern = r'(?:0\\.)?(?:\\d{1,4}) ?(?:mM|µM|nM|%) (?:w/v\\)? |\\(?v/v\\)? )?(?:A|B|at|by|to|for|around|gel|gradient|solvent)'\n",
    "\n",
    "for i in range(50):\n",
    "    text = method.iloc[i]\n",
    "    res = [s for s in re.findall(remove_pattern, text)]\n",
    "    print('===== for text %d =====' % i)\n",
    "    for b in res:\n",
    "        print(b)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "It seems like this regex is picking up things like '0.1% formic acid' and '100% ACN' so try the one below with $"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "===== for text 0 =====\n",
      "===== for text 1 =====\n",
      "===== for text 2 =====\n",
      "===== for text 3 =====\n",
      "10% to\n",
      "50% gradient\n",
      "90% by\n",
      "90% solvent\n",
      "10% at\n",
      "10% solvent\n",
      "===== for text 4 =====\n",
      "===== for text 5 =====\n",
      "50% B\n",
      "===== for text 6 =====\n",
      "2% to\n",
      "2% to\n",
      "===== for text 7 =====\n",
      "===== for text 8 =====\n",
      "===== for text 9 =====\n",
      "===== for text 10 =====\n",
      "===== for text 11 =====\n",
      "28% solvent\n",
      "50% solvent\n",
      "100% solvent\n",
      "100% solvent\n",
      "35% for\n",
      "===== for text 12 =====\n",
      "===== for text 13 =====\n",
      "===== for text 14 =====\n",
      "===== for text 15 =====\n",
      "===== for text 16 =====\n",
      "12% gradient\n",
      "===== for text 17 =====\n",
      "===== for text 18 =====\n",
      "===== for text 19 =====\n",
      "12% gel\n",
      "===== for text 20 =====\n",
      "===== for text 21 =====\n",
      "===== for text 22 =====\n",
      "===== for text 23 =====\n",
      "===== for text 24 =====\n",
      "===== for text 25 =====\n",
      "===== for text 26 =====\n",
      "===== for text 27 =====\n",
      "===== for text 28 =====\n",
      "===== for text 29 =====\n",
      "4 mM at\n",
      "8 mM at\n",
      "100% solvent\n",
      "7% to\n",
      "30% solvent\n",
      "===== for text 30 =====\n",
      "===== for text 31 =====\n",
      "===== for text 32 =====\n",
      "===== for text 33 =====\n",
      "===== for text 34 =====\n",
      "===== for text 35 =====\n",
      "===== for text 36 =====\n",
      "===== for text 37 =====\n",
      "===== for text 38 =====\n",
      "===== for text 39 =====\n",
      "===== for text 40 =====\n",
      "===== for text 41 =====\n",
      "===== for text 42 =====\n",
      "===== for text 43 =====\n",
      "===== for text 44 =====\n",
      "===== for text 45 =====\n",
      "===== for text 46 =====\n",
      "===== for text 47 =====\n",
      "===== for text 48 =====\n",
      "0.8% to\n",
      "35% B\n",
      "48% B\n",
      "97% at\n",
      "0% around\n",
      "5% to\n",
      "26% to\n",
      "===== for text 49 =====\n"
     ]
    }
   ],
   "source": [
    "# with remove_pattern with $\n",
    "buffer_pattern = r'(?:(?:0\\.)?(?:\\d{1,4}) ?(?:mM|µM|nM|%) (?:w/v\\)? |\\(?v/v\\)? )?(?:and )?(?:of )?[A-Za-z0-9-]+(?: [Aa]cid)?(?: with | ?, )?)+'\n",
    "remove_pattern = r'(?:0\\.)?(?:\\d{1,4}) ?(?:mM|µM|nM|%) (?:w/v\\)? |\\(?v/v\\)? )?(?:A|B|at|by|to|for|around|gel|gradient|solvent){1}$'\n",
    "remove_pattern_short = r'.*(?:A|B|at|by|to|for|around|gel|gradient|solvent){1}$'\n",
    "\n",
    "for i in range(50):\n",
    "    text = method.iloc[i]\n",
    "    res = [s for s in re.findall(buffer_pattern, text) if re.match(remove_pattern, s)]\n",
    "    print('===== for text %d =====' % i)\n",
    "    for b in res:\n",
    "        print(b)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "===== for text 0 =====\n",
      "===== for text 1 =====\n",
      "===== for text 2 =====\n",
      "===== for text 3 =====\n",
      "10% to\n",
      "50% gradient\n",
      "90% by\n",
      "90% solvent\n",
      "10% at\n",
      "10% solvent\n",
      "===== for text 4 =====\n",
      "===== for text 5 =====\n",
      "50% B\n",
      "===== for text 6 =====\n",
      "2% to\n",
      "2% to\n",
      "===== for text 7 =====\n",
      "===== for text 8 =====\n",
      "===== for text 9 =====\n",
      "===== for text 10 =====\n",
      "===== for text 11 =====\n",
      "28% solvent\n",
      "50% solvent\n",
      "100% solvent\n",
      "100% solvent\n",
      "35% for\n",
      "===== for text 12 =====\n",
      "===== for text 13 =====\n",
      "===== for text 14 =====\n",
      "===== for text 15 =====\n",
      "===== for text 16 =====\n",
      "12% gradient\n",
      "===== for text 17 =====\n",
      "===== for text 18 =====\n",
      "===== for text 19 =====\n",
      "12% gel\n",
      "===== for text 20 =====\n",
      "===== for text 21 =====\n",
      "===== for text 22 =====\n",
      "===== for text 23 =====\n",
      "===== for text 24 =====\n",
      "===== for text 25 =====\n",
      "===== for text 26 =====\n",
      "===== for text 27 =====\n",
      "===== for text 28 =====\n",
      "===== for text 29 =====\n",
      "4 mM at\n",
      "8 mM at\n",
      "100% solvent\n",
      "7% to\n",
      "30% solvent\n",
      "===== for text 30 =====\n",
      "===== for text 31 =====\n",
      "===== for text 32 =====\n",
      "===== for text 33 =====\n",
      "===== for text 34 =====\n",
      "===== for text 35 =====\n",
      "===== for text 36 =====\n",
      "===== for text 37 =====\n",
      "===== for text 38 =====\n",
      "===== for text 39 =====\n",
      "===== for text 40 =====\n",
      "===== for text 41 =====\n",
      "===== for text 42 =====\n",
      "===== for text 43 =====\n",
      "===== for text 44 =====\n",
      "===== for text 45 =====\n",
      "===== for text 46 =====\n",
      "===== for text 47 =====\n",
      "===== for text 48 =====\n",
      "0.8% to\n",
      "8% of B\n",
      "35% B\n",
      "48% B\n",
      "97% at\n",
      "0% around\n",
      "5% to\n",
      "26% to\n",
      "===== for text 49 =====\n"
     ]
    }
   ],
   "source": [
    "# with shorter remove_pattern\n",
    "buffer_pattern = r'(?:(?:0\\.)?(?:\\d{1,4}) ?(?:mM|µM|nM|%) (?:w/v\\)? |\\(?v/v\\)? )?(?:and )?(?:of )?[A-Za-z0-9-]+(?: [Aa]cid)?(?: with | ?, )?)+'\n",
    "remove_pattern_short = r'^.* (?:A|B|at|by|to|for|around|gel|gradient|solvent){1}$'\n",
    "\n",
    "for i in range(50):\n",
    "    text = method.iloc[i]\n",
    "    res = [s for s in re.findall(buffer_pattern, text) if re.match(remove_pattern_short, s)]\n",
    "    print('===== for text %d =====' % i)\n",
    "    for b in res:\n",
    "        print(b)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This shorter pattern seems to be equally effective."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "===== for text 0 =====\n",
      "2% deoxycholate, \n",
      "0.37% w/v sodium\n",
      "50% v/v acetonitrile\n",
      "2% v/v formic acid, 2% v/v acetonitrile\n",
      "===== for text 1 =====\n",
      "===== for text 2 =====\n",
      "10 mM DTT\n",
      "55 mM iodoacetamide\n",
      "100% acetonitrile\n",
      "40 mM ammoniumbicarbonate\n",
      "===== for text 3 =====\n",
      "10% FBS\n",
      "0.1% SDS, \n",
      "50 mM Tris\n",
      "200 mM methyl\n",
      "1% trifluoroacetic acid\n",
      "2% v/v acetonitrile\n",
      "0.1% v/v TFA\n",
      "98% v/v ACN, 0.1% v/v TFA\n",
      "70% ACN\n",
      "===== for text 4 =====\n",
      "10% Bis-Tris\n",
      "5% formic acid\n",
      "50% acetonitrile, \n",
      "0.1% trifluoroacetic acid\n",
      "40% acetonitrile\n",
      "0.1% formic acid\n",
      "0.67% min\n",
      "===== for text 5 =====\n",
      "100 mM solution\n",
      "250 mM sucrose\n",
      "10 mM triethanolamine\n",
      "12% SDS\n",
      "0.1% formic acid\n",
      "0.1% formic acid\n",
      "===== for text 6 =====\n",
      "2% acetonitrile\n",
      "0.1% formic acid\n",
      "50% Acetonitrile with 0.1% formic acid\n",
      "20 mM KH2PO4, 20% ACN, \n",
      "1000 mM KCl\n",
      "2% ACN, 0.1% FA\n",
      "36% ACN, 0.1% FA\n",
      "===== for text 7 =====\n",
      "===== for text 8 =====\n",
      "20 mM Tris\n",
      "20 mM NaCl\n",
      "25 mM ammonium\n",
      "30% acetonitrile\n",
      "25 mM ABC\n",
      "80% acetonitrile\n",
      "25 mM ABC\n",
      "10 mM DTT\n",
      "25 mM ABC\n",
      "55 mM iodoacetamide\n",
      "25 mM ABC with \n",
      "5% trifluoroacetic acid\n",
      "0.1% TFA\n",
      "===== for text 9 =====\n",
      "===== for text 10 =====\n",
      "70 mM dichlorodiphenyltrichloroethane\n",
      "1% SDS\n",
      "70% EtOH\n",
      "1% SDS\n",
      "12% SDS\n",
      "50% (v/v) methanol\n",
      "100 mM NH4HCO3\n",
      "50 mM NH4HCO3\n",
      "10 mM DTT\n",
      "50 mM NH4HCO3\n",
      "50 mM iodoacetamide\n",
      "100% ACN\n",
      "50 mM Tris\n",
      "1% (v/v) formic acid\n",
      "10% (v/v) formic acid\n",
      "5% (v/v) ACN, 0.1% (v/v) formic acid\n",
      "===== for text 11 =====\n",
      "50 mM Tris-HCl\n",
      "150 mM NaCl, 5 mM EDTA\n",
      "0.1% Triton\n",
      "5 mM NaF\n",
      "1 mM NaVO4\n",
      "12% Bis-Tris\n",
      "100% ACN\n",
      "80% acetonitrile\n",
      "===== for text 12 =====\n",
      "10 mM dithiothreitol\n",
      "55 mM iodoacetamide\n",
      "25 mM ammonium\n",
      "===== for text 13 =====\n",
      "0.1% RapiGest, \n",
      "5 mM TCEP\n",
      "10 mM iodoacetamide\n",
      "5 mM N-acetyl-cysteine\n",
      "100 mM ammoniumbicarbonate\n",
      "0.5% and HCl\n",
      "===== for text 14 =====\n",
      "===== for text 15 =====\n",
      "===== for text 16 =====\n",
      "50 mM ammonium\n",
      "50 mM ammonium\n",
      "50% acetonitrile\n",
      "1% formic acid\n",
      "5% acetonitrile, 0.1% formic acid\n",
      "5% acetonitrile, 0.1% formic acid\n",
      "===== for text 17 =====\n",
      "0.1% RapiGest\n",
      "===== for text 18 =====\n",
      "10% acetonitrile\n",
      "5% acetonitrile-0\n",
      "1% trifluoroacetic acid\n",
      "3% acetonitrile\n",
      "0.1% trifluoroacetic acid\n",
      "===== for text 19 =====\n",
      "4% CHAPS, 2% DTT, 0.5% Bio-Lyte\n",
      "0.5% CHAPS, 0.2% DTT\n",
      "0.2% Bio-Lyte\n",
      "0.1% TFA\n",
      "5% acetonitrile\n",
      "0.1% formic acid\n",
      "95% acetonitrile\n",
      "0.1% formic acid\n",
      "===== for text 20 =====\n",
      "===== for text 21 =====\n",
      "1% PSPEP\n",
      "===== for text 22 =====\n",
      "===== for text 23 =====\n",
      "0.5% SDS\n",
      "===== for text 24 =====\n",
      "===== for text 25 =====\n",
      "200mM of IAM\n",
      "100 mM of IAM\n",
      "20 mM of ascorbate\n",
      "1 mM of iodoTMT1\n",
      "5mM of TCEP\n",
      "1mM of iodoTMT2\n",
      "25mM of ammonium\n",
      "0.5% of RapiGest\n",
      "===== for text 26 =====\n",
      "5% charcoal-treated\n",
      "4% FA\n",
      "===== for text 27 =====\n",
      "5 mM glucose\n",
      "0.05% bovine\n",
      "10 % SDS-polyacrylamide\n",
      "35% and wideband-activation\n",
      "===== for text 28 =====\n",
      "===== for text 29 =====\n",
      "10% fetal\n",
      "10% fetal\n",
      "50 mM ammonium\n",
      "1 mM sodium\n",
      "50 mM ammonium\n",
      "5% formic acid\n",
      "80% acetonitrile\n",
      "6% trifluoroacetic acid\n",
      "80% ACN\n",
      "6% TFA\n",
      "50% ACN, 0.5% TFA\n",
      "200 mM NaCl\n",
      "50% ACN\n",
      "0.1% TFA, \n",
      "10% formic acid\n",
      "10% ammonia\n",
      "80% ACN\n",
      "2% formic acid\n",
      "100% formic acid\n",
      "0.1 % formic acid\n",
      "0.1% formic acid\n",
      "35% and supplemental\n",
      "===== for text 30 =====\n",
      "10 mM DTT\n",
      "55mM IA\n",
      "===== for text 31 =====\n",
      "12% (v/v) acrylamide\n",
      "12% (v/v) acrylamide\n",
      "===== for text 32 =====\n",
      "===== for text 33 =====\n",
      "20 mM ammonium\n",
      "2% w/v CHAPS\n",
      "65 mM DTT\n",
      "0.5% (v/v) IPG\n",
      "===== for text 34 =====\n",
      "50 mM DTT\n",
      "120 mM iodoacetamide\n",
      "12% Bis-Tris\n",
      "100 mM ammonium\n",
      "5% formic acid\n",
      "3% acetonitrile\n",
      "===== for text 35 =====\n",
      "===== for text 36 =====\n",
      "20% (v/v) H2O2\n",
      "50 mM Tris-HCl, \n",
      "10% glycerol, 1% Triton\n",
      "150 mM NaCl, 0.1% SDS, \n",
      "5 mM EDTA\n",
      "1 mM PMSF\n",
      "10 mM DTT\n",
      "50 mM iodoacetamide\n",
      "===== for text 37 =====\n",
      "===== for text 38 =====\n",
      "50 mM TEAB\n",
      "0.1% RapiGest\n",
      "500mM TCEP\n",
      "===== for text 39 =====\n",
      "20% dialyzed\n",
      "1% of tryptic\n",
      "0.3% of identified\n",
      "===== for text 40 =====\n",
      "0.1% Rapigest\n",
      "50 mM ammonium\n",
      "5 mM dithiothreitol\n",
      "15 mM iodoacetamide\n",
      "1% trifluoroacetic acid\n",
      "1% acetonitrile, 0.1% formic acid\n",
      "40% acetonitrile, 0.1% formic acid\n",
      "70% of starting\n",
      "1% of retention\n",
      "40% acetonitrile\n",
      "0.1% formic acid\n",
      "===== for text 41 =====\n",
      "===== for text 42 =====\n",
      "95% as\n",
      "===== for text 43 =====\n",
      "0.25% sodium\n",
      "0.2 mM iodoacetamide, 1 mM EDTA, \n",
      "1 mM PMSF, 1% octyl-\n",
      "150 mM NaCl, 20 mM Tris\n",
      "400 mM NaCl, 20 mM Tris\n",
      "20 mM Tris\n",
      "12% SDS-PAGE\n",
      "80% acetonitrile\n",
      "0.1% trifluoroacetic acid\n",
      "0.1% TFA\n",
      "0.1% TFA\n",
      "30% ACN\n",
      "0.1% TFA\n",
      "30% ACN\n",
      "0.1% TFA\n",
      "0.5% acetic acid\n",
      "80% ACN\n",
      "0.5% acetic acid\n",
      "===== for text 44 =====\n",
      "35% normalized\n",
      "===== for text 45 =====\n",
      "21% oxygen\n",
      "1% oxygen\n",
      "98% incorporation, \n",
      "10mM Hepes\n",
      "1mM DTT, \n",
      "5mM Chloroacetamide, \n",
      "60% of 80\n",
      "0.1% Formic acid, \n",
      "===== for text 46 =====\n",
      "===== for text 47 =====\n",
      "0.1% formic acid\n",
      "===== for text 48 =====\n",
      "0.2% formic acid, 1% acetonitrile\n",
      "0.2% formic acid, 100% acetonitrile\n",
      "40% and an\n",
      "26% acetonitrile\n",
      "48% acetonitrile\n",
      "0.5% acetic acid\n",
      "===== for text 49 =====\n"
     ]
    }
   ],
   "source": [
    "# same as above but this time remove the ones that match remove_\n",
    "buffer_pattern = r'(?:(?:0\\.)?(?:\\d{1,4}) ?(?:mM|µM|nM|%) (?:w/v\\)? |\\(?v/v\\)? )?(?:and )?(?:of )?[A-Za-z0-9-]+(?: [Aa]cid)?(?: with | ?, )?)+'\n",
    "remove_pattern_short = r'^.* (?:A|B|at|by|to|for|around|gel|gradient|solvent|NuPAGE|buffer|in){1}$'\n",
    "\n",
    "for i in range(50):\n",
    "    text = method.iloc[i, 0]\n",
    "    res = [s for s in re.findall(buffer_pattern, text) if not re.match(remove_pattern_short, s)]\n",
    "    print('===== for text %d =====' % i)\n",
    "    for b in res:\n",
    "        print(b)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [],
   "source": [
    "def extract_buffers(text):\n",
    "    buffer_pattern = r'(?:(?:0\\.)?(?:\\d{1,4}) ?(?:mM|µM|nM|%) (?:w/v\\)? |\\(?v/v\\)? )?(?:and )?(?:of )?[A-Za-z0-9-]+(?: [Aa]cid)?(?: with | ?, )?)+'\n",
    "    remove_pattern_short = r'^.* (?:A|B|at|by|to|for|around|gel|gradient|solvent|NuPAGE|buffer|in){1}$'\n",
    "    \n",
    "    # some mapping for acronyms\n",
    "    acronym_map = {\n",
    "        'formic acid': 'FA',\n",
    "        'acetonitrile': 'ACN',\n",
    "        'Acetonitrile': 'ACN',\n",
    "        'trifluoroacetic acid': 'TFA'\n",
    "    }\n",
    "    \n",
    "    if type(text) == str:\n",
    "        buffers = ['(' + s + ')' for s in re.findall(buffer_pattern, text) if not re.match(remove_pattern_short, s)]\n",
    "        return buffers\n",
    "    else:\n",
    "        return []"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [],
   "source": [
    "method = method.to_frame()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [],
   "source": [
    "method['buffers'] = method['sample_protocol'].apply(lambda x: extract_buffers(x))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_protocol</th>\n",
       "      <th>buffers</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>The crude membranes from 5 P56-P70 Glun1TAP/TA...</td>\n",
       "      <td>[(2% deoxycholate, ), (0.37% w/v sodium), (50%...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Breast cancer tissue lysates, reduction, alkyl...</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>In-gel digests were performed as described in ...</td>\n",
       "      <td>[(10 mM DTT), (55 mM iodoacetamide), (100% ace...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Conditioned media preparation and HPLC-isobari...</td>\n",
       "      <td>[(10% FBS), (0.1% SDS, ), (50 mM Tris), (200 m...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Proteins were eluted from washed beads by addi...</td>\n",
       "      <td>[(10% Bis-Tris), (5% formic acid), (50% aceton...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Exosomes were isolated from 10 healthy volunte...</td>\n",
       "      <td>[(100 mM solution), (250 mM sucrose), (10 mM t...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Low resolution mass spectrometry: For the shot...</td>\n",
       "      <td>[(2% acetonitrile), (0.1% formic acid), (50% A...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Arabidopsis plants, Wassilewskija background (...</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Zebrafish larvae were ground with a pestle in ...</td>\n",
       "      <td>[(20 mM Tris), (20 mM NaCl), (25 mM ammonium),...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Affinity purification and nano-liquid chromato...</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Protein Extraction from Catheter Biofilm Small...</td>\n",
       "      <td>[(70 mM dichlorodiphenyltrichloroethane), (1% ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Generation of mDia2-Based Immunocomplexes and ...</td>\n",
       "      <td>[(50 mM Tris-HCl), (150 mM NaCl, 5 mM EDTA), (...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Bands of interest were dehydrated using aceton...</td>\n",
       "      <td>[(10 mM dithiothreitol), (55 mM iodoacetamide)...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Whole Cell Lysis: 109 cells were lysed in 50 m...</td>\n",
       "      <td>[(0.1% RapiGest, ), (5 mM TCEP), (10 mM iodoac...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>See the corresponding sections of the associat...</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>See details in reference PMID: 24460188</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Exponentially-growing cultures of HeLa S3 cell...</td>\n",
       "      <td>[(50 mM ammonium), (50 mM ammonium), (50% acet...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>For each condition, two 6-well plates of HeLa ...</td>\n",
       "      <td>[(0.1% RapiGest)]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>For peptide extraction from moss tissues and p...</td>\n",
       "      <td>[(10% acetonitrile), (5% acetonitrile-0), (1% ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>The kidney mitochondrial fraction was obtained...</td>\n",
       "      <td>[(4% CHAPS, 2% DTT, 0.5% Bio-Lyte), (0.5% CHAP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>In-solution tryptic digestion of plasma membra...</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>In order to assess the presence of repair syst...</td>\n",
       "      <td>[(1% PSPEP)]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Carbon metabolism in Mycobacterium smegmatis w...</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>100 mg of frontal lobe brain sample from five ...</td>\n",
       "      <td>[(0.5% SDS)]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>Glomeruli were isolated from wildtype mouse ki...</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>For examining the NO-mediated protective effec...</td>\n",
       "      <td>[(200mM of IAM), (100 mM of IAM), (20 mM of as...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>SILAC-labeled MCF7 cells were cultured for thr...</td>\n",
       "      <td>[(5% charcoal-treated), (4% FA)]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>Blood was obtained from mice euthanized with C...</td>\n",
       "      <td>[(5 mM glucose), (0.05% bovine), (10 % SDS-pol...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>SDS lysate from skeletal muscle and C2C12 cell...</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>HeLa cells were grown to confluence in Dulbecc...</td>\n",
       "      <td>[(10% fetal), (10% fetal), (50 mM ammonium), (...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>Extracted proteins from liver mouse tissue wer...</td>\n",
       "      <td>[(10 mM DTT), (55mM IA)]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>Datasets of RH strain of T. gondii one-dimensi...</td>\n",
       "      <td>[(12% (v/v) acrylamide), (12% (v/v) acrylamide)]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>Datasets of N. caninum Orbitrap 1DE and Orbitr...</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>Culture media were collected, supplemented wit...</td>\n",
       "      <td>[(20 mM ammonium), (2% w/v CHAPS), (65 mM DTT)...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>Stone matrix protein (SMP) was isolated from k...</td>\n",
       "      <td>[(50 mM DTT), (120 mM iodoacetamide), (12% Bis...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>To identify proteins that are differentially e...</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>Cells from the lysine auxotrophic Saccharomyce...</td>\n",
       "      <td>[(20% (v/v) H2O2), (50 mM Tris-HCl, ), (10% gl...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>Samples were digested with trypsin then labele...</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>Prior to digestion with trypsin 100 µg of prot...</td>\n",
       "      <td>[(50 mM TEAB), (0.1% RapiGest), (500mM TCEP)]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>Cell lines from which we generated the super-S...</td>\n",
       "      <td>[(20% dialyzed), (1% of tryptic), (0.3% of ide...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>Total plasma protein concentration was assayed...</td>\n",
       "      <td>[(0.1% Rapigest), (50 mM ammonium), (5 mM dith...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>Data analysis.  Spot detection and matching we...</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>Naïve CD4+ T cells were isolated using magneti...</td>\n",
       "      <td>[(95% as)]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>Purification of HLA-I complexes. HLA-I peptido...</td>\n",
       "      <td>[(0.25% sodium), (0.2 mM iodoacetamide, 1 mM E...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>The files were obtained from proteomic prepara...</td>\n",
       "      <td>[(35% normalized)]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>Cells were cultured for 24 hours in serum free...</td>\n",
       "      <td>[(21% oxygen), (1% oxygen), (98% incorporation...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>Soluble protein extracts were prepared from th...</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>Duplicate ATCC12228 and PM221 samples acquired...</td>\n",
       "      <td>[(0.1% formic acid)]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>Tryptic peptides were generated using a modifi...</td>\n",
       "      <td>[(0.2% formic acid, 1% acetonitrile), (0.2% fo...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>26S proteasome was affinity-purified with anti...</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                      sample_protocol  \\\n",
       "0   The crude membranes from 5 P56-P70 Glun1TAP/TA...   \n",
       "1   Breast cancer tissue lysates, reduction, alkyl...   \n",
       "2   In-gel digests were performed as described in ...   \n",
       "3   Conditioned media preparation and HPLC-isobari...   \n",
       "4   Proteins were eluted from washed beads by addi...   \n",
       "5   Exosomes were isolated from 10 healthy volunte...   \n",
       "6   Low resolution mass spectrometry: For the shot...   \n",
       "7   Arabidopsis plants, Wassilewskija background (...   \n",
       "8   Zebrafish larvae were ground with a pestle in ...   \n",
       "9   Affinity purification and nano-liquid chromato...   \n",
       "10  Protein Extraction from Catheter Biofilm Small...   \n",
       "11  Generation of mDia2-Based Immunocomplexes and ...   \n",
       "12  Bands of interest were dehydrated using aceton...   \n",
       "13  Whole Cell Lysis: 109 cells were lysed in 50 m...   \n",
       "14  See the corresponding sections of the associat...   \n",
       "15            See details in reference PMID: 24460188   \n",
       "16  Exponentially-growing cultures of HeLa S3 cell...   \n",
       "17  For each condition, two 6-well plates of HeLa ...   \n",
       "18  For peptide extraction from moss tissues and p...   \n",
       "19  The kidney mitochondrial fraction was obtained...   \n",
       "20  In-solution tryptic digestion of plasma membra...   \n",
       "21  In order to assess the presence of repair syst...   \n",
       "22  Carbon metabolism in Mycobacterium smegmatis w...   \n",
       "23  100 mg of frontal lobe brain sample from five ...   \n",
       "24  Glomeruli were isolated from wildtype mouse ki...   \n",
       "25  For examining the NO-mediated protective effec...   \n",
       "26  SILAC-labeled MCF7 cells were cultured for thr...   \n",
       "27  Blood was obtained from mice euthanized with C...   \n",
       "28  SDS lysate from skeletal muscle and C2C12 cell...   \n",
       "29  HeLa cells were grown to confluence in Dulbecc...   \n",
       "30  Extracted proteins from liver mouse tissue wer...   \n",
       "31  Datasets of RH strain of T. gondii one-dimensi...   \n",
       "32  Datasets of N. caninum Orbitrap 1DE and Orbitr...   \n",
       "33  Culture media were collected, supplemented wit...   \n",
       "34  Stone matrix protein (SMP) was isolated from k...   \n",
       "35  To identify proteins that are differentially e...   \n",
       "36  Cells from the lysine auxotrophic Saccharomyce...   \n",
       "37  Samples were digested with trypsin then labele...   \n",
       "38  Prior to digestion with trypsin 100 µg of prot...   \n",
       "39  Cell lines from which we generated the super-S...   \n",
       "40  Total plasma protein concentration was assayed...   \n",
       "41  Data analysis.  Spot detection and matching we...   \n",
       "42  Naïve CD4+ T cells were isolated using magneti...   \n",
       "43  Purification of HLA-I complexes. HLA-I peptido...   \n",
       "44  The files were obtained from proteomic prepara...   \n",
       "45  Cells were cultured for 24 hours in serum free...   \n",
       "46  Soluble protein extracts were prepared from th...   \n",
       "47  Duplicate ATCC12228 and PM221 samples acquired...   \n",
       "48  Tryptic peptides were generated using a modifi...   \n",
       "49  26S proteasome was affinity-purified with anti...   \n",
       "\n",
       "                                              buffers  \n",
       "0   [(2% deoxycholate, ), (0.37% w/v sodium), (50%...  \n",
       "1                                                  []  \n",
       "2   [(10 mM DTT), (55 mM iodoacetamide), (100% ace...  \n",
       "3   [(10% FBS), (0.1% SDS, ), (50 mM Tris), (200 m...  \n",
       "4   [(10% Bis-Tris), (5% formic acid), (50% aceton...  \n",
       "5   [(100 mM solution), (250 mM sucrose), (10 mM t...  \n",
       "6   [(2% acetonitrile), (0.1% formic acid), (50% A...  \n",
       "7                                                  []  \n",
       "8   [(20 mM Tris), (20 mM NaCl), (25 mM ammonium),...  \n",
       "9                                                  []  \n",
       "10  [(70 mM dichlorodiphenyltrichloroethane), (1% ...  \n",
       "11  [(50 mM Tris-HCl), (150 mM NaCl, 5 mM EDTA), (...  \n",
       "12  [(10 mM dithiothreitol), (55 mM iodoacetamide)...  \n",
       "13  [(0.1% RapiGest, ), (5 mM TCEP), (10 mM iodoac...  \n",
       "14                                                 []  \n",
       "15                                                 []  \n",
       "16  [(50 mM ammonium), (50 mM ammonium), (50% acet...  \n",
       "17                                  [(0.1% RapiGest)]  \n",
       "18  [(10% acetonitrile), (5% acetonitrile-0), (1% ...  \n",
       "19  [(4% CHAPS, 2% DTT, 0.5% Bio-Lyte), (0.5% CHAP...  \n",
       "20                                                 []  \n",
       "21                                       [(1% PSPEP)]  \n",
       "22                                                 []  \n",
       "23                                       [(0.5% SDS)]  \n",
       "24                                                 []  \n",
       "25  [(200mM of IAM), (100 mM of IAM), (20 mM of as...  \n",
       "26                   [(5% charcoal-treated), (4% FA)]  \n",
       "27  [(5 mM glucose), (0.05% bovine), (10 % SDS-pol...  \n",
       "28                                                 []  \n",
       "29  [(10% fetal), (10% fetal), (50 mM ammonium), (...  \n",
       "30                           [(10 mM DTT), (55mM IA)]  \n",
       "31   [(12% (v/v) acrylamide), (12% (v/v) acrylamide)]  \n",
       "32                                                 []  \n",
       "33  [(20 mM ammonium), (2% w/v CHAPS), (65 mM DTT)...  \n",
       "34  [(50 mM DTT), (120 mM iodoacetamide), (12% Bis...  \n",
       "35                                                 []  \n",
       "36  [(20% (v/v) H2O2), (50 mM Tris-HCl, ), (10% gl...  \n",
       "37                                                 []  \n",
       "38      [(50 mM TEAB), (0.1% RapiGest), (500mM TCEP)]  \n",
       "39  [(20% dialyzed), (1% of tryptic), (0.3% of ide...  \n",
       "40  [(0.1% Rapigest), (50 mM ammonium), (5 mM dith...  \n",
       "41                                                 []  \n",
       "42                                         [(95% as)]  \n",
       "43  [(0.25% sodium), (0.2 mM iodoacetamide, 1 mM E...  \n",
       "44                                 [(35% normalized)]  \n",
       "45  [(21% oxygen), (1% oxygen), (98% incorporation...  \n",
       "46                                                 []  \n",
       "47                               [(0.1% formic acid)]  \n",
       "48  [(0.2% formic acid, 1% acetonitrile), (0.2% fo...  \n",
       "49                                                 []  "
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "method.head(50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Breast cancer tissue lysates, reduction, alkylation by methyl-methanethiosulfonate, trypsin digestion, iTRAQ 8-plex labeling, HILIC fractionation, LC-MS/MS on Orbitrap Velos.  Transcriptomics analysis, immunohistochemistry. Detailed description provided in the manuscript.'"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "method.iloc[1, 0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
