{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import re\n",
    "import numpy\n",
    "import pickle"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "from nltk.corpus import stopwords\n",
    "import spacy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.feature_extraction.text import TfidfVectorizer, CountVectorizer"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "df =  pd.read_csv('base_data/pride_table.csv').astype(str)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>dataset_id</th>\n",
       "      <th>sample_protocol</th>\n",
       "      <th>data_protocol</th>\n",
       "      <th>description</th>\n",
       "      <th>instruments</th>\n",
       "      <th>exp_types</th>\n",
       "      <th>quant_methods</th>\n",
       "      <th>labhead_fullname</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PXD000001</td>\n",
       "      <td>Not available</td>\n",
       "      <td>Two extra files have been added post-publicati...</td>\n",
       "      <td>Expected reporter ion ratios: Erwinia peptides...</td>\n",
       "      <td>LTQ Orbitrap Velos, instrument model</td>\n",
       "      <td>Bottom-up proteomics</td>\n",
       "      <td>nan</td>\n",
       "      <td>nan</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PXD000002</td>\n",
       "      <td>Not available</td>\n",
       "      <td>Not available</td>\n",
       "      <td>Human saliva samples from a 36 year old male w...</td>\n",
       "      <td>instrument model, LTQ Velos</td>\n",
       "      <td>Bottom-up proteomics</td>\n",
       "      <td>nan</td>\n",
       "      <td>nan</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  dataset_id sample_protocol  \\\n",
       "0  PXD000001   Not available   \n",
       "1  PXD000002   Not available   \n",
       "\n",
       "                                       data_protocol  \\\n",
       "0  Two extra files have been added post-publicati...   \n",
       "1                                      Not available   \n",
       "\n",
       "                                         description  \\\n",
       "0  Expected reporter ion ratios: Erwinia peptides...   \n",
       "1  Human saliva samples from a 36 year old male w...   \n",
       "\n",
       "                            instruments             exp_types quant_methods  \\\n",
       "0  LTQ Orbitrap Velos, instrument model  Bottom-up proteomics           nan   \n",
       "1           instrument model, LTQ Velos  Bottom-up proteomics           nan   \n",
       "\n",
       "  labhead_fullname  \n",
       "0              nan  \n",
       "1              nan  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "method = df['sample_protocol'][(df['sample_protocol'] != 'Not available') & (df['sample_protocol'] != 'nan')].reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# list of stop words\n",
    "stop_words = stopwords.words('english')\n",
    "\n",
    "# Import spacy model\n",
    "nlp = spacy.load('en_core_web_lg')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "def lemmatize_text(text):\n",
    "    doc = nlp(text)\n",
    "    lemm_text = [token.lemma_.lower() for token in doc]\n",
    "    \n",
    "    lemm_text = ' '.join(lemm_text)\n",
    "    lemm_text = lemm_text.replace(' - ', '-')\n",
    "    lemm_text = lemm_text.replace(' .', '.')\n",
    "    lemm_text = lemm_text.replace(' ,', ',')\n",
    "    lemm_text = lemm_text.replace('( ', '(')\n",
    "    lemm_text = lemm_text.replace(' )', ')')\n",
    "    lemm_text = lemm_text.replace(' / ', '/')\n",
    "    \n",
    "    lemm_text = lemm_text.replace('\\u2009', '')    # This is a special case applicable to iloc[0]\n",
    "    \n",
    "    return lemm_text"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'In-gel digests were performed as described in standard protocols. Briefly, following the SDS-PAGE and washing of the excised gel slices proteins were reduced by adding 10 mM DTT (Sigma Aldrich) prior to alkylation with 55 mM iodoacetamide (Sigma Aldrich). After washing and shrinking of the gel pieces with 100% acetonitrile trypsin (Sequencing Grade Modified, Promega) was added and proteins were digested overnight in 40 mM ammoniumbicarbonate at 37°C. For protein identification samples were directly used for nano-ESI-LC-MS/MS. Each sample was first desalted on-line by a C18 microcolumn (300 µm i.d. x 5 mm, packed with C18 PepMap™, 5 µm, 100 Å by LC Packings). Peptides were then separated on a C18 reversed phase column via a linear acetonitrile gradient (UltiMate 3000 system (Dionex) and column (75 µm i.d. x 15 cm, packed with C18 PepMap™, 3 µm, 100 Å by LC Packings) before MS and MS/MS spectra were recorded on an Oribitrap mass spectrometer (Thermo Electron). '"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test_text = method.iloc[2]\n",
    "test_text"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'in-gel digest be perform as describe in standard protocol. briefly, follow the sds-page and washing of the excise gel slice protein be reduce by add 10 mm dtt (sigma aldrich) prior to alkylation with 55 mm iodoacetamide (sigma aldrich). after wash and shrink of the gel piece with 100 % acetonitrile trypsin (sequencing grade modified, promega) be add and protein be digest overnight in 40 mm ammoniumbicarbonate at 37 ° c. for protein identification sample be directly use for nano-esi-lc-ms/ms. each sample be first desalt on-line by a c18 microcolumn (300 µm i.d. x 5 mm, pack with c18 pepmap ™, 5 µm, 100 å by lc packings). peptide be then separate on a c18 reverse phase column via a linear acetonitrile gradient (ultimate 3000 system (dionex) and column (75 µm i.d. x 15 cm, pack with c18 pepmap ™, 3 µm, 100 å by lc packings) before ms and ms/ms spectra be record on an oribitrap mass spectrometer (thermo electron).'"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "lemmatize_text(test_text)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "test_doc = [lemmatize_text(method.iloc[i]) for i in range(10)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['the crude membrane from 5 p56-p70 glun1tap/tap mouse forebrain be re-suspend in 12.5  ml buffer h and extract with 12.5  ml 2 % deoxycholate, 100  mm nacl, 50  mm tris. cl ph8 for 1  h at 6  ° c. total extract be centrifuge at 120,000 g. for 40  min at 8  ° c. condition for immuno-capture, wash and peptide-antigen exchange elution be screen use a high-throughput purification robot (magic sample processor, invitrogen). for 25  ml glun1tap/tap extract supernatant, 80  μg mouse flag antibody be couple to 30  mg (500  μl) protein g magnetic bead (invitrogen). receptor be capture from extract supernatant for 2  h. the bead be wash three time with 5  ml wash buffer (0.37 % w/v sodium deoxycholate, 0.05  mg.ml−1 lipid (1:1:3 popc : pope : pog), 150  mm nacl, 50  mm tris. cl ph8). flag capture complex be elute with 600  μl wash buffer supplement with 0.2  mg.ml−1 flag peptide for 2  h at 6  ° c. eluate be buffer exchange to remove the flag elution peptide and concentrate with a 100-kda mwco centrifugal filter (amicon) to 35  μl 0.1–0.25  mg.ml−1 native nmdar. bnp be run accord to schägger. excise bnp gel slice contain the purify protein be prepare for mass spectrometric analysis by manual in situ enzymatic digestion. briefly, the excise protein gel piece be place in a well of a 96-well microtitre plate and de-stain with 50 % v/v acetonitrile and 50  mm ammonium bicarbonate, reduce with 10  mm dtt and alkylat with 55  mm iodoacetamide. after alkylation, protein be digest with 6  ng  μl−1 trypsin (promega, uk) overnight at 37  ° c. the result peptide be extract in 2 % v/v formic acid, 2 % v/v acetonitrile. the digest be analyse by nano-scale capillary lc-ms/ms use an ultimate u3000 hplc (thermoscientific dionex, san jose, usa) to deliver a flow of ~300  nl  min−1. a c18 acclaim pepmap100 5  μm, 100  μm × 20  mm nanoviper (thermoscientific dionex, san jose, usa), trap the peptide before separation on a c18 acclaim pepmap100 3  μm, 75  μm × 150  mm nanoviper (thermoscientific dionex, san jose, usa). peptide be elute with a gradient of acetonitrile. the analytical column outlet be directly interfac via a modify nano-flow electrospray ionization source, with a hybrid dual pressure linear ion trap mass spectrometer (orbitrap velos, thermoscientific, san jose, usa).',\n",
       " 'breast cancer tissue lysate, reduction, alkylation by methyl-methanethiosulfonate, trypsin digestion, itraq 8-plex labeling, hilic fractionation, lc-ms/ms on orbitrap velos.   transcriptomics analysis, immunohistochemistry. detailed description provide in the manuscript.',\n",
       " 'in-gel digest be perform as describe in standard protocol. briefly, follow the sds-page and washing of the excise gel slice protein be reduce by add 10 mm dtt (sigma aldrich) prior to alkylation with 55 mm iodoacetamide (sigma aldrich). after wash and shrink of the gel piece with 100 % acetonitrile trypsin (sequencing grade modified, promega) be add and protein be digest overnight in 40 mm ammoniumbicarbonate at 37 ° c. for protein identification sample be directly use for nano-esi-lc-ms/ms. each sample be first desalt on-line by a c18 microcolumn (300 µm i.d. x 5 mm, pack with c18 pepmap ™, 5 µm, 100 å by lc packings). peptide be then separate on a c18 reverse phase column via a linear acetonitrile gradient (ultimate 3000 system (dionex) and column (75 µm i.d. x 15 cm, pack with c18 pepmap ™, 3 µm, 100 å by lc packings) before ms and ms/ms spectra be record on an oribitrap mass spectrometer (thermo electron).',\n",
       " 'conditioned medium preparation and hplc-isobaric tag for relative and absolute quantitation mass-spectrometry (itraq ms). twenty million cell be transfect with either control or gαs sirna, then be seed in 100-mm cell culture plate in dmem + 10 % fbs. two day later, cell be wash twice with pre-warm pbs. conditioned medium be harvest another 24 h later, and centrifuge at 20,000 g for 10 min at 4 ° c to remove cell debris. to enrich for secrete protein in the condition medium, condition medium sample be centrifuge use amicon centrifugal filter with a 3kda cutoff (millipore, billerica, ma). a total of 8 concentrate condition medium sample be independently prepare : 4 sample from control sirna-treat cell, and 4 sample from gαs sirna-treat cell. sample be store at –70 ° c until -pron- be analyze by itraq labeling and lc-ms/ms. itraq label – total protein concentration of each sample be measure use the bradford assay. a quantity of 100 µg be take from each sample, denature in 0.1 % sds, reduce in 50 mm tris(2-carboxyethyl)phosphine (tcep), alkylat in 200 mm methyl methanethiosulfonate (mmts), and digest overnight with trypsin. each sample ’s peptide be label with different itraq reagent (ab sciex, framingham, ma) follow the manufacturer ’s protocol, combine, dry down to a 50 µl volume, acidify with 1 % trifluoroacetic acid (tfa), and desalt with a 100 mg-c18 seppak cartridge (waters, milford, ma). peptide fractionation – combine itraq label peptide be first separate into 12 fraction use a 3100 offgel fractionator (agilent, santa clara, ca) with a 12-well setup and a 12-cm ph 3–10 immobilized ph-gradient (ipg) strip, accord to manufacturer ’s recommendation. peptide be collect after the focus voltage reach a maximum plateau (> 50 kvh). the recovered fraction be lyophilize. lc-maldi-tof/tof analysis – all solvent use in this section be aqueous unless specify and obtain from thermofisher scientific (san jose, ca). dry sample be reconstitute in 30 µl of solvent a (2 % v/v acetonitrile [ acn ], 0.1 % v/v tfa), and 10 µl of the result solution be inject into an ultimate ™ hplc system with uv detection (lc packings, sunnyvale, ca). solvent b contain 98 % v/v acn, 0.1 % v/v tfa. a c18 trap (5.0 mm x 300 µm id, lc packings) and a 15 cm x 100 µm id column pack in-house with magic c18 5 µm bead (michrom bioresources, auburn, ca) be use for chromatography. solvent flow rate be set to 0.700 µl/min. after injection, each sample be load and wash with solvent a for 5 min at a flow rate of 0.030 ml/min, follow by a 10 % to 50 % gradient over 55 min, then ramp to 90 % by 65 min. each sample run be maintain in 90 % solvent b until 75 min, return to 10 % at 75.1 min, follow by a 10 min equilibration in 10 % solvent b. the hplc system be connect to a modify probot ™ automate maldi spotter (lc packings) to spot eluate on an lc maldi plate with on-line mixing of α-cyano-4-hydroxycinnamic acid (chca) matrix (7.5 mg/ml in 70 % acn). consider the dead time for the hplc and spot system, spot collection time be start 10 min from injection time with a spot time of 0.33 min per spot. a total of 176 spot be collect. the sample plate with the result spot be load onto a 5800 maldi time-of-flight/time-of-flight (tof/tof) mass spectrometer (ab sciex, framingham, ma) for ms and ms/ms analysis. batch-mode mass calibration with plate alignment be do before datum acquisition, thus a separate plate calibration file be generate for each individual plate contain sample spot. full scan ms spectra be obtain with a total of 1000 laser shot per spectrum. from each spot, the top 20 most intense peak, weak precursor first, that be above a signal-to-noise threshold of 10, be select for successful ms/ms analysis. each ms/ms spectrum be an average of 1600 laser shot. ms/ms spectra be acquire in the presence of collision gas. the sequence of the vgf-nerp1 peptide be confirm use the ltq-orbitrap mass spectrometer (thermofisher scientific, san jose, ca).',\n",
       " 'protein be elute from washed bead by add 2-fold-concentrate sds-gel sample loading buffer and heating to 80 ° c for 10 min. the supernatant be load onto a novex gel (10 % bis-tris sds gel ; life technologies/invitrogen, carlsbad, ca) and run about 1/3 of the length of the lane. the lane be cut out and stain with instantblue (expedeon ltd., harston, united kingdom) in separate tray until band be just visible. the area cover the band be cut into 5 to 6 slice, which be wash, reduce, and alkylat and treat with trypsin accord to standard procedure. peptide be extract with 5 % formic acid–50 % acetonitrile, dry down, and redissolv in 0.1 % trifluoroacetic acid (tfa). for liquid chromatography-tandem mass spectrometry (lc-ms/ms) analysis, a sample aliquot be apply via a nanoacquity (waters, manchester, united kingdom) ultraperformance liquid chromatography (uplc) system run at a flow rate of 250 nl min−1 to an ltq-orbitrap mass spectrometer (thermo fisher, waltham, ma). peptide be trap use a precolumn (symmetry c18, 5 μm, 180 μm by 20 mm ; waters), which be then switch in-line to an analytical column (beh c18, 1.7 μm, 75 μm by 250 mm ; waters) for separation. peptide be elute with a gradient of 3 to 40 % acetonitrile in water–0.1 % formic acid at a rate of 0.67 % min−1. the column be connect to a 10-μm silicatip nanospray emitter (new objective, woburn, ma, usa) attach to a nanospray interface (proxeon, odense, denmark) for infusion into the mass spectrometer. the mass spectrometer be operate in positive ion mode at a capillary temperature of 200 ° c. the source voltage and focus voltage be tune for the transmission of met-arg-phe-ala (mrfa) peptide (m/z 524) (sigma-aldrich, st. louis, mo). datum-dependent analysis be carry out in orbitrap-ion trap parallel mode use collision-induce fragmentation (cid) on the 6 most abundant ion in each cycle. the orbitrap be run with a resolution of 30,000 over the ms range from m/z 350 to m/z 1,800 and an ms target of 106 and 1-s maximum scan time. collision energy be 35, and an isolation width of 2 be use. only monoisotopic 2 + and 3 + charge precursor be select for ms2 fragmentation stage. the ms2 be trigger by a minimal signal of 1,000 with an automatic gain control target of 3 × 104 ion and 150-ms scan time use the chromatography function for peak apex detection. dynamic exclusion be set to 1 count and 60-s exclusion with an exclusion mass window of ±20 ppm. ms scan be save in profile mode, while ms/ms scan be save in centroid mode.',\n",
       " 'exosome be isolate from 10 healthy volunteer urine sample as previously describe. the five man and five woman age 23–36 year be on no regular medication and have not consume antibiotic or other medication within the previous 1 month. current uti be exclude. briefly, subject urinate directly into a container with protease inhibitor, include pmsf (500 μl 0.5 m solution), leupeptin (450 μg), and sodium azide (15 ml 100 mm solution). urine be centrifuge within 30 minute of collection (beckman avanti j26-xp centrifuge ; ja-17 fix angle rotor ; polyallomer 50-ml centrifuge bottle) for 20 minute at 17,000×g. the supernatant be pass through a sterile 0.22-µm filter and ultracentrifuged (beckman optim l-100 xp vac ultracentrifuge ; ti45 fix-angle titanium rotor ; beckman 70-ml polycarbonate ultracentrifuge bottle) for 135 minute at 235,000×g and 4 ° c. each ultracentrifugation pellet be suspend in 50 µl suspension buffer (250 mm sucrose and 10 mm triethanolamine [ ph 7.6 ]) and pool with the other pellet from the same urine sample. to avoid confound effect on bacterial growth, urine sample for these experiment be collect without protease inhibitor or sodium azide, triethanolamine be remove from the suspension buffer, and exosomal pellet be wash one time by resuspension and ultracentrifugation. protein from resuspended exosomal pellet be concentrate by precipitation and quantify use a bradford protein bind colorimetric assay (bio-rad). protein pellet be store at −80 ° c until use. for ms, protein pellet be suspend in laemmli sample buffer and incubate at 95 ° c ; 50 µg protein from each sample be fragment on a 4%/12 % sds – polyacrylamide gel. after stain, each gel track be separate into 28 equal section, which be process individually for the remainder of the workflow. after destain, protein be reduce and alkylat in gel, wash with nh4hco3, and dehydrate in acetonitrile, and protein be digest with modify trypsin. liquid chromatography – ms/ms be perform use an eksigent nanolc-1d plus (eksigent technologies) hplc system and an ltq orbitrap mass spectrometer (thermo fisher scientific). peptide be separate by reverse-phase chromatography (dionex). peptide be load onto a 5-cm c18 precolumn (300 μm inner diameter ; lc packings) from the autosampler. peptide be elute onto the analytical column use gradient for solvent a (water+0.1 % formic acid) and b (acetonitrile+0.1 % formic acid) of 5%–50 % b over 40 minute. a new-objective nanospray source be use for electrospray ionization. m/z value of elute ion be measure in the orbitrap mass analyzer with a mass range of 350–1600, and the resolution be set at 7500.',\n",
       " \"low resolution mass spectrometry : for the shotgun proteomic analysis, six cell pellet be subject to differential detergent fractionation as describe by mccarthy et al. use the detergent digitonin, tween 20 and sds. after each detergent application, sample be centrifuge to separate solubliz protein from cellular debris. the insoluble pellet leave after treatment be subject to trypsin digestion along with the soluble fraction, but could not be quantify. fraction be normalize to 20 μg each and trypsin digestion as describe by mccarthy et al. follow digestion, each fraction be desalt use a peptide microtrap (michrom bioresources) accord to the manufacturer 's instruction. after desalt, each fraction be further clean use a strong cation exchange (scx) microtrap (michrom bioresources) to remove any residual detergent, which could interfere with the mass spectrometry. fraction be dry and resuspend in 10 μl of 2 % acetonitrile (acn), 0.1 % formic acid (fa) and transfer to low retention vial in preparation for separation use 1d-lc. the high performance liquid chromatography (hplc) equipment use for peptide separation be an ultimate 3000 (dionex) operate in 1d-lc mode at a flow rate of 333 nl per min and equip with a 0.075 mm x 100 mm column pack with halo c18 material (michrom bioresources) for reverse phase separation. each sample be separate use a 4 h gradient from 2 % to 50 % acetonitrile with 0.1 % formic acid as a proton source. the column be locate on the ion source and connect directly to a nanospray emitter to minimize peak broadening. scan parameter for the ltq velos pro be one ms scan follow by 20 ms/ms scan of the 20 most intense peak use high energy collisional dissociation as the fragmentation method. dynamic exclusion be enable with a mass exclusion time of 3 min and a repeat count of 1 within 30 sec of initial m/z measurement. high resolution mass spectrometry : the two cell pellet reserve for high resolution analysis be lys and digest as describe by wilton et al. digest peptide be dry, resuspend in 20 mm kh2po4, 20 % acn, ph 3 (buffer a) in 2.5 μl and transfer to low retention vial in preparation for separation use an ultimate 3000 configure for 2d-lc. each sample be load at 15 μl/min onto an scx microtrap (michrom bioresources) for the first dimension of separation, involve scx step of buffer a plus 0, 5, 10, 15, 20, 25, 30, 40, 50, 100, 250, 500, and 1000 mm kcl. for the second dimension of separation, each elute salt step be desalt with an inline peptide microtrap (michrom bioresources) with 2 % acn, 0.1 % fa at 5 μl/min. once desalt, the microtrap be switch into line with a fritless nano column (75 μm x ~10 cm) contain c18 medium (5 μ, 200 å magic, michrom) manufacture accord to gatlin. peptide be elute use a gradient of 2 % to 36 % acn, 0.1 % fa at 350 nl/min over 60 min and electrospray ionize for analysis use an ltq ft ultra mass spectrometer. a survey scan m/z 350–1750 be acquire in the ft ion cyclotron resonance cell (resolution = 100,000 at m/z 400, with an accumulation target value of 1,000,000 ion). up to the 6 most abundant ion (> 3,000 count) with charge state > + 2 be sequentially isolate and fragment within the linear ion trap use collisionally induce dissociation with an activation q = 0.25 and activation time of 30 ms at a target value of 30,000 ion. m/z ratio select for ms/ ms be dynamically exclude for 30 second.\",\n",
       " 'arabidopsis plant, wassilewskija background (ws), be grow in culture chamber at 23 ° c (12-h light cycle) with a light intensity of 150 μmol·m−2·s−1 in standard condition. percoll-purify chloroplast be obtain from 100–200 g of a. thaliana leaf. purify intact chloroplast be then lys in hypotonic medium, and envelope, thylakoid, and stroma subfraction be purify on a sucrose gradient. sample be separate by sds-page in the conventional separate mode. after coomassie blue stain to reveal protein, the gel be cut into discrete band. in-gel digestion with trypsin (sequencing grade ; promega, madison, wi) be carry out. approximately 500 ng of digest sample be inject for each lc-ms/ms analysis perform on a 7-tesla hybrid linear ion trap fourier transform mass spectrometer (ltq-ft, thermo, bremen, germany). the experimental sequence consist of one high resolution ms acquisition in the icr cell and three ms/ms scan in the linear ion trap in parallel with the ms acquisition. dynamic exclusion be activate for ion within 5 ppm of a select peak and eluting in a 3-min window, and one repeat scan be allow within 30 s. an “ ultimate 3000 ” nano-hplc system (lc packings, amsterdam, the netherlands) equip with a dynamic flow control and a pepmap (lc packings) column (15 cm, 75-μm diameter, 3-μm c18 particle, 100-å pore size) be couple to the ft-icr instrument.',\n",
       " 'zebrafish larvae be grind with a pestle in liquid nitrogen in 1.5 ml eppendorf tube and vortex for 30 second in a lysis buffer consist of 9 part 20 mm tris – hcl, ph 8.5, 20 mm nacl and 1 part protease inhibitor cocktail (p 8340, sigma-aldrich). the sample be place on a shake table for 20 minute at room temperature before spin down cellular debris at 16,100  ×  g for 10 minute at 4 ° c. the supernatant be transfer to a fresh tube and the supernatant as well as the remain pellet freeze in −35 ° c. five μl nupage ® lsd sample buffer (invitrogen) be add to 20 μl, 20 μg protein, as measure by the bicinchoninic acid (bca) protein assay (thermoscientific), and load in each lane on a 10-lane 4-12 % nupage ® bis-tris gel. the assemble invitrogen xcell surelock ™ mini-cell be fill with sds-mop run buffer and the separation run for 1 hour at 200 v. the gel be release from -pron- casing and immerse in nupage ® surestain colloidal blue (55 ml deionize water, 20 ml methanol, 20 ml “ stain a ” and 5 ml “ stain b ”) and leave on a shaker overnight at room temperature. the stain solution be decant and the gel be wash with deionized water before be scan in an optigo uv imager (isogen life science, de meern, the netherlands). the gel lane be cut into 48 identical slice fraction use a custom-make onetouch mount and lane picker (the gel company, san francisco, ca). the gel slice be then remove with a acu-min ® stainless steel precision probe with a 35 ° single bend tip (# 6) (moody tools inc., warwick, ri) and each slice place into one well in a 96-well polypropylene pcr plate (greiner bio-one, frickenhausen germany). use 50 and 250 μl rainin 8-channel multi-pipettor, each gel slice be wash for 5 minute at room temperature with 100 μl 25 mm ammonium bicarbonate (abc), follow by a wash with 100 μl 30 % acetonitrile in 25 mm abc for 10 minute and a final wash of 100 μl 80 % acetonitrile in 25 mm abc for 10 minute, discard the solution between each wash. cystine be reduce by addition of 75 μl 10 mm dtt in 25 mm abc and the plate incubate at 56 ° c for 20 minute. after discard the dtt solution, cysteine be alkylat by addition of 75 μl 55 mm iodoacetamide in 25 mm abc with incubation at room temperature in darkness for 20 minute. the gel piece be wash again as above and the supernatant discard. the protein be digest in-gel by addition of 15 μl 5 ng/μl porcine trypsin (promega, madison, wi) and incubation for 6 hour at 37 ° c, after which the digestion be quench with the addition of 1 μl 5 % trifluoroacetic acid (tfa). the solution be remove, follow by a second extraction with 20 μl 0.1 % tfa for 1 h and the solution pool with the first extraction. the plate be store at −35 ° c until analysis by lc-ms/ms.',\n",
       " 'affinity purification and nano-liquid chromatography/tandem mass spectrometry be use for explore novel sirt6 and sirt7-interact partner which be validate by co-immunoprecipitation, immunoblot and confocal microscopy. take together, -pron- study have identify novel sirt6 and sirt7-interact protein.']"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test_doc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "vectorizer = TfidfVectorizer(\n",
    "    analyzer='word', \n",
    "    strip_accents='ascii',\n",
    "    ngram_range=(2, 2),\n",
    "    stop_words='english')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "vectorizer.fit(test_doc)\n",
    "bigram_names = vectorizer.get_feature_names()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1877\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "['000 000',\n",
       " '000 10',\n",
       " '000 40',\n",
       " '000 400',\n",
       " '000 automatic',\n",
       " '000 count',\n",
       " '000 ion',\n",
       " '000 ms',\n",
       " '000g supernatant',\n",
       " '000g ultracentrifugation',\n",
       " '030 ml',\n",
       " '05 mg',\n",
       " '075 mm',\n",
       " '10 15',\n",
       " '10 25',\n",
       " '10 50',\n",
       " '10 75',\n",
       " '10 acetonitrile',\n",
       " '10 bis',\n",
       " '10 cm',\n",
       " '10 fbs',\n",
       " '10 healthy',\n",
       " '10 lane',\n",
       " '10 min',\n",
       " '10 minute',\n",
       " '10 mm',\n",
       " '10 result',\n",
       " '10 select',\n",
       " '10 silicatip',\n",
       " '10 solvent',\n",
       " '100 000',\n",
       " '100 10',\n",
       " '100 20',\n",
       " '100 25',\n",
       " '100 250',\n",
       " '100 30',\n",
       " '100 80',\n",
       " '100 acetonitrile',\n",
       " '100 id',\n",
       " '100 kda',\n",
       " '100 lc',\n",
       " '100 mg',\n",
       " '100 mm',\n",
       " '100 pore',\n",
       " '100 sample',\n",
       " '100 xp',\n",
       " '1000 laser',\n",
       " '1000 mm',\n",
       " '100200 thaliana',\n",
       " '104 ion',\n",
       " '106 maximum',\n",
       " '12 cm',\n",
       " '12 fraction',\n",
       " '12 light',\n",
       " '12 ml',\n",
       " '12 nupage',\n",
       " '12 sds',\n",
       " '12 setup',\n",
       " '120 000',\n",
       " '135 minute',\n",
       " '15 20',\n",
       " '15 cm',\n",
       " '15 min',\n",
       " '15 ml',\n",
       " '15 ng',\n",
       " '150 mm',\n",
       " '150 molm2s1',\n",
       " '150 ms',\n",
       " '16 100',\n",
       " '1600 laser',\n",
       " '17 000g',\n",
       " '17 fix',\n",
       " '176 spot',\n",
       " '180 20',\n",
       " '1d lc',\n",
       " '1d plus',\n",
       " '20 000',\n",
       " '20 20',\n",
       " '20 25',\n",
       " '20 acn',\n",
       " '20 intense',\n",
       " '20 minute',\n",
       " '20 ml',\n",
       " '20 mm',\n",
       " '20 ms',\n",
       " '20 ppm',\n",
       " '20 protein',\n",
       " '20 sds',\n",
       " '20 tfa',\n",
       " '20 trypsin',\n",
       " '200 gel',\n",
       " '200 magic',\n",
       " '200 mm',\n",
       " '200 source',\n",
       " '22 filter',\n",
       " '23 12',\n",
       " '2336 year',\n",
       " '235 000g',\n",
       " '24 later',\n",
       " '25 30',\n",
       " '25 activation',\n",
       " '25 mg',\n",
       " '25 ml',\n",
       " '25 mm',\n",
       " '250 500',\n",
       " '250 mm',\n",
       " '250 nl',\n",
       " '250 rainin',\n",
       " '28 equal',\n",
       " '2d lc',\n",
       " '30 000',\n",
       " '30 40',\n",
       " '30 acetonitrile',\n",
       " '30 mg',\n",
       " '30 minute',\n",
       " '30 ms',\n",
       " '30 sec',\n",
       " '30 second',\n",
       " '30 solvent',\n",
       " '30 ultimate',\n",
       " '300 id',\n",
       " '300 inner',\n",
       " '300 mm',\n",
       " '300 nl',\n",
       " '3000 configure',\n",
       " '3000 dionex',\n",
       " '3000 nano',\n",
       " '310 immobilized',\n",
       " '3100 offgel',\n",
       " '33 min',\n",
       " '333 nl',\n",
       " '35 10',\n",
       " '35 analysis',\n",
       " '35 isolation',\n",
       " '35 nupage',\n",
       " '35 single',\n",
       " '350 800',\n",
       " '350 nl',\n",
       " '3501600 resolution',\n",
       " '3501750 acquire',\n",
       " '36 acn',\n",
       " '37 digestion',\n",
       " '37 protein',\n",
       " '37 result',\n",
       " '37 sodium',\n",
       " '3kda cutoff',\n",
       " '40 50',\n",
       " '40 acetonitrile',\n",
       " '40 min',\n",
       " '40 minute',\n",
       " '40 mm',\n",
       " '400 accumulation',\n",
       " '450 sodium',\n",
       " '48 identical',\n",
       " '50 100',\n",
       " '50 250',\n",
       " '50 40',\n",
       " '50 acetonitrile',\n",
       " '50 gradient',\n",
       " '50 kvh',\n",
       " '50 ml',\n",
       " '50 mm',\n",
       " '50 protein',\n",
       " '50 suspension',\n",
       " '50 volume',\n",
       " '500 1000',\n",
       " '500 ng',\n",
       " '500 protein',\n",
       " '500 solution',\n",
       " '524 sigma',\n",
       " '55 min',\n",
       " '55 ml',\n",
       " '55 mm',\n",
       " '56 20',\n",
       " '5800 maldi',\n",
       " '60 exclusion',\n",
       " '60 min',\n",
       " '600 wash',\n",
       " '65 min',\n",
       " '67 min1',\n",
       " '70 acn',\n",
       " '70 ml',\n",
       " '70 pron',\n",
       " '700 min',\n",
       " '75 10',\n",
       " '75 15',\n",
       " '75 150',\n",
       " '75 250',\n",
       " '75 55',\n",
       " '75 diameter',\n",
       " '75 min',\n",
       " '80 10',\n",
       " '80 acetonitrile',\n",
       " '80 mouse',\n",
       " '80 use',\n",
       " '800 ms',\n",
       " '8340 sigma',\n",
       " '90 65',\n",
       " '90 solvent',\n",
       " '95 50',\n",
       " '96 microtitre',\n",
       " '96 polypropylene',\n",
       " '98 acn',\n",
       " 'TM 100',\n",
       " 'TM automate',\n",
       " 'TM hplc',\n",
       " 'TM mini',\n",
       " 'ab sciex',\n",
       " 'abc 10',\n",
       " 'abc follow',\n",
       " 'abc incubation',\n",
       " 'abc plate',\n",
       " 'absolute quantitation',\n",
       " 'abundant ion',\n",
       " 'acclaim pepmap100',\n",
       " 'accord gatlin',\n",
       " 'accord manufacturer',\n",
       " 'accord schagger',\n",
       " 'accord standard',\n",
       " 'accumulation target',\n",
       " 'acetonitrile 25',\n",
       " 'acetonitrile 50',\n",
       " 'acetonitrile acn',\n",
       " 'acetonitrile analytical',\n",
       " 'acetonitrile digest',\n",
       " 'acetonitrile dry',\n",
       " 'acetonitrile formic',\n",
       " 'acetonitrile gradient',\n",
       " 'acetonitrile protein',\n",
       " 'acetonitrile trypsin',\n",
       " 'acetonitrile water0',\n",
       " 'acid 50',\n",
       " 'acid acetonitrile',\n",
       " 'acid bca',\n",
       " 'acid chca',\n",
       " 'acid fa',\n",
       " 'acid proton',\n",
       " 'acid rate',\n",
       " 'acid tfa',\n",
       " 'acid50 acetonitrile',\n",
       " 'acidify trifluoroacetic',\n",
       " 'acn consider',\n",
       " 'acn fa',\n",
       " 'acn formic',\n",
       " 'acn ph',\n",
       " 'acn tfa',\n",
       " 'acquire ft',\n",
       " 'acquire presence',\n",
       " 'acquisition dynamic',\n",
       " 'acquisition icr',\n",
       " 'acquisition separate',\n",
       " 'activate ion',\n",
       " 'activation 25',\n",
       " 'activation time',\n",
       " 'acu min',\n",
       " 'add 10',\n",
       " 'add 20',\n",
       " 'add fold',\n",
       " 'add protein',\n",
       " 'addition 15',\n",
       " 'addition 75',\n",
       " 'addition trifluoroacetic',\n",
       " 'affinity purification',\n",
       " 'age 2336',\n",
       " 'agilent santa',\n",
       " 'al digest',\n",
       " 'al follow',\n",
       " 'al use',\n",
       " 'ala mrfa',\n",
       " 'aldrich prior',\n",
       " 'aldrich sample',\n",
       " 'aldrich st',\n",
       " 'aldrich wash',\n",
       " 'alignment datum',\n",
       " 'aliquot apply',\n",
       " 'alkylat 200',\n",
       " 'alkylat 55',\n",
       " 'alkylat addition',\n",
       " 'alkylat gel',\n",
       " 'alkylat treat',\n",
       " 'alkylation 55',\n",
       " 'alkylation methyl',\n",
       " 'alkylation protein',\n",
       " 'allow 30',\n",
       " 'amicon 35',\n",
       " 'amicon centrifugal',\n",
       " 'ammonium bicarbonate',\n",
       " 'ammoniumbicarbonate 37',\n",
       " 'amsterdam netherlands',\n",
       " 'analyse nano',\n",
       " 'analysis batch',\n",
       " 'analysis carry',\n",
       " 'analysis cell',\n",
       " 'analysis immunohistochemistry',\n",
       " 'analysis lc',\n",
       " 'analysis lys',\n",
       " 'analysis manual',\n",
       " 'analysis ms',\n",
       " 'analysis perform',\n",
       " 'analysis sample',\n",
       " 'analysis solvent',\n",
       " 'analysis use',\n",
       " 'analytical column',\n",
       " 'analyze itraq',\n",
       " 'analyzer mass',\n",
       " 'angle rotor',\n",
       " 'angle titanium',\n",
       " 'antibiotic medication',\n",
       " 'antibody couple',\n",
       " 'antigen exchange',\n",
       " 'apex detection',\n",
       " 'application sample',\n",
       " 'apply nanoacquity',\n",
       " 'approximately 500',\n",
       " 'aqueous unless',\n",
       " 'arabidopsis plant',\n",
       " 'area cover',\n",
       " 'arg phe',\n",
       " 'assay bio',\n",
       " 'assay quantity',\n",
       " 'assay thermoscientific',\n",
       " 'assemble invitrogen',\n",
       " 'attach nanospray',\n",
       " 'auburn ca',\n",
       " 'automate maldi',\n",
       " 'automatic gain',\n",
       " 'autosampler peptide',\n",
       " 'avanti j26',\n",
       " 'average 1600',\n",
       " 'avoid confound',\n",
       " 'azide 15',\n",
       " 'azide triethanolamine',\n",
       " 'background ws',\n",
       " 'bacterial growth',\n",
       " 'band cut',\n",
       " 'band gel',\n",
       " 'band just',\n",
       " 'batch mode',\n",
       " 'bca protein',\n",
       " 'bead add',\n",
       " 'bead invitrogen',\n",
       " 'bead michrom',\n",
       " 'bead wash',\n",
       " 'beckman 70',\n",
       " 'beckman avanti',\n",
       " 'beckman optim',\n",
       " 'beh c18',\n",
       " 'bend tip',\n",
       " 'bicarbonate abc',\n",
       " 'bicarbonate reduce',\n",
       " 'bicinchoninic acid',\n",
       " 'billerica ma',\n",
       " 'bind colorimetric',\n",
       " 'bio frickenhausen',\n",
       " 'bio rad',\n",
       " 'bioresources accord',\n",
       " 'bioresources acn',\n",
       " 'bioresources auburn',\n",
       " 'bioresources dimension',\n",
       " 'bioresources remove',\n",
       " 'bioresources reverse',\n",
       " 'bis tris',\n",
       " 'blue 55',\n",
       " 'blue stain',\n",
       " 'bnp gel',\n",
       " 'bnp run',\n",
       " 'bottle 135',\n",
       " 'bottle 20',\n",
       " 'bradford assay',\n",
       " 'bradford protein',\n",
       " 'breast cancer',\n",
       " 'bremen germany',\n",
       " 'briefly excise',\n",
       " 'briefly follow',\n",
       " 'briefly subject',\n",
       " 'broadening scan',\n",
       " 'buffer 250',\n",
       " 'buffer 37',\n",
       " 'buffer consist',\n",
       " 'buffer exchange',\n",
       " 'buffer exosomal',\n",
       " 'buffer extract',\n",
       " 'buffer heating',\n",
       " 'buffer incubate',\n",
       " 'buffer invitrogen',\n",
       " 'buffer plus',\n",
       " 'buffer separation',\n",
       " 'buffer supplement',\n",
       " 'buffer transfer',\n",
       " 'c18 180',\n",
       " 'c18 75',\n",
       " 'c18 acclaim',\n",
       " 'c18 bead',\n",
       " 'c18 material',\n",
       " 'c18 medium',\n",
       " 'c18 microcolumn',\n",
       " 'c18 particle',\n",
       " 'c18 pepmap',\n",
       " 'c18 precolumn',\n",
       " 'c18 reverse',\n",
       " 'c18 seppak',\n",
       " 'c18 trap',\n",
       " 'ca 12',\n",
       " 'ca dry',\n",
       " 'ca gel',\n",
       " 'ca run',\n",
       " 'ca solvent',\n",
       " 'ca use',\n",
       " 'calibration file',\n",
       " 'calibration plate',\n",
       " 'cancer tissue',\n",
       " 'capillary lc',\n",
       " 'capillary temperature',\n",
       " 'capture complex',\n",
       " 'capture extract',\n",
       " 'capture wash',\n",
       " 'carboxyethyl phosphine',\n",
       " 'carlsbad ca',\n",
       " 'carry approximately',\n",
       " 'carry orbitrap',\n",
       " 'cartridge waters',\n",
       " 'casing immerse',\n",
       " 'cation exchange',\n",
       " 'cell culture',\n",
       " 'cell debris',\n",
       " 'cell ms',\n",
       " 'cell pellet',\n",
       " 'cell resolution',\n",
       " 'cell sample',\n",
       " 'cell sds',\n",
       " 'cell transfect',\n",
       " 'cell wash',\n",
       " 'cellular debris',\n",
       " 'centrifugal filter',\n",
       " 'centrifuge 120',\n",
       " 'centrifuge 20',\n",
       " 'centrifuge 30',\n",
       " 'centrifuge bottle',\n",
       " 'centrifuge ja',\n",
       " 'centrifuge separate',\n",
       " 'centrifuge use',\n",
       " 'centroid mode',\n",
       " 'chamber 23',\n",
       " 'channel multi',\n",
       " 'charge precursor',\n",
       " 'charge state',\n",
       " 'chca matrix',\n",
       " 'chloroplast lys',\n",
       " 'chloroplast obtain',\n",
       " 'chromatography dionex',\n",
       " 'chromatography function',\n",
       " 'chromatography hplc',\n",
       " 'chromatography ms',\n",
       " 'chromatography solvent',\n",
       " 'chromatography tandem',\n",
       " 'chromatography uplc',\n",
       " 'cid abundant',\n",
       " 'cl ph8',\n",
       " 'clara ca',\n",
       " 'clean use',\n",
       " 'cm 100',\n",
       " 'cm 75',\n",
       " 'cm c18',\n",
       " 'cm contain',\n",
       " 'cm pack',\n",
       " 'cm ph',\n",
       " 'cocktail 8340',\n",
       " 'collect focus',\n",
       " 'collect protease',\n",
       " 'collect sample',\n",
       " 'collection beckman',\n",
       " 'collection time',\n",
       " 'collision energy',\n",
       " 'collision gas',\n",
       " 'collision induce',\n",
       " 'collisional dissociation',\n",
       " 'collisionally induce',\n",
       " 'colloidal blue',\n",
       " 'colorimetric assay',\n",
       " 'column 15',\n",
       " 'column 75',\n",
       " 'column beh',\n",
       " 'column connect',\n",
       " 'column linear',\n",
       " 'column locate',\n",
       " 'column outlet',\n",
       " 'column pack',\n",
       " 'column use',\n",
       " 'combine dry',\n",
       " 'combine itraq',\n",
       " 'company san',\n",
       " 'complex elute',\n",
       " 'concentrate 100',\n",
       " 'concentrate condition',\n",
       " 'concentrate precipitation',\n",
       " 'concentrate sds',\n",
       " 'concentration sample',\n",
       " 'condition immuno',\n",
       " 'condition medium',\n",
       " 'condition percoll',\n",
       " 'conditioned medium',\n",
       " 'configure 2d',\n",
       " 'confirm use',\n",
       " 'confocal microscopy',\n",
       " 'confound effect',\n",
       " 'connect 10',\n",
       " 'connect directly',\n",
       " 'connect modify',\n",
       " 'consider dead',\n",
       " 'consist 20',\n",
       " 'consist high',\n",
       " 'consume antibiotic',\n",
       " 'contain 98',\n",
       " 'contain c18',\n",
       " 'contain purify',\n",
       " 'contain sample',\n",
       " 'container protease',\n",
       " 'control gs',\n",
       " 'control pepmap',\n",
       " 'control sirna',\n",
       " 'control target',\n",
       " 'conventional separate',\n",
       " 'coomassie blue',\n",
       " 'count 30',\n",
       " 'count 60',\n",
       " 'count charge',\n",
       " 'couple 30',\n",
       " 'couple ft',\n",
       " 'cover band',\n",
       " 'crude membrane',\n",
       " 'culture chamber',\n",
       " 'culture plate',\n",
       " 'current uti',\n",
       " 'custom make',\n",
       " 'cut 48',\n",
       " 'cut discrete',\n",
       " 'cut slice',\n",
       " 'cut stain',\n",
       " 'cutoff millipore',\n",
       " 'cyano hydroxycinnamic',\n",
       " 'cycle light',\n",
       " 'cycle orbitrap',\n",
       " 'cyclotron resonance',\n",
       " 'cysteine alkylat',\n",
       " 'cystine reduce',\n",
       " 'darkness 20',\n",
       " 'datum acquisition',\n",
       " 'datum dependent',\n",
       " 'day later',\n",
       " 'dead time',\n",
       " 'debris 16',\n",
       " 'debris enrich',\n",
       " 'debris insoluble',\n",
       " 'decant gel',\n",
       " 'dehydrate acetonitrile',\n",
       " 'deionize water',\n",
       " 'deionized water',\n",
       " 'deliver flow',\n",
       " 'denature sds',\n",
       " 'denmark infusion',\n",
       " 'deoxycholate 05',\n",
       " 'deoxycholate 100',\n",
       " 'dependent analysis',\n",
       " 'desalt 100',\n",
       " 'desalt fraction',\n",
       " 'desalt inline',\n",
       " 'desalt line',\n",
       " 'desalt microtrap',\n",
       " 'desalt use',\n",
       " 'description provide',\n",
       " 'destain protein',\n",
       " 'detailed description',\n",
       " 'detection dynamic',\n",
       " 'detection lc',\n",
       " 'detergent application',\n",
       " 'detergent digitonin',\n",
       " 'detergent fractionation',\n",
       " 'detergent interfere',\n",
       " 'diameter c18',\n",
       " 'diameter lc',\n",
       " 'different itraq',\n",
       " 'differential detergent',\n",
       " 'digest analyse',\n",
       " 'digest gel',\n",
       " 'digest modify',\n",
       " 'digest ng',\n",
       " 'digest overnight',\n",
       " 'digest peptide',\n",
       " 'digest perform',\n",
       " 'digest sample',\n",
       " 'digest wilton',\n",
       " 'digestion briefly',\n",
       " 'digestion fraction',\n",
       " 'digestion itraq',\n",
       " 'digestion mccarthy',\n",
       " 'digestion quench',\n",
       " 'digestion soluble',\n",
       " 'digestion trypsin',\n",
       " 'digitonin tween',\n",
       " 'dimension separation',\n",
       " 'dionex column',\n",
       " 'dionex operate',\n",
       " 'dionex peptide',\n",
       " 'dionex san',\n",
       " 'directly container',\n",
       " 'directly interfac',\n",
       " 'directly nanospray',\n",
       " 'directly use',\n",
       " 'discard dtt',\n",
       " 'discard protein',\n",
       " 'discard solution',\n",
       " 'discrete band',\n",
       " 'dissociation activation',\n",
       " 'dissociation fragmentation',\n",
       " 'dmem 10',\n",
       " 'dry 50',\n",
       " 'dry redissolv',\n",
       " 'dry resuspend',\n",
       " 'dry sample',\n",
       " 'dtt 25',\n",
       " 'dtt alkylat',\n",
       " 'dtt sigma',\n",
       " 'dtt solution',\n",
       " 'dual pressure',\n",
       " 'dynamic exclusion',\n",
       " 'dynamic flow',\n",
       " 'dynamically exclude',\n",
       " 'effect bacterial',\n",
       " 'eksigent nanolc',\n",
       " 'eksigent technologies',\n",
       " 'electrospray ionization',\n",
       " 'electrospray ionize',\n",
       " 'eluate buffer',\n",
       " 'eluate lc',\n",
       " 'elute 600',\n",
       " 'elute analytical',\n",
       " 'elute gradient',\n",
       " 'elute ion',\n",
       " 'elute salt',\n",
       " 'elute use',\n",
       " 'elute washed',\n",
       " 'eluting min',\n",
       " 'elution peptide',\n",
       " 'elution screen',\n",
       " 'emitter minimize',\n",
       " 'emitter new',\n",
       " 'enable mass',\n",
       " 'energy 35',\n",
       " 'energy collisional',\n",
       " 'enrich secrete',\n",
       " 'envelope thylakoid',\n",
       " 'enzymatic digestion',\n",
       " 'eppendorf tube',\n",
       " 'equal section',\n",
       " 'equilibration 10',\n",
       " 'equip 075',\n",
       " 'equip dynamic',\n",
       " 'equipment use',\n",
       " 'esi lc',\n",
       " 'et al',\n",
       " 'exchange elution',\n",
       " 'exchange remove',\n",
       " 'exchange scx',\n",
       " 'excise bnp',\n",
       " 'excise gel',\n",
       " 'excise protein',\n",
       " 'exclude 30',\n",
       " 'exclude briefly',\n",
       " 'exclusion activate',\n",
       " 'exclusion enable',\n",
       " 'exclusion exclusion',\n",
       " 'exclusion mass',\n",
       " 'exclusion set',\n",
       " 'exclusion time',\n",
       " 'exosomal pellet',\n",
       " 'exosome isolate',\n",
       " 'expedeon harston',\n",
       " 'experiment collect',\n",
       " 'experimental sequence',\n",
       " 'explore novel',\n",
       " 'extract 12',\n",
       " 'extract centrifuge',\n",
       " 'extract formic',\n",
       " 'extract supernatant',\n",
       " 'extraction 20',\n",
       " 'extraction plate',\n",
       " 'fa 350',\n",
       " 'fa min',\n",
       " 'fa transfer',\n",
       " 'fbs day',\n",
       " 'file generate',\n",
       " 'filter 3kda',\n",
       " 'filter amicon',\n",
       " 'filter ultracentrifuged',\n",
       " 'final wash',\n",
       " 'fisher scientific',\n",
       " 'fisher waltham',\n",
       " 'fix angle',\n",
       " 'flag antibody',\n",
       " 'flag capture',\n",
       " 'flag elution',\n",
       " 'flag peptide',\n",
       " 'flight time',\n",
       " 'flight tof',\n",
       " 'flow 300',\n",
       " 'flow control',\n",
       " 'flow electrospray',\n",
       " 'flow rate',\n",
       " 'focus voltage',\n",
       " 'fold concentrate',\n",
       " 'follow 10',\n",
       " 'follow 20',\n",
       " 'follow digestion',\n",
       " 'follow manufacturer',\n",
       " 'follow sds',\n",
       " 'follow second',\n",
       " 'follow wash',\n",
       " 'forebrain suspend',\n",
       " 'formic acid',\n",
       " 'formic acid50',\n",
       " 'fourier transform',\n",
       " 'fraction clean',\n",
       " 'fraction desalt',\n",
       " 'fraction dry',\n",
       " 'fraction lyophilize',\n",
       " 'fraction normalize',\n",
       " 'fraction quantify',\n",
       " 'fraction use',\n",
       " 'fractionation combine',\n",
       " 'fractionation lc',\n",
       " 'fractionation mccarthy',\n",
       " 'fractionator agilent',\n",
       " 'fragment 12',\n",
       " 'fragment linear',\n",
       " 'fragmentation cid',\n",
       " 'fragmentation method',\n",
       " 'fragmentation stage',\n",
       " 'framingham ma',\n",
       " 'francisco ca',\n",
       " 'freeze 35',\n",
       " 'fresh tube',\n",
       " 'frickenhausen germany',\n",
       " 'fritless nano',\n",
       " 'ft icr',\n",
       " 'ft ion',\n",
       " 'ft thermo',\n",
       " 'ft ultra',\n",
       " 'function peak',\n",
       " 'gain control',\n",
       " 'gas sequence',\n",
       " 'gatlin peptide',\n",
       " 'gel 10',\n",
       " 'gel addition',\n",
       " 'gel assemble',\n",
       " 'gel company',\n",
       " 'gel cut',\n",
       " 'gel digest',\n",
       " 'gel digestion',\n",
       " 'gel lane',\n",
       " 'gel life',\n",
       " 'gel piece',\n",
       " 'gel release',\n",
       " 'gel sample',\n",
       " 'gel slice',\n",
       " 'gel stain',\n",
       " 'gel track',\n",
       " 'gel wash',\n",
       " 'generate individual',\n",
       " 'germany experimental',\n",
       " 'germany use',\n",
       " 'glun1tap tap',\n",
       " 'grade modified',\n",
       " 'grade promega',\n",
       " 'gradient 36',\n",
       " 'gradient 40',\n",
       " 'gradient 50',\n",
       " 'gradient 55',\n",
       " 'gradient acetonitrile',\n",
       " 'gradient ipg',\n",
       " 'gradient sample',\n",
       " 'gradient solvent',\n",
       " 'gradient ultimate',\n",
       " 'greiner bio',\n",
       " 'grind pestle',\n",
       " 'grow culture',\n",
       " 'growth urine',\n",
       " 'gs sirna',\n",
       " 'halo c18',\n",
       " 'harston united',\n",
       " 'harvest 24',\n",
       " 'hcl ph',\n",
       " 'healthy volunteer',\n",
       " 'heating 80',\n",
       " 'high energy',\n",
       " 'high performance',\n",
       " 'high resolution',\n",
       " 'high throughput',\n",
       " 'hilic fractionation',\n",
       " 'hour 200',\n",
       " 'hour 37',\n",
       " 'house magic',\n",
       " 'hplc connect',\n",
       " 'hplc equipment',\n",
       " 'hplc isobaric',\n",
       " 'hplc lc',\n",
       " 'hplc ltq',\n",
       " 'hplc spot',\n",
       " 'hplc thermoscientific',\n",
       " 'hplc uv',\n",
       " 'hybrid dual',\n",
       " 'hybrid linear',\n",
       " 'hydroxycinnamic acid',\n",
       " 'hypotonic medium',\n",
       " 'icr cell',\n",
       " 'icr instrument',\n",
       " 'id column',\n",
       " 'id lc',\n",
       " 'identical slice',\n",
       " 'identification sample',\n",
       " 'identify novel',\n",
       " 'imager isogen',\n",
       " 'immerse nupage',\n",
       " 'immobilized ph',\n",
       " 'immuno capture',\n",
       " 'immunoblot confocal',\n",
       " 'immunohistochemistry detailed',\n",
       " 'immunoprecipitation immunoblot',\n",
       " 'include pmsf',\n",
       " 'incubate 56',\n",
       " 'incubate 95',\n",
       " 'incubation hour',\n",
       " 'incubation room',\n",
       " 'independently prepare',\n",
       " 'individual plate',\n",
       " 'individually remainder',\n",
       " 'induce dissociation',\n",
       " 'induce fragmentation',\n",
       " 'infusion mass',\n",
       " 'inhibitor cocktail',\n",
       " 'inhibitor include',\n",
       " 'inhibitor sodium',\n",
       " 'initial measurement',\n",
       " 'inject lc',\n",
       " 'inject ultimate',\n",
       " 'injection sample',\n",
       " 'injection time',\n",
       " 'inline peptide',\n",
       " 'inner diameter',\n",
       " 'insoluble pellet',\n",
       " 'instantblue expedeon',\n",
       " 'instruction desalt',\n",
       " 'intact chloroplast',\n",
       " 'intense peak',\n",
       " 'intensity 150',\n",
       " 'interact partner',\n",
       " 'interact protein',\n",
       " 'interfac modify',\n",
       " 'interface proxeon',\n",
       " 'interfere mass',\n",
       " 'invitrogen 25',\n",
       " 'invitrogen add',\n",
       " 'invitrogen carlsbad',\n",
       " 'invitrogen receptor',\n",
       " 'invitrogen xcell',\n",
       " 'involve scx',\n",
       " 'iodoacetamide 25',\n",
       " 'iodoacetamide alkylation',\n",
       " 'iodoacetamide sigma',\n",
       " 'ion 000',\n",
       " 'ion 150',\n",
       " 'ion abundant',\n",
       " 'ion cycle',\n",
       " 'ion cyclotron',\n",
       " 'ion measure',\n",
       " 'ion mode',\n",
       " 'ion ppm',\n",
       " 'ion ratio',\n",
       " 'ion source',\n",
       " 'ion trap',\n",
       " 'ionization source',\n",
       " 'ionization value',\n",
       " 'ionize analysis',\n",
       " 'ipg strip',\n",
       " 'isobaric tag',\n",
       " 'isogen life',\n",
       " 'isolate 10',\n",
       " 'isolate fragment',\n",
       " 'isolation width',\n",
       " 'itraq label',\n",
       " 'itraq labeling',\n",
       " 'itraq ms',\n",
       " 'itraq plex',\n",
       " 'itraq reagent',\n",
       " 'j26 xp',\n",
       " 'ja 17',\n",
       " 'jose ca',\n",
       " 'jose usa',\n",
       " 'just visible',\n",
       " 'kcl second',\n",
       " 'kda mwco',\n",
       " 'kh2po4 20',\n",
       " 'kingdom separate',\n",
       " 'kingdom ultraperformance',\n",
       " 'kvh recovered',\n",
       " 'l1 trypsin',\n",
       " 'label different',\n",
       " 'label peptide',\n",
       " 'label total',\n",
       " 'labeling hilic',\n",
       " 'labeling lc',\n",
       " 'laemmli sample',\n",
       " 'lane 10',\n",
       " 'lane 12',\n",
       " 'lane cut',\n",
       " 'lane lane',\n",
       " 'lane picker',\n",
       " 'larvae grind',\n",
       " 'laser shot',\n",
       " 'later cell',\n",
       " 'later centrifuge',\n",
       " 'lc high',\n",
       " 'lc maldi',\n",
       " 'lc mode',\n",
       " 'lc ms',\n",
       " 'lc packings',\n",
       " 'lc sample',\n",
       " 'leaf purify',\n",
       " 'leave shaker',\n",
       " 'leave treatment',\n",
       " 'length lane',\n",
       " 'leupeptin 450',\n",
       " 'life science',\n",
       " 'life technologies',\n",
       " 'light cycle',\n",
       " 'light intensity',\n",
       " 'line analytical',\n",
       " 'line c18',\n",
       " 'line fritless',\n",
       " 'line mixing',\n",
       " 'linear acetonitrile',\n",
       " 'linear ion',\n",
       " 'lipid popc',\n",
       " 'liquid chromatography',\n",
       " 'liquid nitrogen',\n",
       " 'load 15',\n",
       " 'load 5800',\n",
       " 'load cm',\n",
       " 'load lane',\n",
       " 'load novex',\n",
       " 'load wash',\n",
       " 'loading buffer',\n",
       " 'locate ion',\n",
       " 'louis mo',\n",
       " 'low resolution',\n",
       " 'low retention',\n",
       " 'lsd sample',\n",
       " 'ltq ft',\n",
       " 'ltq orbitrap',\n",
       " 'ltq velos',\n",
       " 'lyophilize lc',\n",
       " 'lys digest',\n",
       " 'lys hypotonic',\n",
       " 'lysate reduction',\n",
       " 'lysis buffer',\n",
       " 'ma follow',\n",
       " 'ma ms',\n",
       " 'ma peptide',\n",
       " 'ma total',\n",
       " 'ma usa',\n",
       " 'madison wi',\n",
       " 'magic c18',\n",
       " 'magic michrom',\n",
       " 'magic sample',\n",
       " 'magnetic bead',\n",
       " 'maintain 90',\n",
       " 'make onetouch',\n",
       " 'maldi plate',\n",
       " 'maldi spotter',\n",
       " 'maldi time',\n",
       " 'maldi tof',\n",
       " 'man woman',\n",
       " 'manchester united',\n",
       " 'manual situ',\n",
       " 'manufacture accord',\n",
       " 'manufacturer instruction',\n",
       " 'manufacturer protocol',\n",
       " 'manufacturer recommendation',\n",
       " 'mass analyzer',\n",
       " 'mass calibration',\n",
       " 'mass exclusion',\n",
       " 'mass range',\n",
       " 'mass spectrometer',\n",
       " 'mass spectrometric',\n",
       " 'mass spectrometry',\n",
       " 'mass window',\n",
       " 'material michrom',\n",
       " 'matrix mg',\n",
       " 'maximum plateau',\n",
       " 'maximum scan',\n",
       " 'mccarthy et',\n",
       " ...]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(len(bigram_names))\n",
    "bigram_names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'crude membrane': 529,\n",
       " 'membrane p56': 1013,\n",
       " 'p56 p70': 1202,\n",
       " 'p70 glun1tap': 1203,\n",
       " 'glun1tap tap': 770,\n",
       " 'tap mouse': 1678,\n",
       " 'mouse forebrain': 1124,\n",
       " 'forebrain suspend': 717,\n",
       " 'suspend 12': 1668,\n",
       " '12 ml': 54,\n",
       " 'ml buffer': 1071,\n",
       " 'buffer extract': 381,\n",
       " 'extract 12': 680,\n",
       " 'ml deoxycholate': 1074,\n",
       " 'deoxycholate 100': 561,\n",
       " '100 mm': 42,\n",
       " 'mm nacl': 1097,\n",
       " 'nacl 50': 1146,\n",
       " '50 mm': 161,\n",
       " 'mm tris': 1103,\n",
       " 'tris cl': 1753,\n",
       " 'cl ph8': 457,\n",
       " 'ph8 total': 1271,\n",
       " 'total extract': 1729,\n",
       " 'extract centrifuge': 681,\n",
       " 'centrifuge 120': 434,\n",
       " '120 000': 58,\n",
       " '000 40': 2,\n",
       " '40 min': 148,\n",
       " 'min condition': 1035,\n",
       " 'condition immuno': 497,\n",
       " 'immuno capture': 823,\n",
       " 'capture wash': 415,\n",
       " 'wash peptide': 1843,\n",
       " 'peptide antigen': 1244,\n",
       " 'antigen exchange': 309,\n",
       " 'exchange elution': 660,\n",
       " 'elution screen': 643,\n",
       " 'screen use': 1495,\n",
       " 'use high': 1804,\n",
       " 'high throughput': 796,\n",
       " 'throughput purification': 1705,\n",
       " 'purification robot': 1367,\n",
       " 'robot magic': 1436,\n",
       " 'magic sample': 972,\n",
       " 'sample processor': 1468,\n",
       " 'processor invitrogen': 1335,\n",
       " 'invitrogen 25': 859,\n",
       " '25 ml': 102,\n",
       " 'ml glun1tap': 1076,\n",
       " 'tap extract': 1677,\n",
       " 'extract supernatant': 683,\n",
       " 'supernatant 80': 1657,\n",
       " '80 mouse': 193,\n",
       " 'mouse flag': 1123,\n",
       " 'flag antibody': 698,\n",
       " 'antibody couple': 308,\n",
       " 'couple 30': 526,\n",
       " '30 mg': 113,\n",
       " 'mg 500': 1020,\n",
       " '500 protein': 167,\n",
       " 'protein magnetic': 1353,\n",
       " 'magnetic bead': 973,\n",
       " 'bead invitrogen': 340,\n",
       " 'invitrogen receptor': 862,\n",
       " 'receptor capture': 1390,\n",
       " 'capture extract': 414,\n",
       " 'supernatant bead': 1658,\n",
       " 'bead wash': 342,\n",
       " 'wash time': 1848,\n",
       " 'time ml': 1714,\n",
       " 'ml wash': 1081,\n",
       " 'wash buffer': 1838,\n",
       " 'buffer 37': 377,\n",
       " '37 sodium': 144,\n",
       " 'sodium deoxycholate': 1567,\n",
       " 'deoxycholate 05': 560,\n",
       " '05 mg': 11,\n",
       " 'mg ml1': 1023,\n",
       " 'ml1 lipid': 1083,\n",
       " 'lipid popc': 941,\n",
       " 'popc pope': 1307,\n",
       " 'pope pog': 1308,\n",
       " 'pog 150': 1300,\n",
       " '150 mm': 65,\n",
       " 'ph8 flag': 1270,\n",
       " 'flag capture': 699,\n",
       " 'capture complex': 413,\n",
       " 'complex elute': 491,\n",
       " 'elute 600': 634,\n",
       " '600 wash': 177,\n",
       " 'buffer supplement': 387,\n",
       " 'supplement mg': 1664,\n",
       " 'ml1 flag': 1082,\n",
       " 'flag peptide': 701,\n",
       " 'peptide eluate': 1249,\n",
       " 'eluate buffer': 632,\n",
       " 'buffer exchange': 379,\n",
       " 'exchange remove': 661,\n",
       " 'remove flag': 1409,\n",
       " 'flag elution': 700,\n",
       " 'elution peptide': 642,\n",
       " 'peptide concentrate': 1246,\n",
       " 'concentrate 100': 492,\n",
       " '100 kda': 39,\n",
       " 'kda mwco': 899,\n",
       " 'mwco centrifugal': 1145,\n",
       " 'centrifugal filter': 433,\n",
       " 'filter amicon': 692,\n",
       " 'amicon 35': 284,\n",
       " '35 10': 131,\n",
       " '10 25': 14,\n",
       " '25 mg': 101,\n",
       " 'ml1 native': 1084,\n",
       " 'native nmdar': 1160,\n",
       " 'nmdar bnp': 1172,\n",
       " 'bnp run': 365,\n",
       " 'run accord': 1440,\n",
       " 'accord schagger': 217,\n",
       " 'schagger excise': 1490,\n",
       " 'excise bnp': 663,\n",
       " 'bnp gel': 364,\n",
       " 'gel slice': 763,\n",
       " 'slice contain': 1560,\n",
       " 'contain purify': 514,\n",
       " 'purify protein': 1370,\n",
       " 'protein prepare': 1356,\n",
       " 'prepare mass': 1323,\n",
       " 'mass spectrometric': 992,\n",
       " 'spectrometric analysis': 1603,\n",
       " 'analysis manual': 296,\n",
       " 'manual situ': 982,\n",
       " 'situ enzymatic': 1558,\n",
       " 'enzymatic digestion': 651,\n",
       " 'digestion briefly': 591,\n",
       " 'briefly excise': 372,\n",
       " 'excise protein': 665,\n",
       " 'protein gel': 1351,\n",
       " 'gel piece': 760,\n",
       " 'piece place': 1279,\n",
       " 'place 96': 1282,\n",
       " '96 microtitre': 200,\n",
       " 'microtitre plate': 1028,\n",
       " 'plate stain': 1293,\n",
       " 'stain 50': 1625,\n",
       " '50 acetonitrile': 157,\n",
       " 'acetonitrile 50': 221,\n",
       " 'mm ammonium': 1088,\n",
       " 'ammonium bicarbonate': 286,\n",
       " 'bicarbonate reduce': 349,\n",
       " 'reduce 10': 1396,\n",
       " '10 mm': 25,\n",
       " 'mm dtt': 1092,\n",
       " 'dtt alkylat': 620,\n",
       " 'alkylat 55': 276,\n",
       " '55 mm': 172,\n",
       " 'mm iodoacetamide': 1093,\n",
       " 'iodoacetamide alkylation': 866,\n",
       " 'alkylation protein': 282,\n",
       " 'protein digest': 1349,\n",
       " 'digest ng': 585,\n",
       " 'ng l1': 1166,\n",
       " 'l1 trypsin': 904,\n",
       " 'trypsin promega': 1760,\n",
       " 'promega uk': 1339,\n",
       " 'uk overnight': 1769,\n",
       " 'overnight 37': 1198,\n",
       " '37 result': 143,\n",
       " 'result peptide': 1424,\n",
       " 'peptide extract': 1251,\n",
       " 'extract formic': 682,\n",
       " 'formic acid': 718,\n",
       " 'acid acetonitrile': 232,\n",
       " 'acetonitrile digest': 224,\n",
       " 'digest analyse': 582,\n",
       " 'analyse nano': 289,\n",
       " 'nano scale': 1153,\n",
       " 'scale capillary': 1479,\n",
       " 'capillary lc': 411,\n",
       " 'lc ms': 923,\n",
       " 'ms ms': 1131,\n",
       " 'ms use': 1141,\n",
       " 'use ultimate': 1813,\n",
       " 'ultimate u3000': 1772,\n",
       " 'u3000 hplc': 1768,\n",
       " 'hplc thermoscientific': 807,\n",
       " 'thermoscientific dionex': 1701,\n",
       " 'dionex san': 603,\n",
       " 'san jose': 1475,\n",
       " 'jose usa': 896,\n",
       " 'usa deliver': 1787,\n",
       " 'deliver flow': 557,\n",
       " 'flow 300': 704,\n",
       " '300 nl': 123,\n",
       " 'nl min1': 1171,\n",
       " 'min1 c18': 1053,\n",
       " 'c18 acclaim': 391,\n",
       " 'acclaim pepmap100': 214,\n",
       " 'pepmap100 100': 1241,\n",
       " '100 20': 32,\n",
       " '20 mm': 83,\n",
       " 'mm nanoviper': 1098,\n",
       " 'nanoviper thermoscientific': 1159,\n",
       " 'usa trap': 1789,\n",
       " 'trap peptide': 1742,\n",
       " 'peptide separation': 1257,\n",
       " 'separation c18': 1526,\n",
       " 'pepmap100 75': 1242,\n",
       " '75 150': 186,\n",
       " 'usa peptide': 1788,\n",
       " 'peptide elute': 1250,\n",
       " 'elute gradient': 636,\n",
       " 'gradient acetonitrile': 777,\n",
       " 'acetonitrile analytical': 223,\n",
       " 'analytical column': 302,\n",
       " 'column outlet': 485,\n",
       " 'outlet directly': 1197,\n",
       " 'directly interfac': 605,\n",
       " 'interfac modify': 856,\n",
       " 'modify nano': 1114,\n",
       " 'nano flow': 1150,\n",
       " 'flow electrospray': 706,\n",
       " 'electrospray ionization': 630,\n",
       " 'ionization source': 879,\n",
       " 'source hybrid': 1588,\n",
       " 'hybrid dual': 809,\n",
       " 'dual pressure': 623,\n",
       " 'pressure linear': 1326,\n",
       " 'linear ion': 940,\n",
       " 'ion trap': 878,\n",
       " 'trap mass': 1739,\n",
       " 'mass spectrometer': 991,\n",
       " 'spectrometer orbitrap': 1599,\n",
       " 'orbitrap velos': 1195,\n",
       " 'velos thermoscientific': 1823,\n",
       " 'thermoscientific san': 1703,\n",
       " 'breast cancer': 370,\n",
       " 'cancer tissue': 410,\n",
       " 'tissue lysate': 1720,\n",
       " 'lysate reduction': 962,\n",
       " 'reduction alkylation': 1401,\n",
       " 'alkylation methyl': 281,\n",
       " 'methyl methanethiosulfonate': 1019,\n",
       " 'methanethiosulfonate trypsin': 1016,\n",
       " 'trypsin digestion': 1758,\n",
       " 'digestion itraq': 593,\n",
       " 'itraq plex': 891,\n",
       " 'plex labeling': 1296,\n",
       " 'labeling hilic': 908,\n",
       " 'hilic fractionation': 797,\n",
       " 'fractionation lc': 729,\n",
       " 'ms orbitrap': 1132,\n",
       " 'velos transcriptomics': 1824,\n",
       " 'transcriptomics analysis': 1732,\n",
       " 'analysis immunohistochemistry': 293,\n",
       " 'immunohistochemistry detailed': 825,\n",
       " 'detailed description': 571,\n",
       " 'description provide': 569,\n",
       " 'provide manuscript': 1364,\n",
       " 'gel digest': 756,\n",
       " 'digest perform': 588,\n",
       " 'perform standard': 1260,\n",
       " 'standard protocol': 1635,\n",
       " 'protocol briefly': 1361,\n",
       " 'briefly follow': 373,\n",
       " 'follow sds': 714,\n",
       " 'sds page': 1501,\n",
       " 'page washing': 1216,\n",
       " 'washing excise': 1851,\n",
       " 'excise gel': 664,\n",
       " 'slice protein': 1563,\n",
       " 'protein reduce': 1357,\n",
       " 'reduce add': 1398,\n",
       " 'add 10': 255,\n",
       " 'dtt sigma': 621,\n",
       " 'sigma aldrich': 1549,\n",
       " 'aldrich prior': 269,\n",
       " 'prior alkylation': 1329,\n",
       " 'alkylation 55': 280,\n",
       " 'iodoacetamide sigma': 867,\n",
       " 'aldrich wash': 272,\n",
       " 'wash shrink': 1845,\n",
       " 'shrink gel': 1548,\n",
       " 'piece 100': 1278,\n",
       " '100 acetonitrile': 37,\n",
       " 'acetonitrile trypsin': 229,\n",
       " 'trypsin sequencing': 1762,\n",
       " 'sequencing grade': 1537,\n",
       " 'grade modified': 771,\n",
       " 'modified promega': 1113,\n",
       " 'promega add': 1337,\n",
       " 'add protein': 258,\n",
       " 'digest overnight': 586,\n",
       " 'overnight 40': 1199,\n",
       " '40 mm': 150,\n",
       " 'mm ammoniumbicarbonate': 1089,\n",
       " 'ammoniumbicarbonate 37': 287,\n",
       " '37 protein': 142,\n",
       " 'protein identification': 1352,\n",
       " 'identification sample': 818,\n",
       " 'sample directly': 1455,\n",
       " 'directly use': 607,\n",
       " 'use nano': 1808,\n",
       " 'nano esi': 1149,\n",
       " 'esi lc': 658,\n",
       " 'ms sample': 1136,\n",
       " 'sample desalt': 1454,\n",
       " 'desalt line': 566,\n",
       " 'line c18': 936,\n",
       " 'c18 microcolumn': 395,\n",
       " 'microcolumn 300': 1026,\n",
       " '300 mm': 122,\n",
       " 'mm pack': 1099,\n",
       " 'pack c18': 1204,\n",
       " 'c18 pepmap': 397,\n",
       " 'pepmap TM': 1239,\n",
       " 'TM 100': 203,\n",
       " '100 lc': 40,\n",
       " 'lc packings': 924,\n",
       " 'packings peptide': 1212,\n",
       " 'peptide separate': 1256,\n",
       " 'separate c18': 1518,\n",
       " 'c18 reverse': 399,\n",
       " 'reverse phase': 1434,\n",
       " 'phase column': 1273,\n",
       " 'column linear': 483,\n",
       " 'linear acetonitrile': 939,\n",
       " 'acetonitrile gradient': 227,\n",
       " 'gradient ultimate': 781,\n",
       " 'ultimate 3000': 1770,\n",
       " '3000 dionex': 125,\n",
       " 'dionex column': 600,\n",
       " 'column 75': 480,\n",
       " '75 15': 185,\n",
       " '15 cm': 61,\n",
       " 'cm pack': 464,\n",
       " 'packings ms': 1211,\n",
       " 'ms spectra': 1138,\n",
       " 'spectra record': 1594,\n",
       " 'record oribitrap': 1393,\n",
       " 'oribitrap mass': 1196,\n",
       " 'spectrometer thermo': 1601,\n",
       " 'thermo electron': 1698,\n",
       " 'conditioned medium': 500,\n",
       " 'medium preparation': 1010,\n",
       " 'preparation hplc': 1321,\n",
       " 'hplc isobaric': 803,\n",
       " 'isobaric tag': 883,\n",
       " 'tag relative': 1675,\n",
       " 'relative absolute': 1403,\n",
       " 'absolute quantitation': 212,\n",
       " 'quantitation mass': 1374,\n",
       " 'mass spectrometry': 993,\n",
       " 'spectrometry itraq': 1606,\n",
       " 'itraq ms': 890,\n",
       " 'ms million': 1130,\n",
       " 'million cell': 1032,\n",
       " 'cell transfect': 430,\n",
       " 'transfect control': 1733,\n",
       " 'control gs': 517,\n",
       " 'gs sirna': 786,\n",
       " 'sirna seed': 1554,\n",
       " 'seed 100': 1511,\n",
       " 'mm cell': 1090,\n",
       " 'cell culture': 423,\n",
       " 'culture plate': 531,\n",
       " 'plate dmem': 1288,\n",
       " 'dmem 10': 614,\n",
       " '10 fbs': 20,\n",
       " 'fbs day': 689,\n",
       " 'day later': 548,\n",
       " 'later cell': 918,\n",
       " 'cell wash': 431,\n",
       " 'wash twice': 1849,\n",
       " 'twice pre': 1767,\n",
       " 'pre warm': 1314,\n",
       " 'warm pbs': 1835,\n",
       " 'pbs conditioned': 1223,\n",
       " 'medium harvest': 1009,\n",
       " 'harvest 24': 789,\n",
       " '24 later': 98,\n",
       " 'later centrifuge': 919,\n",
       " 'centrifuge 20': 435,\n",
       " '20 000': 76,\n",
       " '000 10': 1,\n",
       " '10 min': 23,\n",
       " 'min remove': 1044,\n",
       " 'remove cell': 1408,\n",
       " 'cell debris': 424,\n",
       " 'debris enrich': 551,\n",
       " 'enrich secrete': 649,\n",
       " 'secrete protein': 1508,\n",
       " 'protein condition': 1348,\n",
       " 'condition medium': 498,\n",
       " 'medium condition': 1007,\n",
       " 'medium sample': 1011,\n",
       " 'sample centrifuge': 1451,\n",
       " 'centrifuge use': 440,\n",
       " 'use amicon': 1793,\n",
       " 'amicon centrifugal': 285,\n",
       " 'filter 3kda': 691,\n",
       " '3kda cutoff': 145,\n",
       " 'cutoff millipore': 538,\n",
       " 'millipore billerica': 1033,\n",
       " 'billerica ma': 351,\n",
       " 'ma total': 967,\n",
       " 'total concentrate': 1728,\n",
       " 'concentrate condition': 493,\n",
       " 'sample independently': 1459,\n",
       " 'independently prepare': 832,\n",
       " 'prepare sample': 1324,\n",
       " 'sample control': 1452,\n",
       " 'control sirna': 519,\n",
       " 'sirna treat': 1555,\n",
       " 'treat cell': 1745,\n",
       " 'cell sample': 428,\n",
       " 'sample gs': 1458,\n",
       " 'sample store': 1473,\n",
       " 'store 70': 1643,\n",
       " '70 pron': 182,\n",
       " 'pron analyze': 1340,\n",
       " 'analyze itraq': 303,\n",
       " 'itraq labeling': 889,\n",
       " 'labeling lc': 909,\n",
       " 'ms itraq': 1129,\n",
       " 'itraq label': 888,\n",
       " 'label total': 907,\n",
       " 'total protein': 1730,\n",
       " 'protein concentration': 1347,\n",
       " 'concentration sample': 496,\n",
       " 'sample measure': 1463,\n",
       " 'measure use': 1002,\n",
       " 'use bradford': 1794,\n",
       " 'bradford assay': 368,\n",
       " 'assay quantity': 319,\n",
       " 'quantity 100': 1375,\n",
       " '100 sample': 44,\n",
       " 'sample denature': 1453,\n",
       " 'denature sds': 558,\n",
       " 'sds reduce': 1503,\n",
       " 'reduce 50': 1397,\n",
       " 'tris carboxyethyl': 1752,\n",
       " 'carboxyethyl phosphine': 416,\n",
       " 'phosphine tcep': 1276,\n",
       " 'tcep alkylat': 1682,\n",
       " 'alkylat 200': 275,\n",
       " '200 mm': 92,\n",
       " 'mm methyl': 1096,\n",
       " 'methanethiosulfonate mmts': 1015,\n",
       " 'mmts digest': 1105,\n",
       " 'overnight trypsin': 1201,\n",
       " 'trypsin sample': 1761,\n",
       " 'sample peptide': 1464,\n",
       " 'peptide label': 1253,\n",
       " 'label different': 905,\n",
       " 'different itraq': 580,\n",
       " 'itraq reagent': 892,\n",
       " 'reagent ab': 1389,\n",
       " 'ab sciex': 207,\n",
       " 'sciex framingham': 1494,\n",
       " 'framingham ma': 737,\n",
       " 'ma follow': 964,\n",
       " 'follow manufacturer': 713,\n",
       " 'manufacturer protocol': 985,\n",
       " 'protocol combine': 1362,\n",
       " 'combine dry': 488,\n",
       " 'dry 50': 615,\n",
       " '50 volume': 164,\n",
       " 'volume acidify': 1831,\n",
       " 'acidify trifluoroacetic': 240,\n",
       " 'trifluoroacetic acid': 1750,\n",
       " 'acid tfa': 238,\n",
       " 'tfa desalt': 1693,\n",
       " 'desalt 100': 563,\n",
       " '100 mg': 41,\n",
       " 'mg c18': 1021,\n",
       " 'c18 seppak': 400,\n",
       " 'seppak cartridge': 1534,\n",
       " 'cartridge waters': 420,\n",
       " 'waters milford': 1858,\n",
       " 'milford ma': 1031,\n",
       " 'ma peptide': 966,\n",
       " 'peptide fractionation': 1252,\n",
       " 'fractionation combine': 728,\n",
       " 'combine itraq': 489,\n",
       " 'label peptide': 906,\n",
       " 'separate 12': 1516,\n",
       " '12 fraction': 52,\n",
       " 'fraction use': 727,\n",
       " 'use 3100': 1791,\n",
       " '3100 offgel': 128,\n",
       " 'offgel fractionator': 1186,\n",
       " 'fractionator agilent': 731,\n",
       " 'agilent santa': 264,\n",
       " 'santa clara': 1476,\n",
       " 'clara ca': 458,\n",
       " 'ca 12': 402,\n",
       " '12 setup': 57,\n",
       " 'setup 12': 1542,\n",
       " '12 cm': 51,\n",
       " 'cm ph': 465,\n",
       " 'ph 310': 1266,\n",
       " '310 immobilized': 127,\n",
       " 'immobilized ph': 822,\n",
       " 'ph gradient': 1268,\n",
       " 'gradient ipg': 778,\n",
       " 'ipg strip': 882,\n",
       " 'strip accord': 1645,\n",
       " 'accord manufacturer': 216,\n",
       " 'manufacturer recommendation': 986,\n",
       " 'recommendation peptide': 1391,\n",
       " 'peptide collect': 1245,\n",
       " 'collect focus': 467,\n",
       " 'focus voltage': 708,\n",
       " 'voltage reach': 1829,\n",
       " 'reach maximum': 1388,\n",
       " 'maximum plateau': 997,\n",
       " 'plateau 50': 1295,\n",
       " '50 kvh': 159,\n",
       " 'kvh recovered': 903,\n",
       " 'recovered fraction': 1394,\n",
       " 'fraction lyophilize': 724,\n",
       " 'lyophilize lc': 959,\n",
       " 'lc maldi': 921,\n",
       " 'maldi tof': 979,\n",
       " 'tof tof': 1724,\n",
       " 'tof analysis': 1722,\n",
       " 'analysis solvent': 300,\n",
       " 'solvent use': 1584,\n",
       " 'use section': 1811,\n",
       " 'section aqueous': 1509,\n",
       " 'aqueous unless': 314,\n",
       " 'unless specify': 1781,\n",
       " 'specify obtain': 1591,\n",
       " 'obtain thermofisher': 1183,\n",
       " 'thermofisher scientific': 1700,\n",
       " 'scientific san': 1493,\n",
       " 'jose ca': 895,\n",
       " 'ca dry': 403,\n",
       " 'dry sample': 618,\n",
       " 'sample reconstitute': 1469,\n",
       " 'reconstitute 30': 1392,\n",
       " '30 solvent': 118,\n",
       " 'solvent acetonitrile': 1579,\n",
       " 'acetonitrile acn': 222,\n",
       " 'acn tfa': 245,\n",
       " 'tfa 10': 1691,\n",
       " '10 result': 26,\n",
       " 'result solution': 1425,\n",
       " 'solution inject': 1572,\n",
       " 'inject ultimate': 843,\n",
       " 'ultimate TM': 1771,\n",
       " 'TM hplc': 205,\n",
       " 'hplc uv': 808,\n",
       " 'uv detection': 1815,\n",
       " 'detection lc': 573,\n",
       " 'packings sunnyvale': 1214,\n",
       " 'sunnyvale ca': 1656,\n",
       " 'ca solvent': 406,\n",
       " 'solvent contain': 1580,\n",
       " 'contain 98': 512,\n",
       " '98 acn': 202,\n",
       " 'tfa c18': 1692,\n",
       " 'c18 trap': 401,\n",
       " 'trap mm': 1740,\n",
       " 'mm 300': 1086,\n",
       " '300 id': 120,\n",
       " 'id lc': 816,\n",
       " 'packings 15': 1207,\n",
       " 'cm 100': 460,\n",
       " '100 id': 38,\n",
       " 'id column': 815,\n",
       " 'column pack': 486,\n",
       " 'pack house': 1206,\n",
       " 'house magic': 800,\n",
       " 'magic c18': 970,\n",
       " 'c18 bead': 392,\n",
       " 'bead michrom': 341,\n",
       " 'michrom bioresources': 1024,\n",
       " 'bioresources auburn': 357,\n",
       " 'auburn ca': 323,\n",
       " 'ca use': 407,\n",
       " 'use chromatography': 1795,\n",
       " 'chromatography solvent': 453,\n",
       " 'solvent flow': 1581,\n",
       " 'flow rate': 707,\n",
       " 'rate set': 1386,\n",
       " 'set 700': 1539,\n",
       " '700 min': 183,\n",
       " 'min injection': 1042,\n",
       " 'injection sample': 844,\n",
       " 'sample load': 1461,\n",
       " 'load wash': 949,\n",
       " 'wash solvent': 1846,\n",
       " 'solvent min': 1583,\n",
       " 'min flow': 1040,\n",
       " 'rate 030': 1382,\n",
       " '030 ml': 10,\n",
       " 'ml min': 1078,\n",
       " 'min follow': 1041,\n",
       " 'follow 10': 710,\n",
       " '10 50': 15,\n",
       " '50 gradient': 158,\n",
       " 'gradient 55': 776,\n",
       " '55 min': 170,\n",
       " 'min ramp': 1043,\n",
       " 'ramp 90': 1379,\n",
       " '90 65': 197,\n",
       " '65 min': 178,\n",
       " 'min sample': 1047,\n",
       " 'sample run': 1470,\n",
       " 'run maintain': 1445,\n",
       " 'maintain 90': 974,\n",
       " '90 solvent': 198,\n",
       " 'solvent 75': 1578,\n",
       " '75 min': 190,\n",
       " 'min return': 1046,\n",
       " 'return 10': 1432,\n",
       " '10 75': 16,\n",
       " 'min equilibration': 1038,\n",
       " 'equilibration 10': 654,\n",
       " '10 solvent': 29,\n",
       " 'solvent hplc': 1582,\n",
       " 'hplc connect': 801,\n",
       " 'connect modify': 507,\n",
       " 'modify probot': 1115,\n",
       " 'probot TM': 1332,\n",
       " 'TM automate': 204,\n",
       " 'automate maldi': 324,\n",
       " 'maldi spotter': 977,\n",
       " 'spotter lc': 1622,\n",
       " 'packings spot': 1213,\n",
       " 'spot eluate': 1616,\n",
       " 'eluate lc': 633,\n",
       " 'maldi plate': 976,\n",
       " 'plate line': 1291,\n",
       " 'line mixing': 938,\n",
       " 'mixing cyano': 1068,\n",
       " 'cyano hydroxycinnamic': 539,\n",
       " 'hydroxycinnamic acid': 811,\n",
       " 'acid chca': 234,\n",
       " 'chca matrix': 446,\n",
       " 'matrix mg': 996,\n",
       " 'mg ml': 1022,\n",
       " 'ml 70': 1070,\n",
       " '70 acn': 180,\n",
       " 'acn consider': 241,\n",
       " 'consider dead': 508,\n",
       " 'dead time': 549,\n",
       " 'time hplc': 1712,\n",
       " 'hplc spot': 806,\n",
       " 'spot spot': 1619,\n",
       " 'spot collection': 1615,\n",
       " 'collection time': 471,\n",
       " 'time start': 1717,\n",
       " 'start 10': 1636,\n",
       " 'injection time': 845,\n",
       " 'time spot': 1716,\n",
       " 'spot time': 1620,\n",
       " 'time 33': 1709,\n",
       " '33 min': 129,\n",
       " 'min spot': 1049,\n",
       " 'spot total': 1621,\n",
       " 'total 176': 1727,\n",
       " '176 spot': 72,\n",
       " 'spot collect': 1614,\n",
       " 'collect sample': 469,\n",
       " 'sample plate': 1466,\n",
       " 'plate result': 1292,\n",
       " 'result spot': 1426,\n",
       " 'spot load': 1617,\n",
       " 'load 5800': 945,\n",
       " '5800 maldi': 174,\n",
       " 'maldi time': 978,\n",
       " 'time flight': 1711,\n",
       " 'flight time': 702,\n",
       " 'flight tof': 703,\n",
       " 'tof mass': 1723,\n",
       " 'spectrometer ab': 1595,\n",
       " 'ma ms': 965,\n",
       " 'ms analysis': 1127,\n",
       " 'analysis batch': 290,\n",
       " 'batch mode': 337,\n",
       " 'mode mass': 1110,\n",
       " 'mass calibration': 988,\n",
       " 'calibration plate': 409,\n",
       " 'plate alignment': 1285,\n",
       " 'alignment datum': 273,\n",
       " 'datum acquisition': 546,\n",
       " 'acquisition separate': 250,\n",
       " 'separate plate': 1520,\n",
       " 'plate calibration': 1286,\n",
       " 'calibration file': 408,\n",
       " 'file generate': 690,\n",
       " 'generate individual': 767,\n",
       " 'individual plate': 833,\n",
       " 'plate contain': 1287,\n",
       " 'contain sample': 515,\n",
       " 'sample spot': 1472,\n",
       " 'spot scan': 1618,\n",
       " 'scan ms': 1485,\n",
       " 'spectra obtain': 1593,\n",
       " 'obtain total': 1184,\n",
       " 'total 1000': 1726,\n",
       " '1000 laser': 46,\n",
       " 'laser shot': 917,\n",
       " 'shot spectrum': 1546,\n",
       " 'spectrum spot': 1611,\n",
       " 'spot 20': 1613,\n",
       " '20 intense': 80,\n",
       " 'intense peak': 852,\n",
       " 'peak weak': 1229,\n",
       " 'weak precursor': 1861,\n",
       " 'precursor signal': 1320,\n",
       " 'signal noise': 1551,\n",
       " 'noise threshold': 1173,\n",
       " 'threshold 10': 1704,\n",
       " '10 select': 27,\n",
       " 'select successful': 1515,\n",
       " 'successful ms': 1653,\n",
       " 'analysis ms': 297,\n",
       " 'ms spectrum': 1139,\n",
       " 'spectrum average': 1610,\n",
       " 'average 1600': 328,\n",
       " '1600 laser': 69,\n",
       " 'shot ms': 1545,\n",
       " 'spectra acquire': 1592,\n",
       " 'acquire presence': 247,\n",
       " 'presence collision': 1325,\n",
       " 'collision gas': 473,\n",
       " 'gas sequence': 749,\n",
       " 'sequence vgf': 1536,\n",
       " 'vgf nerp1': 1825,\n",
       " 'nerp1 peptide': 1161,\n",
       " 'peptide confirm': 1247,\n",
       " 'confirm use': 502,\n",
       " 'use ltq': 1805,\n",
       " 'ltq orbitrap': 957,\n",
       " 'orbitrap mass': 1193,\n",
       " 'spectrometer thermofisher': 1602,\n",
       " 'protein elute': 1350,\n",
       " 'elute washed': 640,\n",
       " 'washed bead': 1850,\n",
       " 'bead add': 339,\n",
       " 'add fold': 257,\n",
       " 'fold concentrate': 709,\n",
       " 'concentrate sds': 495,\n",
       " 'sds gel': 1499,\n",
       " 'gel sample': 762,\n",
       " 'sample loading': 1462,\n",
       " 'loading buffer': 950,\n",
       " 'buffer heating': 382,\n",
       " 'heating 80': 792,\n",
       " '80 10': 191,\n",
       " 'min supernatant': 1051,\n",
       " 'supernatant load': 1660,\n",
       " 'load novex': 948,\n",
       " 'novex gel': 1176,\n",
       " 'gel 10': 751,\n",
       " '10 bis': 18,\n",
       " 'bis tris': 361,\n",
       " 'tris sds': 1756,\n",
       " 'gel life': 759,\n",
       " 'life technologies': 932,\n",
       " 'technologies invitrogen': 1684,\n",
       " 'invitrogen carlsbad': 861,\n",
       " 'carlsbad ca': 417,\n",
       " 'ca run': 405,\n",
       " 'run length': 1444,\n",
       " 'length lane': 929,\n",
       " 'lane lane': 914,\n",
       " 'lane cut': 913,\n",
       " 'cut stain': 537,\n",
       " 'stain instantblue': 1627,\n",
       " 'instantblue expedeon': 849,\n",
       " 'expedeon harston': 676,\n",
       " 'harston united': 788,\n",
       " 'united kingdom': 1780,\n",
       " 'kingdom separate': 901,\n",
       " 'separate tray': 1524,\n",
       " 'tray band': 1744,\n",
       " 'band just': 336,\n",
       " 'just visible': 897,\n",
       " 'visible area': 1827,\n",
       " 'area cover': 316,\n",
       " 'cover band': 528,\n",
       " 'band cut': 334,\n",
       " 'cut slice': 536,\n",
       " 'slice wash': 1565,\n",
       " 'wash reduce': 1844,\n",
       " 'reduce alkylat': 1400,\n",
       " 'alkylat treat': 279,\n",
       " 'treat trypsin': 1746,\n",
       " 'trypsin accord': 1757,\n",
       " 'accord standard': 218,\n",
       " 'standard procedure': 1634,\n",
       " 'procedure peptide': 1333,\n",
       " 'formic acid50': 719,\n",
       " 'acid50 acetonitrile': 239,\n",
       " 'acetonitrile dry': 225,\n",
       " 'dry redissolv': 616,\n",
       " 'redissolv trifluoroacetic': 1395,\n",
       " 'tfa liquid': 1694,\n",
       " 'liquid chromatography': 942,\n",
       " 'chromatography tandem': 454,\n",
       " 'tandem mass': 1676,\n",
       " 'spectrometry lc': 1607,\n",
       " 'analysis sample': 299,\n",
       " 'sample aliquot': 1448,\n",
       " 'aliquot apply': 274,\n",
       " 'apply nanoacquity': 312,\n",
       " 'nanoacquity waters': 1154,\n",
       " 'waters manchester': 1857,\n",
       " 'manchester united': 981,\n",
       " 'kingdom ultraperformance': 902,\n",
       " 'ultraperformance liquid': 1779,\n",
       " 'chromatography uplc': 455,\n",
       " 'uplc run': 1782,\n",
       " 'run flow': 1442,\n",
       " 'rate 250': 1383,\n",
       " '250 nl': 106,\n",
       " 'min1 ltq': 1055,\n",
       " 'thermo fisher': 1699,\n",
       " 'fisher waltham': 696,\n",
       " 'waltham ma': 1834,\n",
       " 'peptide trap': 1258,\n",
       " 'trap use': 1743,\n",
       " 'use precolumn': 1810,\n",
       " 'precolumn symmetry': 1318,\n",
       " 'symmetry c18': 1673,\n",
       " 'c18 180': 389,\n",
       " '180 20': 73,\n",
       " 'mm waters': 1104,\n",
       " 'waters switch': 1860,\n",
       " 'switch line': 1672,\n",
       " 'line analytical': 935,\n",
       " 'column beh': 481,\n",
       " 'beh c18': 346,\n",
       " 'c18 75': 390,\n",
       " '75 250': 187,\n",
       " '250 mm': 105,\n",
       " 'waters separation': 1859,\n",
       " 'separation peptide': 1529,\n",
       " 'gradient 40': 774,\n",
       " '40 acetonitrile': 147,\n",
       " 'acetonitrile water0': 230,\n",
       " 'water0 formic': 1856,\n",
       " 'acid rate': 237,\n",
       " 'rate 67': 1385,\n",
       " '67 min1': 179,\n",
       " 'min1 column': 1054,\n",
       " 'column connect': 482,\n",
       " 'connect 10': 505,\n",
       " '10 silicatip': 28,\n",
       " 'silicatip nanospray': 1552,\n",
       " 'nanospray emitter': 1156,\n",
       " 'emitter new': 645,\n",
       " 'new objective': 1164,\n",
       " 'objective woburn': 1181,\n",
       " 'woburn ma': 1868,\n",
       " 'ma usa': 968,\n",
       " 'usa attach': 1786,\n",
       " 'attach nanospray': 322,\n",
       " 'nanospray interface': 1157,\n",
       " 'interface proxeon': 857,\n",
       " 'proxeon odense': 1365,\n",
       " 'odense denmark': 1185,\n",
       " 'denmark infusion': 559,\n",
       " 'infusion mass': 837,\n",
       " 'spectrometer mass': 1597,\n",
       " 'spectrometer operate': 1598,\n",
       " 'operate positive': 1189,\n",
       " 'positive ion': 1311,\n",
       " 'ion mode': 874,\n",
       " 'mode capillary': 1107,\n",
       " 'capillary temperature': 412,\n",
       " 'temperature 200': 1686,\n",
       " '200 source': 93,\n",
       " 'source voltage': 1590,\n",
       " 'voltage focus': 1828,\n",
       " 'voltage tune': 1830,\n",
       " 'tune transmission': 1765,\n",
       " 'transmission met': 1737,\n",
       " 'met arg': 1014,\n",
       " 'arg phe': 317,\n",
       " 'phe ala': 1275,\n",
       " 'ala mrfa': 268,\n",
       " 'mrfa peptide': 1125,\n",
       " 'peptide 524': 1243,\n",
       " '524 sigma': 169,\n",
       " 'aldrich st': 271,\n",
       " 'st louis': 1623,\n",
       " 'louis mo': 952,\n",
       " 'mo datum': 1106,\n",
       " 'datum dependent': 547,\n",
       " 'dependent analysis': 562,\n",
       " 'analysis carry': 291,\n",
       " 'carry orbitrap': 419,\n",
       " 'orbitrap ion': 1192,\n",
       " 'trap parallel': 1741,\n",
       " 'parallel mode': 1217,\n",
       " 'mode use': 1112,\n",
       " 'use collision': 1796,\n",
       " 'collision induce': 474,\n",
       " 'induce fragmentation': 836,\n",
       " 'fragmentation cid': 734,\n",
       " 'cid abundant': 456,\n",
       " 'abundant ion': 213,\n",
       " 'ion cycle': 871,\n",
       " 'cycle orbitrap': 541,\n",
       " 'orbitrap run': 1194,\n",
       " 'run resolution': 1446,\n",
       " 'resolution 30': 1418,\n",
       " '30 000': 110,\n",
       " '000 ms': 7,\n",
       " 'ms range': 1135,\n",
       " 'range 350': 1380,\n",
       " '350 800': 136,\n",
       " '800 ms': 195,\n",
       " 'ms target': 1140,\n",
       " 'target 106': 1680,\n",
       " '106 maximum': 50,\n",
       " 'maximum scan': 998,\n",
       " 'scan time': 1489,\n",
       " 'time collision': 1710,\n",
       " 'collision energy': 472,\n",
       " 'energy 35': 647,\n",
       " '35 isolation': 133,\n",
       " 'isolation width': 887,\n",
       " 'width use': 1864,\n",
       " 'use monoisotopic': 1806,\n",
       " 'monoisotopic charge': 1118,\n",
       " 'charge precursor': 444,\n",
       " 'precursor select': 1319,\n",
       " 'select ms2': 1513,\n",
       " 'ms2 fragmentation': 1142,\n",
       " 'fragmentation stage': 736,\n",
       " 'stage ms2': 1624,\n",
       " 'ms2 trigger': 1143,\n",
       " 'trigger minimal': 1751,\n",
       " 'minimal signal': 1057,\n",
       " 'signal 000': 1550,\n",
       " '000 automatic': 4,\n",
       " 'automatic gain': 325,\n",
       " 'gain control': 748,\n",
       " 'control target': 520,\n",
       " 'target 104': 1679,\n",
       " '104 ion': 49,\n",
       " 'ion 150': 869,\n",
       " '150 ms': 67,\n",
       " 'ms scan': 1137,\n",
       " 'time use': 1718,\n",
       " 'chromatography function': 450,\n",
       " 'function peak': 747,\n",
       " 'peak apex': 1225,\n",
       " 'apex detection': 310,\n",
       " 'detection dynamic': 572,\n",
       " 'dynamic exclusion': 624,\n",
       " 'exclusion set': 672,\n",
       " 'set count': 1541,\n",
       " 'count 60': 524,\n",
       " '60 exclusion': 175,\n",
       " 'exclusion exclusion': 670,\n",
       " 'exclusion mass': 671,\n",
       " 'mass window': 994,\n",
       " 'window 20': 1866,\n",
       " '20 ppm': 85,\n",
       " 'ppm ms': 1312,\n",
       " 'scan save': 1488,\n",
       " 'save profile': 1478,\n",
       " 'profile mode': 1336,\n",
       " 'mode ms': 1111,\n",
       " 'save centroid': 1477,\n",
       " 'centroid mode': 441,\n",
       " 'exosome isolate': 675,\n",
       " 'isolate 10': 885,\n",
       " '10 healthy': 21,\n",
       " 'healthy volunteer': 791,\n",
       " 'volunteer urine': 1832,\n",
       " 'urine sample': 1785,\n",
       " 'sample previously': 1467,\n",
       " 'previously man': 1328,\n",
       " 'man woman': 980,\n",
       " 'woman age': 1869,\n",
       " 'age 2336': 263,\n",
       " '2336 year': 96,\n",
       " 'year regular': 1875,\n",
       " 'regular medication': 1402,\n",
       " 'medication consume': 1004,\n",
       " 'consume antibiotic': 511,\n",
       " 'antibiotic medication': 307,\n",
       " 'medication previous': 1005,\n",
       " 'previous month': 1327,\n",
       " 'month current': 1119,\n",
       " 'current uti': 532,\n",
       " 'uti exclude': 1814,\n",
       " 'exclude briefly': 667,\n",
       " 'briefly subject': 374,\n",
       " 'subject urinate': 1652,\n",
       " 'urinate directly': 1783,\n",
       " ...}"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vectorizer.vocabulary_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([2.70474809, 2.70474809, 2.70474809, ..., 2.70474809, 2.70474809,\n",
       "       2.70474809])"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vectorizer.idf_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "tfidf = vectorizer.transform(test_doc)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(10, 1877)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "array([[0.        , 0.        , 0.05627674, ..., 0.        , 0.        ,\n",
       "        0.        ],\n",
       "       [0.        , 0.        , 0.        , ..., 0.        , 0.        ,\n",
       "        0.        ],\n",
       "       [0.        , 0.        , 0.        , ..., 0.        , 0.        ,\n",
       "        0.        ],\n",
       "       ...,\n",
       "       [0.        , 0.        , 0.        , ..., 0.        , 0.        ,\n",
       "        0.        ],\n",
       "       [0.        , 0.        , 0.        , ..., 0.        , 0.        ,\n",
       "        0.05281834],\n",
       "       [0.        , 0.        , 0.        , ..., 0.        , 0.        ,\n",
       "        0.        ]])"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(tfidf.toarray().shape)\n",
    "tfidf.toarray()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Test case to figure out processing time"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "4390"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(method)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "n = len(method)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 23min 38s, sys: 21min 58s, total: 45min 36s\n",
      "Wall time: 5min 54s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "doc = [lemmatize_text(method.iloc[i]) for i in range(n)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 1.5 s, sys: 277 ms, total: 1.78 s\n",
      "Wall time: 1.43 s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "vectorizer = TfidfVectorizer(analyzer='word',  strip_accents='ascii', ngram_range=(2, 2), stop_words='english')\n",
    "bigram_tfidf = vectorizer.fit_transform(doc)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 1.45 s, sys: 9.29 ms, total: 1.46 s\n",
      "Wall time: 1.46 s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "vectorizer2 = CountVectorizer(analyzer='word',  strip_accents='ascii', ngram_range=(2, 2), stop_words='english')\n",
    "bigram_count= vectorizer2.fit_transform(doc)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "x = pd.DataFrame(bigram_count.toarray(), columns=vectorizer2.get_feature_names())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(4390, 219664)"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "x.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ms ms                    5004\n",
       "formic acid              3378\n",
       "mass spectrometer        2594\n",
       "lc ms                    2479\n",
       "50 mm                    1995\n",
       "thermo fisher            1805\n",
       "thermo scientific        1801\n",
       "flow rate                1663\n",
       "ms analysis              1617\n",
       "fisher scientific        1604\n",
       "et al                    1526\n",
       "10 mm                    1519\n",
       "30 min                   1465\n",
       "10 min                   1438\n",
       "room temperature         1432\n",
       "mass spectrometry        1316\n",
       "nl min                   1163\n",
       "mm tris                  1154\n",
       "100 mm                   1143\n",
       "ammonium bicarbonate     1121\n",
       "reverse phase            1102\n",
       "ms scan                  1097\n",
       "collision energy         1078\n",
       "ltq orbitrap             1073\n",
       "mm ammonium              1031\n",
       "sds page                  917\n",
       "data dependent            916\n",
       "tris hcl                  916\n",
       "lysis buffer              914\n",
       "peptide elute             885\n",
       "20 mm                     882\n",
       "spectrometer thermo       869\n",
       "dynamic exclusion         866\n",
       "final concentration       840\n",
       "mm dtt                    823\n",
       "protein concentration     808\n",
       "15 min                    800\n",
       "acetonitrile formic       799\n",
       "mobile phase              789\n",
       "analytical column         769\n",
       "protease inhibitor        763\n",
       "digest trypsin            753\n",
       "ultimate 3000             723\n",
       "hcl ph                    710\n",
       "sigma aldrich             692\n",
       "ion trap                  688\n",
       "peptide separate          686\n",
       "charge state              683\n",
       "orbitrap velos            662\n",
       "20 min                    656\n",
       "dtype: int64"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "x.sum(axis=0).sort_values(ascending=False).head(50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['04 label',\n",
       " '09 label',\n",
       " '10 label',\n",
       " '100 unlabeled',\n",
       " '10plex label',\n",
       " '10plex labeling',\n",
       " '10plex labelling',\n",
       " '11 label',\n",
       " '116 label',\n",
       " '117 label',\n",
       " '118 label',\n",
       " '12 label',\n",
       " '12 labelling',\n",
       " '121 label',\n",
       " '126 label',\n",
       " '127 label',\n",
       " '127 labelling',\n",
       " '128 label',\n",
       " '129n label',\n",
       " '130 label',\n",
       " '131 label',\n",
       " '131 labeling',\n",
       " '13c label',\n",
       " '13c15n label',\n",
       " '13cd3 label',\n",
       " '14g label',\n",
       " '14n label',\n",
       " '14n labeling',\n",
       " '14n unlabelled',\n",
       " '150 label',\n",
       " '15mm unlabeled',\n",
       " '15n label',\n",
       " '15n labeling',\n",
       " '15n labelling',\n",
       " '15n2 label',\n",
       " '15n4 labell',\n",
       " '16 label',\n",
       " '16 labelling',\n",
       " '16h label',\n",
       " '17odya label',\n",
       " '1830 label',\n",
       " '18o label',\n",
       " '18o labeling',\n",
       " '18o labelling',\n",
       " '20 label',\n",
       " '200 label',\n",
       " '2005 label',\n",
       " '2009 label',\n",
       " '2012 label',\n",
       " '2013 label',\n",
       " '213 label',\n",
       " '24 labelling',\n",
       " '2425 label',\n",
       " '24h label',\n",
       " '25 label',\n",
       " '28 label',\n",
       " '2dge label',\n",
       " '2mega labeling',\n",
       " '30 labeling',\n",
       " '30 labelling',\n",
       " '300 unlabeled',\n",
       " '300ul labeling',\n",
       " '35 label',\n",
       " '350 unlabelled',\n",
       " '37 label',\n",
       " '37 labeling',\n",
       " '37oc labeling',\n",
       " '3a labeling',\n",
       " '40 label',\n",
       " '494 label',\n",
       " '4h label',\n",
       " '500 label',\n",
       " '50ug label',\n",
       " '55 label',\n",
       " '5min label',\n",
       " '60 label',\n",
       " '600 label',\n",
       " '630 labeling',\n",
       " '64 label',\n",
       " '66 label',\n",
       " '67c label',\n",
       " '6_r2 label',\n",
       " '6plex label',\n",
       " '70 label',\n",
       " '72 labeling',\n",
       " '80 label',\n",
       " '800 labeling',\n",
       " '800labeled secondary',\n",
       " '80oc label',\n",
       " '8plex labeling',\n",
       " '95 labelling',\n",
       " '97 label',\n",
       " '98 label',\n",
       " '99 label',\n",
       " 'a03 label',\n",
       " 'abl labeld',\n",
       " 'access label',\n",
       " 'accidently label',\n",
       " 'accommodate label',\n",
       " 'accord label',\n",
       " 'acetonitrile label',\n",
       " 'acetonitrile labeling',\n",
       " 'acetylation labeling',\n",
       " 'achieve label',\n",
       " 'acid label',\n",
       " 'acid labeling',\n",
       " 'acid labelling',\n",
       " 'acid unlabeled',\n",
       " 'acn label',\n",
       " 'acn unlabeled',\n",
       " 'acquisition label',\n",
       " 'add labeling',\n",
       " 'add labelling',\n",
       " 'add unlabelled',\n",
       " 'addition radiolabel',\n",
       " 'additional label',\n",
       " 'aha label',\n",
       " 'aliquot label',\n",
       " 'aliquot unlabelled',\n",
       " 'alkylat label',\n",
       " 'allow label',\n",
       " 'allow labelling',\n",
       " 'amination labeling',\n",
       " 'amine label',\n",
       " 'amine labeling',\n",
       " 'ampylator label',\n",
       " 'analyse label',\n",
       " 'analysis label',\n",
       " 'analysis labeling',\n",
       " 'analyze label',\n",
       " 'animal label',\n",
       " 'antibody label',\n",
       " 'apply label',\n",
       " 'approximately label',\n",
       " 'arg0 label',\n",
       " 'arg10 label',\n",
       " 'arginine label',\n",
       " 'authors8 label',\n",
       " 'avoid label',\n",
       " 'azide label',\n",
       " 'base label',\n",
       " 'basically label',\n",
       " 'basis label',\n",
       " 'batch label',\n",
       " 'bead label',\n",
       " 'bead labeling',\n",
       " 'benzonase label',\n",
       " 'benzonase labeling',\n",
       " 'benzophenone label',\n",
       " 'bicarbonate label',\n",
       " 'biosystems label',\n",
       " 'biotin label',\n",
       " 'biotin labeling',\n",
       " 'biotin labelling',\n",
       " 'biotinylated unlabeled',\n",
       " 'biotinylation labeling',\n",
       " 'br label',\n",
       " 'br labeling',\n",
       " 'bradford unlabeled',\n",
       " 'briefly label',\n",
       " 'briefly labeling',\n",
       " 'briefly labelling',\n",
       " 'buffer label',\n",
       " 'buffer labeling',\n",
       " 'c13 label',\n",
       " 'c18 label',\n",
       " 'c3 label',\n",
       " 'cancer label',\n",
       " 'carry unlabeled',\n",
       " 'cartridge label',\n",
       " 'case label',\n",
       " 'cell label',\n",
       " 'centrifugation label',\n",
       " 'centrifugation labeling',\n",
       " 'centrifuge label',\n",
       " 'centriole label',\n",
       " 'cerevisiae label',\n",
       " 'chamber label',\n",
       " 'channel label',\n",
       " 'check label',\n",
       " 'check labeling',\n",
       " 'chemical label',\n",
       " 'chemical labeling',\n",
       " 'chemically label',\n",
       " 'chromatography label',\n",
       " 'cicat label',\n",
       " 'cicatlabel peptide',\n",
       " 'cleanup label',\n",
       " 'cm2 label',\n",
       " 'cm3 label',\n",
       " 'co2 labelling',\n",
       " 'cold labeling',\n",
       " 'collect label',\n",
       " 'collection label',\n",
       " 'colon label',\n",
       " 'column label',\n",
       " 'column labeling',\n",
       " 'column labelling',\n",
       " 'combine label',\n",
       " 'combine labeling',\n",
       " 'combine unlabeled',\n",
       " 'compare unlabeled',\n",
       " 'complete label',\n",
       " 'complete labeling',\n",
       " 'complete labelling',\n",
       " 'completely label',\n",
       " 'completeness labeling',\n",
       " 'completion labeling',\n",
       " 'concentration label',\n",
       " 'condition label',\n",
       " 'condition photolabel',\n",
       " 'condition unlabeled',\n",
       " 'contain label',\n",
       " 'contain unlabeled',\n",
       " 'containing label',\n",
       " 'control label',\n",
       " 'control labeling',\n",
       " 'conventional label',\n",
       " 'correspond unlabelled',\n",
       " 'correspondingly label',\n",
       " 'covalent labeling',\n",
       " 'covalently label',\n",
       " 'creation label',\n",
       " 'cross label',\n",
       " 'csc label',\n",
       " 'ctl label',\n",
       " 'cultivate label',\n",
       " 'culture label',\n",
       " 'cure unlabeled',\n",
       " 'cy2 label',\n",
       " 'cy2 labelled',\n",
       " 'cy3 labell',\n",
       " 'cy5 label',\n",
       " 'cy5 labell',\n",
       " 'cycle label',\n",
       " 'cydye label',\n",
       " 'cysteine labeling',\n",
       " 'cystmt6 labeling',\n",
       " 'd12 labelled',\n",
       " 'd3 label',\n",
       " 'da label',\n",
       " 'da labeling',\n",
       " 'dark label',\n",
       " 'dark labeling',\n",
       " 'dark unlabeled',\n",
       " 'day label',\n",
       " 'day labelling',\n",
       " 'day unlabelled',\n",
       " 'dependent label',\n",
       " 'desalt label',\n",
       " 'desalt labeling',\n",
       " 'desthiobiotin label',\n",
       " 'detailed labeling',\n",
       " 'determine label',\n",
       " 'determine labeling',\n",
       " 'deuterium labeling',\n",
       " 'dia label',\n",
       " 'diet label',\n",
       " 'diet unlabeled',\n",
       " 'different label',\n",
       " 'differentially label',\n",
       " 'differently label',\n",
       " 'dige labeling',\n",
       " 'digest label',\n",
       " 'digestion label',\n",
       " 'digestion labeling',\n",
       " 'digestion labelling',\n",
       " 'dimedone label',\n",
       " 'dimethyl label',\n",
       " 'dimethyl labeling',\n",
       " 'dimethyl labelling',\n",
       " 'dimethylation label',\n",
       " 'dimethylation labeling',\n",
       " 'dimethyllabel sample',\n",
       " 'discovery label',\n",
       " 'dish label',\n",
       " 'dmem label',\n",
       " 'dmso labeling',\n",
       " 'dmso unlabeled',\n",
       " 'doubling label',\n",
       " 'doxycycline labeling',\n",
       " 'dry label',\n",
       " 'dry labeling',\n",
       " 'dryness label',\n",
       " 'dtt label',\n",
       " 'duplex labeling',\n",
       " 'duplicate label',\n",
       " 'dyn labeling',\n",
       " 'each15n label',\n",
       " 'efficacy label',\n",
       " 'efficiency label',\n",
       " 'efficiency labeling',\n",
       " 'ehis photolabel',\n",
       " 'elite label',\n",
       " 'eluate label',\n",
       " 'eluent label',\n",
       " 'end labeling',\n",
       " 'energy label',\n",
       " 'enrich label',\n",
       " 'enrichment label',\n",
       " 'enzymatically label',\n",
       " 'equal label',\n",
       " 'equal unlabeled',\n",
       " 'evaporator label',\n",
       " 'excess labeling',\n",
       " 'excess labelling',\n",
       " 'exchanger label',\n",
       " 'exercise label',\n",
       " 'exist label',\n",
       " 'experiment label',\n",
       " 'experiment labeling',\n",
       " 'experiment labelling',\n",
       " 'extract label',\n",
       " 'fa label',\n",
       " 'facilitate label',\n",
       " 'fcs unlabeled',\n",
       " 'file label',\n",
       " 'finally label',\n",
       " 'fix label',\n",
       " 'fluid label',\n",
       " 'fluorescein label',\n",
       " 'fluorescently label',\n",
       " 'fmol label',\n",
       " 'follow label',\n",
       " 'follow labeling',\n",
       " 'follow labelling',\n",
       " 'form labeling',\n",
       " 'form unlabeled',\n",
       " 'formaldehyde labeling',\n",
       " 'forward label',\n",
       " 'fourplex label',\n",
       " 'fraction label',\n",
       " 'fractionate label',\n",
       " 'fractionation label',\n",
       " 'free label',\n",
       " 'freezability label',\n",
       " 'fully label',\n",
       " 'fusion label',\n",
       " 'g3 label',\n",
       " 'galnaz label',\n",
       " 'galnaz labeling',\n",
       " 'gcb label',\n",
       " 'gcjd label',\n",
       " 'gel label',\n",
       " 'generate label',\n",
       " 'germany label',\n",
       " 'germany labeling',\n",
       " 'glycoprotein label',\n",
       " 'group label',\n",
       " 'group labeling',\n",
       " 'grow unlabeled',\n",
       " 'grow unlabelled',\n",
       " 'h2o labels',\n",
       " 'half label',\n",
       " 'hcl label',\n",
       " 'heavy label',\n",
       " 'heavy labeled',\n",
       " 'heavy labeling',\n",
       " 'heavy labelled',\n",
       " 'heavy labelling',\n",
       " 'hek293 unlabeled',\n",
       " 'heteroconjugates label',\n",
       " 'hg001prsa label',\n",
       " 'hippocampus label',\n",
       " 'homogeneity labeling',\n",
       " 'hour label',\n",
       " 'hour labeling',\n",
       " 'house label',\n",
       " 'hpi label',\n",
       " 'hr label',\n",
       " 'hr labeling',\n",
       " 'hrp label',\n",
       " 'huvecs label',\n",
       " 'hydrazide label',\n",
       " 'i14l labelling',\n",
       " 'iam label',\n",
       " 'icat label',\n",
       " 'icat labeling',\n",
       " 'icat labelling',\n",
       " 'icdid label',\n",
       " 'ice label',\n",
       " 'icpl labeling',\n",
       " 'identification label',\n",
       " 'identify label',\n",
       " 'ii label',\n",
       " 'immunofluorescent labeling',\n",
       " 'immunopurification unlabeled',\n",
       " 'implantation labeling',\n",
       " 'include label',\n",
       " 'include labeling',\n",
       " 'incorporate label',\n",
       " 'incorporation label',\n",
       " 'incubate label',\n",
       " 'incubate unlabeled',\n",
       " 'independent labeling',\n",
       " 'indirect labeling',\n",
       " 'individually label',\n",
       " 'infect label',\n",
       " 'inhibition label',\n",
       " 'inhomogenous labeling',\n",
       " 'insert label',\n",
       " 'instruction label',\n",
       " 'instruction labeling',\n",
       " 'intermediate label',\n",
       " 'internal labeling',\n",
       " 'internalisation label',\n",
       " 'introduce label',\n",
       " 'invert label',\n",
       " 'iodoacetamide labelling',\n",
       " 'iodotmt label',\n",
       " 'iodotmt6 labeling',\n",
       " 'ion labeling',\n",
       " 'irdye 800labeled',\n",
       " 'isobaric label',\n",
       " 'isobaric labeling',\n",
       " 'isobaric labelling',\n",
       " 'isobarically label',\n",
       " 'isolation label',\n",
       " 'isopropanol label',\n",
       " 'isotope label',\n",
       " 'isotope labeling',\n",
       " 'isotope labelling',\n",
       " 'isotopic label',\n",
       " 'isotopic labeling',\n",
       " 'isotopic labelling',\n",
       " 'isotopically label',\n",
       " 'isotopicaly label',\n",
       " 'it121 label',\n",
       " 'itraq label',\n",
       " 'itraq labeling',\n",
       " 'itraq labelling',\n",
       " 'jpt label',\n",
       " 'k8 label',\n",
       " 'k8r10 label',\n",
       " 'kit label',\n",
       " 'kit labelled',\n",
       " 'label 02',\n",
       " 'label 10',\n",
       " 'label 100',\n",
       " 'label 10276',\n",
       " 'label 10plex',\n",
       " 'label 11',\n",
       " 'label 113',\n",
       " 'label 114',\n",
       " 'label 115',\n",
       " 'label 116',\n",
       " 'label 117',\n",
       " 'label 118',\n",
       " 'label 119',\n",
       " 'label 12',\n",
       " 'label 120',\n",
       " 'label 121',\n",
       " 'label 126',\n",
       " 'label 127',\n",
       " 'label 128',\n",
       " 'label 128c',\n",
       " 'label 128n',\n",
       " 'label 129',\n",
       " 'label 12c6',\n",
       " 'label 12ch2o',\n",
       " 'label 130n',\n",
       " 'label 13c2d3',\n",
       " 'label 13c6',\n",
       " 'label 13c615n2',\n",
       " 'label 13c615n4',\n",
       " 'label 13cd2o',\n",
       " 'label 15',\n",
       " 'label 15n',\n",
       " 'label 16',\n",
       " 'label 17',\n",
       " 'label 1ml',\n",
       " 'label 20',\n",
       " 'label 200',\n",
       " 'label 200k',\n",
       " 'label 20l',\n",
       " 'label 21',\n",
       " 'label 22',\n",
       " 'label 24',\n",
       " 'label 25',\n",
       " 'label 2d',\n",
       " 'label 30',\n",
       " 'label 32',\n",
       " 'label 35s',\n",
       " 'label 3t3',\n",
       " 'label 40',\n",
       " 'label 400',\n",
       " 'label 42',\n",
       " 'label 48',\n",
       " 'label 4plex',\n",
       " 'label 50',\n",
       " 'label 62',\n",
       " 'label 63',\n",
       " 'label 6plex',\n",
       " 'label 70',\n",
       " 'label 80',\n",
       " 'label 8plex',\n",
       " 'label a20',\n",
       " 'label a431',\n",
       " 'label a549',\n",
       " 'label accord',\n",
       " 'label acetone',\n",
       " 'label acetylated',\n",
       " 'label activation',\n",
       " 'label activity',\n",
       " 'label add',\n",
       " 'label addition',\n",
       " 'label adipocyte',\n",
       " 'label adult',\n",
       " 'label agent',\n",
       " 'label aha',\n",
       " 'label akt1',\n",
       " 'label alexa',\n",
       " 'label alkyne',\n",
       " 'label allow',\n",
       " 'label amine',\n",
       " 'label amino',\n",
       " 'label aminooxy',\n",
       " 'label analyse',\n",
       " 'label analysis',\n",
       " 'label analyze',\n",
       " 'label animal',\n",
       " 'label anti',\n",
       " 'label anxa1',\n",
       " 'label apoa',\n",
       " 'label approach',\n",
       " 'label appropriate',\n",
       " 'label approximately',\n",
       " 'label arg',\n",
       " 'label arg6',\n",
       " 'label arginine',\n",
       " 'label assess',\n",
       " 'label associate',\n",
       " 'label atom40',\n",
       " 'label atp',\n",
       " 'label azido',\n",
       " 'label bacteria',\n",
       " 'label base',\n",
       " 'label bias',\n",
       " 'label biocytin',\n",
       " 'label biological',\n",
       " 'label biomass',\n",
       " 'label blend',\n",
       " 'label bloodstream',\n",
       " 'label boersema',\n",
       " 'label brain',\n",
       " 'label breast',\n",
       " 'label bt',\n",
       " 'label buffer',\n",
       " 'label c18',\n",
       " 'label carbonylat',\n",
       " 'label cd3',\n",
       " 'label cd4',\n",
       " 'label cell',\n",
       " 'label cellular',\n",
       " 'label ch2o',\n",
       " 'label ch3',\n",
       " 'label channel',\n",
       " 'label chd2',\n",
       " 'label check',\n",
       " 'label chromatin',\n",
       " 'label clean',\n",
       " 'label cleavable',\n",
       " 'label cm',\n",
       " 'label coli',\n",
       " 'label collect',\n",
       " 'label column',\n",
       " 'label combine',\n",
       " 'label combined',\n",
       " 'label comparison',\n",
       " 'label compound',\n",
       " 'label condition',\n",
       " 'label confirm',\n",
       " 'label confluent',\n",
       " 'label conidia',\n",
       " 'label consider',\n",
       " 'label contain',\n",
       " 'label control',\n",
       " 'label correspond',\n",
       " 'label corresponding',\n",
       " 'label counterpart',\n",
       " 'label cr',\n",
       " 'label ctl',\n",
       " 'label ctps',\n",
       " 'label ctr7',\n",
       " 'label ctrl',\n",
       " 'label culture',\n",
       " 'label cultured',\n",
       " 'label custom',\n",
       " 'label cy',\n",
       " 'label cy2',\n",
       " 'label cy3',\n",
       " 'label cydye2',\n",
       " 'label cydyes',\n",
       " 'label cys',\n",
       " 'label d4',\n",
       " 'label d5',\n",
       " 'label datum',\n",
       " 'label day',\n",
       " 'label desthiobiotinylat',\n",
       " 'label deuterated',\n",
       " 'label dg75',\n",
       " 'label diet',\n",
       " 'label different',\n",
       " 'label differently',\n",
       " 'label digest',\n",
       " 'label digestion',\n",
       " 'label dilute',\n",
       " 'label dimethyl',\n",
       " 'label dimethylation',\n",
       " 'label directly',\n",
       " 'label disuccinimidyl',\n",
       " 'label dmem',\n",
       " 'label dna',\n",
       " 'label drug',\n",
       " 'label dss',\n",
       " 'label du145',\n",
       " 'label duplicate',\n",
       " 'label e8',\n",
       " 'label ea',\n",
       " 'label efficiency',\n",
       " 'label egm2',\n",
       " 'label eluate',\n",
       " 'label enable',\n",
       " 'label enrich',\n",
       " 'label esc',\n",
       " 'label experiment',\n",
       " 'label extensive',\n",
       " 'label extract',\n",
       " 'label female',\n",
       " 'label fibroblast',\n",
       " 'label figure',\n",
       " 'label filter',\n",
       " 'label final',\n",
       " 'label fix',\n",
       " 'label fluorescein',\n",
       " 'label fluorophore',\n",
       " 'label flush',\n",
       " 'label follow',\n",
       " 'label formaldehyde',\n",
       " 'label fp',\n",
       " 'label fraction',\n",
       " 'label fractionate',\n",
       " 'label free',\n",
       " 'label freshly',\n",
       " 'label fumigatus',\n",
       " 'label g1s',\n",
       " 'label g2',\n",
       " 'label ga',\n",
       " 'label galnac',\n",
       " 'label gfp',\n",
       " 'label glcnac',\n",
       " 'label glycan',\n",
       " 'label glycoprotein',\n",
       " 'label goat',\n",
       " 'label h2o2',\n",
       " 'label half',\n",
       " 'label hbri',\n",
       " 'label hcmv',\n",
       " 'label hct116',\n",
       " 'label heavy',\n",
       " 'label hek293',\n",
       " 'label hela',\n",
       " 'label hepa',\n",
       " 'label high',\n",
       " 'label hk',\n",
       " 'label hour',\n",
       " 'label hrde',\n",
       " 'label hsp21',\n",
       " 'label hufa',\n",
       " 'label huvecs',\n",
       " 'label hvff',\n",
       " 'label hydroxysuccinimide',\n",
       " 'label hydroxysuccinimidobiotin',\n",
       " 'label iaa',\n",
       " 'label icpl',\n",
       " 'label im',\n",
       " 'label imcrx',\n",
       " 'label incorporation',\n",
       " 'label incubation',\n",
       " 'label independent',\n",
       " 'label individually',\n",
       " 'label inhibitor',\n",
       " 'label intermediate',\n",
       " 'label internal',\n",
       " 'label iodotmt6',\n",
       " 'label ip',\n",
       " 'label isg15',\n",
       " 'label isobaric',\n",
       " 'label isotope',\n",
       " 'label isotopic',\n",
       " 'label itraq',\n",
       " 'label itraq114',\n",
       " 'label itraq115',\n",
       " 'label jk418',\n",
       " 'label jurkat',\n",
       " 'label k0r0',\n",
       " 'label k8',\n",
       " 'label keratinocyte',\n",
       " 'label kit',\n",
       " 'label ko',\n",
       " 'label labelling',\n",
       " 'label lcl',\n",
       " 'label leucine',\n",
       " 'label light',\n",
       " 'label line',\n",
       " 'label linearized',\n",
       " 'label liquid',\n",
       " 'label live',\n",
       " 'label liver',\n",
       " 'label lm2',\n",
       " 'label lncap',\n",
       " 'label lyophilize',\n",
       " 'label lys',\n",
       " 'label lys4',\n",
       " 'label lys8',\n",
       " 'label lysate',\n",
       " 'label lysine',\n",
       " 'label manufacturer',\n",
       " 'label mapklk1',\n",
       " 'label mass',\n",
       " 'label material',\n",
       " 'label maximum',\n",
       " 'label mc58',\n",
       " 'label mcf',\n",
       " 'label mcf7',\n",
       " 'label mda',\n",
       " 'label mdck',\n",
       " 'label measure',\n",
       " 'label medium',\n",
       " 'label mefs',\n",
       " 'label membrane',\n",
       " 'label methionine',\n",
       " 'label mg',\n",
       " 'label micrograms',\n",
       " 'label mino',\n",
       " 'label mitochondria',\n",
       " 'label mix',\n",
       " 'label mixed',\n",
       " 'label mixture',\n",
       " 'label ml',\n",
       " 'label mm',\n",
       " 'label mock',\n",
       " 'label mono',\n",
       " 'label monophosphate',\n",
       " 'label mouse',\n",
       " 'label mrc',\n",
       " 'label ms',\n",
       " 'label ms151906xl_emmanuelle_hang',\n",
       " 'label ms152077xl_emmanuelle_hang',\n",
       " 'label mtraq',\n",
       " 'label mtraqd0',\n",
       " 'label mtraqd4',\n",
       " 'label myoblast',\n",
       " 'label neurobllastoma',\n",
       " 'label neuron',\n",
       " 'label newly',\n",
       " 'label nhs',\n",
       " 'label non',\n",
       " 'label normal',\n",
       " 'label nucleosome',\n",
       " 'label nucleotide',\n",
       " 'label okf6',\n",
       " 'label ong',\n",
       " 'label opposite',\n",
       " 'label oxygen',\n",
       " 'label parallel',\n",
       " 'label passage',\n",
       " 'label pbs',\n",
       " 'label pd145',\n",
       " 'label peptide',\n",
       " 'label perform',\n",
       " 'label pex14',\n",
       " 'label phosphopeptide',\n",
       " 'label phosphopetide',\n",
       " 'label photoaffinity',\n",
       " 'label pierce',\n",
       " 'label place',\n",
       " 'label plex',\n",
       " 'label pool',\n",
       " 'label post',\n",
       " 'label ppeptides',\n",
       " 'label pre',\n",
       " 'label precede',\n",
       " 'label precursor',\n",
       " 'label prepare',\n",
       " 'label previously',\n",
       " 'label prior',\n",
       " 'label probe',\n",
       " 'label procedure',\n",
       " 'label process',\n",
       " 'label product',\n",
       " 'label pron',\n",
       " 'label protein',\n",
       " 'label proteolytic',\n",
       " 'label proteome',\n",
       " 'label proteomic',\n",
       " 'label provide',\n",
       " 'label psd95',\n",
       " 'label purification',\n",
       " 'label purify',\n",
       " 'label qcat',\n",
       " 'label qconcat',\n",
       " 'label qiao',\n",
       " 'label quantification',\n",
       " 'label quantitation',\n",
       " 'label quench',\n",
       " 'label quenching',\n",
       " 'label r10',\n",
       " 'label r79c',\n",
       " 'label radioactively',\n",
       " 'label ratio',\n",
       " 'label react',\n",
       " 'label reaction',\n",
       " 'label reagent',\n",
       " 'label reagents',\n",
       " 'label reconstitute',\n",
       " 'label reductive',\n",
       " 'label reference',\n",
       " 'label regular',\n",
       " 'label relative',\n",
       " 'label repeat',\n",
       " 'label replicate',\n",
       " 'label replication',\n",
       " 'label reporter',\n",
       " 'label resolve',\n",
       " 'label respective',\n",
       " 'label respectively',\n",
       " 'label result',\n",
       " 'label resuspend',\n",
       " 'label reverse',\n",
       " 'label reversely',\n",
       " 'label rna',\n",
       " 'label room',\n",
       " 'label rpe1s',\n",
       " 'label rsp5',\n",
       " 'label run',\n",
       " 'label sam50',\n",
       " 'label sample',\n",
       " 'label scr5',\n",
       " 'label scr6',\n",
       " 'label sdc',\n",
       " 'label sds',\n",
       " 'label secondary',\n",
       " 'label selectively',\n",
       " 'label separate',\n",
       " 'label separately',\n",
       " 'label set',\n",
       " 'label sham',\n",
       " 'label siglec',\n",
       " 'label sil',\n",
       " 'label silac',\n",
       " 'label single',\n",
       " 'label sixplex',\n",
       " 'label smem',\n",
       " 'label solution',\n",
       " 'label spike',\n",
       " 'label splitting',\n",
       " 'label stable',\n",
       " 'label stably',\n",
       " 'label stage',\n",
       " 'label stagetip',\n",
       " 'label standard',\n",
       " 'label stimulate',\n",
       " 'label strain',\n",
       " 'label strategy',\n",
       " 'label strong',\n",
       " 'label study',\n",
       " 'label subject',\n",
       " 'label subsequent',\n",
       " 'label subsequently',\n",
       " 'label suggest',\n",
       " 'label suject',\n",
       " 'label sulfenic',\n",
       " 'label sulfo',\n",
       " 'label superparamagnetic',\n",
       " 'label swap',\n",
       " 'label swapping',\n",
       " 'label switch',\n",
       " 'label switching',\n",
       " 'label synthetic',\n",
       " 'label t8',\n",
       " 'label tag',\n",
       " 'label tags',\n",
       " 'label tandem',\n",
       " 'label tap',\n",
       " 'label target',\n",
       " 'label tdtomato',\n",
       " 'label technical',\n",
       " 'label technique',\n",
       " 'label terminal',\n",
       " 'label terminally',\n",
       " 'label termini',\n",
       " 'label terminus',\n",
       " 'label test',\n",
       " 'label tetracycline',\n",
       " 'label thp',\n",
       " 'label time',\n",
       " 'label tissue',\n",
       " 'label tmt',\n",
       " 'label tmt0',\n",
       " 'label tmt10plex',\n",
       " 'label tmt6',\n",
       " 'label tmt6plex',\n",
       " 'label total',\n",
       " 'label transect',\n",
       " 'label transfect',\n",
       " 'label trap3high',\n",
       " 'label trypsin',\n",
       " 'label tryptic',\n",
       " 'label u251',\n",
       " 'label u2os',\n",
       " 'label uncut',\n",
       " 'label unique',\n",
       " 'label unlabeled',\n",
       " 'label unlabelled',\n",
       " 'label urea',\n",
       " 'label use',\n",
       " 'label various',\n",
       " 'label vector',\n",
       " 'label version',\n",
       " 'label virus',\n",
       " 'label vol',\n",
       " 'label water',\n",
       " 'label wild',\n",
       " 'label wt',\n",
       " 'label wts',\n",
       " 'label xx',\n",
       " 'label yeast',\n",
       " 'labeld silac',\n",
       " 'labeled 13c6',\n",
       " 'labeled atlas',\n",
       " 'labeled cell',\n",
       " 'labeled hek293',\n",
       " 'labeled medium',\n",
       " 'labeled sample',\n",
       " 'labeled silac',\n",
       " 'labelfree enrichment',\n",
       " 'labeling 10',\n",
       " 'labeling 100',\n",
       " 'labeling 13c6',\n",
       " 'labeling 16',\n",
       " 'labeling 200',\n",
       " 'labeling 25',\n",
       " 'labeling 3t3',\n",
       " 'labeling 500',\n",
       " 'labeling 5l',\n",
       " 'labeling 97',\n",
       " 'labeling 98',\n",
       " 'labeling 99',\n",
       " 'labeling ab',\n",
       " 'labeling accord',\n",
       " 'labeling acidify',\n",
       " 'labeling add',\n",
       " 'labeling affinity',\n",
       " 'labeling aliquot',\n",
       " 'labeling allow',\n",
       " 'labeling amino',\n",
       " 'labeling aminoacids',\n",
       " 'labeling ammonium',\n",
       " 'labeling analysis',\n",
       " 'labeling analyze',\n",
       " 'labeling appropriate',\n",
       " 'labeling approximately',\n",
       " 'labeling arg0',\n",
       " 'labeling available',\n",
       " 'labeling bacterial',\n",
       " 'labeling base',\n",
       " 'labeling biological',\n",
       " 'labeling boersema',\n",
       " 'labeling brief',\n",
       " 'labeling briefly',\n",
       " 'labeling buffer',\n",
       " 'labeling c57',\n",
       " 'labeling carry',\n",
       " 'labeling cd2o',\n",
       " 'labeling cell',\n",
       " 'labeling check',\n",
       " 'labeling click',\n",
       " 'labeling column',\n",
       " 'labeling combine',\n",
       " 'labeling condition',\n",
       " 'labeling conduct',\n",
       " 'labeling conidia',\n",
       " 'labeling continue',\n",
       " 'labeling control',\n",
       " 'labeling controlcell',\n",
       " 'labeling derivatization',\n",
       " 'labeling desalt',\n",
       " 'labeling desalting',\n",
       " 'labeling deuterated',\n",
       " 'labeling different',\n",
       " 'labeling digest',\n",
       " 'labeling dimethylation',\n",
       " 'labeling directly',\n",
       " 'labeling dna',\n",
       " 'labeling dry',\n",
       " 'labeling dt40',\n",
       " 'labeling dye',\n",
       " 'labeling e8',\n",
       " 'labeling efficiency',\n",
       " ...]"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "[col for col in x.columns if 'label' in col]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "183"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "x['label free'].sum(axis=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['000 spectra',\n",
       " '10 spectra',\n",
       " '100ms spectra',\n",
       " '145 spectra',\n",
       " '1500 spectra',\n",
       " '1600 spectra',\n",
       " '2017 spectral',\n",
       " '250 spectra',\n",
       " '27 spectra',\n",
       " '275 spectra',\n",
       " '28 spectra',\n",
       " '2kv spectra',\n",
       " '30 spectra',\n",
       " '30 spectral',\n",
       " '3spectra maximum',\n",
       " '45 spectra',\n",
       " '50 spectra',\n",
       " '60 spectra',\n",
       " '6965 spectral',\n",
       " '70 spectra',\n",
       " '8902345 spectra',\n",
       " 'abundant spectra',\n",
       " 'acquire spectra',\n",
       " 'acquisition spectral',\n",
       " 'active spectra',\n",
       " 'additional spectral',\n",
       " 'agilent spectra',\n",
       " 'aldrich spectra',\n",
       " 'analysis spectra',\n",
       " 'analysis spectral',\n",
       " 'assay spectramax',\n",
       " 'assign spectra',\n",
       " 'base spectral',\n",
       " 'cad spectra',\n",
       " 'charge spectra',\n",
       " 'cid spectra',\n",
       " 'cluster spectra',\n",
       " 'collect spectra',\n",
       " 'configuration spectra',\n",
       " 'construct spectral',\n",
       " 'coverage spectral',\n",
       " 'dda spectra',\n",
       " 'dda spectral',\n",
       " 'deconvolute spectra',\n",
       " 'dependent spectra',\n",
       " 'dependent spectral',\n",
       " 'dionex spectra',\n",
       " 'disabled spectra',\n",
       " 'dissociation spectra',\n",
       " 'duration spectra',\n",
       " 'ecd spectra',\n",
       " 'energy spectra',\n",
       " 'etd spectra',\n",
       " 'exclude spectra',\n",
       " 'exclusion spectra',\n",
       " 'export spectral',\n",
       " 'file spectra',\n",
       " 'fragment spectra',\n",
       " 'fragmentation spectra',\n",
       " 'ftms spectra',\n",
       " 'ftms1 spectra',\n",
       " 'ftms3 spectra',\n",
       " 'generation spectral',\n",
       " 'grouping spectra',\n",
       " 'gs spectra',\n",
       " 'hcd spectra',\n",
       " 'ims spectra',\n",
       " 'injection spectra',\n",
       " 'intense spectra',\n",
       " 'intensity spectra',\n",
       " 'ion spectra',\n",
       " 'ion spectral',\n",
       " 'itms2 spectra',\n",
       " 'kv spectra',\n",
       " 'lc spectra',\n",
       " 'level spectra',\n",
       " 'level1 spectra',\n",
       " 'library spectra',\n",
       " 'light spectral',\n",
       " 'line spectra',\n",
       " 'mass spectra',\n",
       " 'mass spectral',\n",
       " 'mean spectral',\n",
       " 'measurement spectra',\n",
       " 'min spectra',\n",
       " 'mode spectra',\n",
       " 'mode spectral',\n",
       " 'modification spectra',\n",
       " 'modification spectral',\n",
       " 'ms spectra',\n",
       " 'ms spectral',\n",
       " 'ms spectrawere',\n",
       " 'ms1 spectra',\n",
       " 'ms1 spectral',\n",
       " 'ms2 spectra',\n",
       " 'ms3 spectra',\n",
       " 'msms spectra',\n",
       " 'normalization spectral',\n",
       " 'normalized spectral',\n",
       " 'objective spectra',\n",
       " 'obtain spectra',\n",
       " 'optimize spectra',\n",
       " 'orbitrap spectra',\n",
       " 'peptide spectra',\n",
       " 'peptide spectral',\n",
       " 'perform spectra',\n",
       " 'picoview spectra',\n",
       " 'precursor spectra',\n",
       " 'previously spectra',\n",
       " 'pro spectra',\n",
       " 'pro spectral',\n",
       " 'product spectra',\n",
       " 'profile spectra',\n",
       " 'protein spectra',\n",
       " 'protein spectral',\n",
       " 'quantification spectra',\n",
       " 'rate 3spectra',\n",
       " 'rate spectra',\n",
       " 'raw spectra',\n",
       " 'raw spectral',\n",
       " 'reduce spectra',\n",
       " 'redundant spectra',\n",
       " 'reference spectral',\n",
       " 'remain spectra',\n",
       " 'resolution spectra',\n",
       " 'respectively spectra',\n",
       " 'result spectra',\n",
       " 'run spectra',\n",
       " 'scale spectra',\n",
       " 'scan spectra',\n",
       " 'scientific spectra',\n",
       " 'search spectra',\n",
       " 'source spectra',\n",
       " 'spectra 000',\n",
       " 'spectra 10',\n",
       " 'spectra 100',\n",
       " 'spectra 100ms',\n",
       " 'spectra 106',\n",
       " 'spectra 12',\n",
       " 'spectra 138',\n",
       " 'spectra 15',\n",
       " 'spectra 17',\n",
       " 'spectra 17500',\n",
       " 'spectra 20',\n",
       " 'spectra 200',\n",
       " 'spectra 2002000',\n",
       " 'spectra 2501',\n",
       " 'spectra 30',\n",
       " 'spectra 300',\n",
       " 'spectra 3001350',\n",
       " 'spectra 3001500',\n",
       " 'spectra 3001650',\n",
       " 'spectra 3001700',\n",
       " 'spectra 3001800',\n",
       " 'spectra 3002000',\n",
       " 'spectra 300a2000',\n",
       " 'spectra 3102000',\n",
       " 'spectra 335',\n",
       " 'spectra 3351800',\n",
       " 'spectra 350',\n",
       " 'spectra 3501400',\n",
       " 'spectra 3501550',\n",
       " 'spectra 3501800',\n",
       " 'spectra 3502000',\n",
       " 'spectra 375',\n",
       " 'spectra 3751500',\n",
       " 'spectra 3751600',\n",
       " 'spectra 3751650',\n",
       " 'spectra 3751800',\n",
       " 'spectra 400',\n",
       " 'spectra 4001600',\n",
       " 'spectra 4002000',\n",
       " 'spectra 400m',\n",
       " 'spectra 50',\n",
       " 'spectra 500ms',\n",
       " 'spectra 5e4',\n",
       " 'spectra 60000',\n",
       " 'spectra 7500',\n",
       " 'spectra 8plex',\n",
       " 'spectra abundant',\n",
       " 'spectra accumulate',\n",
       " 'spectra accumulation',\n",
       " 'spectra acquire',\n",
       " 'spectra acquisition',\n",
       " 'spectra addition',\n",
       " 'spectra agc',\n",
       " 'spectra allow',\n",
       " 'spectra analyse',\n",
       " 'spectra analyze',\n",
       " 'spectra apply',\n",
       " 'spectra assign',\n",
       " 'spectra automatic',\n",
       " 'spectra automsn',\n",
       " 'spectra average',\n",
       " 'spectra base',\n",
       " 'spectra baseline',\n",
       " 'spectra calculate',\n",
       " 'spectra calibrate',\n",
       " 'spectra chimeric',\n",
       " 'spectra choose',\n",
       " 'spectra chromatographic',\n",
       " 'spectra cid',\n",
       " 'spectra collect',\n",
       " 'spectra combine',\n",
       " 'spectra compare',\n",
       " 'spectra compound',\n",
       " 'spectra concurrently',\n",
       " 'spectra convert',\n",
       " 'spectra correspond',\n",
       " 'spectra cycle',\n",
       " 'spectra data',\n",
       " 'spectra dda',\n",
       " 'spectra deisotop',\n",
       " 'spectra derive',\n",
       " 'spectra determine',\n",
       " 'spectra doubly',\n",
       " 'spectra dynamic',\n",
       " 'spectra elute',\n",
       " 'spectra enrich',\n",
       " 'spectra entire',\n",
       " 'spectra enzyme',\n",
       " 'spectra exactive',\n",
       " 'spectra exclude',\n",
       " 'spectra exclusion',\n",
       " 'spectra experiment',\n",
       " 'spectra export',\n",
       " 'spectra extract',\n",
       " 'spectra f15',\n",
       " 'spectra fix',\n",
       " 'spectra fixed',\n",
       " 'spectra fragment',\n",
       " 'spectra fragmentation',\n",
       " 'spectra fragmented',\n",
       " 'spectra generate',\n",
       " 'spectra hcd',\n",
       " 'spectra high',\n",
       " 'spectra identify',\n",
       " 'spectra incremental',\n",
       " 'spectra intact',\n",
       " 'spectra intense',\n",
       " 'spectra internally',\n",
       " 'spectra ion',\n",
       " 'spectra isolation',\n",
       " 'spectra lc',\n",
       " 'spectra linear',\n",
       " 'spectra low',\n",
       " 'spectra ltq',\n",
       " 'spectra mass',\n",
       " 'spectra maximum',\n",
       " 'spectra measure',\n",
       " 'spectra microscan',\n",
       " 'spectra min',\n",
       " 'spectra minimum',\n",
       " 'spectra ms',\n",
       " 'spectra ms1',\n",
       " 'spectra ms2',\n",
       " 'spectra noising',\n",
       " 'spectra normalize',\n",
       " 'spectra normalized',\n",
       " 'spectra obtain',\n",
       " 'spectra open',\n",
       " 'spectra orbitrap',\n",
       " 'spectra parent',\n",
       " 'spectra peak',\n",
       " 'spectra peptide',\n",
       " 'spectra perform',\n",
       " 'spectra permanent',\n",
       " 'spectra phosphopeptide',\n",
       " 'spectra pool',\n",
       " 'spectra precursor',\n",
       " 'spectra predict',\n",
       " 'spectra process',\n",
       " 'spectra produce',\n",
       " 'spectra product',\n",
       " 'spectra production',\n",
       " 'spectra profile',\n",
       " 'spectra pron',\n",
       " 'spectra protein',\n",
       " 'spectra quadrupole',\n",
       " 'spectra range',\n",
       " 'spectra rate',\n",
       " 'spectra record',\n",
       " 'spectra release',\n",
       " 'spectra repeat',\n",
       " 'spectra reporter',\n",
       " 'spectra resolution',\n",
       " 'spectra respectively',\n",
       " 'spectra resultant',\n",
       " 'spectra sample',\n",
       " 'spectra save',\n",
       " 'spectra scan',\n",
       " 'spectra search',\n",
       " 'spectra sec',\n",
       " 'spectra second',\n",
       " 'spectra select',\n",
       " 'spectra selection',\n",
       " 'spectra set',\n",
       " 'spectra signal',\n",
       " 'spectra silac',\n",
       " 'spectra speed',\n",
       " 'spectra store',\n",
       " 'spectra subject',\n",
       " 'spectra sum',\n",
       " 'spectra supplementary',\n",
       " 'spectra survey',\n",
       " 'spectra trypsin',\n",
       " 'spectra use',\n",
       " 'spectra variable',\n",
       " 'spectra velos',\n",
       " 'spectra viral',\n",
       " 'spectra vitro',\n",
       " 'spectra zee',\n",
       " 'spectral abundance',\n",
       " 'spectral acquisition',\n",
       " 'spectral analysis',\n",
       " 'spectral collect',\n",
       " 'spectral count',\n",
       " 'spectral counting',\n",
       " 'spectral counting3',\n",
       " 'spectral datum',\n",
       " 'spectral density',\n",
       " 'spectral histopathology',\n",
       " 'spectral library',\n",
       " 'spectral match',\n",
       " 'spectral peak',\n",
       " 'spectral quality',\n",
       " 'spectramax m2',\n",
       " 'spectramax m5',\n",
       " 'spectrawere record',\n",
       " 'spectrometer spectra',\n",
       " 'spectrometry spectra',\n",
       " 'state spectra',\n",
       " 'survey spectra',\n",
       " 'swath spectral',\n",
       " 'synthetic spectral',\n",
       " 'thermofisher spectral',\n",
       " 'time spectra',\n",
       " 'trap spectra',\n",
       " 'usa spectramax',\n",
       " 'use spectra',\n",
       " 'use spectral',\n",
       " 'use spectramax',\n",
       " 'uv spectra',\n",
       " 'velos spectra',\n",
       " 'voltage spectra',\n",
       " 'waters spectra',\n",
       " 'work spectral']"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "[col for col in x.columns if 'spectra' in col]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "spectral count        3\n",
       "spectral counting     5\n",
       "spectral abundance    1\n",
       "dtype: int64"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "x[['spectral count', 'spectral counting', 'spectral abundance']].sum(axis=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Some regex to filter columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [],
   "source": [
    "with_nubers = [col for col in x.columns if re.match(r'(?:.*\\d+.* [A-z]*)|(?:[A-z]* .*\\d+.*)', col)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "62258\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "['00 00',\n",
       " '00 01',\n",
       " '00 12',\n",
       " '00 15',\n",
       " '00 1800',\n",
       " '00 2000',\n",
       " '00 2425',\n",
       " '00 2426',\n",
       " '00 38',\n",
       " '00 50',\n",
       " '00 5l',\n",
       " '00 65',\n",
       " '00 assess',\n",
       " '00 centrifugation',\n",
       " '00 concentrated',\n",
       " '00 correspond',\n",
       " '00 flow',\n",
       " '00 ftms',\n",
       " '00 hrs',\n",
       " '00 instrument',\n",
       " '00 kv',\n",
       " '00 ld',\n",
       " '00 measurement',\n",
       " '00 ml',\n",
       " '00 mobile',\n",
       " '00 noon',\n",
       " '00 ph',\n",
       " '00 place',\n",
       " '00 pron',\n",
       " '00 sample',\n",
       " '00 soft',\n",
       " '00 sun',\n",
       " '00 tryptic',\n",
       " '000 000',\n",
       " '000 0000',\n",
       " '000 0e4',\n",
       " '000 0e5',\n",
       " '000 10',\n",
       " '000 100',\n",
       " '000 10min',\n",
       " '000 10minutes',\n",
       " '000 110',\n",
       " '000 12',\n",
       " '000 120',\n",
       " '000 15',\n",
       " '000 150',\n",
       " '000 15min',\n",
       " '000 15oc',\n",
       " '000 17',\n",
       " '000 17500',\n",
       " '000 18',\n",
       " '000 1h',\n",
       " '000 1h30',\n",
       " '000 20',\n",
       " '000 200',\n",
       " '000 200m',\n",
       " '000 20a',\n",
       " '000 20min',\n",
       " '000 21',\n",
       " '000 22',\n",
       " '000 24',\n",
       " '000 25',\n",
       " '000 250',\n",
       " '000 2aa',\n",
       " '000 2min',\n",
       " '000 30',\n",
       " '000 300',\n",
       " '000 30min',\n",
       " '000 35',\n",
       " '000 350',\n",
       " '000 375',\n",
       " '000 3801',\n",
       " '000 3e6',\n",
       " '000 3min',\n",
       " '000 40',\n",
       " '000 400',\n",
       " '000 4001',\n",
       " '000 400ms',\n",
       " '000 400th',\n",
       " '000 45',\n",
       " '000 4a',\n",
       " '000 4oc',\n",
       " '000 50',\n",
       " '000 500',\n",
       " '000 60',\n",
       " '000 60min',\n",
       " '000 70',\n",
       " '000 75',\n",
       " '000 7500',\n",
       " '000 800',\n",
       " '000 90',\n",
       " '000 abundant',\n",
       " '000 accumulation',\n",
       " '000 acetone',\n",
       " '000 acg',\n",
       " '000 acid',\n",
       " '000 acquire',\n",
       " '000 acquisition',\n",
       " '000 activation',\n",
       " '000 add',\n",
       " '000 additional',\n",
       " '000 aforementioned',\n",
       " '000 agc',\n",
       " '000 aliquot',\n",
       " '000 alkylat',\n",
       " '000 allow',\n",
       " '000 analysis',\n",
       " '000 animal',\n",
       " '000 apply',\n",
       " '000 auto',\n",
       " '000 automate',\n",
       " '000 automated',\n",
       " '000 automatic',\n",
       " '000 avoid',\n",
       " '000 base',\n",
       " '000 baseline',\n",
       " '000 bead',\n",
       " '000 biofuge',\n",
       " '000 buffer',\n",
       " '000 c2c12s',\n",
       " '000 cancer',\n",
       " '000 capillary',\n",
       " '000 capture',\n",
       " '000 case',\n",
       " '000 cat',\n",
       " '000 cell',\n",
       " '000 centrifugation',\n",
       " '000 centrifuge',\n",
       " '000 charge',\n",
       " '000 choose',\n",
       " '000 cid',\n",
       " '000 clear',\n",
       " '000 cleared',\n",
       " '000 collect',\n",
       " '000 collision',\n",
       " '000 concentrated',\n",
       " '000 concentration',\n",
       " '000 consist',\n",
       " '000 count',\n",
       " '000 cover',\n",
       " '000 cp',\n",
       " '000 cps',\n",
       " '000 crude',\n",
       " '000 cycle',\n",
       " '000 cytosolic',\n",
       " '000 da',\n",
       " '000 data',\n",
       " '000 datadependent',\n",
       " '000 datum',\n",
       " '000 dd',\n",
       " '000 debris',\n",
       " '000 define',\n",
       " '000 delay',\n",
       " '000 detect',\n",
       " '000 detection',\n",
       " '000 detergent',\n",
       " '000 determine',\n",
       " '000 digestion',\n",
       " '000 dilute',\n",
       " '000 diluted',\n",
       " '000 dilution',\n",
       " '000 dirt',\n",
       " '000 dissect',\n",
       " '000 dissolve',\n",
       " '000 double',\n",
       " '000 dynamic',\n",
       " '000 eighty',\n",
       " '000 eliminate',\n",
       " '000 elite',\n",
       " '000 elute',\n",
       " '000 employ',\n",
       " '000 endomembrane',\n",
       " '000 enzyme',\n",
       " '000 eppendorf',\n",
       " '000 etd',\n",
       " '000 exactive',\n",
       " '000 exclude',\n",
       " '000 exclusion',\n",
       " '000 execute',\n",
       " '000 exosome',\n",
       " '000 fhmw',\n",
       " '000 filter',\n",
       " '000 filtration',\n",
       " '000 final',\n",
       " '000 fix',\n",
       " '000 float',\n",
       " '000 flow',\n",
       " '000 follow',\n",
       " '000 foot',\n",
       " '000 fractions',\n",
       " '000 fragment',\n",
       " '000 fragmentation',\n",
       " '000 fragmented',\n",
       " '000 ft',\n",
       " '000 ftms',\n",
       " '000 ftms2',\n",
       " '000 fusion',\n",
       " '000 fw',\n",
       " '000 fwhm',\n",
       " '000 groundwater',\n",
       " '000 h2o2',\n",
       " '000 hcd',\n",
       " '000 high',\n",
       " '000 higher',\n",
       " '000 hla',\n",
       " '000 hour',\n",
       " '000 hr',\n",
       " '000 ief',\n",
       " '000 injection',\n",
       " '000 intact',\n",
       " '000 intense',\n",
       " '000 intensity',\n",
       " '000 internal',\n",
       " '000 ion',\n",
       " '000 ip',\n",
       " '000 isolate',\n",
       " '000 isolation',\n",
       " '000 itms2',\n",
       " '000 ja',\n",
       " '000 layer',\n",
       " '000 lb',\n",
       " '000 linear',\n",
       " '000 lit',\n",
       " '000 load',\n",
       " '000 lock',\n",
       " '000 lockmass',\n",
       " '000 low',\n",
       " '000 ltq',\n",
       " '000 lux',\n",
       " '000 lysates',\n",
       " '000 m2',\n",
       " '000 mass',\n",
       " '000 max',\n",
       " '000 maximal',\n",
       " '000 maximum',\n",
       " '000 measure',\n",
       " '000 membrane',\n",
       " '000 membranes',\n",
       " '000 method',\n",
       " '000 microliter',\n",
       " '000 microscan',\n",
       " '000 microsomal',\n",
       " '000 millipore',\n",
       " '000 min',\n",
       " '000 minimum',\n",
       " '000 minute',\n",
       " '000 ml',\n",
       " '000 mm',\n",
       " '000 molecular',\n",
       " '000 ms',\n",
       " '000 ms1',\n",
       " '000 ms2',\n",
       " '000 msn',\n",
       " '000 multiply',\n",
       " '000 mw',\n",
       " '000 mwco',\n",
       " '000 nanolc',\n",
       " '000 nce',\n",
       " '000 new',\n",
       " '000 nl',\n",
       " '000 nominal',\n",
       " '000 nondenaturing',\n",
       " '000 normalize',\n",
       " '000 normalized',\n",
       " '000 notably',\n",
       " '000 obtain',\n",
       " '000 oc',\n",
       " '000 orbitrap',\n",
       " '000 orbitrapxl',\n",
       " '000 order',\n",
       " '000 overnight',\n",
       " '000 parallel',\n",
       " '000 parent',\n",
       " '000 peak',\n",
       " '000 pellet',\n",
       " '000 pelletise',\n",
       " '000 pellets',\n",
       " '000 peptide',\n",
       " '000 perform',\n",
       " '000 phosphopeptides',\n",
       " '000 pi',\n",
       " '000 pore',\n",
       " '000 positive',\n",
       " '000 precipitate',\n",
       " '000 preclude',\n",
       " '000 precursor',\n",
       " '000 predictive',\n",
       " '000 preparation',\n",
       " '000 prepare',\n",
       " '000 preview',\n",
       " '000 previously',\n",
       " '000 prior',\n",
       " '000 prm',\n",
       " '000 product',\n",
       " '000 profile',\n",
       " '000 pron',\n",
       " '000 protein',\n",
       " '000 psi',\n",
       " '000 range',\n",
       " '000 rapid',\n",
       " '000 rcf',\n",
       " '000 reach',\n",
       " '000 record',\n",
       " '000 reduce',\n",
       " '000 regular',\n",
       " '000 relative',\n",
       " '000 remove',\n",
       " '000 replicate',\n",
       " '000 resin',\n",
       " '000 resolution',\n",
       " '000 resolve',\n",
       " '000 result',\n",
       " '000 resuspend',\n",
       " '000 rf',\n",
       " '000 room',\n",
       " '000 rp',\n",
       " '000 rpm',\n",
       " '000 rt',\n",
       " '000 run',\n",
       " '000 sample',\n",
       " '000 scan',\n",
       " '000 second',\n",
       " '000 seconde',\n",
       " '000 sediment',\n",
       " '000 select',\n",
       " '000 self',\n",
       " '000 separate',\n",
       " '000 sequence',\n",
       " '000 sh',\n",
       " '000 shrna',\n",
       " '000 signal',\n",
       " '000 silac',\n",
       " '000 simultaneously',\n",
       " '000 singly',\n",
       " '000 solid',\n",
       " '000 solubilization',\n",
       " '000 soluble',\n",
       " '000 sonicat',\n",
       " '000 source',\n",
       " '000 spectra',\n",
       " '000 spin',\n",
       " '000 spore',\n",
       " '000 sps3',\n",
       " '000 stage',\n",
       " '000 strip',\n",
       " '000 subject',\n",
       " '000 subsequent',\n",
       " '000 subsequently',\n",
       " '000 supernatant',\n",
       " '000 supernatants',\n",
       " '000 supplemental',\n",
       " '000 survey',\n",
       " '000 sw',\n",
       " '000 swath',\n",
       " '000 synaptosome',\n",
       " '000 tandem',\n",
       " '000 target',\n",
       " '000 temperature',\n",
       " '000 th',\n",
       " '000 thompson',\n",
       " '000 threshold',\n",
       " '000 time',\n",
       " '000 tmt',\n",
       " '000 top10',\n",
       " '000 top12',\n",
       " '000 total',\n",
       " '000 transfer',\n",
       " '000 treat',\n",
       " '000 trypsin',\n",
       " '000 tsim',\n",
       " '000 ubiquitylated',\n",
       " '000 ultra',\n",
       " '000 ultracentrifugation',\n",
       " '000 ultracentrifuge',\n",
       " '000 undissolved',\n",
       " '000 unique',\n",
       " '000 unit',\n",
       " '000 upper',\n",
       " '000 urea',\n",
       " '000 use',\n",
       " '000 value',\n",
       " '000 velos',\n",
       " '000 vh',\n",
       " '000 vhr',\n",
       " '000 vhrs',\n",
       " '000 volt',\n",
       " '000 voltage',\n",
       " '000 volume',\n",
       " '000 wash',\n",
       " '000 washed',\n",
       " '000 washing',\n",
       " '000 width',\n",
       " '000 window',\n",
       " '000 worm',\n",
       " '000 xg',\n",
       " '000 yield',\n",
       " '000 z200',\n",
       " '0000 fhmw',\n",
       " '0000 ion',\n",
       " '0000 resolution',\n",
       " '0000 rpm',\n",
       " '00001 flow',\n",
       " '0001 sample',\n",
       " '0002 use',\n",
       " '0005 dharmacon',\n",
       " '0005 mm',\n",
       " '0006 bio',\n",
       " '0006 room',\n",
       " '000664 house',\n",
       " '0008 correspond',\n",
       " '000g 10',\n",
       " '000g 15',\n",
       " '000g 15min',\n",
       " '000g 20',\n",
       " '000g 30',\n",
       " '000g 30min',\n",
       " '000g 40',\n",
       " '000g 80',\n",
       " '000g 90',\n",
       " '000g aqueous',\n",
       " '000g beckman',\n",
       " '000g centrifugation',\n",
       " '000g concentrate',\n",
       " '000g desalt',\n",
       " '000g hour',\n",
       " '000g min',\n",
       " '000g protein',\n",
       " '000g result',\n",
       " '000g supernatant',\n",
       " '000g ultracentrifugation',\n",
       " '000g use',\n",
       " '000gav 1hr',\n",
       " '000gs 10',\n",
       " '000rcf remove',\n",
       " '000rpm 10min',\n",
       " '000rpm 10mins',\n",
       " '000rpm 30min',\n",
       " '000rpm minute',\n",
       " '000rpm quantify',\n",
       " '000x 10',\n",
       " '000x 15',\n",
       " '000x 20',\n",
       " '000x 30',\n",
       " '000x bead',\n",
       " '000x hr',\n",
       " '000x min',\n",
       " '000x minute',\n",
       " '000x ms',\n",
       " '000x rpm',\n",
       " '000x sample',\n",
       " '000x supernatant',\n",
       " '000xg 10',\n",
       " '000xg 15',\n",
       " '000xg 15min',\n",
       " '000xg 16h',\n",
       " '000xg 20',\n",
       " '000xg 30',\n",
       " '000xg 3hr',\n",
       " '000xg 45',\n",
       " '000xg 45min',\n",
       " '000xg 4oc',\n",
       " '000xg 70ti',\n",
       " '000xg 90',\n",
       " '000xg beckman',\n",
       " '000xg filter',\n",
       " '000xg guanidinium',\n",
       " '000xg hr',\n",
       " '000xg hrs',\n",
       " '000xg load',\n",
       " '000xg min',\n",
       " '000xg minute',\n",
       " '000xg protein',\n",
       " '000xg quantify',\n",
       " '000xg remove',\n",
       " '000xg rt',\n",
       " '000xg supernatant',\n",
       " '001 1mm',\n",
       " '001 bromophenol',\n",
       " '001 collision',\n",
       " '001 gel',\n",
       " '001 gonad',\n",
       " '001 ha',\n",
       " '001 ph',\n",
       " '001 st',\n",
       " '0018 znso4',\n",
       " '001810 01',\n",
       " '002 bromophenol',\n",
       " '002 bromphenol',\n",
       " '002 collect',\n",
       " '002 protein',\n",
       " '002 zwittergent',\n",
       " '00237 ramagli',\n",
       " '0027 kcl',\n",
       " '002980 mdsc',\n",
       " '003 bromophenol',\n",
       " '003 ruthenium',\n",
       " '0033 mm',\n",
       " '004 bromophenol',\n",
       " '0041 mm',\n",
       " '00417 search',\n",
       " '004233 06',\n",
       " '004nm fe',\n",
       " '005 bromophenol',\n",
       " '005 cm',\n",
       " '005 da',\n",
       " '005 dtt',\n",
       " '005 edta',\n",
       " '005 hepes',\n",
       " '005 heptafluorbutyric',\n",
       " '005 heptafluorobutyric',\n",
       " '005 hfba',\n",
       " '005 ng',\n",
       " '005 rapig',\n",
       " '005 tfa',\n",
       " '0050258 obtain',\n",
       " '0050258 purpose',\n",
       " '005microg microl',\n",
       " '006 abemaciclib',\n",
       " '006 carry',\n",
       " '006 leupeptin',\n",
       " '0061 21',\n",
       " '0062 run',\n",
       " '0068 mm',\n",
       " '0075 fa',\n",
       " '0075 mm',\n",
       " '0075 silwet',\n",
       " '00757 write',\n",
       " '0082 tryptic',\n",
       " '009 mm',\n",
       " '009506 2011',\n",
       " '00965 mgml1',\n",
       " '0098 tandem',\n",
       " '009a remove',\n",
       " '00e 03',\n",
       " '00e 04',\n",
       " '00e 05',\n",
       " '00e 06',\n",
       " '00e ltq',\n",
       " '00e orbitrap',\n",
       " '00g 4168',\n",
       " '01 0005',\n",
       " '01 01',\n",
       " '01 015',\n",
       " '01 02',\n",
       " '01 03',\n",
       " '01 04',\n",
       " '01 05',\n",
       " '01 07',\n",
       " '01 08',\n",
       " '01 10',\n",
       " '01 100',\n",
       " '01 14',\n",
       " '01 16h',\n",
       " '01 1h',\n",
       " '01 28',\n",
       " '01 35',\n",
       " '01 40',\n",
       " '01 51',\n",
       " '01 60',\n",
       " '01 88',\n",
       " '01 accord',\n",
       " '01 acetic',\n",
       " '01 add',\n",
       " '01 adjust',\n",
       " '01 agilent',\n",
       " '01 allele',\n",
       " '01 antioxidant',\n",
       " '01 aqueous',\n",
       " '01 bromophenol',\n",
       " '01 bromophenolblue',\n",
       " '01 cocl2',\n",
       " '01 cultivate',\n",
       " '01 da',\n",
       " '01 decyl',\n",
       " '01 details',\n",
       " '01 dg',\n",
       " '01 dharmacon',\n",
       " '01 digitonin',\n",
       " '01 dithiothreitol',\n",
       " '01 enhance',\n",
       " '01 fa',\n",
       " '01 formic',\n",
       " '01 freshly',\n",
       " '01 ge',\n",
       " '01 global',\n",
       " '01 h3po4',\n",
       " '01 hac',\n",
       " '01 hfba',\n",
       " '01 hla',\n",
       " '01 hour',\n",
       " '01 hplc',\n",
       " '01 hr',\n",
       " '01 igepal',\n",
       " '01 include',\n",
       " '01 inhibitor',\n",
       " '01 iodoacetamide',\n",
       " '01 kcl',\n",
       " '01 lys',\n",
       " '01 mass',\n",
       " '01 matrix',\n",
       " '01 maximum',\n",
       " '01 mg',\n",
       " '01 min',\n",
       " '01 ml',\n",
       " '01 mm',\n",
       " '01 mms',\n",
       " '01 morpholino',\n",
       " '01 nacl',\n",
       " '01 naco3',\n",
       " '01 nan3',\n",
       " '01 nh4oh',\n",
       " '01 np',\n",
       " '01 obtain',\n",
       " '01 pbs',\n",
       " '01 peg',\n",
       " '01 peptide',\n",
       " '01 phosphate',\n",
       " '01 porcine',\n",
       " '01 positive',\n",
       " '01 protease',\n",
       " '01 proteasemax',\n",
       " '01 proteasemaxtm',\n",
       " '01 protein',\n",
       " '01 rapigest',\n",
       " '01 rehydrat',\n",
       " '01 rev',\n",
       " '01 room',\n",
       " '01 sds',\n",
       " '01 set',\n",
       " '01 sodium',\n",
       " '01 solution',\n",
       " '01 sorafenib',\n",
       " '01 supplementary',\n",
       " '01 surfactant',\n",
       " '01 tandem',\n",
       " '01 tfa',\n",
       " '01 transfect',\n",
       " '01 trifluoroacetic',\n",
       " '01 tris',\n",
       " '01 triton',\n",
       " '01 trypsin',\n",
       " '01 tween',\n",
       " '01 ug',\n",
       " '01 washing',\n",
       " '010 10',\n",
       " '010 dynamic',\n",
       " '0103 mbar',\n",
       " '0104 cell',\n",
       " '010565 da',\n",
       " '010min 10',\n",
       " '0110 min',\n",
       " '0111 b4',\n",
       " '0114 01',\n",
       " '011429 briefly',\n",
       " '0115 0125',\n",
       " '01214 cystein',\n",
       " '0122089 purpose',\n",
       " '0125 10',\n",
       " '0125 25',\n",
       " '0125 ii',\n",
       " '0126 0130',\n",
       " '0127 0132',\n",
       " '0128 min',\n",
       " '0130 iii',\n",
       " '0132 process',\n",
       " '013334 119',\n",
       " '013334 1a19',\n",
       " '015 1414',\n",
       " '015 200',\n",
       " '015 digitonin',\n",
       " '015 heavy',\n",
       " '015 light',\n",
       " '015 min',\n",
       " '015 mm',\n",
       " '015 sec',\n",
       " '015804 01',\n",
       " '016 0293',\n",
       " '016 formic',\n",
       " '016 purify',\n",
       " '017077 g5',\n",
       " '017087 dilute',\n",
       " '017087 hsa',\n",
       " '018 01',\n",
       " '01800 use',\n",
       " '01801 usa',\n",
       " '018672 01',\n",
       " '019 mm',\n",
       " '019927 schiller',\n",
       " '01g final',\n",
       " '01g trypsin',\n",
       " '02 01',\n",
       " '02 018',\n",
       " '02 03',\n",
       " '02 07positive',\n",
       " '02 2014',\n",
       " '02 23',\n",
       " '02 allow',\n",
       " '02 antibody',\n",
       " '02 bound',\n",
       " '02 bromophenol',\n",
       " '02 bromphenol',\n",
       " '02 cell',\n",
       " '02 cm',\n",
       " '02 cross',\n",
       " '02 da',\n",
       " '02 edta',\n",
       " '02 formic',\n",
       " '02 frame',\n",
       " '02 incubate',\n",
       " '02 list',\n",
       " '02 lys',\n",
       " '02 mascot',\n",
       " '02 matrix',\n",
       " '02 mg',\n",
       " '02 min',\n",
       " '02 ml',\n",
       " '02 mm',\n",
       " '02 mms',\n",
       " '02 na2s2o3',\n",
       " '02 naf',\n",
       " '02 np',\n",
       " '02 pepstatin',\n",
       " '02 potassium',\n",
       " '02 prior',\n",
       " '02 proline',\n",
       " '02 proteasemax',\n",
       " '02 rapigest',\n",
       " '02 result',\n",
       " '02 sds',\n",
       " '02 sodium',\n",
       " '02 standard',\n",
       " '02 syngene',\n",
       " '02 tfa',\n",
       " '02 thermo',\n",
       " '02 thimerosal',\n",
       " '02 tween',\n",
       " '02 tween20',\n",
       " '020 incubate',\n",
       " '020 irradiate',\n",
       " '020 sample',\n",
       " '020 solvent',\n",
       " '020 supplement',\n",
       " '020061 02',\n",
       " '021 da',\n",
       " '021464 da',\n",
       " '022431102 aliquot',\n",
       " '022808 091211',\n",
       " '0230 plant',\n",
       " '024 min',\n",
       " '024 mm',\n",
       " '02451 add',\n",
       " '02454 usa',\n",
       " '02454920 succinic',\n",
       " '025 agc',\n",
       " '025 coomassie',\n",
       " '025 formaldehyde',\n",
       " '025 mg',\n",
       " '025 ms',\n",
       " '025 np',\n",
       " '025 proteasemax',\n",
       " '025 resolution',\n",
       " '025 serva',\n",
       " '025 tfa',\n",
       " '025 triethylphosphine',\n",
       " '025 trifluoroacetic',\n",
       " '025 trypsin',\n",
       " '0251m fix',\n",
       " '02541 hour',\n",
       " '026549 da',\n",
       " '0289 85',\n",
       " '0293 available',\n",
       " '0294885 22',\n",
       " '02ug ul',\n",
       " '03 00',\n",
       " '03 004nm',\n",
       " '03 01',\n",
       " '03 04',\n",
       " '03 07',\n",
       " '03 100',\n",
       " '03 110',\n",
       " '03 26',\n",
       " '03 27',\n",
       " '03 agilent',\n",
       " '03 cm',\n",
       " '03 count',\n",
       " '03 da',\n",
       " '03 glutamax',\n",
       " '03 glutamine',\n",
       " '03 glutmax',\n",
       " '03 mixture',\n",
       " '03 ml',\n",
       " '03 mmol',\n",
       " '03 panc',\n",
       " '03 panc05',\n",
       " '03 rapigest',\n",
       " '03 sds',\n",
       " '03 trifluoroacetic',\n",
       " '030 50',\n",
       " '030 60',\n",
       " '030 buffer',\n",
       " '030 min',\n",
       " '030 ml',\n",
       " '030 scx',\n",
       " '0300 gradient',\n",
       " '0307 4412',\n",
       " '0313 da',\n",
       " '0313 heavy',\n",
       " '031300 da',\n",
       " '0316 internal',\n",
       " '032 1525',\n",
       " '032mg ml',\n",
       " '035 participant',\n",
       " '035 sample',\n",
       " '036 mobile',\n",
       " '03n 103',\n",
       " '04 03',\n",
       " '04 05',\n",
       " '04 06',\n",
       " '04 13',\n",
       " '04 150',\n",
       " '04 36',\n",
       " '04 bromophenol',\n",
       " '04 buffer',\n",
       " '04 count',\n",
       " '04 cultivate',\n",
       " '04 ddm',\n",
       " '04 dtt',\n",
       " '04 fa',\n",
       " '04 greenbaum',\n",
       " '04 hemolymph',\n",
       " '04 label',\n",
       " '04 load',\n",
       " '04 ml',\n",
       " '04 naoh',\n",
       " '04 panc',\n",
       " '04 set',\n",
       " '04 sodium',\n",
       " '04 tfa',\n",
       " '04 tkcc',\n",
       " '04 trifluoroacetic',\n",
       " '040 110117',\n",
       " '040 20',\n",
       " '040 acetonitrile',\n",
       " '040 gateway',\n",
       " '040 solvent',\n",
       " '0415 kefir',\n",
       " '042 701',\n",
       " '042 mm',\n",
       " '0425 cerebral',\n",
       " '043 tfa',\n",
       " '044 10',\n",
       " '044 100',\n",
       " '045 mass',\n",
       " '045 min',\n",
       " '04693132001 mm',\n",
       " '04693159001 dpbs',\n",
       " '04693159001 resuspended',\n",
       " '049 82',\n",
       " '04a 4507',\n",
       " '04a 5452',\n",
       " '05 00e',\n",
       " '05 01',\n",
       " '05 056',\n",
       " '05 10',\n",
       " '05 125',\n",
       " '05 2013',\n",
       " '05 50',\n",
       " '05 accept',\n",
       " '05 acetic',\n",
       " '05 addition',\n",
       " '05 aerobically',\n",
       " '05 ambi',\n",
       " '05 ammonium',\n",
       " '05 anova',\n",
       " '05 aps',\n",
       " '05 aqueous',\n",
       " '05 bovine',\n",
       " '05 brij',\n",
       " '05 bromophenol',\n",
       " '05 cell',\n",
       " '05 choose',\n",
       " '05 ci',\n",
       " '05 coomassie',\n",
       " '05 count',\n",
       " '05 culture',\n",
       " '05 da',\n",
       " '05 determine',\n",
       " '05 dharmacon',\n",
       " '05 differentially',\n",
       " '05 digest',\n",
       " '05 dithiothreitol',\n",
       " '05 dm',\n",
       " '05 dmso',\n",
       " '05 dtt',\n",
       " '05 edta',\n",
       " '05 fa',\n",
       " '05 fdr',\n",
       " '05 final',\n",
       " '05 fold',\n",
       " '05 formaldehyde',\n",
       " '05 formic',\n",
       " '05 fwhm',\n",
       " '05 gel',\n",
       " '05 glucose',\n",
       " '05 guanidinehcl',\n",
       " '05 heptafluorobutyric',\n",
       " '05 human',\n",
       " '05 igepal',\n",
       " '05 igpal',\n",
       " '05 incubate',\n",
       " '05 iodoacetamide',\n",
       " '05 ion',\n",
       " '05 k2hpo4',\n",
       " '05 log2',\n",
       " '05 maximum',\n",
       " '05 mg',\n",
       " '05 mgso4',\n",
       " '05 mgso47h2o',\n",
       " '05 micro',\n",
       " '05 million',\n",
       " '05 min',\n",
       " '05 ml',\n",
       " '05 mm',\n",
       " '05 mnh4hco3',\n",
       " '05 muromachi',\n",
       " '05 na',\n",
       " '05 nacl',\n",
       " '05 naoh',\n",
       " '05 nh4hco3',\n",
       " '05 nh4oh',\n",
       " '05 np',\n",
       " '05 np40',\n",
       " '05 panc',\n",
       " '05 pellicle',\n",
       " '05 power',\n",
       " '05 present',\n",
       " '05 prior',\n",
       " '05 product',\n",
       " '05 proteasemax',\n",
       " '05 protein',\n",
       " '05 rapig',\n",
       " '05 rapigest',\n",
       " '05 respectively',\n",
       " '05 room',\n",
       " '05 sample',\n",
       " '05 saponin',\n",
       " '05 sds',\n",
       " '05 second',\n",
       " '05 sodium',\n",
       " '05 sonation',\n",
       " '05 student',\n",
       " '05 sucrose',\n",
       " '05 tcep',\n",
       " '05 temed',\n",
       " '05 tfa',\n",
       " '05 tkcc',\n",
       " '05 treat',\n",
       " '05 triethylammonium',\n",
       " '05 triethylammoniumbicarbonate',\n",
       " '05 triflouroacetic',\n",
       " '05 trifluoroacetic',\n",
       " '05 triply',\n",
       " '05 tris',\n",
       " '05 triton',\n",
       " '05 tritonx',\n",
       " '05 trypsin',\n",
       " '05 tween',\n",
       " '05 ul',\n",
       " '05 uniformly',\n",
       " '05 unit',\n",
       " '05 use',\n",
       " '05 ut',\n",
       " '05 yeast',\n",
       " '050 37',\n",
       " '050 buffer',\n",
       " '050 incubate',\n",
       " '050 mm',\n",
       " '050 solvent',\n",
       " '050 tfa',\n",
       " '05061 24',\n",
       " '0530 250',\n",
       " '0539 internal',\n",
       " '0545 20',\n",
       " '0545 366',\n",
       " '0545 hexnac',\n",
       " '0549 microvial',\n",
       " '055 mobile',\n",
       " '056 489',\n",
       " '056 aqueous',\n",
       " '0564 da',\n",
       " '05892791001 phosphatase',\n",
       " '05892791001 roche',\n",
       " '05da fix',\n",
       " '05mm iodoacetamide',\n",
       " '05mol sodium',\n",
       " '05mpa young',\n",
       " '05s production',\n",
       " '06 010',\n",
       " '06 20',\n",
       " '06 abemaciclib',\n",
       " ...]"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(len(with_nubers))\n",
    "with_nubers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.28342377449195133"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(with_nubers)/x.shape[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "62258\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0.28342377449195133"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "with_nubers2 = [col for col in x.columns if re.match(r'.*\\d+.*', col)]\n",
    "print(len(with_nubers2))\n",
    "len(with_nubers2)/x.shape[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['00 00',\n",
       " '00 01',\n",
       " '00 12',\n",
       " '00 15',\n",
       " '00 1800',\n",
       " '00 2000',\n",
       " '00 2425',\n",
       " '00 2426',\n",
       " '00 38',\n",
       " '00 50',\n",
       " '00 5l',\n",
       " '00 65',\n",
       " '00 assess',\n",
       " '00 centrifugation',\n",
       " '00 concentrated',\n",
       " '00 correspond',\n",
       " '00 flow',\n",
       " '00 ftms',\n",
       " '00 hrs',\n",
       " '00 instrument',\n",
       " '00 kv',\n",
       " '00 ld',\n",
       " '00 measurement',\n",
       " '00 ml',\n",
       " '00 mobile',\n",
       " '00 noon',\n",
       " '00 ph',\n",
       " '00 place',\n",
       " '00 pron',\n",
       " '00 sample',\n",
       " '00 soft',\n",
       " '00 sun',\n",
       " '00 tryptic',\n",
       " '000 000',\n",
       " '000 0000',\n",
       " '000 0e4',\n",
       " '000 0e5',\n",
       " '000 10',\n",
       " '000 100',\n",
       " '000 10min',\n",
       " '000 10minutes',\n",
       " '000 110',\n",
       " '000 12',\n",
       " '000 120',\n",
       " '000 15',\n",
       " '000 150',\n",
       " '000 15min',\n",
       " '000 15oc',\n",
       " '000 17',\n",
       " '000 17500',\n",
       " '000 18',\n",
       " '000 1h',\n",
       " '000 1h30',\n",
       " '000 20',\n",
       " '000 200',\n",
       " '000 200m',\n",
       " '000 20a',\n",
       " '000 20min',\n",
       " '000 21',\n",
       " '000 22',\n",
       " '000 24',\n",
       " '000 25',\n",
       " '000 250',\n",
       " '000 2aa',\n",
       " '000 2min',\n",
       " '000 30',\n",
       " '000 300',\n",
       " '000 30min',\n",
       " '000 35',\n",
       " '000 350',\n",
       " '000 375',\n",
       " '000 3801',\n",
       " '000 3e6',\n",
       " '000 3min',\n",
       " '000 40',\n",
       " '000 400',\n",
       " '000 4001',\n",
       " '000 400ms',\n",
       " '000 400th',\n",
       " '000 45',\n",
       " '000 4a',\n",
       " '000 4oc',\n",
       " '000 50',\n",
       " '000 500',\n",
       " '000 60',\n",
       " '000 60min',\n",
       " '000 70',\n",
       " '000 75',\n",
       " '000 7500',\n",
       " '000 800',\n",
       " '000 90',\n",
       " '000 abundant',\n",
       " '000 accumulation',\n",
       " '000 acetone',\n",
       " '000 acg',\n",
       " '000 acid',\n",
       " '000 acquire',\n",
       " '000 acquisition',\n",
       " '000 activation',\n",
       " '000 add',\n",
       " '000 additional',\n",
       " '000 aforementioned',\n",
       " '000 agc',\n",
       " '000 aliquot',\n",
       " '000 alkylat',\n",
       " '000 allow',\n",
       " '000 analysis',\n",
       " '000 animal',\n",
       " '000 apply',\n",
       " '000 auto',\n",
       " '000 automate',\n",
       " '000 automated',\n",
       " '000 automatic',\n",
       " '000 avoid',\n",
       " '000 base',\n",
       " '000 baseline',\n",
       " '000 bead',\n",
       " '000 biofuge',\n",
       " '000 buffer',\n",
       " '000 c2c12s',\n",
       " '000 cancer',\n",
       " '000 capillary',\n",
       " '000 capture',\n",
       " '000 case',\n",
       " '000 cat',\n",
       " '000 cell',\n",
       " '000 centrifugation',\n",
       " '000 centrifuge',\n",
       " '000 charge',\n",
       " '000 choose',\n",
       " '000 cid',\n",
       " '000 clear',\n",
       " '000 cleared',\n",
       " '000 collect',\n",
       " '000 collision',\n",
       " '000 concentrated',\n",
       " '000 concentration',\n",
       " '000 consist',\n",
       " '000 count',\n",
       " '000 cover',\n",
       " '000 cp',\n",
       " '000 cps',\n",
       " '000 crude',\n",
       " '000 cycle',\n",
       " '000 cytosolic',\n",
       " '000 da',\n",
       " '000 data',\n",
       " '000 datadependent',\n",
       " '000 datum',\n",
       " '000 dd',\n",
       " '000 debris',\n",
       " '000 define',\n",
       " '000 delay',\n",
       " '000 detect',\n",
       " '000 detection',\n",
       " '000 detergent',\n",
       " '000 determine',\n",
       " '000 digestion',\n",
       " '000 dilute',\n",
       " '000 diluted',\n",
       " '000 dilution',\n",
       " '000 dirt',\n",
       " '000 dissect',\n",
       " '000 dissolve',\n",
       " '000 double',\n",
       " '000 dynamic',\n",
       " '000 eighty',\n",
       " '000 eliminate',\n",
       " '000 elite',\n",
       " '000 elute',\n",
       " '000 employ',\n",
       " '000 endomembrane',\n",
       " '000 enzyme',\n",
       " '000 eppendorf',\n",
       " '000 etd',\n",
       " '000 exactive',\n",
       " '000 exclude',\n",
       " '000 exclusion',\n",
       " '000 execute',\n",
       " '000 exosome',\n",
       " '000 fhmw',\n",
       " '000 filter',\n",
       " '000 filtration',\n",
       " '000 final',\n",
       " '000 fix',\n",
       " '000 float',\n",
       " '000 flow',\n",
       " '000 follow',\n",
       " '000 foot',\n",
       " '000 fractions',\n",
       " '000 fragment',\n",
       " '000 fragmentation',\n",
       " '000 fragmented',\n",
       " '000 ft',\n",
       " '000 ftms',\n",
       " '000 ftms2',\n",
       " '000 fusion',\n",
       " '000 fw',\n",
       " '000 fwhm',\n",
       " '000 groundwater',\n",
       " '000 h2o2',\n",
       " '000 hcd',\n",
       " '000 high',\n",
       " '000 higher',\n",
       " '000 hla',\n",
       " '000 hour',\n",
       " '000 hr',\n",
       " '000 ief',\n",
       " '000 injection',\n",
       " '000 intact',\n",
       " '000 intense',\n",
       " '000 intensity',\n",
       " '000 internal',\n",
       " '000 ion',\n",
       " '000 ip',\n",
       " '000 isolate',\n",
       " '000 isolation',\n",
       " '000 itms2',\n",
       " '000 ja',\n",
       " '000 layer',\n",
       " '000 lb',\n",
       " '000 linear',\n",
       " '000 lit',\n",
       " '000 load',\n",
       " '000 lock',\n",
       " '000 lockmass',\n",
       " '000 low',\n",
       " '000 ltq',\n",
       " '000 lux',\n",
       " '000 lysates',\n",
       " '000 m2',\n",
       " '000 mass',\n",
       " '000 max',\n",
       " '000 maximal',\n",
       " '000 maximum',\n",
       " '000 measure',\n",
       " '000 membrane',\n",
       " '000 membranes',\n",
       " '000 method',\n",
       " '000 microliter',\n",
       " '000 microscan',\n",
       " '000 microsomal',\n",
       " '000 millipore',\n",
       " '000 min',\n",
       " '000 minimum',\n",
       " '000 minute',\n",
       " '000 ml',\n",
       " '000 mm',\n",
       " '000 molecular',\n",
       " '000 ms',\n",
       " '000 ms1',\n",
       " '000 ms2',\n",
       " '000 msn',\n",
       " '000 multiply',\n",
       " '000 mw',\n",
       " '000 mwco',\n",
       " '000 nanolc',\n",
       " '000 nce',\n",
       " '000 new',\n",
       " '000 nl',\n",
       " '000 nominal',\n",
       " '000 nondenaturing',\n",
       " '000 normalize',\n",
       " '000 normalized',\n",
       " '000 notably',\n",
       " '000 obtain',\n",
       " '000 oc',\n",
       " '000 orbitrap',\n",
       " '000 orbitrapxl',\n",
       " '000 order',\n",
       " '000 overnight',\n",
       " '000 parallel',\n",
       " '000 parent',\n",
       " '000 peak',\n",
       " '000 pellet',\n",
       " '000 pelletise',\n",
       " '000 pellets',\n",
       " '000 peptide',\n",
       " '000 perform',\n",
       " '000 phosphopeptides',\n",
       " '000 pi',\n",
       " '000 pore',\n",
       " '000 positive',\n",
       " '000 precipitate',\n",
       " '000 preclude',\n",
       " '000 precursor',\n",
       " '000 predictive',\n",
       " '000 preparation',\n",
       " '000 prepare',\n",
       " '000 preview',\n",
       " '000 previously',\n",
       " '000 prior',\n",
       " '000 prm',\n",
       " '000 product',\n",
       " '000 profile',\n",
       " '000 pron',\n",
       " '000 protein',\n",
       " '000 psi',\n",
       " '000 range',\n",
       " '000 rapid',\n",
       " '000 rcf',\n",
       " '000 reach',\n",
       " '000 record',\n",
       " '000 reduce',\n",
       " '000 regular',\n",
       " '000 relative',\n",
       " '000 remove',\n",
       " '000 replicate',\n",
       " '000 resin',\n",
       " '000 resolution',\n",
       " '000 resolve',\n",
       " '000 result',\n",
       " '000 resuspend',\n",
       " '000 rf',\n",
       " '000 room',\n",
       " '000 rp',\n",
       " '000 rpm',\n",
       " '000 rt',\n",
       " '000 run',\n",
       " '000 sample',\n",
       " '000 scan',\n",
       " '000 second',\n",
       " '000 seconde',\n",
       " '000 sediment',\n",
       " '000 select',\n",
       " '000 self',\n",
       " '000 separate',\n",
       " '000 sequence',\n",
       " '000 sh',\n",
       " '000 shrna',\n",
       " '000 signal',\n",
       " '000 silac',\n",
       " '000 simultaneously',\n",
       " '000 singly',\n",
       " '000 solid',\n",
       " '000 solubilization',\n",
       " '000 soluble',\n",
       " '000 sonicat',\n",
       " '000 source',\n",
       " '000 spectra',\n",
       " '000 spin',\n",
       " '000 spore',\n",
       " '000 sps3',\n",
       " '000 stage',\n",
       " '000 strip',\n",
       " '000 subject',\n",
       " '000 subsequent',\n",
       " '000 subsequently',\n",
       " '000 supernatant',\n",
       " '000 supernatants',\n",
       " '000 supplemental',\n",
       " '000 survey',\n",
       " '000 sw',\n",
       " '000 swath',\n",
       " '000 synaptosome',\n",
       " '000 tandem',\n",
       " '000 target',\n",
       " '000 temperature',\n",
       " '000 th',\n",
       " '000 thompson',\n",
       " '000 threshold',\n",
       " '000 time',\n",
       " '000 tmt',\n",
       " '000 top10',\n",
       " '000 top12',\n",
       " '000 total',\n",
       " '000 transfer',\n",
       " '000 treat',\n",
       " '000 trypsin',\n",
       " '000 tsim',\n",
       " '000 ubiquitylated',\n",
       " '000 ultra',\n",
       " '000 ultracentrifugation',\n",
       " '000 ultracentrifuge',\n",
       " '000 undissolved',\n",
       " '000 unique',\n",
       " '000 unit',\n",
       " '000 upper',\n",
       " '000 urea',\n",
       " '000 use',\n",
       " '000 value',\n",
       " '000 velos',\n",
       " '000 vh',\n",
       " '000 vhr',\n",
       " '000 vhrs',\n",
       " '000 volt',\n",
       " '000 voltage',\n",
       " '000 volume',\n",
       " '000 wash',\n",
       " '000 washed',\n",
       " '000 washing',\n",
       " '000 width',\n",
       " '000 window',\n",
       " '000 worm',\n",
       " '000 xg',\n",
       " '000 yield',\n",
       " '000 z200',\n",
       " '0000 fhmw',\n",
       " '0000 ion',\n",
       " '0000 resolution',\n",
       " '0000 rpm',\n",
       " '00001 flow',\n",
       " '0001 sample',\n",
       " '0002 use',\n",
       " '0005 dharmacon',\n",
       " '0005 mm',\n",
       " '0006 bio',\n",
       " '0006 room',\n",
       " '000664 house',\n",
       " '0008 correspond',\n",
       " '000g 10',\n",
       " '000g 15',\n",
       " '000g 15min',\n",
       " '000g 20',\n",
       " '000g 30',\n",
       " '000g 30min',\n",
       " '000g 40',\n",
       " '000g 80',\n",
       " '000g 90',\n",
       " '000g aqueous',\n",
       " '000g beckman',\n",
       " '000g centrifugation',\n",
       " '000g concentrate',\n",
       " '000g desalt',\n",
       " '000g hour',\n",
       " '000g min',\n",
       " '000g protein',\n",
       " '000g result',\n",
       " '000g supernatant',\n",
       " '000g ultracentrifugation',\n",
       " '000g use',\n",
       " '000gav 1hr',\n",
       " '000gs 10',\n",
       " '000rcf remove',\n",
       " '000rpm 10min',\n",
       " '000rpm 10mins',\n",
       " '000rpm 30min',\n",
       " '000rpm minute',\n",
       " '000rpm quantify',\n",
       " '000x 10',\n",
       " '000x 15',\n",
       " '000x 20',\n",
       " '000x 30',\n",
       " '000x bead',\n",
       " '000x hr',\n",
       " '000x min',\n",
       " '000x minute',\n",
       " '000x ms',\n",
       " '000x rpm',\n",
       " '000x sample',\n",
       " '000x supernatant',\n",
       " '000xg 10',\n",
       " '000xg 15',\n",
       " '000xg 15min',\n",
       " '000xg 16h',\n",
       " '000xg 20',\n",
       " '000xg 30',\n",
       " '000xg 3hr',\n",
       " '000xg 45',\n",
       " '000xg 45min',\n",
       " '000xg 4oc',\n",
       " '000xg 70ti',\n",
       " '000xg 90',\n",
       " '000xg beckman',\n",
       " '000xg filter',\n",
       " '000xg guanidinium',\n",
       " '000xg hr',\n",
       " '000xg hrs',\n",
       " '000xg load',\n",
       " '000xg min',\n",
       " '000xg minute',\n",
       " '000xg protein',\n",
       " '000xg quantify',\n",
       " '000xg remove',\n",
       " '000xg rt',\n",
       " '000xg supernatant',\n",
       " '001 1mm',\n",
       " '001 bromophenol',\n",
       " '001 collision',\n",
       " '001 gel',\n",
       " '001 gonad',\n",
       " '001 ha',\n",
       " '001 ph',\n",
       " '001 st',\n",
       " '0018 znso4',\n",
       " '001810 01',\n",
       " '002 bromophenol',\n",
       " '002 bromphenol',\n",
       " '002 collect',\n",
       " '002 protein',\n",
       " '002 zwittergent',\n",
       " '00237 ramagli',\n",
       " '0027 kcl',\n",
       " '002980 mdsc',\n",
       " '003 bromophenol',\n",
       " '003 ruthenium',\n",
       " '0033 mm',\n",
       " '004 bromophenol',\n",
       " '0041 mm',\n",
       " '00417 search',\n",
       " '004233 06',\n",
       " '004nm fe',\n",
       " '005 bromophenol',\n",
       " '005 cm',\n",
       " '005 da',\n",
       " '005 dtt',\n",
       " '005 edta',\n",
       " '005 hepes',\n",
       " '005 heptafluorbutyric',\n",
       " '005 heptafluorobutyric',\n",
       " '005 hfba',\n",
       " '005 ng',\n",
       " '005 rapig',\n",
       " '005 tfa',\n",
       " '0050258 obtain',\n",
       " '0050258 purpose',\n",
       " '005microg microl',\n",
       " '006 abemaciclib',\n",
       " '006 carry',\n",
       " '006 leupeptin',\n",
       " '0061 21',\n",
       " '0062 run',\n",
       " '0068 mm',\n",
       " '0075 fa',\n",
       " '0075 mm',\n",
       " '0075 silwet',\n",
       " '00757 write',\n",
       " '0082 tryptic',\n",
       " '009 mm',\n",
       " '009506 2011',\n",
       " '00965 mgml1',\n",
       " '0098 tandem',\n",
       " '009a remove',\n",
       " '00e 03',\n",
       " '00e 04',\n",
       " '00e 05',\n",
       " '00e 06',\n",
       " '00e ltq',\n",
       " '00e orbitrap',\n",
       " '00g 4168',\n",
       " '01 0005',\n",
       " '01 01',\n",
       " '01 015',\n",
       " '01 02',\n",
       " '01 03',\n",
       " '01 04',\n",
       " '01 05',\n",
       " '01 07',\n",
       " '01 08',\n",
       " '01 10',\n",
       " '01 100',\n",
       " '01 14',\n",
       " '01 16h',\n",
       " '01 1h',\n",
       " '01 28',\n",
       " '01 35',\n",
       " '01 40',\n",
       " '01 51',\n",
       " '01 60',\n",
       " '01 88',\n",
       " '01 accord',\n",
       " '01 acetic',\n",
       " '01 add',\n",
       " '01 adjust',\n",
       " '01 agilent',\n",
       " '01 allele',\n",
       " '01 antioxidant',\n",
       " '01 aqueous',\n",
       " '01 bromophenol',\n",
       " '01 bromophenolblue',\n",
       " '01 cocl2',\n",
       " '01 cultivate',\n",
       " '01 da',\n",
       " '01 decyl',\n",
       " '01 details',\n",
       " '01 dg',\n",
       " '01 dharmacon',\n",
       " '01 digitonin',\n",
       " '01 dithiothreitol',\n",
       " '01 enhance',\n",
       " '01 fa',\n",
       " '01 formic',\n",
       " '01 freshly',\n",
       " '01 ge',\n",
       " '01 global',\n",
       " '01 h3po4',\n",
       " '01 hac',\n",
       " '01 hfba',\n",
       " '01 hla',\n",
       " '01 hour',\n",
       " '01 hplc',\n",
       " '01 hr',\n",
       " '01 igepal',\n",
       " '01 include',\n",
       " '01 inhibitor',\n",
       " '01 iodoacetamide',\n",
       " '01 kcl',\n",
       " '01 lys',\n",
       " '01 mass',\n",
       " '01 matrix',\n",
       " '01 maximum',\n",
       " '01 mg',\n",
       " '01 min',\n",
       " '01 ml',\n",
       " '01 mm',\n",
       " '01 mms',\n",
       " '01 morpholino',\n",
       " '01 nacl',\n",
       " '01 naco3',\n",
       " '01 nan3',\n",
       " '01 nh4oh',\n",
       " '01 np',\n",
       " '01 obtain',\n",
       " '01 pbs',\n",
       " '01 peg',\n",
       " '01 peptide',\n",
       " '01 phosphate',\n",
       " '01 porcine',\n",
       " '01 positive',\n",
       " '01 protease',\n",
       " '01 proteasemax',\n",
       " '01 proteasemaxtm',\n",
       " '01 protein',\n",
       " '01 rapigest',\n",
       " '01 rehydrat',\n",
       " '01 rev',\n",
       " '01 room',\n",
       " '01 sds',\n",
       " '01 set',\n",
       " '01 sodium',\n",
       " '01 solution',\n",
       " '01 sorafenib',\n",
       " '01 supplementary',\n",
       " '01 surfactant',\n",
       " '01 tandem',\n",
       " '01 tfa',\n",
       " '01 transfect',\n",
       " '01 trifluoroacetic',\n",
       " '01 tris',\n",
       " '01 triton',\n",
       " '01 trypsin',\n",
       " '01 tween',\n",
       " '01 ug',\n",
       " '01 washing',\n",
       " '010 10',\n",
       " '010 dynamic',\n",
       " '0103 mbar',\n",
       " '0104 cell',\n",
       " '010565 da',\n",
       " '010min 10',\n",
       " '0110 min',\n",
       " '0111 b4',\n",
       " '0114 01',\n",
       " '011429 briefly',\n",
       " '0115 0125',\n",
       " '01214 cystein',\n",
       " '0122089 purpose',\n",
       " '0125 10',\n",
       " '0125 25',\n",
       " '0125 ii',\n",
       " '0126 0130',\n",
       " '0127 0132',\n",
       " '0128 min',\n",
       " '0130 iii',\n",
       " '0132 process',\n",
       " '013334 119',\n",
       " '013334 1a19',\n",
       " '015 1414',\n",
       " '015 200',\n",
       " '015 digitonin',\n",
       " '015 heavy',\n",
       " '015 light',\n",
       " '015 min',\n",
       " '015 mm',\n",
       " '015 sec',\n",
       " '015804 01',\n",
       " '016 0293',\n",
       " '016 formic',\n",
       " '016 purify',\n",
       " '017077 g5',\n",
       " '017087 dilute',\n",
       " '017087 hsa',\n",
       " '018 01',\n",
       " '01800 use',\n",
       " '01801 usa',\n",
       " '018672 01',\n",
       " '019 mm',\n",
       " '019927 schiller',\n",
       " '01g final',\n",
       " '01g trypsin',\n",
       " '02 01',\n",
       " '02 018',\n",
       " '02 03',\n",
       " '02 07positive',\n",
       " '02 2014',\n",
       " '02 23',\n",
       " '02 allow',\n",
       " '02 antibody',\n",
       " '02 bound',\n",
       " '02 bromophenol',\n",
       " '02 bromphenol',\n",
       " '02 cell',\n",
       " '02 cm',\n",
       " '02 cross',\n",
       " '02 da',\n",
       " '02 edta',\n",
       " '02 formic',\n",
       " '02 frame',\n",
       " '02 incubate',\n",
       " '02 list',\n",
       " '02 lys',\n",
       " '02 mascot',\n",
       " '02 matrix',\n",
       " '02 mg',\n",
       " '02 min',\n",
       " '02 ml',\n",
       " '02 mm',\n",
       " '02 mms',\n",
       " '02 na2s2o3',\n",
       " '02 naf',\n",
       " '02 np',\n",
       " '02 pepstatin',\n",
       " '02 potassium',\n",
       " '02 prior',\n",
       " '02 proline',\n",
       " '02 proteasemax',\n",
       " '02 rapigest',\n",
       " '02 result',\n",
       " '02 sds',\n",
       " '02 sodium',\n",
       " '02 standard',\n",
       " '02 syngene',\n",
       " '02 tfa',\n",
       " '02 thermo',\n",
       " '02 thimerosal',\n",
       " '02 tween',\n",
       " '02 tween20',\n",
       " '020 incubate',\n",
       " '020 irradiate',\n",
       " '020 sample',\n",
       " '020 solvent',\n",
       " '020 supplement',\n",
       " '020061 02',\n",
       " '021 da',\n",
       " '021464 da',\n",
       " '022431102 aliquot',\n",
       " '022808 091211',\n",
       " '0230 plant',\n",
       " '024 min',\n",
       " '024 mm',\n",
       " '02451 add',\n",
       " '02454 usa',\n",
       " '02454920 succinic',\n",
       " '025 agc',\n",
       " '025 coomassie',\n",
       " '025 formaldehyde',\n",
       " '025 mg',\n",
       " '025 ms',\n",
       " '025 np',\n",
       " '025 proteasemax',\n",
       " '025 resolution',\n",
       " '025 serva',\n",
       " '025 tfa',\n",
       " '025 triethylphosphine',\n",
       " '025 trifluoroacetic',\n",
       " '025 trypsin',\n",
       " '0251m fix',\n",
       " '02541 hour',\n",
       " '026549 da',\n",
       " '0289 85',\n",
       " '0293 available',\n",
       " '0294885 22',\n",
       " '02ug ul',\n",
       " '03 00',\n",
       " '03 004nm',\n",
       " '03 01',\n",
       " '03 04',\n",
       " '03 07',\n",
       " '03 100',\n",
       " '03 110',\n",
       " '03 26',\n",
       " '03 27',\n",
       " '03 agilent',\n",
       " '03 cm',\n",
       " '03 count',\n",
       " '03 da',\n",
       " '03 glutamax',\n",
       " '03 glutamine',\n",
       " '03 glutmax',\n",
       " '03 mixture',\n",
       " '03 ml',\n",
       " '03 mmol',\n",
       " '03 panc',\n",
       " '03 panc05',\n",
       " '03 rapigest',\n",
       " '03 sds',\n",
       " '03 trifluoroacetic',\n",
       " '030 50',\n",
       " '030 60',\n",
       " '030 buffer',\n",
       " '030 min',\n",
       " '030 ml',\n",
       " '030 scx',\n",
       " '0300 gradient',\n",
       " '0307 4412',\n",
       " '0313 da',\n",
       " '0313 heavy',\n",
       " '031300 da',\n",
       " '0316 internal',\n",
       " '032 1525',\n",
       " '032mg ml',\n",
       " '035 participant',\n",
       " '035 sample',\n",
       " '036 mobile',\n",
       " '03n 103',\n",
       " '04 03',\n",
       " '04 05',\n",
       " '04 06',\n",
       " '04 13',\n",
       " '04 150',\n",
       " '04 36',\n",
       " '04 bromophenol',\n",
       " '04 buffer',\n",
       " '04 count',\n",
       " '04 cultivate',\n",
       " '04 ddm',\n",
       " '04 dtt',\n",
       " '04 fa',\n",
       " '04 greenbaum',\n",
       " '04 hemolymph',\n",
       " '04 label',\n",
       " '04 load',\n",
       " '04 ml',\n",
       " '04 naoh',\n",
       " '04 panc',\n",
       " '04 set',\n",
       " '04 sodium',\n",
       " '04 tfa',\n",
       " '04 tkcc',\n",
       " '04 trifluoroacetic',\n",
       " '040 110117',\n",
       " '040 20',\n",
       " '040 acetonitrile',\n",
       " '040 gateway',\n",
       " '040 solvent',\n",
       " '0415 kefir',\n",
       " '042 701',\n",
       " '042 mm',\n",
       " '0425 cerebral',\n",
       " '043 tfa',\n",
       " '044 10',\n",
       " '044 100',\n",
       " '045 mass',\n",
       " '045 min',\n",
       " '04693132001 mm',\n",
       " '04693159001 dpbs',\n",
       " '04693159001 resuspended',\n",
       " '049 82',\n",
       " '04a 4507',\n",
       " '04a 5452',\n",
       " '05 00e',\n",
       " '05 01',\n",
       " '05 056',\n",
       " '05 10',\n",
       " '05 125',\n",
       " '05 2013',\n",
       " '05 50',\n",
       " '05 accept',\n",
       " '05 acetic',\n",
       " '05 addition',\n",
       " '05 aerobically',\n",
       " '05 ambi',\n",
       " '05 ammonium',\n",
       " '05 anova',\n",
       " '05 aps',\n",
       " '05 aqueous',\n",
       " '05 bovine',\n",
       " '05 brij',\n",
       " '05 bromophenol',\n",
       " '05 cell',\n",
       " '05 choose',\n",
       " '05 ci',\n",
       " '05 coomassie',\n",
       " '05 count',\n",
       " '05 culture',\n",
       " '05 da',\n",
       " '05 determine',\n",
       " '05 dharmacon',\n",
       " '05 differentially',\n",
       " '05 digest',\n",
       " '05 dithiothreitol',\n",
       " '05 dm',\n",
       " '05 dmso',\n",
       " '05 dtt',\n",
       " '05 edta',\n",
       " '05 fa',\n",
       " '05 fdr',\n",
       " '05 final',\n",
       " '05 fold',\n",
       " '05 formaldehyde',\n",
       " '05 formic',\n",
       " '05 fwhm',\n",
       " '05 gel',\n",
       " '05 glucose',\n",
       " '05 guanidinehcl',\n",
       " '05 heptafluorobutyric',\n",
       " '05 human',\n",
       " '05 igepal',\n",
       " '05 igpal',\n",
       " '05 incubate',\n",
       " '05 iodoacetamide',\n",
       " '05 ion',\n",
       " '05 k2hpo4',\n",
       " '05 log2',\n",
       " '05 maximum',\n",
       " '05 mg',\n",
       " '05 mgso4',\n",
       " '05 mgso47h2o',\n",
       " '05 micro',\n",
       " '05 million',\n",
       " '05 min',\n",
       " '05 ml',\n",
       " '05 mm',\n",
       " '05 mnh4hco3',\n",
       " '05 muromachi',\n",
       " '05 na',\n",
       " '05 nacl',\n",
       " '05 naoh',\n",
       " '05 nh4hco3',\n",
       " '05 nh4oh',\n",
       " '05 np',\n",
       " '05 np40',\n",
       " '05 panc',\n",
       " '05 pellicle',\n",
       " '05 power',\n",
       " '05 present',\n",
       " '05 prior',\n",
       " '05 product',\n",
       " '05 proteasemax',\n",
       " '05 protein',\n",
       " '05 rapig',\n",
       " '05 rapigest',\n",
       " '05 respectively',\n",
       " '05 room',\n",
       " '05 sample',\n",
       " '05 saponin',\n",
       " '05 sds',\n",
       " '05 second',\n",
       " '05 sodium',\n",
       " '05 sonation',\n",
       " '05 student',\n",
       " '05 sucrose',\n",
       " '05 tcep',\n",
       " '05 temed',\n",
       " '05 tfa',\n",
       " '05 tkcc',\n",
       " '05 treat',\n",
       " '05 triethylammonium',\n",
       " '05 triethylammoniumbicarbonate',\n",
       " '05 triflouroacetic',\n",
       " '05 trifluoroacetic',\n",
       " '05 triply',\n",
       " '05 tris',\n",
       " '05 triton',\n",
       " '05 tritonx',\n",
       " '05 trypsin',\n",
       " '05 tween',\n",
       " '05 ul',\n",
       " '05 uniformly',\n",
       " '05 unit',\n",
       " '05 use',\n",
       " '05 ut',\n",
       " '05 yeast',\n",
       " '050 37',\n",
       " '050 buffer',\n",
       " '050 incubate',\n",
       " '050 mm',\n",
       " '050 solvent',\n",
       " '050 tfa',\n",
       " '05061 24',\n",
       " '0530 250',\n",
       " '0539 internal',\n",
       " '0545 20',\n",
       " '0545 366',\n",
       " '0545 hexnac',\n",
       " '0549 microvial',\n",
       " '055 mobile',\n",
       " '056 489',\n",
       " '056 aqueous',\n",
       " '0564 da',\n",
       " '05892791001 phosphatase',\n",
       " '05892791001 roche',\n",
       " '05da fix',\n",
       " '05mm iodoacetamide',\n",
       " '05mol sodium',\n",
       " '05mpa young',\n",
       " '05s production',\n",
       " '06 010',\n",
       " '06 20',\n",
       " '06 abemaciclib',\n",
       " ...]"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "with_nubers2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "So use the short form of regex..."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
